---
title: "Final version of Leaf Heterosis proteomics for Paper"
output: github_document
---
#Setup
```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
set.seed(586)
```
```{r libraries}
library(matrixStats)
library(topGO)
library(ggplot2)
library(AnnotationDbi)
library(cowplot)
library(dplyr)
library(gage)
library(datapasta)
library(tibble)
library(RColorBrewer)
library(gganimate)
library(pathview)
library(GO.db)
```

#Import data
```{r read csv}
tmt <- read.csv("data/Leaf Paper Data 050520-processed.csv", stringsAsFactors = F)
#re-define protein column to shortened name
tmt <- tmt[,-182]
colnames(tmt)[182] <- "Protein"
#make new column for gene accession without transcript id
tmt$Gene <- substr(tmt$Accession, start = 1, stop = 14)
tmt.withPolymorphic <- tmt
#Remove polymorphic peptides
tmt <- tmt[-(grep("only", tmt$Accession)),]
```
```{r}
for(i in 1:170){
  tmt[,i] <- as.numeric(tmt)
}
```

```{r T-test function}
ttestFun <- function(df, col1, col2){
  result <- NULL
  for(i in 1:(nrow(df))){
  v1 = (as.numeric(df[i, col1]))
  v2 = (as.numeric(df[i, col2]))
  t_test <- t.test(v1, v2, var.equal = TRUE)
  pval <- t_test$p.value
  result[i] <- pval
  }
  result
}
```

##Gamer GO
```{r read in gamer}
# Read in Maize GAMER GO dataset, which lists maize V4 accessions with plastid associated GO terms (Wimalanathan et al. 2018) downloaded from https://datacommons.cyverse.org/browse/iplant/home/shared/commons_repo/curated/Carolyn_Lawrence-Dill_maize-GAMER_maize.B73_RefGen_v4_Zm00001d.2_Oct_2017.r1/e.agg_data
GAMER <- read.delim("maize.B73.AGPv4.aggregate.gaf", sep = "\t", header = FALSE)
colnames(GAMER) <- c("db", "db_object_id", "db_object_symbol", "qualifier", "term_accession", "db_reference", "evidence_code", "with", "aspect", "db_object_name", "db_object_synonym", "db_object_type", "taxon", "date", "assigned_by", "annotation_extension", "gene_product_form_id")
gamer <- GAMER[-(1:2),]
gamer$Term <- Term(as.character(gamer$term_accession))
```

#Format dfs
###original BxM SL
```{r format df}
#Subset
tmt.OG.SL <- tmt[,c(171:176,181:183,12:20)]
#Remove missing genes
tmt.OG.SL <- tmt.OG.SL[!is.na(tmt.OG.SL$SeedlingLeaf.B73.r1),]
#Calculate averages of bio reps
tmt.OG.SL$B73avg <- rowMeans(tmt.OG.SL[,10:12])
tmt.OG.SL$Mo17avg <- rowMeans(tmt.OG.SL[,13:15])
tmt.OG.SL$HYBavg <- rowMeans(tmt.OG.SL[,16:18])
tmt.OG.SL$MPavg <- rowMeans(tmt.OG.SL[,c(10:15)])
tmt.OG.SL$MP.1 <- rowMeans(tmt.OG.SL[,c(10,13)])
tmt.OG.SL$MP.2 <- rowMeans(tmt.OG.SL[,c(11,15)])
tmt.OG.SL$MP.3 <- rowMeans(tmt.OG.SL[,c(12,15)])
#calculate ratio
tmt.OG.SL$HYB.MP <- tmt.OG.SL$HYBavg/tmt.OG.SL$MPavg
#calculate t-test
tmt.OG.SL$Sig <- ttestFun(tmt.OG.SL, 16:18, 23:25)
#calculate log2(hyb/MP)
tmt.OG.SL$l2.BM.MP <- log2(tmt.OG.SL$HYB.MP)
#Calculate log2(B73/Mo17)
tmt.OG.SL$B73.Mo17 <- tmt.OG.SL$B73avg/tmt.OG.SL$Mo17avg
tmt.OG.SL$l2.B73.Mo17 <-log2(tmt.OG.SL$B73.Mo17)
saveRDS(tmt.OG.SL, "tmt.OG.SL")
```

###6hybrid BxM SL
```{r}
#subset
tmt.6h.SL <- tmt[,c(171:176,181:183,31,41,51,33,43,53,35,45,55)]
#Remove missing genes
tmt.6h.SL <- tmt.6h.SL[!is.na(tmt.6h.SL$X6Hybrid.B73.r1),]
tmt.6h.SL <- tmt.6h.SL[!is.na(tmt.6h.SL$X6Hybrid.B73.r2),]
tmt.6h.SL <- tmt.6h.SL[!is.na(tmt.6h.SL$X6Hybrid.B73.r3),]
#Calculate averages of bio reps
tmt.6h.SL$B73avg <- rowMeans(tmt.6h.SL[,10:12])
tmt.6h.SL$Mo17avg <- rowMeans(tmt.6h.SL[,13:15])
tmt.6h.SL$HYBavg <- rowMeans(tmt.6h.SL[,16:18])
tmt.6h.SL$MPavg <- rowMeans(tmt.6h.SL[,c(10:15)])
tmt.6h.SL$MP.1 <- rowMeans(tmt.6h.SL[,c(10,13)])
tmt.6h.SL$MP.2 <- rowMeans(tmt.6h.SL[,c(11,14)])
tmt.6h.SL$MP.3 <- rowMeans(tmt.6h.SL[,c(12,15)])
#calculate ratio
tmt.6h.SL$HYB.MP <- tmt.6h.SL$HYBavg/tmt.6h.SL$MPavg
#calculate t-test
tmt.6h.SL$Sig <- ttestFun(tmt.6h.SL, 16:18, 23:25)
#calculate log2(hyb/MP)
tmt.6h.SL$l2.BM.MP <- log2(tmt.6h.SL$HYB.MP)
#Calculate log2(B73/Mo17)
tmt.6h.SL$B73.Mo17 <- tmt.6h.SL$B73avg/tmt.6h.SL$Mo17avg
tmt.6h.SL$l2.B73.Mo17 <-log2(tmt.6h.SL$B73.Mo17)
saveRDS(tmt.6h.SL, "tmt.6h.SL")
```
###Day/night BxM SL
```{r}
#subset
tmt.dn.SL <- tmt[,c(171:176,181:183,62:70)]
#Remove missing genes
tmt.dn.SL <- tmt.dn.SL[!is.na(tmt.dn.SL$Day.B73.r1),]
#Calculate averages of bio reps
tmt.dn.SL$B73avg <- rowMeans(tmt.dn.SL[,10:12])
tmt.dn.SL$Mo17avg <- rowMeans(tmt.dn.SL[,13:15])
tmt.dn.SL$HYBavg <- rowMeans(tmt.dn.SL[,16:18])
tmt.dn.SL$MPavg <- rowMeans(tmt.dn.SL[,c(10:15)])
tmt.dn.SL$MP.1 <- rowMeans(tmt.dn.SL[,c(10,13)])
tmt.dn.SL$MP.2 <- rowMeans(tmt.dn.SL[,c(11,14)])
tmt.dn.SL$MP.3 <- rowMeans(tmt.dn.SL[,c(12,15)])
#calculate ratio
tmt.dn.SL$HYB.MP <- tmt.dn.SL$HYBavg/tmt.dn.SL$MPavg
#calculate t-test
tmt.dn.SL$Sig <- ttestFun(tmt.dn.SL, 16:18, 23:25)
#calculate log2(hyb/MP)
tmt.dn.SL$l2.BM.MP <- log2(tmt.dn.SL$HYB.MP)
#Calculate log2(B73/Mo17)
tmt.dn.SL$B73.Mo17 <- tmt.dn.SL$B73avg/tmt.dn.SL$Mo17avg
tmt.dn.SL$l2.B73.Mo17 <-log2(tmt.dn.SL$B73.Mo17)
saveRDS(tmt.dn.SL, "tmt.dn.SL")
```
###RIL BxM SL
```{r}
#subset
tmt.ril.SL <- tmt[,c(171:176,181:183,81:89)]
#Remove missing genes
tmt.ril.SL <- tmt.ril.SL[!is.na(tmt.ril.SL$B73.1),]
#Calculate averages of bio reps
tmt.ril.SL$B73avg <- rowMeans(tmt.ril.SL[,10:12])
tmt.ril.SL$Mo17avg <- rowMeans(tmt.ril.SL[,13:15])
tmt.ril.SL$HYBavg <- rowMeans(tmt.ril.SL[,16:18])
tmt.ril.SL$MPavg <- rowMeans(tmt.ril.SL[,c(10:15)])
tmt.ril.SL$MP.1 <- rowMeans(tmt.ril.SL[,c(10,13)])
tmt.ril.SL$MP.2 <- rowMeans(tmt.ril.SL[,c(11,14)])
tmt.ril.SL$MP.3 <- rowMeans(tmt.ril.SL[,c(12,15)])
#calculate ratio
tmt.ril.SL$HYB.MP <- tmt.ril.SL$HYBavg/tmt.ril.SL$MPavg
#calculate t-test
tmt.ril.SL$Sig <- ttestFun(tmt.ril.SL, 16:18, 23:25)
#calculate log2(hyb/MP)
tmt.ril.SL$l2.BM.MP <- log2(tmt.ril.SL$HYB.MP)
#Calculate log2(B73/Mo17)
tmt.ril.SL$B73.Mo17 <- tmt.ril.SL$B73avg/tmt.ril.SL$Mo17avg
tmt.ril.SL$l2.B73.Mo17 <-log2(tmt.ril.SL$B73.Mo17)
saveRDS(tmt.ril.SL, "tmt.ril.SL")
```
```{r}
tmt.OG.SL <- readRDS("tmt.OG.SL")
tmt.6h.SL <- readRDS("tmt.6h.SL")
tmt.dn.SL <- readRDS("tmt.dn.SL")
tmt.ril.SL <- readRDS("tmt.ril.SL")
```

###original BxM SL + acs
```{r format df}
#Subset
tmt.OG.acs.SL <- tmt[,c(171:176,181:183,12:20, 1:10)]
#Remove missing genes
tmt.OG.acs.SL <- tmt.OG.acs.SL[!is.na(tmt.OG.acs.SL$SeedlingLeaf.B73.r1),]
tmt.OG.acs.SL <- tmt.OG.acs.SL[!is.na(tmt.OG.acs.SL$B73.acs.B73.r1),]
#Calculate averages of bio reps - OG
tmt.OG.acs.SL$B73avg <- rowMeans(tmt.OG.acs.SL[,10:12])
tmt.OG.acs.SL$Mo17avg <- rowMeans(tmt.OG.acs.SL[,13:15])
tmt.OG.acs.SL$HYBavg <- rowMeans(tmt.OG.acs.SL[,16:18])
tmt.OG.acs.SL$MPavg <- rowMeans(tmt.OG.acs.SL[,c(10:15)])
tmt.OG.acs.SL$MP.1 <- rowMeans(tmt.OG.acs.SL[,c(10,13)])
tmt.OG.acs.SL$MP.2 <- rowMeans(tmt.OG.acs.SL[,c(11,15)])
tmt.OG.acs.SL$MP.3 <- rowMeans(tmt.OG.acs.SL[,c(12,15)])
#calculate ratio - OG
tmt.OG.acs.SL$HYB.MP <- tmt.OG.acs.SL$HYBavg/tmt.OG.acs.SL$MPavg
#Calculate averages of bio reps - acs
tmt.OG.acs.SL$acsB73avg <- rowMeans(tmt.OG.acs.SL[,19:23])
tmt.OG.acs.SL$ACSavg <- rowMeans(tmt.OG.acs.SL[,24:28])
#calculate ratio - acs
tmt.OG.acs.SL$acs.B73 <- tmt.OG.acs.SL$ACSavg/tmt.OG.acs.SL$acsB73avg
#calculate log(hyb/MP)
tmt.OG.acs.SL$l2.HYB.MP <- log2(tmt.OG.acs.SL$HYB.MP)
#calculate log(acs/B73)
tmt.OG.acs.SL$l2.ACS.B73 <- log2(tmt.OG.acs.SL$acs.B73)
#save
saveRDS(tmt.OG.acs.SL, "tmt.OG.acs.SL")
```

###6hybrid BxM SL + acs
```{r format df}
#Subset
tmt.6h.acs.SL <- tmt[,c(171:176,181:183,31,41,51,33,43,53,35,45,55, 1:10)]
#Remove missing genes
tmt.6h.acs.SL <- tmt.6h.acs.SL[!is.na(tmt.6h.acs.SL$X6Hybrid.B73.r1),]
tmt.6h.acs.SL <- tmt.6h.acs.SL[!is.na(tmt.6h.acs.SL$B73.acs.B73.r1),]
#Calculate averages of bio reps - OG
tmt.6h.acs.SL$B73avg <- rowMeans(tmt.6h.acs.SL[,10:12])
tmt.6h.acs.SL$Mo17avg <- rowMeans(tmt.6h.acs.SL[,13:15])
tmt.6h.acs.SL$HYBavg <- rowMeans(tmt.6h.acs.SL[,16:18])
tmt.6h.acs.SL$MPavg <- rowMeans(tmt.6h.acs.SL[,c(10:15)])
tmt.6h.acs.SL$MP.1 <- rowMeans(tmt.6h.acs.SL[,c(10,13)])
tmt.6h.acs.SL$MP.2 <- rowMeans(tmt.6h.acs.SL[,c(11,15)])
tmt.6h.acs.SL$MP.3 <- rowMeans(tmt.6h.acs.SL[,c(12,15)])
#calculate ratio - OG
tmt.6h.acs.SL$HYB.MP <- tmt.6h.acs.SL$HYBavg/tmt.6h.acs.SL$MPavg
#Calculate averages of bio reps - acs
tmt.6h.acs.SL$acsB73avg <- rowMeans(tmt.6h.acs.SL[,19:23])
tmt.6h.acs.SL$ACSavg <- rowMeans(tmt.6h.acs.SL[,24:28])
#calculate ratio - acs
tmt.6h.acs.SL$acs.B73 <- tmt.6h.acs.SL$ACSavg/tmt.6h.acs.SL$acsB73avg
#calculate log(hyb/MP)
tmt.6h.acs.SL$l2.HYB.MP <- log2(tmt.6h.acs.SL$HYB.MP)
#calculate log(acs/B73)
tmt.6h.acs.SL$l2.ACS.B73 <- log2(tmt.6h.acs.SL$acs.B73)
#save
saveRDS(tmt.6h.acs.SL, "tmt.6h.acs.SL")
```

###Day/night BxM SL + acs
```{r format df}
#Subset
tmt.dn.acs.SL <- tmt[,c(171:176,181:183,62:70,1:10)]
#Remove missing genes
tmt.dn.acs.SL <- tmt.dn.acs.SL[!is.na(tmt.dn.acs.SL$Day.B73.r1),]
tmt.dn.acs.SL <- tmt.dn.acs.SL[!is.na(tmt.dn.acs.SL$B73.acs.B73.r1),]
#Calculate averages of bio reps - OG
tmt.dn.acs.SL$B73avg <- rowMeans(tmt.dn.acs.SL[,10:12])
tmt.dn.acs.SL$Mo17avg <- rowMeans(tmt.dn.acs.SL[,13:15])
tmt.dn.acs.SL$HYBavg <- rowMeans(tmt.dn.acs.SL[,16:18])
tmt.dn.acs.SL$MPavg <- rowMeans(tmt.dn.acs.SL[,c(10:15)])
tmt.dn.acs.SL$MP.1 <- rowMeans(tmt.dn.acs.SL[,c(10,13)])
tmt.dn.acs.SL$MP.2 <- rowMeans(tmt.dn.acs.SL[,c(11,15)])
tmt.dn.acs.SL$MP.3 <- rowMeans(tmt.dn.acs.SL[,c(12,15)])
#calculate ratio - OG
tmt.dn.acs.SL$HYB.MP <- tmt.dn.acs.SL$HYBavg/tmt.dn.acs.SL$MPavg
#Calculate averages of bio reps - acs
tmt.dn.acs.SL$acsB73avg <- rowMeans(tmt.dn.acs.SL[,19:23])
tmt.dn.acs.SL$ACSavg <- rowMeans(tmt.dn.acs.SL[,24:28])
#calculate ratio - acs
tmt.dn.acs.SL$acs.B73 <- tmt.dn.acs.SL$ACSavg/tmt.dn.acs.SL$acsB73avg
#calculate log(hyb/MP)
tmt.dn.acs.SL$l2.HYB.MP <- log2(tmt.dn.acs.SL$HYB.MP)
#calculate log(acs/B73)
tmt.dn.acs.SL$l2.ACS.B73 <- log2(tmt.dn.acs.SL$acs.B73)
#save
saveRDS(tmt.dn.acs.SL, "tmt.dn.acs.SL")
```

###RIL BxM SL + acs
```{r format df}
#Subset
tmt.ril.acs.SL <- tmt[,c(171:176,181:183,81:89, 1:10)]
#Remove missing genes
tmt.ril.acs.SL <- tmt.ril.acs.SL[!is.na(tmt.ril.acs.SL$B73.1),]
tmt.ril.acs.SL <- tmt.ril.acs.SL[!is.na(tmt.ril.acs.SL$B73.acs.B73.r1),]
#Calculate averages of bio reps - OG
tmt.ril.acs.SL$B73avg <- rowMeans(tmt.ril.acs.SL[,10:12])
tmt.ril.acs.SL$Mo17avg <- rowMeans(tmt.ril.acs.SL[,13:15])
tmt.ril.acs.SL$HYBavg <- rowMeans(tmt.ril.acs.SL[,16:18])
tmt.ril.acs.SL$MPavg <- rowMeans(tmt.ril.acs.SL[,c(10:15)])
tmt.ril.acs.SL$MP.1 <- rowMeans(tmt.ril.acs.SL[,c(10,13)])
tmt.ril.acs.SL$MP.2 <- rowMeans(tmt.ril.acs.SL[,c(11,15)])
tmt.ril.acs.SL$MP.3 <- rowMeans(tmt.ril.acs.SL[,c(12,15)])
#calculate ratio - OG
tmt.ril.acs.SL$HYB.MP <- tmt.ril.acs.SL$HYBavg/tmt.ril.acs.SL$MPavg
#Calculate averages of bio reps - acs
tmt.ril.acs.SL$acsB73avg <- rowMeans(tmt.ril.acs.SL[,19:23])
tmt.ril.acs.SL$ACSavg <- rowMeans(tmt.ril.acs.SL[,24:28])
#calculate ratio - acs
tmt.ril.acs.SL$acs.B73 <- tmt.ril.acs.SL$ACSavg/tmt.ril.acs.SL$acsB73avg
#calculate log(hyb/MP)
tmt.ril.acs.SL$l2.HYB.MP <- log2(tmt.ril.acs.SL$HYB.MP)
#calculate log(acs/B73)
tmt.ril.acs.SL$l2.ACS.B73 <- log2(tmt.ril.acs.SL$acs.B73)
#save
saveRDS(tmt.ril.acs.SL, "tmt.ril.acs.SL")
```

#Define GOIs
##corncyc
```{r}
cornCyc <- read.csv("cornCycGeneLists01042020.csv", stringsAsFactors = F)
```
##MaizeGDB accession conversion
```{r}
mgdb <- read.csv("gene_model_xref_v4.csv", stringsAsFactors = F, header = F)
colnames(mgdb) <- mgdb[5,]
mgdb <- mgdb[-c(1:5),]
```
##Photosynthesis
###g1
```{r}
g1tib <-tibble::tribble(
              ~V1,                                                                                    ~V2,
          845199L,                            "psbA; photosystem II protein D1 [KO:K02703] [EC:1.10.3.9]",
          845200L,                                      "psbB; photosystem II 47 kDa protein [KO:K02704]",
          845201L,                                      "psbC; photosystem II 44 kDa protein [KO:K02705]",
          845202L,                            "psbD; photosystem II protein D2 [KO:K02706] [EC:1.10.3.9]",
          845203L,                                           "psbE; photosystem II protein V [KO:K02707]",
          845204L,                                          "psbF; photosystem II protein VI [KO:K02708]",
          845205L,                                           "psbH; photosystem II protein H [KO:K02709]",
          845206L,                                           "psbI; photosystem II protein I [KO:K02710]",
          845207L,                                           "psbJ; photosystem II protein J [KO:K02711]",
          845208L,                                           "psbK; photosystem II protein K [KO:K02712]",
          845209L,                                           "psbL; photosystem II protein L [KO:K02713]",
          845210L,                                           "psbM; photosystem II protein M [KO:K02714]",
       100191684L,                                     "uncharacterized protein LOC100191684 [KO:K02716]",
       100272890L,                                             "uncharacterized LOC100272890 [KO:K02716]",
       100281199L,    "chloroplast oxygen-evolving complex/thylakoid lumenal 25.6kDa protein [KO:K02717]",
       107648855L,                                     "uncharacterized protein LOC107648855 [KO:K02717]",
       100273117L,                        "oxygen-evolving enhancer protein 2, chloroplastic [KO:K02717]",
          542279L,               "PSBQ1; oxygen-evolving enhancer protein 3-1, chloroplastic [KO:K08901]",
       100273120L,                    "oxygen evolving enhancer protein 3 containing protein [KO:K08901]",
       103647735L,                 "oxygen-evolving enhancer protein 3-1, chloroplastic-like [KO:K08901]",
       103627333L,          "photosynthetic NDH subunit of lumenal location 2, chloroplastic [KO:K08901]",
       103653672L,          "photosynthetic NDH subunit of lumenal location 2, chloroplastic [KO:K08901]",
       100281646L,                         "photosystem II 10 kDa polypeptide, chloroplastic [KO:K03541]",
          542126L,                                      "psbs1; photosystem II subunit PsbS1 [KO:K03542]",
         1466364L,                                           "psbT; photosystem II protein T [KO:K02718]",
       100283816L,                          "pco070877; uncharacterized protein LOC100283816 [KO:K02721]",
       100283835L,                           "cl5838_2; uncharacterized protein LOC100283835 [KO:K02721]",
       100280994L,                                "photosystem II core complex proteins psbY [KO:K02723]",
       100281349L,                                "photosystem II core complex proteins psbY [KO:K02723]",
         1466358L,                                           "psbZ; photosystem II protein Z [KO:K02724]",
       100282043L,                                             "photosystem II 11 kD protein [KO:K08902]",
       100281235L,                                             "photosystem II 11 kD protein [KO:K08902]",
       103636309L,         "photosystem II reaction center PSB28 protein, chloroplastic-like [KO:K08903]",
       100284876L,                                 "photosystem II reaction center W protein [KO:K08903]",
       103650780L,              "photosystem II reaction center PSB28 protein, chloroplastic [KO:K08903]",
       109945128L,              "photosystem II reaction center PSB28 protein, chloroplastic [KO:K08903]",
       103650783L,              "photosystem II reaction center PSB28 protein, chloroplastic [KO:K08903]",
          845195L,                     "psaA; photosystem I P700 chlorophyll a apoprotein A1 [KO:K02689]",
          845196L,                     "psaB; photosystem I P700 chlorophyll a apoprotein A2 [KO:K02690]",
         1466379L,                                          "psaC; photosystem I subunit VII [KO:K02691]",
       100191984L,                                 "photosystem I reaction center subunit II [KO:K02692]",
          541791L,                  "photosystem I reaction center subunit II, chloroplastic [KO:K02692]",
       100283327L,                               "photosystem I reaction center subunit IV A [KO:K02693]",
       100283085L,                                     "uncharacterized protein LOC100283085 [KO:K02693]",
       100282027L,                                "photosystem I reaction center subunit III [KO:K02694]",
       103625835L,                 "photosystem I reaction center subunit III, chloroplastic [KO:K02694]",
       100285458L,                                  "photosystem I reaction center subunit V [KO:K08905]",
       100284847L,                                  "photosystem I reaction center subunit V [KO:K08905]",
          541685L, "psah1; photosystem I reaction center subunit VI, chloroplastic precursor [KO:K02695]",
          845197L,                                         "psaI; photosystem I subunit VIII [KO:K02696]",
          845198L,                                           "psaJ; photosystem I subunit IX [KO:K02697]",
       100286020L,                                    "psa6; photosystem I reaction center 6 [KO:K02698]",
       100281679L,                                 "photosystem I reaction center subunit XI [KO:K02699]",
       100383080L,                            "umc1974; uncharacterized protein LOC100383080 [KO:K02699]",
          542605L,                   "photosystem I reaction center subunit N, chloroplastic [KO:K02701]",
       103640891L,              "photosystem I reaction center subunit N, chloroplastic-like [KO:K02701]",
       100283848L,                          "pco139896; uncharacterized protein LOC100283848 [KO:K14332]",
          845191L,                                                       "petA; cytochrome f [KO:K02634]",
       103643185L,                                     "uncharacterized protein LOC103643185 [KO:K02634]",
          845192L,                                                      "petB; cytochrome b6 [KO:K02635]",
       100273354L,                                 "ris2; iron-sulfur protein 2 [KO:K02636] [EC:7.1.1.6]",
       100273026L,                 "cytochrome b6-f complex iron-sulfur subunit [KO:K02636] [EC:7.1.1.6]",
          845193L,                                 "petD; cytochrome b6/f complex subunit IV [KO:K02637]",
          845194L,                                  "petG; cytochrome b6/f complex subunit V [KO:K02640]",
         1466360L,                               "petN; cytochrome b6/f complex subunit VIII [KO:K03689]",
       100192779L,                                     "uncharacterized protein LOC100192779 [KO:K02638]",
       103629356L,                                              "plastocyanin, chloroplastic [KO:K02638]",
       100191704L,                                     "uncharacterized protein LOC100191704 [KO:K02639]",
       103628935L,                                        "fdx5; ferredoxin-5, chloroplastic [KO:K02639]",
          541619L,                                        "fdx2; ferredoxin-2, chloroplastic [KO:K02639]",
       100283643L,                                                             "ferredoxin-3 [KO:K02639]",
          541612L,                              "pco072676(750); ferredoxin-6, chloroplastic [KO:K02639]",
       100382495L,                                      "putative ferredoxin-4 chloroplastic [KO:K02639]",
       100281226L,                                                             "ferredoxin-6 [KO:K02639]",
          542276L,                                        "fdx3; ferredoxin-3, chloroplastic [KO:K02639]",
       100284745L,                                                               "ferredoxin [KO:K02639]",
       101027099L,                                         "FDX1; adrenodoxin, mitochondrial [KO:K02639]",
       103653165L,                                                               "ferredoxin [KO:K02639]",
          541626L,                                        "L-FNRII; ferredoxin [KO:K02641] [EC:1.18.1.2]",
       100194249L,               "IDP808; uncharacterized protein LOC100194249 [KO:K02641] [EC:1.18.1.2]",
          542713L,                          "fnr1; ferredoxin NADP reductase 1 [KO:K02641] [EC:1.18.1.2]",
       100282643L,                   "ferredoxin--NADP reductase, leaf isozyme [KO:K02641] [EC:1.18.1.2]",
          542558L,                                         "L-FNRI; ferredoxin [KO:K02641] [EC:1.18.1.2]",
       103644035L,  "ferredoxin--NADP reductase, leaf isozyme 1, chloroplastic [KO:K02641] [EC:1.18.1.2]",
       100382419L,                                                            "cytochrome c6 [KO:K08906]",
          845171L,                                   "atpE; ATP synthase CF1 epsilon subunit [KO:K02114]",
          845170L,                 "atpB; ATP synthase CF1 beta subunit [KO:K02112] [EC:7.1.2.2 7.2.2.1]",
       100284505L,        "csu309(atpc); ATP synthase subunit gamma, chloroplastic precursor [KO:K02115]",
       103637166L,                                "ATP synthase subunit gamma, chloroplastic [KO:K02115]",
       103655982L,                           "ATP synthase subunit gamma, chloroplastic-like [KO:K02115]",
       103649250L,                           "ATP synthase subunit gamma, chloroplastic-like [KO:K02115]",
          845169L,                "atpA; ATP synthase CF1 alpha subunit [KO:K02111] [EC:7.1.2.2 7.2.2.1]",
       100281924L,                                                 "ATP synthase delta chain [KO:K02113]",
       103627748L,                                  "ATP synthase delta chain, chloroplastic [KO:K02113]",
       100282566L,                         "pco080036b; uncharacterized protein LOC100282566 [KO:K02109]",
          845172L,                                         "atpF; ATP synthase CF0 B subunit [KO:K02109]",
          845173L,                                         "atpH; ATP synthase CF0 C subunit [KO:K02110]",
          845174L,                                         "atpI; ATP synthase CF0 A subunit [KO:K02108]"
       )

```
```{r}
g1entrez <- as.character(g1tib$V1)
```
```{r}
g1 <- mgdb$v4_gene_model[mgdb$Entrez%in%g1entrez]
```
```{r}
g1entrez
```


###g2
```{r}
g2 <- as.vector(gamer$db_object_id[gamer$term_accession=="GO:0015979"])
```
###g3
```{r}
g3 <- as.vector(gamer$db_object_id[gamer$term_accession=="GO:1905157"])
```
###g4
```{r}
g4 <- as.vector(gamer$db_object_id[gamer$term_accession=="GO:0010109"])
```
###g5
```{r}
g5 <- as.vector(gamer$db_object_id[gamer$term_accession=="GO:0080005"])
```
###g6
```{r}
g6 <- as.vector(gamer$db_object_id[gamer$term_accession=="GO:0019684"])
```
###g7
```{r}
g7 <- as.vector(gamer$db_object_id[gamer$term_accession=="GO:0042548"])
```
###g8
```{r}
g8 <- as.vector(gamer$db_object_id[gamer$term_accession=="GO:0010205"])
```
###g9
```{r}
g9 <- as.vector(gamer$db_object_id[gamer$term_accession=="GO:0042549"])
```
###g10
```{r}
g10 <- as.vector(gamer$db_object_id[gamer$term_accession=="GO:0042550"])
```
###g11
```{r}
g11 <- as.vector(gamer$db_object_id[gamer$term_accession=="GO:0009522"])
```
###g12
```{r}
g12 <- as.vector(gamer$db_object_id[gamer$term_accession=="GO:0030093"])
```
###g13
```{r}
g13 <- as.vector(gamer$db_object_id[gamer$term_accession=="GO:0009782"])
```
###g14
```{r}
g14 <- as.vector(gamer$db_object_id[gamer$term_accession=="GO:0009538"])
```
###g15
```{r}
g15entrez <- g1entrez[38:57]
```
```{r}
g15 <- mgdb$v4_gene_model[mgdb$Entrez%in%g15entrez]
```



###g16
```{r}
CornCycLightRxns <- tibble::tribble(
                          ~Gene.ID,  ~Gene.Accession,       ~Gene.name,                               ~Reaction.id, ~Reaction.EC,                   ~Enzymatic.activity, ~Evidence,
                       "GDQC-8387", "Zm00001d010715", "Zm00001d010715",                                 "PSII-RXN",   "1.10.3.9",                      "photosystem II", "EV-COMP",
                       "GDQC-8546", "Zm00001d020002", "Zm00001d020002",                                 "PSII-RXN",   "1.10.3.9",                      "photosystem II", "EV-COMP",
                       "GDQC-8747", "Zm00001d000417", "Zm00001d000417",                                 "PSII-RXN",   "1.10.3.9",                      "photosystem II", "EV-COMP",
                       "GDQC-9170", "Zm00001d039715", "Zm00001d039715",                                 "PSII-RXN",   "1.10.3.9",                      "photosystem II", "EV-COMP",
                       "GDQC-9254", "Zm00001d011362", "Zm00001d011362",                                 "PSII-RXN",   "1.10.3.9",                      "photosystem II", "EV-COMP",
                       "GDQC-9445", "Zm00001d008681", "Zm00001d008681",                                 "PSII-RXN",   "1.10.3.9",                      "photosystem II", "EV-COMP",
                       "GDQC-9899", "Zm00001d043167", "Zm00001d043167",                                 "PSII-RXN",   "1.10.3.9",                      "photosystem II", "EV-COMP",
                      "GDQC-10138", "Zm00001d008625", "Zm00001d008625",                                 "PSII-RXN",   "1.10.3.9",                      "photosystem II", "EV-COMP",
                      "GDQC-10311", "Zm00001d041824", "Zm00001d041824",                                 "PSII-RXN",   "1.10.3.9",                      "photosystem II", "EV-COMP",
                      "GDQC-10950", "Zm00001d022464", "Zm00001d022464",                                 "PSII-RXN",   "1.10.3.9",                      "photosystem II", "EV-COMP",
                      "GDQC-11782", "Zm00001d044217", "Zm00001d044217",                                 "PSII-RXN",   "1.10.3.9",                      "photosystem II", "EV-COMP",
                      "GDQC-11860", "Zm00001d030638", "Zm00001d030638",                                 "PSII-RXN",   "1.10.3.9",                      "photosystem II", "EV-COMP",
                      "GDQC-11887",  "GRMZM5G808939",  "GRMZM5G808939",                                 "PSII-RXN",   "1.10.3.9",                      "photosystem II", "EV-COMP",
                      "GDQC-11985", "Zm00001d043299", "Zm00001d043299",                                 "PSII-RXN",   "1.10.3.9",                      "photosystem II", "EV-COMP",
                      "GDQC-11996", "Zm00001d034153", "Zm00001d034153",                                 "PSII-RXN",   "1.10.3.9",                      "photosystem II", "EV-COMP",
                      "GDQC-12078", "Zm00001d000260", "Zm00001d000260",                                 "PSII-RXN",   "1.10.3.9",                      "photosystem II", "EV-COMP",
                      "GDQC-13214", "Zm00001d031484", "Zm00001d031484",                                 "PSII-RXN",   "1.10.3.9",                      "photosystem II", "EV-COMP",
                      "GDQC-14006", "Zm00001d039852", "Zm00001d039852",                                 "PSII-RXN",   "1.10.3.9",                      "photosystem II", "EV-COMP",
                       "GDQC-8316", "Zm00001d016134", "Zm00001d016134", "PLASTOQUINOL--PLASTOCYANIN-REDUCTASE-RXN",   "1.10.9.1", "plastoquinol-plastocyanin reductase", "EV-COMP",
                       "GDQC-9168", "Zm00001d018162", "Zm00001d018162", "PLASTOQUINOL--PLASTOCYANIN-REDUCTASE-RXN",   "1.10.9.1", "plastoquinol-plastocyanin reductase", "EV-COMP",
                      "GDQC-11523", "Zm00001d053432", "Zm00001d053432", "PLASTOQUINOL--PLASTOCYANIN-REDUCTASE-RXN",   "1.10.9.1", "plastoquinol-plastocyanin reductase", "EV-COMP",
                      "GDQC-13682",  "GRMZM5G800780",  "GRMZM5G800780", "PLASTOQUINOL--PLASTOCYANIN-REDUCTASE-RXN",   "1.10.9.1", "plastoquinol-plastocyanin reductase", "EV-COMP",
                       "GDQC-8046", "Zm00001d003767", "Zm00001d003767",                                "RXN-15479",  "1.97.1.12",                       "photosystem I", "EV-COMP",
                       "GDQC-8673", "Zm00001d034283", "Zm00001d034283",                                "RXN-15479",  "1.97.1.12",                       "photosystem I", "EV-COMP",
                       "GDQC-9216",  "GRMZM5G815553",  "GRMZM5G815553",                                "RXN-15479",  "1.97.1.12",                       "photosystem I", "EV-COMP",
                      "GDQC-10703", "Zm00001d013146", "Zm00001d013146",                                "RXN-15479",  "1.97.1.12",                       "photosystem I", "EV-COMP",
                      "GDQC-12161", "Zm00001d000272", "Zm00001d000272",                                "RXN-15479",  "1.97.1.12",                       "photosystem I", "EV-COMP",
                       "GDQC-7635", "Zm00001d011833", "Zm00001d011833",                             "1.18.1.2-RXN",   "1.18.1.2",           "ferredoxin NADP reductase", "EV-COMP",
                       "GDQC-7812", "Zm00001d023296", "Zm00001d023296",                             "1.18.1.2-RXN",   "1.18.1.2",           "ferredoxin NADP reductase", "EV-COMP",
                       "GDQC-8192", "Zm00001d045575", "Zm00001d045575",                             "1.18.1.2-RXN",   "1.18.1.2",           "ferredoxin NADP reductase", "EV-COMP",
                       "GDQC-8377", "Zm00001d051680", "Zm00001d051680",                             "1.18.1.2-RXN",   "1.18.1.2",           "ferredoxin NADP reductase", "EV-COMP",
                       "GDQC-8916", "Zm00001d048452", "Zm00001d048452",                             "1.18.1.2-RXN",   "1.18.1.2",           "ferredoxin NADP reductase", "EV-COMP",
                      "GDQC-10511", "Zm00001d005458", "Zm00001d005458",                             "1.18.1.2-RXN",   "1.18.1.2",           "ferredoxin NADP reductase", "EV-COMP",
                      "GDQC-10877", "Zm00001d018794", "Zm00001d018794",                             "1.18.1.2-RXN",   "1.18.1.2",           "ferredoxin NADP reductase", "EV-COMP",
                      "GDQC-10917", "Zm00001d034099", "Zm00001d034099",                             "1.18.1.2-RXN",   "1.18.1.2",           "ferredoxin NADP reductase", "EV-COMP",
                      "GDQC-11276", "Zm00001d042049", "Zm00001d042049",                             "1.18.1.2-RXN",   "1.18.1.2",           "ferredoxin NADP reductase", "EV-COMP",
                      "GDQC-11941", "Zm00001d039519", "Zm00001d039519",                             "1.18.1.2-RXN",   "1.18.1.2",           "ferredoxin NADP reductase", "EV-COMP",
                      "GDQC-11960", "Zm00001d034345", "Zm00001d034345",                             "1.18.1.2-RXN",   "1.18.1.2",           "ferredoxin NADP reductase", "EV-COMP",
                      "GDQC-12126", "Zm00001d051407", "Zm00001d051407",                             "1.18.1.2-RXN",   "1.18.1.2",           "ferredoxin NADP reductase", "EV-COMP",
                      "GDQC-12735", "Zm00001d045574", "Zm00001d045574",                             "1.18.1.2-RXN",   "1.18.1.2",           "ferredoxin NADP reductase", "EV-COMP",
                      "GDQC-13931", "Zm00001d036041", "Zm00001d036041",                             "1.18.1.2-RXN",   "1.18.1.2",           "ferredoxin NADP reductase", "EV-COMP",
                      "GDQC-14028", "Zm00001d037231", "Zm00001d037231",                             "1.18.1.2-RXN",   "1.18.1.2",           "ferredoxin NADP reductase", "EV-COMP"
                      )

```
```{r}
g16 <- CornCycLightRxns$Gene.Accession[CornCycLightRxns$Reaction.EC=="1.97.1.12"]
```
```{r}
g16 <- substr(g16, start = 1, stop = 14)
```


###g17
```{r}
g17 <- as.vector(gamer$db_object_id[gamer$term_accession=="GO:0009768"])
```
###g18
```{r}
g18tib <- tibble::tribble(
                   ~V1,                                                                              ~V2,
            100281791L,                                 "chlorophyll a-b binding protein 6A [KO:K08907]",
            100283212L,                                 "chlorophyll a-b binding protein 6A [KO:K08907]",
            100273427L,      "Photosystem I chlorophyll a/b-binding protein 6 chloroplastic [KO:K08908]",
            100191715L,                                    "chlorophyll a-b binding protein [KO:K08908]",
            100282512L,                                    "chlorophyll a-b binding protein [KO:K08908]",
            100282214L,                                  "chlorophyll a-b binding protein 8 [KO:K08909]",
            100284001L,                       "gpm571; uncharacterized protein LOC100284001 [KO:K08910]",
            100280410L,                               "uncharacterized protein LOC100280410 [KO:K08910]",
            100281795L,                "csu1028(lhcb); uncharacterized protein LOC100281795 [KO:K08911]",
            100281248L,         "chlorophyll a-b binding protein 1, chloroplastic precursor [KO:K08912]",
            100193833L,              "lhcb7; chlorophyll a-b binding protein, chloroplastic [KO:K08912]",
            100282054L,                                  "chlorophyll a-b binding protein 2 [KO:K08912]",
               542530L, "lhcb3; chlorophyll a-b binding protein M9, chloroplastic precursor [KO:K08912]",
            100274453L,                  "light harvesting chlorophyll a/b binding protein1 [KO:K08912]",
            103651575L,                  "chlorophyll a-b binding protein 48, chloroplastic [KO:K08912]",
            103651577L,                    "chlorophyll a-b binding protein of LHCII type 1 [KO:K08912]",
            103651578L,                    "chlorophyll a-b binding protein of LHCII type 1 [KO:K08912]",
            100281879L,                                    "chlorophyll a-b binding protein [KO:K08913]",
            103643653L,                     "chlorophyll a-b binding protein, chloroplastic [KO:K08913]",
            100284154L,                  "chlorophyll a-b binding protein of LHCII type III [KO:K08914]",
            100216729L,                               "uncharacterized protein LOC100216729 [KO:K08915]",
               542478L,                                    "psb29; photosystem II subunit29 [KO:K08915]",
               542320L,         "Lhcb5-2; chlorophyll a/b-binding apoprotein CP26 precursor [KO:K08916]",
               542716L,          "lhcb5; light harvesting chlorophyll a/b binding protein 5 [KO:K08916]",
            100273752L,                               "uncharacterized protein LOC100273752 [KO:K08917]",
               542321L,          "lhcb6; light harvesting chlorophyll a/b binding protein 6 [KO:K08917]",
            109944964L,            "chlorophyll a-b binding protein CP24 10B, chloroplastic [KO:K08917]",
            113561367L,                               "chlorophyll a-b binding protein CP24 [KO:K08917]",
            100274085L,                  "si486069e02; uncharacterized protein LOC100274085 [KO:K14172]"
            )

```
```{r}
g18entrez <- as.character(g18tib$V1)
```
```{r}
g18entrez.lhc1 <- g18entrez[1:10]
```
```{r}
g18 <- mgdb$v4_gene_model[mgdb$Entrez%in%g18entrez.lhc1]
```


###g19
```{r}
g19 <- as.vector(gamer$db_object_id[gamer$term_accession=="GO:0048564"])
```
###g20
```{r}
g20 <- as.vector(gamer$db_object_id[gamer$term_accession=="GO:0009523"])
```
###g21
```{r}
g21 <- as.vector(gamer$db_object_id[gamer$term_accession=="GO:0030095"])
```
###g22
```{r}
g22 <- as.vector(gamer$db_object_id[gamer$term_accession=="GO:0009783"])
```
###g23
```{r}
g23 <- as.vector(gamer$db_object_id[gamer$term_accession=="GO:0009654"])
```
###g24
```{r}
g24 <- as.vector(gamer$db_object_id[gamer$term_accession=="GO:0009539"])
```

###g25
```{r}
g25entrez <- g1entrez[1:37]
```
```{r}
g25 <- mgdb$v4_gene_model[mgdb$Entrez%in%g25entrez]
```
###g26
```{r}
g26 <- CornCycLightRxns$Gene.Accession[CornCycLightRxns$Reaction.EC=="1.10.3.9"]
```
```{r}
g26 <- substr(g26, start = 1, stop = 14)
```


###g27
```{r}
g27 <- as.vector(gamer$db_object_id[gamer$term_accession=="GO:0009769"])
```
###g28
```{r}
g28entrez.lhc2 <- g18entrez[11:29]
```
```{r}
g28 <- mgdb$v4_gene_model[mgdb$Entrez%in%g28entrez.lhc2]
```

###g29
```{r}
g29 <- as.vector(gamer$db_object_id[gamer$term_accession=="GO:0010206"])
```
###g30
```{r}
g30 <- as.vector(gamer$db_object_id[gamer$term_accession=="GO:0010207"])
```
###g31
```{r}
g31 <- as.vector(gamer$db_object_id[gamer$term_accession=="GO:0010270"])
```
###g32
```{r}
g32 <- as.vector(gamer$db_object_id[gamer$term_accession=="GO:0009767"])
```
###g33
```{r}
g33 <- as.vector(gamer$db_object_id[gamer$term_accession=="GO:0009773"])
```
###g34
```{r}
g34 <- as.vector(gamer$db_object_id[gamer$term_accession=="GO:0009772"])
```
###g35
```{r}
g35entrez <- g1entrez[66:85]
```
```{r}
g35 <- mgdb$v4_gene_model[mgdb$Entrez%in%g35entrez]
```
###g36
```{r}
g36 <- CornCycLightRxns$Gene.Accession[CornCycLightRxns$Reaction.EC=="1.10.9.1"]
```
```{r}
g36 <- substr(g36, start = 1, stop = 14)
```

###g37
```{r}
g37entrez <- g1entrez[58:65]
```
```{r}
g37entrez
```

```{r}
g37 <- mgdb$v4_gene_model[mgdb$Entrez%in%g37entrez]
```
```{r}
g37
```

###g38
```{r}
g38 <- as.vector(gamer$db_object_id[gamer$term_accession=="GO:0009780"])
```
###g39
```{r}
g39 <- CornCycLightRxns$Gene.Accession[CornCycLightRxns$Reaction.EC=="1.18.1.2"]
```
```{r}
g39 <- substr(g39, start = 1, stop = 14)
```

###g40
```{r}
g40entrez <- g1entrez[86:98]
```
```{r}
g40 <- mgdb$v4_gene_model[mgdb$Entrez%in%g40entrez]
```
###g41
```{r}
g41 <- as.vector(gamer$db_object_id[gamer$term_accession=="GO:0019685"])
```
###g42
```{r}
CornCycC4NADP <- tibble::tribble(
                        ~Gene.ID,         ~Gene.Accession,         ~Gene.name,                          ~Reaction.id, ~Reaction.EC,                                           ~Enzymatic.activity, ~Evidence,
                   "GDQC-112543",        "Zm00001d044096",   "Zm00001d044096",                           "RXN0-5224",    "4.2.1.1",                                       "carbonate dehydratase",  "EV-EXP",
                   "GDQC-113944",        "Zm00001d044099",             "cah1",                           "RXN0-5224",    "4.2.1.1",                                       "carbonate dehydratase",  "EV-EXP",
                     "GDQC-7639",        "Zm00001d024980",   "Zm00001d024980",                       "PEPCARBOX-RXN",   "4.1.1.31",                             "phosphoenolpyruvate carboxylase", "EV-COMP",
                     "GDQC-8952",        "Zm00001d016166",   "Zm00001d016166",                       "PEPCARBOX-RXN",   "4.1.1.31",                             "phosphoenolpyruvate carboxylase", "EV-COMP",
                     "GDQC-9159",        "Zm00001d040129",   "Zm00001d040129",                       "PEPCARBOX-RXN",   "4.1.1.31",                             "phosphoenolpyruvate carboxylase", "EV-COMP",
                     "GDQC-9830",        "Zm00001d046754",   "Zm00001d046754",                       "PEPCARBOX-RXN",   "4.1.1.31",                             "phosphoenolpyruvate carboxylase", "EV-COMP",
                    "GDQC-10258",        "Zm00001d020057",   "Zm00001d020057",                       "PEPCARBOX-RXN",   "4.1.1.31",                             "phosphoenolpyruvate carboxylase", "EV-COMP",
                    "GDQC-11167",        "Zm00001d053453",   "Zm00001d053453",                       "PEPCARBOX-RXN",   "4.1.1.31",                             "phosphoenolpyruvate carboxylase", "EV-COMP",
                    "GDQC-13992",        "Zm00001d037294",   "Zm00001d037294",                       "PEPCARBOX-RXN",   "4.1.1.31",                             "phosphoenolpyruvate carboxylase", "EV-COMP",
                    "GDQC-14269",        "Zm00001d012702",   "Zm00001d012702",                       "PEPCARBOX-RXN",   "4.1.1.31",                             "phosphoenolpyruvate carboxylase", "EV-COMP",
                          "G-41",        "Zm00001d046170",             "pep1",                       "PEPCARBOX-RXN",   "4.1.1.31",                             "phosphoenolpyruvate carboxylase",  "EV-EXP",
                          "G-61",        "Zm00001d031899",   "Zm00001d031899",      "MALATE-DEHYDROGENASE-NADP+-RXN",   "1.1.1.82",                                   "NADP malate dehydrogenase",  "EV-EXP",
                           "ME1",        "Zm00001d000316",   "Zm00001d000316",                      "MALIC-NADP-RXN",   "1.1.1.40",                                        "malate dehydrogenase", "EV-COMP",
                     "GDQC-7940",        "Zm00001d010358",   "Zm00001d010358",                      "MALIC-NADP-RXN",   "1.1.1.40",                                        "malate dehydrogenase", "EV-COMP",
                     "GDQC-8014",        "Zm00001d037693",   "Zm00001d037693",                      "MALIC-NADP-RXN",   "1.1.1.40",                                        "malate dehydrogenase", "EV-COMP",
                     "GDQC-9044",        "Zm00001d003090",   "Zm00001d003090",                      "MALIC-NADP-RXN",   "1.1.1.40",                                        "malate dehydrogenase", "EV-COMP",
                     "GDQC-9094",        "Zm00001d037961",   "Zm00001d037961",                      "MALIC-NADP-RXN",   "1.1.1.40",                                        "malate dehydrogenase", "EV-COMP",
                     "GDQC-9862",        "Zm00001d011289",   "Zm00001d011289",                      "MALIC-NADP-RXN",   "1.1.1.40",                                        "malate dehydrogenase", "EV-COMP",
                     "GDQC-9900",        "Zm00001d042392",   "Zm00001d042392",                      "MALIC-NADP-RXN",   "1.1.1.40",                                        "malate dehydrogenase", "EV-COMP",
                    "GDQC-11405",        "Zm00001d036873",   "Zm00001d036873",                      "MALIC-NADP-RXN",   "1.1.1.40",                                        "malate dehydrogenase", "EV-COMP",
                    "GDQC-12576",        "Zm00001d043601",   "Zm00001d043601",                      "MALIC-NADP-RXN",   "1.1.1.40",                                        "malate dehydrogenase", "EV-COMP",
                    "GDQC-12623",        "Zm00001d034292",   "Zm00001d034292",                      "MALIC-NADP-RXN",   "1.1.1.40",                                        "malate dehydrogenase", "EV-COMP",
                    "GDQC-12650",        "Zm00001d039633",   "Zm00001d039633",                      "MALIC-NADP-RXN",   "1.1.1.40",                                        "malate dehydrogenase", "EV-COMP",
                    "GDQC-13022",        "Zm00001d004962",   "Zm00001d004962",                      "MALIC-NADP-RXN",   "1.1.1.40",                                        "malate dehydrogenase", "EV-COMP",
                    "GDQC-14112",        "Zm00001d012764",   "Zm00001d012764",                      "MALIC-NADP-RXN",   "1.1.1.40",                                        "malate dehydrogenase", "EV-COMP",
                    "GDQC-14700",        "Zm00001d051182",   "Zm00001d051182",                      "MALIC-NADP-RXN",   "1.1.1.40",                                        "malate dehydrogenase", "EV-COMP",
                   "GDQC-108581",        "Zm00001d037962",              "me2",                      "MALIC-NADP-RXN",   "1.1.1.40", "malate dehydrogenase (oxaloacetate-decarboxylating) (NADP+)",  "EV-EXP",
                           "me1", "Zm00001d000316_T001.1",              "me3",                      "MALIC-NADP-RXN",   "1.1.1.40",   "malate dehydrogenase NADP+ (oxaloacetate-decarboxylating)",  "EV-EXP",
                     "GDQC-8439",        "Zm00001d010321",   "Zm00001d010321", "PYRUVATEORTHOPHOSPHATE-DIKINASE-RXN",    "2.7.9.1",                                 "pyruvate phosphate dikinase", "EV-COMP",
                     "GDQC-9261",        "Zm00001d038161",   "Zm00001d038161", "PYRUVATEORTHOPHOSPHATE-DIKINASE-RXN",    "2.7.9.1",                                 "pyruvate phosphate dikinase", "EV-COMP",
                   "GDQC-112401",        "Zm00001d038163", "pdk1 chloroplast", "PYRUVATEORTHOPHOSPHATE-DIKINASE-RXN",    "2.7.9.1",                                "pyruvate, phosphate dikinase",  "EV-EXP"
                   )

```
```{r}
g42 <- CornCycC4NADP$Gene.Accession
```
```{r}
g42 <- substr(g42, start = 1, stop = 14)
```

###g43
```{r}
CornCycC4PEPCK <- tibble::tribble(
                         ~Gene.ID,         ~Gene.Accession,         ~Gene.name,                          ~Reaction.id, ~Reaction.EC,                                           ~Enzymatic.activity, ~Evidence,
                    "GDQC-112543",        "Zm00001d044096",   "Zm00001d044096",                           "RXN0-5224",    "4.2.1.1",                                       "carbonate dehydratase",  "EV-EXP",
                    "GDQC-113944",        "Zm00001d044099",             "cah1",                           "RXN0-5224",    "4.2.1.1",                                       "carbonate dehydratase",  "EV-EXP",
                    "GDQC-108139",        "Zm00001d018429",            "ppck1",                   "PEPCARBOXYKIN-RXN",   "4.1.1.49",                     "phosphoenolpyruvate carboxykinase (ATP)", "EV-COMP",
                      "GDQC-8305",        "Zm00001d008318",   "Zm00001d008318",                   "PEPCARBOXYKIN-RXN",   "4.1.1.49",                     "phosphoenolpyruvate carboxykinase (ATP)", "EV-COMP",
                     "GDQC-13994",        "Zm00001d026192",   "Zm00001d026192",                   "PEPCARBOXYKIN-RXN",   "4.1.1.49",                     "phosphoenolpyruvate carboxykinase (ATP)", "EV-COMP",
                     "GDQC-14364",        "Zm00001d047893",   "Zm00001d047893",                   "PEPCARBOXYKIN-RXN",   "4.1.1.49",                     "phosphoenolpyruvate carboxykinase (ATP)", "EV-COMP",
                     "GDQC-14687",        "Zm00001d029398",   "Zm00001d029398",                   "PEPCARBOXYKIN-RXN",   "4.1.1.49",                     "phosphoenolpyruvate carboxykinase (ATP)", "EV-COMP",
                    "GDQC-113518",        "Zm00001d028471",   "Zm00001d028471",                   "PEPCARBOXYKIN-RXN",   "4.1.1.49",                           "phosphoenolpyruvate carboxykinase",  "EV-EXP",
                      "GDQC-7639",        "Zm00001d024980",   "Zm00001d024980",                       "PEPCARBOX-RXN",   "4.1.1.31",                             "phosphoenolpyruvate carboxylase", "EV-COMP",
                      "GDQC-8952",        "Zm00001d016166",   "Zm00001d016166",                       "PEPCARBOX-RXN",   "4.1.1.31",                             "phosphoenolpyruvate carboxylase", "EV-COMP",
                      "GDQC-9159",        "Zm00001d040129",   "Zm00001d040129",                       "PEPCARBOX-RXN",   "4.1.1.31",                             "phosphoenolpyruvate carboxylase", "EV-COMP",
                      "GDQC-9830",        "Zm00001d046754",   "Zm00001d046754",                       "PEPCARBOX-RXN",   "4.1.1.31",                             "phosphoenolpyruvate carboxylase", "EV-COMP",
                     "GDQC-10258",        "Zm00001d020057",   "Zm00001d020057",                       "PEPCARBOX-RXN",   "4.1.1.31",                             "phosphoenolpyruvate carboxylase", "EV-COMP",
                     "GDQC-11167",        "Zm00001d053453",   "Zm00001d053453",                       "PEPCARBOX-RXN",   "4.1.1.31",                             "phosphoenolpyruvate carboxylase", "EV-COMP",
                     "GDQC-13992",        "Zm00001d037294",   "Zm00001d037294",                       "PEPCARBOX-RXN",   "4.1.1.31",                             "phosphoenolpyruvate carboxylase", "EV-COMP",
                     "GDQC-14269",        "Zm00001d012702",   "Zm00001d012702",                       "PEPCARBOX-RXN",   "4.1.1.31",                             "phosphoenolpyruvate carboxylase", "EV-COMP",
                           "G-41",        "Zm00001d046170",             "pep1",                       "PEPCARBOX-RXN",   "4.1.1.31",                             "phosphoenolpyruvate carboxylase",  "EV-EXP",
                           "G-61",        "Zm00001d031899",   "Zm00001d031899",      "MALATE-DEHYDROGENASE-NADP+-RXN",   "1.1.1.82",                                   "NADP malate dehydrogenase",  "EV-EXP",
                            "ME1",        "Zm00001d000316",   "Zm00001d000316",                      "MALIC-NADP-RXN",   "1.1.1.40",                                        "malate dehydrogenase", "EV-COMP",
                      "GDQC-7940",        "Zm00001d010358",   "Zm00001d010358",                      "MALIC-NADP-RXN",   "1.1.1.40",                                        "malate dehydrogenase", "EV-COMP",
                      "GDQC-8014",        "Zm00001d037693",   "Zm00001d037693",                      "MALIC-NADP-RXN",   "1.1.1.40",                                        "malate dehydrogenase", "EV-COMP",
                      "GDQC-9044",        "Zm00001d003090",   "Zm00001d003090",                      "MALIC-NADP-RXN",   "1.1.1.40",                                        "malate dehydrogenase", "EV-COMP",
                      "GDQC-9094",        "Zm00001d037961",   "Zm00001d037961",                      "MALIC-NADP-RXN",   "1.1.1.40",                                        "malate dehydrogenase", "EV-COMP",
                      "GDQC-9862",        "Zm00001d011289",   "Zm00001d011289",                      "MALIC-NADP-RXN",   "1.1.1.40",                                        "malate dehydrogenase", "EV-COMP",
                      "GDQC-9900",        "Zm00001d042392",   "Zm00001d042392",                      "MALIC-NADP-RXN",   "1.1.1.40",                                        "malate dehydrogenase", "EV-COMP",
                     "GDQC-11405",        "Zm00001d036873",   "Zm00001d036873",                      "MALIC-NADP-RXN",   "1.1.1.40",                                        "malate dehydrogenase", "EV-COMP",
                     "GDQC-12576",        "Zm00001d043601",   "Zm00001d043601",                      "MALIC-NADP-RXN",   "1.1.1.40",                                        "malate dehydrogenase", "EV-COMP",
                     "GDQC-12623",        "Zm00001d034292",   "Zm00001d034292",                      "MALIC-NADP-RXN",   "1.1.1.40",                                        "malate dehydrogenase", "EV-COMP",
                     "GDQC-12650",        "Zm00001d039633",   "Zm00001d039633",                      "MALIC-NADP-RXN",   "1.1.1.40",                                        "malate dehydrogenase", "EV-COMP",
                     "GDQC-13022",        "Zm00001d004962",   "Zm00001d004962",                      "MALIC-NADP-RXN",   "1.1.1.40",                                        "malate dehydrogenase", "EV-COMP",
                     "GDQC-14112",        "Zm00001d012764",   "Zm00001d012764",                      "MALIC-NADP-RXN",   "1.1.1.40",                                        "malate dehydrogenase", "EV-COMP",
                     "GDQC-14700",        "Zm00001d051182",   "Zm00001d051182",                      "MALIC-NADP-RXN",   "1.1.1.40",                                        "malate dehydrogenase", "EV-COMP",
                    "GDQC-108581",        "Zm00001d037962",              "me2",                      "MALIC-NADP-RXN",   "1.1.1.40", "malate dehydrogenase (oxaloacetate-decarboxylating) (NADP+)",  "EV-EXP",
                            "me1", "Zm00001d000316_T001.1",              "me3",                      "MALIC-NADP-RXN",   "1.1.1.40",   "malate dehydrogenase NADP+ (oxaloacetate-decarboxylating)",  "EV-EXP",
                      "GDQC-7790",        "Zm00001d003953",   "Zm00001d003953",        "ALANINE-AMINOTRANSFERASE-RXN",    "2.6.1.2",                         "glutamate-pyruvate aminotransferase", "EV-COMP",
                      "GDQC-7870",        "Zm00001d030557",   "Zm00001d030557",        "ALANINE-AMINOTRANSFERASE-RXN",    "2.6.1.2",                         "glutamate-pyruvate aminotransferase", "EV-COMP",
                      "GDQC-9203",        "Zm00001d008277",   "Zm00001d008277",        "ALANINE-AMINOTRANSFERASE-RXN",    "2.6.1.2",                         "glutamate-pyruvate aminotransferase", "EV-COMP",
                      "GDQC-9262",        "Zm00001d006875",   "Zm00001d006875",        "ALANINE-AMINOTRANSFERASE-RXN",    "2.6.1.2",                         "glutamate-pyruvate aminotransferase", "EV-COMP",
                     "GDQC-10679",        "Zm00001d032984",   "Zm00001d032984",        "ALANINE-AMINOTRANSFERASE-RXN",    "2.6.1.2",                         "glutamate-pyruvate aminotransferase", "EV-COMP",
                     "GDQC-11731",        "Zm00001d020590",   "Zm00001d020590",        "ALANINE-AMINOTRANSFERASE-RXN",    "2.6.1.2",                         "glutamate-pyruvate aminotransferase", "EV-COMP",
                     "GDQC-11867",        "Zm00001d022152",   "Zm00001d022152",        "ALANINE-AMINOTRANSFERASE-RXN",    "2.6.1.2",                         "glutamate-pyruvate aminotransferase", "EV-COMP",
                     "GDQC-12271",        "Zm00001d044135",   "Zm00001d044135",        "ALANINE-AMINOTRANSFERASE-RXN",    "2.6.1.2",                         "glutamate-pyruvate aminotransferase", "EV-COMP",
                     "GDQC-12767",        "Zm00001d014258",   "Zm00001d014258",        "ALANINE-AMINOTRANSFERASE-RXN",    "2.6.1.2",                         "glutamate-pyruvate aminotransferase", "EV-COMP",
                     "GDQC-12961",        "Zm00001d018639",   "Zm00001d018639",        "ALANINE-AMINOTRANSFERASE-RXN",    "2.6.1.2",                         "glutamate-pyruvate aminotransferase", "EV-COMP",
                    "GDQC-104563",        "Zm00001d007937",             "alt8",        "ALANINE-AMINOTRANSFERASE-RXN",    "2.6.1.2",                                        "alanine transaminase",  "EV-EXP",
                    "GDQC-113394",                      NA,             "alt7",        "ALANINE-AMINOTRANSFERASE-RXN",    "2.6.1.2",                                        "alanine transaminase",  "EV-EXP",
                    "GDQC-113394",                      NA,             "alt7",                           "RXN-13698",    "2.6.1.2",                                        "alanine transaminase", "EV-COMP",
                      "GDQC-8439",        "Zm00001d010321",   "Zm00001d010321", "PYRUVATEORTHOPHOSPHATE-DIKINASE-RXN",    "2.7.9.1",                                 "pyruvate phosphate dikinase", "EV-COMP",
                      "GDQC-9261",        "Zm00001d038161",   "Zm00001d038161", "PYRUVATEORTHOPHOSPHATE-DIKINASE-RXN",    "2.7.9.1",                                 "pyruvate phosphate dikinase", "EV-COMP",
                    "GDQC-112401",        "Zm00001d038163", "pdk1 chloroplast", "PYRUVATEORTHOPHOSPHATE-DIKINASE-RXN",    "2.7.9.1",                                "pyruvate, phosphate dikinase",  "EV-EXP",
                    "GDQC-111967",        "Zm00001d038163",     "pdk1 cytosol", "PYRUVATEORTHOPHOSPHATE-DIKINASE-RXN",    "2.7.9.1",                                "pyruvate, phosphate dikinase",  "EV-EXP",
                      "GDQC-9889",        "Zm00001d010190",   "Zm00001d010190",                   "ASPAMINOTRANS-RXN",    "2.6.1.1",                                      "aspartate transaminase", "EV-COMP",
                     "GDQC-10379",        "Zm00001d016198",   "Zm00001d016198",                   "ASPAMINOTRANS-RXN",    "2.6.1.1",                                      "aspartate transaminase", "EV-COMP",
                     "GDQC-10415",        "Zm00001d042685",   "Zm00001d042685",                   "ASPAMINOTRANS-RXN",    "2.6.1.1",                                      "aspartate transaminase", "EV-COMP",
                     "GDQC-12672",        "Zm00001d043382",   "Zm00001d043382",                   "ASPAMINOTRANS-RXN",    "2.6.1.1",                                      "aspartate transaminase", "EV-COMP",
                    "GDQC-108527",        "Zm00001d018386",   "Zm00001d018386",                   "ASPAMINOTRANS-RXN",    "2.6.1.1",                                      "aspartate transaminase",  "EV-EXP"
                    )

```
```{r}
g43 <- CornCycC4PEPCK$Gene.Accession
```
```{r}
g43 <- substr(g43, start = 1, stop = 14)
```

###g44
```{r}
g44 <- as.vector(gamer$db_object_id[gamer$term_accession=="GO:0019253"])
```
###g45
```{r}
g45 <- as.vector(gamer$db_object_id[gamer$term_accession=="GO:0080153"])
```
###g46
```{r}
g46tib <- tibble::tribble(
                   ~V1,                                                                                                      ~V2,
               845212L,          "rbcL; ribulose-1,5-bisphosphate carboxylase/oxygenase large subunit [KO:K01601] [EC:4.1.1.39]",
               542212L, "ssu1; ribulose bisphosphate carboxylase smplastid chain, chloroplastic precursor [KO:K01602] [EC:4.1.1.39]",
            100279574L,                            "ribulose bisphosphate carboxylase smplastid subunit 2 [KO:K01602] [EC:4.1.1.39]",
            100274579L,                                              "umc1110; Phosphoglycerate kinase [KO:K00927] [EC:2.7.2.3]",
            100281237L,                                                       "phosphoglycerate kinase [KO:K00927] [EC:2.7.2.3]",
            100381624L,                                                       "phosphoglycerate kinase [KO:K00927] [EC:2.7.2.3]",
            100383528L,                                                       "phosphoglycerate kinase [KO:K00927] [EC:2.7.2.3]",
            103635817L,                                        "phosphoglycerate kinase, chloroplastic [KO:K00927] [EC:2.7.2.3]",
               542718L,                 "GAPC2; glyceraldehyde-3-phosphate dehydrogenase 2, cytosolic [KO:K00134] [EC:1.2.1.12]",
            100191900L,                          "glyceraldehyde-3-phosphate dehydrogenase, cytosolic [KO:K00134] [EC:1.2.1.12]",
               542333L,                 "GAPC3; glyceraldehyde-3-phosphate dehydrogenase 3, cytosolic [KO:K00134] [EC:1.2.1.12]",
               542367L,                        "glyceraldehyde-3-phosphate dehydrogenase 1, cytosolic [KO:K00134] [EC:1.2.1.12]",
            100037774L,                               "TIDP3317; uncharacterized protein LOC100037774 [KO:K00134] [EC:1.2.1.12]",
            100282981L,                             "pco070235b; uncharacterized protein LOC100282981 [KO:K00134] [EC:1.2.1.12]",
               542368L,    "gpa1; glyceraldehyde-3-phosphate dehydrogenase A, chloroplastic precursor [KO:K05298] [EC:1.2.1.13]",
               542290L,                              "gpb1; glyceraldehyde phosphate dehydrogenase B1 [KO:K05298] [EC:1.2.1.13]",
            103641723L,                    "glyceraldehyde-3-phosphate dehydrogenase A, chloroplastic [KO:K05298] [EC:1.2.1.13]",
            103642790L,                    "glyceraldehyde-3-phosphate dehydrogenase A, chloroplastic [KO:K05298] [EC:1.2.1.13]",
            109943118L,                    "glyceraldehyde-3-phosphate dehydrogenase A, chloroplastic [KO:K05298] [EC:1.2.1.13]",
               542261L,                    "ald1; fructose-bisphosphate aldolase, cytoplasmic isozyme [KO:K01623] [EC:4.1.2.13]",
            100272845L,                               "AY110170; uncharacterized protein LOC100272845 [KO:K01623] [EC:4.1.2.13]",
            100273196L,                                "IDP1438; uncharacterized protein LOC100273196 [KO:K01623] [EC:4.1.2.13]",
            100272913L,                           "fructose-bisphosphate aldolase cytoplasmic isozyme [KO:K01623] [EC:4.1.2.13]",
            100286050L,                                               "fructose-bisphosphate aldolase [KO:K01623] [EC:4.1.2.13]",
            103644539L,                              "fructose-bisphosphate aldolase 3, chloroplastic [KO:K01623] [EC:4.1.2.13]",
            100502020L,                              "fructose-bisphosphate aldolase 3, chloroplastic [KO:K01623] [EC:4.1.2.13]",
            100274515L,                                                 "uncharacterized LOC100274515 [KO:K01623] [EC:4.1.2.13]",
            103653043L,                         "fructose-bisphosphate aldolase 3, chloroplastic-like [KO:K01623] [EC:4.1.2.13]",
            103648726L,                              "fructose-bisphosphate aldolase 3, chloroplastic [KO:K01623] [EC:4.1.2.13]",
            103653987L,                              "fructose-bisphosphate aldolase 3, chloroplastic [KO:K01623] [EC:4.1.2.13]",
            100273700L,                              "pco073762; uncharacterized protein LOC100273700 [KO:K03841] [EC:3.1.3.11]",
            100216767L,                                         "uncharacterized protein LOC100216767 [KO:K03841] [EC:3.1.3.11]",
            100281068L,                                                  "fructose-1,6-bisphosphatase [KO:K03841] [EC:3.1.3.11]",
            100284739L,                                       "fructose-1,6-bisphosphatase, cytosolic [KO:K03841] [EC:3.1.3.11]",
            103631224L,                                       "fructose-1,6-bisphosphatase, cytosolic [KO:K03841] [EC:3.1.3.11]",
            100279513L,                                                  "uncharacterized LOC100279513 [KO:K00615] [EC:2.2.1.1]",
            103646666L,                                             "transketolase, chloroplastic-like [KO:K00615] [EC:2.2.1.1]",
            103640190L,                                                  "transketolase, chloroplastic [KO:K00615] [EC:2.2.1.1]",
            100282017L,                                        "shbp1; sedoheptulose bisphosphatase 1 [KO:K01100] [EC:3.1.3.37]",
            100283465L,                                                  "ribose-5-phosphate isomerase [KO:K01807] [EC:5.3.1.6]",
            100284292L,                                                  "ribose-5-phosphate isomerase [KO:K01807] [EC:5.3.1.6]",
            100284771L,                                          "uncharacterized protein LOC100284771 [KO:K01807] [EC:5.3.1.6]",
            100285298L,                                                  "ribose-5-phosphate isomerase [KO:K01807] [EC:5.3.1.6]",
            100285676L,                                                  "ribose-5-phosphate isomerase [KO:K01807] [EC:5.3.1.6]",
            103626247L,                                       "probable ribose-5-phosphate isomerase 2 [KO:K01807] [EC:5.3.1.6]",
            103655139L,                                       "probable ribose-5-phosphate isomerase 2 [KO:K01807] [EC:5.3.1.6]",
            100279820L,                                                 "uncharacterized LOC100279820 [KO:K00855] [EC:2.7.1.19]",
            100282845L,                              "pco092382; uncharacterized protein LOC100282845 [KO:K00855] [EC:2.7.1.19]",
            100281866L,                            "si707035f08; uncharacterized protein LOC100281866 [KO:K00855] [EC:2.7.1.19]",
            100191222L,                                          "uncharacterized protein LOC100191222 [KO:K01803] [EC:5.3.1.1]",
            100193000L,                                 "gpm397a; uncharacterized protein LOC100193000 [KO:K01803] [EC:5.3.1.1]",
            103645706L,                                     "triosephosphate isomerase, cytosolic-like [KO:K01803] [EC:5.3.1.1]",
            100194114L,                                          "uncharacterized protein LOC100194114 [KO:K01803] [EC:5.3.1.1]",
            100272481L,                                     "TIM; triosephosphate isomerase, cytosolic [KO:K01803] [EC:5.3.1.1]",
            100280823L,                                          "triosephosphate isomerase, cytosolic [KO:K01803] [EC:5.3.1.1]",
            103651898L,                                     "triosephosphate isomerase, cytosolic-like [KO:K01803] [EC:5.3.1.1]",
            103644857L,                                          "triosephosphate isomerase, cytosolic [KO:K01803] [EC:5.3.1.1]",
            100274365L,                                          "uncharacterized protein LOC100274365 [KO:K01783] [EC:5.1.3.1]",
            100282201L,                                                "ribulose-phosphate 3-epimerase [KO:K01783] [EC:5.1.3.1]",
            100283478L,                                                "ribulose-phosphate 3-epimerase [KO:K01783] [EC:5.1.3.1]",
            100191460L,                          "IDP1621; phosphoenolpyruvate carboxylase isoform X1 [KO:K01595] [EC:4.1.1.31]",
               542393L,                                      "pep4; phosphoenolpyruvate carboxylase 2 [KO:K01595] [EC:4.1.1.31]",
               542479L,                                        "pep7; phosphoenolpyruvate carboxylase [KO:K01595] [EC:4.1.1.31]",
               542372L,                                      "pep1; phosphoenolpyruvate carboxylase 1 [KO:K01595] [EC:4.1.1.31]",
            100384708L,                              "uncharacterized protein LOC100384708 isoform X1 [KO:K01595] [EC:4.1.1.31]",
            103649899L,                                 "phosphoenolpyruvate carboxylase 4 isoform X1 [KO:K01595] [EC:4.1.1.31]",
            103642664L,                                       "phosphoenolpyruvate carboxylase 1-like [KO:K01595] [EC:4.1.1.31]",
            100273311L,                               "pco118382; uncharacterized protein LOC100273311 [KO:K14454] [EC:2.6.1.1]",
               541622L,                            "phosphoenolpyruvate carboxykinase (ATP) isoform 1 [KO:K01610] [EC:4.1.1.49]",
            100279748L,                            "pck2; phosphoenolpyruvate carboxykinase homolog 2 [KO:K01610] [EC:4.1.1.49]",
            109939179L,                                      "phosphoenolpyruvate carboxykinase (ATP) [KO:K01610] [EC:4.1.1.49]",
            100216685L,                                          "uncharacterized protein LOC100216685 [KO:K00814] [EC:2.6.1.2]",
            100279149L,                                          "uncharacterized protein LOC100279149 [KO:K00814] [EC:2.6.1.2]",
            100193705L,                                          "uncharacterized protein LOC100193705 [KO:K00814] [EC:2.6.1.2]",
            100384793L,                                                    "alanine aminotransferase 2 [KO:K00814] [EC:2.6.1.2]",
            100282849L,                                          "uncharacterized protein LOC100282849 [KO:K00814] [EC:2.6.1.2]",
            100280822L,                                                    "alanine aminotransferase 2 [KO:K00814] [EC:2.6.1.2]",
            103634667L,                                                    "alanine aminotransferase 2 [KO:K00814] [EC:2.6.1.2]",
            100501386L,                         "uncharacterized protein LOC100501386 [KO:K14272] [EC:2.6.1.4 2.6.1.2 2.6.1.44]",
            100284842L,                         "uncharacterized protein LOC100284842 [KO:K14272] [EC:2.6.1.4 2.6.1.2 2.6.1.44]",
               542759L,                 "pdk1; pyruvate, phosphate dikinase 1, chloroplastic precursor [KO:K01006] [EC:2.7.9.1]",
            103635678L,                                                "pyruvate, phosphate dikinase 2 [KO:K01006] [EC:2.7.9.1]",
               542233L,                    "me3; NADP-dependent malic enzyme, chloroplastic precursor [KO:K00029] [EC:1.1.1.40]",
               542331L,                                             "me2; NADP-dependent malic enzyme [KO:K00029] [EC:1.1.1.40]",
               542209L,                                                     "me4; NADP malic enzyme 4 [KO:K00029] [EC:1.1.1.40]",
            100284598L,                                                  "NADP-dependent malic enzyme [KO:K00029] [EC:1.1.1.40]",
            100286036L,                               "AY104511; uncharacterized protein LOC100286036 [KO:K00029] [EC:1.1.1.40]",
            109939197L,                        "NADP-dependent malic enzyme, chloroplastic isoform X1 [KO:K00029] [EC:1.1.1.40]",
            103635691L,                                   "NADP-dependent malic enzyme, chloroplastic [KO:K00029] [EC:1.1.1.40]",
               542374L,                   "mdh6; malate dehydrogenase [NADP], chloroplastic precursor [KO:K00051] [EC:1.1.1.82]",
            100274119L,                                                    "aspartate aminotransferase [KO:K14455] [EC:2.6.1.1]",
            103648465L,                                            "malate dehydrogenase, cytoplasmic [KO:K00025] [EC:1.1.1.37]",
               542598L,                                      "mdh5; malate dehydrogenase, cytoplasmic [KO:K00025] [EC:1.1.1.37]",
            100280767L,                                            "malate dehydrogenase, cytoplasmic [KO:K00025] [EC:1.1.1.37]",
            103633793L,                                            "malate dehydrogenase, cytoplasmic [KO:K00025] [EC:1.1.1.37]",
            100274264L,                                                       "malate dehydrogenase 2 [KO:K00026] [EC:1.1.1.37]",
            100273428L,                                                 "uncharacterized LOC100273428 [KO:K00026] [EC:1.1.1.37]",
            100272900L,                                         "uncharacterized protein LOC100272900 [KO:K00026] [EC:1.1.1.37]",
            100193491L,                              "pco104294; uncharacterized protein LOC100193491 [KO:K00026] [EC:1.1.1.37]",
            100193663L,                                                         "malate dehydrogenase [KO:K00026] [EC:1.1.1.37]",
            100282134L,                                            "malate dehydrogenase, glyoxysomal [KO:K00026] [EC:1.1.1.37]",
            100193743L,                                             "pco143139c; Malate dehydrogenase [KO:K00026] [EC:1.1.1.37]",
            107305678L,                                         "uncharacterized protein LOC107305678 [KO:K00026] [EC:1.1.1.37]",
            100191942L,                             "pco115183b; uncharacterized protein LOC100191942 [KO:K00028] [EC:1.1.1.39]",
            100501486L,                                         "uncharacterized protein LOC100501486 [KO:K00028] [EC:1.1.1.39]"
            )

```
```{r}
g46entrez <- as.character(g46tib$V1)
```
```{r}
g46 <- mgdb$v4_gene_model[mgdb$Entrez%in%g46entrez]
```

###g47
```{r}
CornCycCalvinCycle <- tibble::tribble(
                            ~Gene.ID,  ~Gene.Accession,       ~Gene.name,                            ~Reaction.id, ~Reaction.EC,                        ~Enzymatic.activity, ~Evidence,
                         "GDQC-7815",  "GRMZM5G815453",  "GRMZM5G815453", "RIBULOSE-BISPHOSPHATE-CARBOXYLASE-RXN",   "4.1.1.39",        "ribulose bisphosphate carboxylase", "EV-COMP",
                         "GDQC-9028", "Zm00001d004894", "Zm00001d004894", "RIBULOSE-BISPHOSPHATE-CARBOXYLASE-RXN",   "4.1.1.39",        "ribulose bisphosphate carboxylase", "EV-COMP",
                        "GDQC-11504", "Zm00001d037514", "Zm00001d037514", "RIBULOSE-BISPHOSPHATE-CARBOXYLASE-RXN",   "4.1.1.39",        "ribulose bisphosphate carboxylase", "EV-COMP",
                        "GDQC-12129", "Zm00001d045698", "Zm00001d045698", "RIBULOSE-BISPHOSPHATE-CARBOXYLASE-RXN",   "4.1.1.39",        "ribulose bisphosphate carboxylase", "EV-COMP",
                        "GDQC-12448", "Zm00001d006402", "Zm00001d006402", "RIBULOSE-BISPHOSPHATE-CARBOXYLASE-RXN",   "4.1.1.39",        "ribulose bisphosphate carboxylase", "EV-COMP",
                        "GDQC-13860", "Zm00001d052595", "Zm00001d052595", "RIBULOSE-BISPHOSPHATE-CARBOXYLASE-RXN",   "4.1.1.39",        "ribulose bisphosphate carboxylase", "EV-COMP",
                        "GDQC-14404", "Zm00001d051485", "Zm00001d051485", "RIBULOSE-BISPHOSPHATE-CARBOXYLASE-RXN",   "4.1.1.39",        "ribulose bisphosphate carboxylase", "EV-COMP",
                        "GDQC-14718", "Zm00001d000279", "Zm00001d000279", "RIBULOSE-BISPHOSPHATE-CARBOXYLASE-RXN",   "4.1.1.39",        "ribulose bisphosphate carboxylase", "EV-COMP",
                         "GDQC-8568", "Zm00001d010672", "Zm00001d010672",                       "PHOSGLYPHOS-RXN",    "2.7.2.3",                  "phosphoglycerate kinase", "EV-COMP",
                         "GDQC-9820", "Zm00001d032867", "Zm00001d032867",                       "PHOSGLYPHOS-RXN",    "2.7.2.3",                  "phosphoglycerate kinase", "EV-COMP",
                         "GDQC-9835", "Zm00001d043194", "Zm00001d043194",                       "PHOSGLYPHOS-RXN",    "2.7.2.3",                  "phosphoglycerate kinase", "EV-COMP",
                        "GDQC-13456", "Zm00001d015376", "Zm00001d015376",                       "PHOSGLYPHOS-RXN",    "2.7.2.3",                  "phosphoglycerate kinase", "EV-COMP",
                        "GDQC-14659", "Zm00001d038579", "Zm00001d038579",                       "PHOSGLYPHOS-RXN",    "2.7.2.3",                  "phosphoglycerate kinase", "EV-COMP",
                         "GDQC-7880", "Zm00001d027488", "Zm00001d027488",                          "1.2.1.13-RXN",   "1.2.1.13", "glyceraldehyde-3-phosphate dehydrogenase", "EV-COMP",
                         "GDQC-9182", "Zm00001d003429", "Zm00001d003429",                          "1.2.1.13-RXN",   "1.2.1.13", "glyceraldehyde-3-phosphate dehydrogenase", "EV-COMP",
                         "GDQC-9880", "Zm00001d025601", "Zm00001d025601",                          "1.2.1.13-RXN",   "1.2.1.13", "glyceraldehyde-3-phosphate dehydrogenase", "EV-COMP",
                        "GDQC-10044", "Zm00001d025607", "Zm00001d025607",                          "1.2.1.13-RXN",   "1.2.1.13", "glyceraldehyde-3-phosphate dehydrogenase", "EV-COMP",
                        "GDQC-13181", "Zm00001d025599", "Zm00001d025599",                          "1.2.1.13-RXN",   "1.2.1.13", "glyceraldehyde-3-phosphate dehydrogenase", "EV-COMP",
                        "GDQC-14696", "Zm00001d025593", "Zm00001d025593",                          "1.2.1.13-RXN",   "1.2.1.13", "glyceraldehyde-3-phosphate dehydrogenase", "EV-COMP",
                         "GDQC-7719", "Zm00001d009835", "Zm00001d009835",              "TRIOSEPISOMERIZATION-RXN",    "5.3.1.1",                "triosephosphate isomerase", "EV-COMP",
                         "GDQC-8254", "Zm00001d006221", "Zm00001d006221",              "TRIOSEPISOMERIZATION-RXN",    "5.3.1.1",                "triosephosphate isomerase", "EV-COMP",
                         "GDQC-8259", "Zm00001d033237", "Zm00001d033237",              "TRIOSEPISOMERIZATION-RXN",    "5.3.1.1",                "triosephosphate isomerase", "EV-COMP",
                         "GDQC-9733", "Zm00001d039865", "Zm00001d039865",              "TRIOSEPISOMERIZATION-RXN",    "5.3.1.1",                "triosephosphate isomerase", "EV-COMP",
                        "GDQC-11094", "Zm00001d030216", "Zm00001d030216",              "TRIOSEPISOMERIZATION-RXN",    "5.3.1.1",                "triosephosphate isomerase", "EV-COMP",
                        "GDQC-11682", "Zm00001d024005", "Zm00001d024005",              "TRIOSEPISOMERIZATION-RXN",    "5.3.1.1",                "triosephosphate isomerase", "EV-COMP",
                        "GDQC-12310", "Zm00001d021310", "Zm00001d021310",              "TRIOSEPISOMERIZATION-RXN",    "5.3.1.1",                "triosephosphate isomerase", "EV-COMP",
                        "GDQC-12367", "Zm00001d008619", "Zm00001d008619",              "TRIOSEPISOMERIZATION-RXN",    "5.3.1.1",                "triosephosphate isomerase", "EV-COMP",
                        "GDQC-12389", "Zm00001d012407", "Zm00001d012407",              "TRIOSEPISOMERIZATION-RXN",    "5.3.1.1",                "triosephosphate isomerase", "EV-COMP",
                        "GDQC-12402", "Zm00001d040375", "Zm00001d040375",              "TRIOSEPISOMERIZATION-RXN",    "5.3.1.1",                "triosephosphate isomerase", "EV-COMP",
                        "GDQC-13866", "Zm00001d029307", "Zm00001d029307",              "TRIOSEPISOMERIZATION-RXN",    "5.3.1.1",                "triosephosphate isomerase", "EV-COMP",
                        "GDQC-11235", "Zm00001d042840", "Zm00001d042840",      "SEDOHEPTULOSE-BISPHOSPHATASE-RXN",   "3.1.3.37",             "sedoheptulose bisphosphatase", "EV-COMP",
                         "GDQC-8410", "Zm00001d046962", "Zm00001d046962",                        "1TRANSKETO-RXN",    "2.2.1.1",                            "transketolase", "EV-COMP",
                         "GDQC-8808", "Zm00001d045451", "Zm00001d045451",                        "1TRANSKETO-RXN",    "2.2.1.1",                            "transketolase", "EV-COMP",
                         "GDQC-9566", "Zm00001d004171", "Zm00001d004171",                        "1TRANSKETO-RXN",    "2.2.1.1",                            "transketolase", "EV-COMP",
                         "GDQC-8788", "Zm00001d030701", "Zm00001d030701",                         "RIB5PISOM-RXN",    "5.3.1.6",             "ribose-5-phosphate isomerase", "EV-COMP",
                         "GDQC-9330", "Zm00001d052482", "Zm00001d052482",                         "RIB5PISOM-RXN",    "5.3.1.6",             "ribose-5-phosphate isomerase", "EV-COMP",
                        "GDQC-11246", "Zm00001d013135", "Zm00001d013135",                         "RIB5PISOM-RXN",    "5.3.1.6",             "ribose-5-phosphate isomerase", "EV-COMP",
                        "GDQC-12396", "Zm00001d018936", "Zm00001d018936",                         "RIB5PISOM-RXN",    "5.3.1.6",             "ribose-5-phosphate isomerase", "EV-COMP",
                        "GDQC-14749", "Zm00001d044548", "Zm00001d044548",                         "RIB5PISOM-RXN",    "5.3.1.6",             "ribose-5-phosphate isomerase", "EV-COMP",
                        "GDQC-11498", "Zm00001d028395", "Zm00001d028395",                       "F16ALDOLASE-RXN",   "4.1.2.13",           "fructose bisphosphate aldolase", "EV-COMP",
                        "GDQC-11630", "Zm00001d028270", "Zm00001d028270",                       "F16ALDOLASE-RXN",   "4.1.2.13",           "fructose bisphosphate aldolase", "EV-COMP",
                        "GDQC-12132", "Zm00001d008491", "Zm00001d008491",                       "F16ALDOLASE-RXN",   "4.1.2.13",           "fructose bisphosphate aldolase", "EV-COMP",
                        "GDQC-12701", "Zm00001d053015", "Zm00001d053015",                       "F16ALDOLASE-RXN",   "4.1.2.13",           "fructose bisphosphate aldolase", "EV-COMP",
                        "GDQC-12807", "Zm00001d037333", "Zm00001d037333",                       "F16ALDOLASE-RXN",   "4.1.2.13",           "fructose bisphosphate aldolase", "EV-COMP",
                        "GDQC-13437", "Zm00001d042453", "Zm00001d042453",                       "F16ALDOLASE-RXN",   "4.1.2.13",           "fructose bisphosphate aldolase", "EV-COMP",
                        "GDQC-13481", "Zm00001d046553", "Zm00001d046553",                       "F16ALDOLASE-RXN",   "4.1.2.13",           "fructose bisphosphate aldolase", "EV-COMP",
                        "GDQC-13554", "Zm00001d012103", "Zm00001d012103",                       "F16ALDOLASE-RXN",   "4.1.2.13",           "fructose bisphosphate aldolase", "EV-COMP",
                        "GDQC-13634", "Zm00001d003968", "Zm00001d003968",                       "F16ALDOLASE-RXN",   "4.1.2.13",           "fructose bisphosphate aldolase", "EV-COMP",
                        "GDQC-13867", "Zm00001d023559", "Zm00001d023559",                       "F16ALDOLASE-RXN",   "4.1.2.13",           "fructose bisphosphate aldolase", "EV-COMP",
                        "GDQC-14596", "Zm00001d040084", "Zm00001d040084",                       "F16ALDOLASE-RXN",   "4.1.2.13",           "fructose bisphosphate aldolase", "EV-COMP",
                        "GDQC-14702", "Zm00001d051622", "Zm00001d051622",                       "F16ALDOLASE-RXN",   "4.1.2.13",           "fructose bisphosphate aldolase", "EV-COMP",
                         "GDQC-9110", "Zm00001d042727", "Zm00001d042727",                        "F16BDEPHOS-RXN",   "3.1.3.11",              "fructose-1,6-bisphosphatase", "EV-COMP",
                        "GDQC-10262", "Zm00001d036552", "Zm00001d036552",                        "F16BDEPHOS-RXN",   "3.1.3.11",              "fructose-1,6-bisphosphatase", "EV-COMP",
                        "GDQC-13231", "Zm00001d010172", "Zm00001d010172",                        "F16BDEPHOS-RXN",   "3.1.3.11",              "fructose-1,6-bisphosphatase", "EV-COMP",
                        "GDQC-13510", "Zm00001d046942", "Zm00001d046942",                        "F16BDEPHOS-RXN",   "3.1.3.11",              "fructose-1,6-bisphosphatase", "EV-COMP",
                        "GDQC-14499", "Zm00001d028562", "Zm00001d028562",                        "F16BDEPHOS-RXN",   "3.1.3.11",              "fructose-1,6-bisphosphatase", "EV-COMP",
                         "GDQC-9420", "Zm00001d027841", "Zm00001d027841",                       "RIBULP3EPIM-RXN",    "5.1.3.1",         "ribulose-5-phosphate 3-epimerase", "EV-COMP",
                        "GDQC-12905", "Zm00001d021050", "Zm00001d021050",                       "RIBULP3EPIM-RXN",    "5.1.3.1",         "ribulose-5-phosphate 3-epimerase", "EV-COMP",
                        "GDQC-14125", "Zm00001d048324", "Zm00001d048324",                       "RIBULP3EPIM-RXN",    "5.1.3.1",         "ribulose-5-phosphate 3-epimerase", "EV-COMP",
                         "GDQC-8653", "Zm00001d002454", "Zm00001d002454",               "PHOSPHORIBULOKINASE-RXN",   "2.7.1.19",                      "phosphoribulokinase", "EV-COMP",
                        "GDQC-10278", "Zm00001d017711", "Zm00001d017711",               "PHOSPHORIBULOKINASE-RXN",   "2.7.1.19",                      "phosphoribulokinase", "EV-COMP",
                        "GDQC-13352", "Zm00001d000164", "Zm00001d000164",               "PHOSPHORIBULOKINASE-RXN",   "2.7.1.19",                      "phosphoribulokinase", "EV-COMP"
                        )

```
```{r}
g47 <- CornCycCalvinCycle$Gene.Accession
```
```{r}
g47 <- substr(g47, start = 1, stop = 14)
```

###g48
```{r}
g48 <- as.vector(gamer$db_object_id[gamer$term_accession=="GO:0009853"])
```
###g49
```{r}
g49 <- as.vector(gamer$db_object_id[gamer$term_accession=="GO:0080093"])
```
###g50
```{r}
g50 <- as.vector(gamer$db_object_id[gamer$term_accession=="GO:0009853"])
```
###g51
```{r}
g51 <- c("Zm00001d034887", "Zm00001d025747", "Zm00001d031403", "Zm00001d031402", "Zm00001d031364", "Zm00001d031379", "Zm00001d031349", "Zm00001d014919", "Zm00001d004451", "Zm00001d024906", "Zm00001d005489", "Zm00001d027450", "Zm00001d018790", "Zm00001d008883", "Zm00001d024907", "Zm00001d035737", "Zm00001d019077", "Zm00001d047085", "Zm00001d000064", "Zm00001d015378", "Zm00001d053800", "Zm00001d048974", "Zm00001d007937", "Zm00001d031403", "Zm00001d031402", "Zm00001d031364", "Zm00001d018639", "Zm00001d031379", "Zm00001d031349", "Zm00001d040457", "Zm00001d012247", "Zm00001d007237", "Zm00001d030859", "Zm00001d002554", "Zm00001d042661", "Zm00001d049234", "Zm00001d009738")
```
###listplastid
```{r}
PhotosynthesisTermsList <- list(g1,g2,g3,g4,g5,g6,g7,g8,g9,g10,g11,g12,g13,g14,g15,g16,g17,g18,g19,g20,g21,g22,g23,g24,g25,g26,g27,g28,g29,g30,g31,g32,g33,g34,g35,g36,g37,g38,g39,g40,g41,g42,g43,g44,g45,g46,g47,g48,g49,g50,g51)
```
```{r}
saveRDS(PhotosynthesisTermsList, "PhotosynthesisTermsList")
```
```{r}
PhotosynthesisTermsList <- readRDS("PhotosynthesisTermsList")
```

###combine terms
```{r}
lightrxn.kc <- unique(unlist(PhotosynthesisTermsList[c(15:16,18,25,26,28,35,36,37,39,40)]))
darkrxn.kc <- unique(unlist(PhotosynthesisTermsList[c(42,43,46,47)]))

lightrxn.PSI.kc <- unique(unlist(PhotosynthesisTermsList[c(15,16)]))

lightrxn.PSII.kc <- unique(unlist(PhotosynthesisTermsList[c(25,26)]))

lightrxn.ETC.kc <- unlist(PhotosynthesisTermsList[35])

lightrxn.b6f.k <- unlist(PhotosynthesisTermsList[37])

lightrxn.NDH <- tmt.withPolymorphic[grep("NDH", tmt.withPolymorphic$Protein),]
lightrxn.NDH <- lightrxn.NDH$Gene[grep("subunit", lightrxn.NDH$Protein)]

lightrxn.ATPase.k <- unlist(PhotosynthesisTermsList[40])
grepATPsynthase <- tmt.withPolymorphic$Gene[grepl("ATP",tmt.withPolymorphic$Protein)&grepl("synthase",tmt.withPolymorphic$Protein)&grepl("chloropl",tmt.withPolymorphic$Protein)]
lightrxn.ATPase.k <- unique(c(lightrxn.ATPase.k,grepATPsynthase))

lightrxn.NucComplexes <- unique(c(lightrxn.PSI.kc,lightrxn.PSII.kc,lightrxn.b6f.k,lightrxn.NDH,lightrxn.ATPase.k))

darkrxn.C4.c <- unique(unlist(PhotosynthesisTermsList[c(42,43)]))
darkrxn.Calvin.kc <- unique(unlist(PhotosynthesisTermsList[c(46,47)]))
darkrxn.Calvin.RubiscoSS <- c("Zm00001d052595", "Zm00001d004894")
```
##chloroplast background
```{r}
plastidCC <- as.list(GOCCCHILDREN)
plastidCC <- as.vector(plastidCC$`GO:0009536`)
plastidCC <- c(plastidCC, "GO:0009536")
plastidCC <- unique(as.vector(gamer$db_object_id[gamer$term_accession%in%plastidCC]))
```
##mitochondria background
```{r}
mitoCC <- as.list(GOCCCHILDREN)
mitoCC <- as.vector(mitoCC$`GO:0005739`)
mitoCC <- c(mitoCC, "GO:0005739")
mitoCC <- as.vector(gamer$db_object_id[gamer$term_accession%in%mitoCC])
```

##Plastid-encoded
```{r}
#Plastid-encoded proteins
Pt<-tmt.withPolymorphic[grep("ZemaCp", tmt.withPolymorphic$Accession),]
#define list of plastid-encoded photosynthesis-associated genes
Pt.PS <- Pt$Gene[grep("psb|psa|ndh|rbc|pet|atp|ccs|cem", Pt$Protein)]
Pt.PS.PSI <- Pt$Gene[grep("psa", Pt$Protein)]
Pt.PS.PSII <- Pt$Gene[grep("psb", Pt$Protein)]
Pt.PS.NADH <- Pt$Gene[grep("ndh", Pt$Protein)]
Pt.PS.b6f <- Pt$Gene[grep("pet", Pt$Protein)]
Pt.PS.ATP <- Pt$Gene[grep("atp", Pt$Protein)]
Pt.PS.rubisco <- Pt$Gene[grep("rbc", Pt$Protein)]
Pt.Complexes <- c(Pt.PS.PSI,Pt.PS.PSII,Pt.PS.NADH,Pt.PS.b6f,Pt.PS.ATP)
#Plastid encoded ribosomal proteins (EnsemblPlants Annotation)
#get subset of Pt encoded genes annotated as ribosomal proteins
Pt.ribo <- Pt$Gene[grep("rps|rpl", Pt$Protein)]
#Plastid encoded RNA polymerases (EnsemblPlants Annotation)
Pt.rpo <- Pt$Gene[grep("rpo", Pt$Protein)]
#Define "other" as everything left
Pt.other <- Pt$Gene[!(Pt$Gene%in%c(Pt.PS.rubisco, Pt.Complexes, Pt.ribo, Pt.rpo))]
Pt <- Pt$Gene
```

##Nuclear-encoded plastid ribosome
```{r}
CRASS <- data.frame("At.acc"= c("AT5G51545", "AT3G60370", "AT1G11430", "AT2G44650", "AT3G08920", "AT3G14110", "AT3G08740", "AT5G546000", "AT5G65220", "AT4G24770", "AT2G33800", "AT5G30510", "AT2G33450", "AT5G47190", "AT1G29070", "AT3G25920", "AT5G13510", "AT1G64510", "AT1G48350", "AT3G15190", "AT1G05190", "AT3G52150", "AT2G43030", "AT1G75350", "AT3G54210", "AT5G14320", "AT1G78630", "AT4G01310", "AT4G17560", "AT1G32990", "AT1G35680"), "At.names"= c("LPA2", "FKBPL", "MORF9", "CPN10", "Rhodanese", "FLU", "EF-P", "PRPL24", "PRPL29", "RBP31", "PRPS5", "PRPS1", "PRPL28", "PRPL19", "PRPL34", "PRPL15", "PRPL10", "PRPS6", "PRPL18", "PRPS20", "PRPL6", "PSRP2", "PRPL3", "PRPL31", "PRPL17", "PRPS13", "PRPL13", "PRPL5", "PRPL19", "PRPL11", "PRPL21"))

CRASS <- merge.data.frame(CRASS, mgdb[,c(1,17)], by.x = "At.acc", by.y = "arabidopsis_ortholog", plastid.x = T, plastid.y = F)
```
```{r}
CRASS.prpl.prps <- CRASS$v4_gene_model[grep("PRPL|PRPS", CRASS$At.names)]
ribosom <- tmt.withPolymorphic[(grep("ribosom", tmt.withPolymorphic$Protein)),]
ribosom.pt <- ribosom[(grep("30S|50S|plastid|Plastid|Chloropl|chloropl", ribosom$Protein)),]
Nuc.PtRibo <- unique(c(ribosom.pt$Gene, CRASS.prpl.prps))
```


##Eukaryotic ribosome
```{r}
#Eukaryotic ribosomal proteins (EnsemblPlants Annotation)
eukRibo<-tmt.withPolymorphic[(grep("40S|60S", tmt.withPolymorphic$Protein)),]
#remove the mitochondrial ones
eukRibo<- eukRibo$Gene[-(grep("mito|Mito", eukRibo$Protein))]
```
##Etyhlene
```{r}
cornCyc <- read.csv("cornCycGeneLists01042020.csv", stringsAsFactors = F)
MAT <- cornCyc$Gene.Accession[(cornCyc$Enzymatic.activity=="methionine adenosyltransferase")]
ACS <- cornCyc$Gene.Accession[(cornCyc$Enzymatic.activity=="1-aminocyclopropane-1-carboxylate synthase")]
ACO <- cornCyc$Gene.Accession[(cornCyc$Enzymatic.activity=="1-aminocyclopropane-1-carboxylate oxidase")]
EtBio <- c(MAT,ACS,ACO)
```

##Mitochondria
```{r}
KEGG.OxPhos.Tib <- tibble::tribble(
                                       ~V1,                                                                                                           ~V2,
                                   845179L,                                                 "ndhA; NADH dehydrogenase subunit 1 [KO:K05572] [EC:7.1.1.2]",
                                   845181L,                                                 "ndhB; NADH dehydrogenase subunit 2 [KO:K05573] [EC:7.1.1.2]",
                                   845180L,                                                 "ndhB; NADH dehydrogenase subunit 2 [KO:K05573] [EC:7.1.1.2]",
                                109943316L,                                               "uncharacterized protein LOC109943316 [KO:K05573] [EC:7.1.1.2]",
                                   845182L,                                                 "ndhC; NADH dehydrogenase subunit 3 [KO:K05574] [EC:7.1.1.2]",
                                   845183L,                                                 "ndhD; NADH dehydrogenase subunit 4 [KO:K05575] [EC:7.1.1.2]",
                                   845184L,                                                "ndhE; NADH dehydrogenase subunit 4L [KO:K05576] [EC:7.1.1.2]",
                                   845185L,                                                 "ndhF; NADH dehydrogenase subunit 5 [KO:K05577] [EC:7.1.1.2]",
                                109940833L,                                               "uncharacterized protein LOC109940833 [KO:K05577] [EC:7.1.1.2]",
                                   845186L,                                                 "ndhG; NADH dehydrogenase subunit 6 [KO:K05578] [EC:7.1.1.2]",
                                   845187L,                                                 "ndhH; NADH dehydrogenase subunit 7 [KO:K05579] [EC:7.1.1.2]",
                                   845188L,                                                 "ndhI; NADH dehydrogenase subunit I [KO:K05580] [EC:7.1.1.2]",
                                   845189L,                                                 "ndhJ; NADH dehydrogenase subunit J [KO:K05581] [EC:7.1.1.2]",
                                   845190L,                                                 "ndhK; NADH dehydrogenase subunit K [KO:K05582] [EC:7.1.1.2]",
                                100191453L,                                    "NADH-ubiquinone oxidoreductase subunit [KO:K03934] [EC:7.1.1.2 1.6.99.3]",
                                109942203L,                                      "uncharacterized protein LOC109942203 [KO:K03935] [EC:7.1.1.2 1.6.99.3]",
                                100193844L,                                                          "NADH-ubiquinone oxidoreductase subunit [KO:K03937]",
                                100281028L,                                                 "pco104752; uncharacterized protein LOC100281028 [KO:K03937]",
                                100285915L,                                                 "pco090267; uncharacterized protein LOC100285915 [KO:K03938]",
                                100280596L,                                  "mitochondrial NADH ubiquinone oxidoreductase 13kD-like subunit [KO:K03939]",
                                100284802L,                                  "mitochondrial NADH ubiquinone oxidoreductase 13kD-like subunit [KO:K03939]",
                                100192583L,                                    "NADH-ubiquinone oxidoreductase subunit [KO:K03940] [EC:7.1.1.2 1.6.99.3]",
                                100127013L,                                                "nad2; NADH dehydrogenase 2 [KO:K03940] [EC:7.1.1.2 1.6.99.3]",
                                100282522L,                             "NADH-ubiquinone oxidoreductase 23 kDa subunit [KO:K03941] [EC:7.1.1.2 1.6.99.3]",
                                100283266L,                           "cl4677_1a; uncharacterized protein LOC100283266 [KO:K03941] [EC:7.1.1.2 1.6.99.3]",
                                100273176L,                                      "uncharacterized protein LOC100273176 [KO:K03942] [EC:7.1.1.2 1.6.99.3]",
                                100282384L,                             "NADH-ubiquinone oxidoreductase 51 kDa subunit [KO:K03942] [EC:7.1.1.2 1.6.99.3]",
                                100272681L,                                      "uncharacterized protein LOC100272681 [KO:K03943] [EC:7.1.1.2 1.6.99.3]",
                                100501943L,                                                            "uncharacterized protein LOC100501943 [KO:K03945]",
                                100283792L,                                                 "NADH-ubiquinone oxidoreductase 10.5 kDa subunit [KO:K03946]",
                                100283003L,                                                 "NADH-ubiquinone oxidoreductase 10.5 kDa subunit [KO:K03946]",
                                100280838L,                                                 "NADH-ubiquinone oxidoreductase 13 kDa-B subunit [KO:K03949]",
                                100285005L,                                                 "NADH-ubiquinone oxidoreductase 13 kDa-B subunit [KO:K03949]",
                                100194239L,                                                                    "uncharacterized LOC100194239 [KO:K03950]",
                                103634509L,                                  "NADH dehydrogenase [ubiquinone] 1 alpha subcomplex subunit 8-B [KO:K03952]",
                                100382870L,                                  "NADH dehydrogenase [ubiquinone] 1 alpha subcomplex subunit 8-B [KO:K03952]",
                                100282690L,                                               "si946088d04; uncharacterized protein LOC100282690 [KO:K03953]",
                                100282353L,                                                 "pco080082; uncharacterized protein LOC100282353 [KO:K03955]",
                                100280460L,                                                                            "acyl carrier protein [KO:K03955]",
                                100283105L,                                                                            "acyl carrier protein [KO:K03955]",
                                100286300L,                                            "pco113177(131); uncharacterized protein LOC100286300 [KO:K03955]",
                                100192845L,                                                            "uncharacterized protein LOC100192845 [KO:K11352]",
                                100285721L,                                         "NADH-ubiquinone oxidoreductase subunit B17.2 isoform X1 [KO:K11352]",
                                100282224L,                                                    "NADH-ubiquinone oxidoreductase B16.6 subunit [KO:K11353]",
                                100282030L,                                                    "NADH-ubiquinone oxidoreductase B16.6 subunit [KO:K11353]",
                                100283425L,                                                   "umc1106; uncharacterized protein LOC100283425 [KO:K03963]",
                                100282004L,                                                 "NADH ubiquinone oxidoreductase B22-like subunit [KO:K03965]",
                                100276847L,                                                            "uncharacterized protein LOC100276847 [KO:K03966]",
                                100191555L,                                                 "pco107093; uncharacterized protein LOC100191555 [KO:K03966]",
                                100279930L,                                     "TIDP3457; uncharacterized protein LOC100279930 [KO:K00234] [EC:1.3.5.1]",
                                100280324L,                                               "uncharacterized protein LOC100280324 [KO:K00234] [EC:1.3.5.1]",
                                100216648L,                                "csu474(rpS14); uncharacterized protein LOC100216648 [KO:K00235] [EC:1.3.5.1]",
                                100502233L, "iron sulfur subunit of succinate dehydrogenase and ribosomal protein S14 Precursor [KO:K00235] [EC:1.3.5.1]",
                                103639746L,          "succinate dehydrogenase [ubiquinone] iron-sulfur subunit 1, mitochondrial [KO:K00235] [EC:1.3.5.1]",
                                103653238L,          "succinate dehydrogenase [ubiquinone] iron-sulfur subunit 1, mitochondrial [KO:K00235] [EC:1.3.5.1]",
                                109944584L,          "succinate dehydrogenase [ubiquinone] iron-sulfur subunit 2, mitochondrial [KO:K00235] [EC:1.3.5.1]",
                                100281355L,                                                  "TIDP2767; uncharacterized protein LOC100281355 [KO:K00236]",
                                100283351L,                                                               "succinate dehydrogenase subunit 3 [KO:K00236]",
                                100192538L,                              "cytochrome b-c1 complex subunit Rieske, mitochondrial [KO:K00411] [EC:7.1.1.8]",
                                100192468L,                                               "uncharacterized protein LOC100192468 [KO:K00411] [EC:7.1.1.8]",
                                   542550L,              "isp1; cytochrome b-c1 complex subunit Rieske, mitochondrial precursor [KO:K00411] [EC:7.1.1.8]",
                                100194151L,                                                                     "cytochrome c1, heme protein [KO:K00413]",
                                100384742L,                                                            "uncharacterized protein LOC100384742 [KO:K00413]",
                                100216756L,                                                            "uncharacterized protein LOC100216756 [KO:K00416]",
                                109945937L,                                                               "cytochrome b-c1 complex subunit 6 [KO:K00416]",
                                100383274L,                                                                    "uncharacterized LOC100383274 [KO:K00417]",
                                100384355L,                                                "ubiquinol-cytochrome c reductase complex protein [KO:K00417]",
                                103649017L,                                                               "cytochrome b-c1 complex subunit 7 [KO:K00417]",
                                100382609L,                                                                    "uncharacterized LOC100382609 [KO:K00419]",
                                100274057L,                                   "uncharacterized protein LOC100274057 precursor [KO:K02257] [EC:2.5.1.141]",
                                100193223L,                                           "uncharacterized LOC100193223 precursor [KO:K02257] [EC:2.5.1.141]",
                                100277708L,                                                            "uncharacterized protein LOC100277708 [KO:K02265]",
                                103651601L,                                                   "putative cytochrome c oxidase subunit 5b-like [KO:K02265]",
                                100280144L,                                                            "uncharacterized protein LOC100280144 [KO:K02265]",
                                103633853L,                                                "cytochrome c oxidase subunit 5b-1, mitochondrial [KO:K02265]",
                                100274770L,                                                            "uncharacterized protein LOC100274770 [KO:K02266]",
                                100272443L,                                                            "uncharacterized protein LOC100272443 [KO:K02267]",
                                100272303L,                                                               "cytochrome c oxidase subunit 6b-3 [KO:K02267]",
                                100192808L,                                                                    "uncharacterized LOC100192808 [KO:K02267]",
                                100384526L,                                                      "cytochrome c oxidase assembly protein ctaG [KO:K02258]",
                                100192499L,                                                                    "uncharacterized LOC100192499 [KO:K02259]",
                                   845169L,                                       "atpA; ATP synthase CF1 alpha subunit [KO:K02111] [EC:7.1.2.2 7.2.2.1]",
                                   845170L,                                        "atpB; ATP synthase CF1 beta subunit [KO:K02112] [EC:7.1.2.2 7.2.2.1]",
                                100284505L,                               "csu309(atpc); ATP synthase subunit gamma, chloroplastic precursor [KO:K02115]",
                                100281924L,                                                                        "ATP synthase delta chain [KO:K02113]",
                                103627748L,                                                         "ATP synthase delta chain, chloroplastic [KO:K02113]",
                                   845171L,                                                          "atpE; ATP synthase CF1 epsilon subunit [KO:K02114]",
                                   845174L,                                                                "atpI; ATP synthase CF0 A subunit [KO:K02108]",
                                100282566L,                                                "pco080036b; uncharacterized protein LOC100282566 [KO:K02109]",
                                   845172L,                                                                "atpF; ATP synthase CF0 B subunit [KO:K02109]",
                                   845173L,                                                                "atpH; ATP synthase CF0 C subunit [KO:K02110]",
                                103650219L,                                                       "ATP synthase subunit alpha, mitochondrial [KO:K02132]",
                                   542267L,                   "csu849(atpb); ATP synthase subunit beta, mitochondrial precursor [KO:K02133] [EC:7.1.2.2]",
                                100285442L,                                                            "ATP synthase beta chain [KO:K02133] [EC:7.1.2.2]",
                                103651342L,                                      "ATP synthase subunit beta, mitochondrial-like [KO:K02133] [EC:7.1.2.2]",
                                103632677L,                                                       "ATP synthase subunit gamma, mitochondrial [KO:K02136]",
                                100191627L,                                                      "csu657(atpd); uncharacterized LOC100191627 [KO:K02134]",
                                100281167L,                                                                        "ATP synthase delta chain [KO:K02134]",
                                   542589L,                                           "meatpase; ATP synthase subunit epsilon, mitochondrial [KO:K02135]",
                                100283198L,                                                "pco133953b; uncharacterized protein LOC100283198 [KO:K02135]",
                                100282837L,                                                 "pco090977; uncharacterized protein LOC100282837 [KO:K02137]",
                                109945965L,                                                            "uncharacterized protein LOC109945965 [KO:K02126]",
                                109939200L,                                                           "ATP synthase subunit d, mitochondrial [KO:K02138]",
                                100285174L,                                        "hydrogen-transporting ATP synthase, rotational mechanism [KO:K02140]",
                                100282853L,                                                 "cl48463_1; uncharacterized protein LOC100282853 [KO:K02140]",
                                100283237L,                                                 "pco141739; uncharacterized protein LOC100283237 [KO:K02140]",
                                   542327L,                                     "vpp3; V-type proton ATPase catalytic subunit A [KO:K02145] [EC:7.1.2.2]",
                                103626681L,                                      "V-type proton ATPase catalytic subunit A-like [KO:K02145] [EC:7.1.2.2]",
                                100282598L,                                                                 "vacuolar ATP synthase subunit B [KO:K02147]",
                                100286306L,                                                                 "vacuolar ATP synthase subunit B [KO:K02147]",
                                100274416L,                                                 "pco086632; uncharacterized protein LOC100274416 [KO:K02148]",
                                100281241L,                                            "pco086635(414); uncharacterized protein LOC100281241 [KO:K02148]",
                                100273246L,                                                                    "uncharacterized LOC100273246 [KO:K02148]",
                                100193069L,                                            "pco119646(297); uncharacterized protein LOC100193069 [KO:K02149]",
                                100283330L,                                                            "uncharacterized protein LOC100283330 [KO:K02149]",
                                103641218L,                                                                  "V-type proton ATPase subunit D [KO:K02149]",
                                100192065L,                                                "pco133385b; uncharacterized protein LOC100192065 [KO:K02150]",
                                100273417L,                                                            "uncharacterized protein LOC100273417 [KO:K02150]",
                                103651402L,                                                             "V-type proton ATPase subunit E-like [KO:K02150]",
                                100501172L,                                                            "uncharacterized protein LOC100501172 [KO:K02150]",
                                111589328L,                                                             "V-type proton ATPase subunit E-like [KO:K02150]",
                                100283102L,                                                                 "vacuolar ATP synthase subunit F [KO:K02151]",
                                100283470L,                                                                 "vacuolar ATP synthase subunit F [KO:K02151]",
                                100193359L,                                                                    "uncharacterized LOC100193359 [KO:K02152]",
                                100282544L,                                                                 "vacuolar ATP synthase subunit G [KO:K02152]",
                                100280574L,                                                               "vacuolar ATPase subunit H protein [KO:K02144]",
                                103647744L,                                                         "probable V-type proton ATPase subunit H [KO:K02144]",
                                100381766L,                                                                       "vacuolar proton ATPase a3 [KO:K02154]",
                                103636525L,                                                                 "V-type proton ATPase subunit a1 [KO:K02154]",
                                103639351L,                                                                 "V-type proton ATPase subunit a3 [KO:K02154]",
                                100280656L,                                                 "V-type proton ATPase 16 kDa proteolipid subunit [KO:K02155]",
                                100282032L,                                                "vacuolar ATP synthase 16 kDa proteolipid subunit [KO:K02155]",
                                100282817L,                                                "vacuolar ATP synthase 16 kDa proteolipid subunit [KO:K02155]",
                                   542441L,                                                 "V-type proton ATPase 16 kDa proteolipid subunit [KO:K02155]",
                                100283904L,                                                 "pco086002; uncharacterized protein LOC100283904 [KO:K02155]",
                                103652900L,                                      "V-type proton ATPase 16 kDa proteolipid subunit isoform X1 [KO:K02155]",
                                103651792L,                                                 "V-type proton ATPase 16 kDa proteolipid subunit [KO:K02155]",
                                109945683L,                                            "V-type proton ATPase 16 kDa proteolipid subunit-like [KO:K02155]",
                                100285771L,                                                 "pco096443; uncharacterized protein LOC100285771 [KO:K03661]",
                                100283291L,                                                "vacuolar ATP synthase 21 kDa proteolipid subunit [KO:K03661]",
                                100280083L,                                                            "uncharacterized protein LOC100280083 [KO:K02146]",
                                100275277L,                                                  "uncharacterized protein LOC100275277 precursor [KO:K02153]",
                                109942225L,                                                                 "V-type proton ATPase subunit e1 [KO:K02153]",
                                100383137L,                                                               "hypothetical protein [KO:K01535] [EC:7.1.2.1]",
                                100501185L,                                               "uncharacterized protein LOC100501185 [KO:K01535] [EC:7.1.2.1]",
                                100216710L,                                    "pco106203; uncharacterized protein LOC100216710 [KO:K01535] [EC:7.1.2.1]",
                                100383904L,                                                               "hypothetical protein [KO:K01535] [EC:7.1.2.1]",
                                   542303L,                                                       "mha1; membrane H(+)-ATPase 1 [KO:K01535] [EC:7.1.2.1]",
                                100502231L,                                               "uncharacterized protein LOC100502231 [KO:K01535] [EC:7.1.2.1]",
                                   542048L,                                                       "mha2; plasma membrane ATPase [KO:K01535] [EC:7.1.2.1]",
                                   542052L,                                                    "mha4; proton-exporting ATPase 4 [KO:K01535] [EC:7.1.2.1]",
                                100383411L,                         "si687062f01(361); ATPase 4 plasma membrane-type isoform X1 [KO:K01535] [EC:7.1.2.1]",
                                103629702L,                                                     "ATPase 2, plasma membrane-type [KO:K01535] [EC:7.1.2.1]",
                                103630585L,                                          "ATPase 9, plasma membrane-type isoform X1 [KO:K01535] [EC:7.1.2.1]",
                                103638716L,                                                           "plasma membrane ATPase 4 [KO:K01535] [EC:7.1.2.1]",
                                103632998L,                                                "plasma membrane ATPase 2 isoform X1 [KO:K01535] [EC:7.1.2.1]",
                                103631462L,                                                             "plasma membrane ATPase [KO:K01535] [EC:7.1.2.1]",
                                   541666L,                                       "pco131913; soluble inorganic pyrophosphatase [KO:K01507] [EC:3.6.1.1]",
                                100216827L,                                               "uncharacterized protein LOC100216827 [KO:K01507] [EC:3.6.1.1]",
                                100272302L,                                                  "soluble inorganic pyrophosphatase [KO:K01507] [EC:3.6.1.1]",
                                100272927L,                                        "ppp1; pyrophosphate-energized proton pump 1 [KO:K01507] [EC:3.6.1.1]",
                                100282768L,                                                  "soluble inorganic pyrophosphatase [KO:K01507] [EC:3.6.1.1]",
                                100284187L,                                      "gpm428a; uncharacterized protein LOC100284187 [KO:K01507] [EC:3.6.1.1]",
                                100383920L,                                               "uncharacterized protein LOC100383920 [KO:K01507] [EC:3.6.1.1]",
                                103626225L,                                                "soluble inorganic pyrophosphatase 1 [KO:K01507] [EC:3.6.1.1]"
                                )
```
```{r}
KEGG.OxPhos.Entrez <- as.character(KEGG.OxPhos.Tib$V1)
```
```{r}
KEGG.OxPhos.V4 <- mgdb$v4_gene_model[mgdb$Entrez%in%KEGG.OxPhos.Entrez]
```
```{r}
CornCycAerRespI.Tib <- tibble::tribble(
                         ~Gene.ID,  ~Gene.Accession,       ~Gene.name,                             ~Reaction.id, ~Reaction.EC,                                ~Enzymatic.activity, ~Evidence,
                      "GDQC-7770", "Zm00001d005695", "Zm00001d005695", "SUCCINATE-DEHYDROGENASE-UBIQUINONE-RXN",    "1.3.5.1",                 "succinate:quinone oxidoreductase", "EV-COMP",
                      "GDQC-7955", "Zm00001d045520", "Zm00001d045520", "SUCCINATE-DEHYDROGENASE-UBIQUINONE-RXN",    "1.3.5.1",                 "succinate:quinone oxidoreductase", "EV-COMP",
                      "GDQC-8018", "Zm00001d013966", "Zm00001d013966", "SUCCINATE-DEHYDROGENASE-UBIQUINONE-RXN",    "1.3.5.1",                 "succinate:quinone oxidoreductase", "EV-COMP",
                      "GDQC-8096", "Zm00001d032588", "Zm00001d032588", "SUCCINATE-DEHYDROGENASE-UBIQUINONE-RXN",    "1.3.5.1",                 "succinate:quinone oxidoreductase", "EV-COMP",
                      "GDQC-8158", "Zm00001d010491", "Zm00001d010491", "SUCCINATE-DEHYDROGENASE-UBIQUINONE-RXN",    "1.3.5.1",                 "succinate:quinone oxidoreductase", "EV-COMP",
                      "GDQC-8318", "Zm00001d013526", "Zm00001d013526", "SUCCINATE-DEHYDROGENASE-UBIQUINONE-RXN",    "1.3.5.1",                 "succinate:quinone oxidoreductase", "EV-COMP",
                      "GDQC-9088", "Zm00001d049691", "Zm00001d049691", "SUCCINATE-DEHYDROGENASE-UBIQUINONE-RXN",    "1.3.5.1",                 "succinate:quinone oxidoreductase", "EV-COMP",
                      "GDQC-9442", "Zm00001d020354", "Zm00001d020354", "SUCCINATE-DEHYDROGENASE-UBIQUINONE-RXN",    "1.3.5.1",                 "succinate:quinone oxidoreductase", "EV-COMP",
                     "GDQC-10013", "Zm00001d033554", "Zm00001d033554", "SUCCINATE-DEHYDROGENASE-UBIQUINONE-RXN",    "1.3.5.1",                 "succinate:quinone oxidoreductase", "EV-COMP",
                     "GDQC-11333", "Zm00001d035054", "Zm00001d035054", "SUCCINATE-DEHYDROGENASE-UBIQUINONE-RXN",    "1.3.5.1",                 "succinate:quinone oxidoreductase", "EV-COMP",
                     "GDQC-12265", "Zm00001d037899", "Zm00001d037899", "SUCCINATE-DEHYDROGENASE-UBIQUINONE-RXN",    "1.3.5.1",                 "succinate:quinone oxidoreductase", "EV-COMP",
                     "GDQC-14714", "Zm00001d005663", "Zm00001d005663", "SUCCINATE-DEHYDROGENASE-UBIQUINONE-RXN",    "1.3.5.1",                          "succinate dehydrogenase", "EV-COMP",
                     "GDQC-13911", "Zm00001d029316", "Zm00001d029316", "SUCCINATE-DEHYDROGENASE-UBIQUINONE-RXN",    "1.3.5.1",                          "succinate dehydrogenase", "EV-COMP",
                     "GDQC-13630", "Zm00001d029250", "Zm00001d029250", "SUCCINATE-DEHYDROGENASE-UBIQUINONE-RXN",    "1.3.5.1",                          "succinate dehydrogenase", "EV-COMP",
                     "GDQC-13185", "Zm00001d035136", "Zm00001d035136", "SUCCINATE-DEHYDROGENASE-UBIQUINONE-RXN",    "1.3.5.1",                          "succinate dehydrogenase", "EV-COMP",
                     "GDQC-13152", "Zm00001d038435", "Zm00001d038435", "SUCCINATE-DEHYDROGENASE-UBIQUINONE-RXN",    "1.3.5.1",                 "succinate:quinone oxidoreductase", "EV-COMP",
                     "GDQC-12667", "Zm00001d007966", "Zm00001d007966", "SUCCINATE-DEHYDROGENASE-UBIQUINONE-RXN",    "1.3.5.1",                          "succinate dehydrogenase", "EV-COMP",
                     "GDQC-12436", "Zm00001d014927", "Zm00001d014927", "SUCCINATE-DEHYDROGENASE-UBIQUINONE-RXN",    "1.3.5.1",                          "succinate dehydrogenase", "EV-COMP",
                     "GDQC-12094", "Zm00001d008855", "Zm00001d008855", "SUCCINATE-DEHYDROGENASE-UBIQUINONE-RXN",    "1.3.5.1",                 "succinate:quinone oxidoreductase", "EV-COMP",
                      "GDQC-9931", "Zm00001d049649", "Zm00001d049649", "SUCCINATE-DEHYDROGENASE-UBIQUINONE-RXN",    "1.3.5.1",                          "succinate dehydrogenase", "EV-COMP",
                      "GDQC-9553", "Zm00001d018758", "Zm00001d018758", "SUCCINATE-DEHYDROGENASE-UBIQUINONE-RXN",    "1.3.5.1",                          "succinate dehydrogenase", "EV-COMP",
                      "GDQC-9434", "Zm00001d038502", "Zm00001d038502", "SUCCINATE-DEHYDROGENASE-UBIQUINONE-RXN",    "1.3.5.1",                          "succinate dehydrogenase", "EV-COMP",
                      "GDQC-9381", "Zm00001d045148", "Zm00001d045148", "SUCCINATE-DEHYDROGENASE-UBIQUINONE-RXN",    "1.3.5.1",                          "succinate dehydrogenase", "EV-COMP",
                      "GDQC-7819", "Zm00001d025385", "Zm00001d025385", "SUCCINATE-DEHYDROGENASE-UBIQUINONE-RXN",    "1.3.5.1",                 "succinate:quinone oxidoreductase", "EV-COMP",
                      "GDQC-7642", "Zm00001d046083", "Zm00001d046083",                    "NADH-DEHYDROG-A-RXN",    "1.6.5.3", "NADH:ubiquinone oxidoreductase (H+-transporting)", "EV-COMP",
                      "GDQC-7715", "Zm00001d012599", "Zm00001d012599",                    "NADH-DEHYDROG-A-RXN",    "1.6.5.3", "NADH:ubiquinone oxidoreductase (H+-transporting)", "EV-COMP",
                      "GDQC-7803", "Zm00001d050384", "Zm00001d050384",                    "NADH-DEHYDROG-A-RXN",    "1.6.5.3", "NADH:ubiquinone oxidoreductase (H+-transporting)", "EV-COMP",
                      "GDQC-7811",  "GRMZM5G810298",  "GRMZM5G810298",                    "NADH-DEHYDROG-A-RXN",    "1.6.5.3", "NADH:ubiquinone oxidoreductase (H+-transporting)", "EV-COMP",
                      "GDQC-7818", "Zm00001d030138", "Zm00001d030138",                    "NADH-DEHYDROG-A-RXN",    "1.6.5.3", "NADH:ubiquinone oxidoreductase (H+-transporting)", "EV-COMP",
                      "GDQC-7876", "Zm00001d022324", "Zm00001d022324",                    "NADH-DEHYDROG-A-RXN",    "1.6.5.3", "NADH:ubiquinone oxidoreductase (H+-transporting)", "EV-COMP",
                      "GDQC-7926", "Zm00001d022341", "Zm00001d022341",                    "NADH-DEHYDROG-A-RXN",    "1.6.5.3", "NADH:ubiquinone oxidoreductase (H+-transporting)", "EV-COMP",
                      "GDQC-7930",  "GRMZM5G876106",  "GRMZM5G876106",                    "NADH-DEHYDROG-A-RXN",    "1.6.5.3", "NADH:ubiquinone oxidoreductase (H+-transporting)", "EV-COMP",
                      "GDQC-8001", "Zm00001d019605", "Zm00001d019605",                    "NADH-DEHYDROG-A-RXN",    "1.6.5.3", "NADH:ubiquinone oxidoreductase (H+-transporting)", "EV-COMP",
                      "GDQC-8044",  "GRMZM5G800980",  "GRMZM5G800980",                    "NADH-DEHYDROG-A-RXN",    "1.6.5.3", "NADH:ubiquinone oxidoreductase (H+-transporting)", "EV-COMP",
                      "GDQC-8072",  "GRMZM5G851708",  "GRMZM5G851708",                    "NADH-DEHYDROG-A-RXN",    "1.6.5.3", "NADH:ubiquinone oxidoreductase (H+-transporting)", "EV-COMP",
                      "GDQC-8084", "Zm00001d007048", "Zm00001d007048",                    "NADH-DEHYDROG-A-RXN",    "1.6.5.3", "NADH:ubiquinone oxidoreductase (H+-transporting)", "EV-COMP",
                      "GDQC-8179", "Zm00001d012742", "Zm00001d012742",                    "NADH-DEHYDROG-A-RXN",    "1.6.5.3", "NADH:ubiquinone oxidoreductase (H+-transporting)", "EV-COMP",
                      "GDQC-8190",  "GRMZM5G856027",  "GRMZM5G856027",                    "NADH-DEHYDROG-A-RXN",    "1.6.5.3", "NADH:ubiquinone oxidoreductase (H+-transporting)", "EV-COMP",
                      "GDQC-8324", "Zm00001d042061", "Zm00001d042061",                    "NADH-DEHYDROG-A-RXN",    "1.6.5.3", "NADH:ubiquinone oxidoreductase (H+-transporting)", "EV-COMP",
                      "GDQC-8488", "Zm00001d043620", "Zm00001d043620",                    "NADH-DEHYDROG-A-RXN",    "1.6.5.3", "NADH:ubiquinone oxidoreductase (H+-transporting)", "EV-COMP",
                      "GDQC-8628", "Zm00001d033531", "Zm00001d033531",                    "NADH-DEHYDROG-A-RXN",    "1.6.5.3", "NADH:ubiquinone oxidoreductase (H+-transporting)", "EV-COMP",
                      "GDQC-8828", "Zm00001d011845", "Zm00001d011845",                    "NADH-DEHYDROG-A-RXN",    "1.6.5.3", "NADH:ubiquinone oxidoreductase (H+-transporting)", "EV-COMP",
                      "GDQC-9059", "Zm00001d043224", "Zm00001d043224",                    "NADH-DEHYDROG-A-RXN",    "1.6.5.3", "NADH:ubiquinone oxidoreductase (H+-transporting)", "EV-COMP",
                      "GDQC-9143", "Zm00001d028918", "Zm00001d028918",                    "NADH-DEHYDROG-A-RXN",    "1.6.5.3", "NADH:ubiquinone oxidoreductase (H+-transporting)", "EV-COMP",
                      "GDQC-9148",  "GRMZM5G876913",  "GRMZM5G876913",                    "NADH-DEHYDROG-A-RXN",    "1.6.5.3", "NADH:ubiquinone oxidoreductase (H+-transporting)", "EV-COMP",
                      "GDQC-9316", "Zm00001d021958", "Zm00001d021958",                    "NADH-DEHYDROG-A-RXN",    "1.6.5.3", "NADH:ubiquinone oxidoreductase (H+-transporting)", "EV-COMP",
                      "GDQC-9414", "Zm00001d043619", "Zm00001d043619",                    "NADH-DEHYDROG-A-RXN",    "1.6.5.3", "NADH:ubiquinone oxidoreductase (H+-transporting)", "EV-COMP",
                      "GDQC-9512", "Zm00001d033753", "Zm00001d033753",                    "NADH-DEHYDROG-A-RXN",    "1.6.5.3", "NADH:ubiquinone oxidoreductase (H+-transporting)", "EV-COMP",
                      "GDQC-9798", "Zm00001d038508", "Zm00001d038508",                    "NADH-DEHYDROG-A-RXN",    "1.6.5.3", "NADH:ubiquinone oxidoreductase (H+-transporting)", "EV-COMP",
                      "GDQC-9807", "Zm00001d006421", "Zm00001d006421",                    "NADH-DEHYDROG-A-RXN",    "1.6.5.3", "NADH:ubiquinone oxidoreductase (H+-transporting)", "EV-COMP",
                     "GDQC-10140", "Zm00001d031975", "Zm00001d031975",                    "NADH-DEHYDROG-A-RXN",    "1.6.5.3", "NADH:ubiquinone oxidoreductase (H+-transporting)", "EV-COMP",
                     "GDQC-10308", "Zm00001d034244", "Zm00001d034244",                    "NADH-DEHYDROG-A-RXN",    "1.6.5.3", "NADH:ubiquinone oxidoreductase (H+-transporting)", "EV-COMP",
                     "GDQC-10380", "Zm00001d012741", "Zm00001d012741",                    "NADH-DEHYDROG-A-RXN",    "1.6.5.3", "NADH:ubiquinone oxidoreductase (H+-transporting)", "EV-COMP",
                     "GDQC-10398", "Zm00001d016705", "Zm00001d016705",                    "NADH-DEHYDROG-A-RXN",    "1.6.5.3", "NADH:ubiquinone oxidoreductase (H+-transporting)", "EV-COMP",
                     "GDQC-10504",  "GRMZM5G834128",  "GRMZM5G834128",                    "NADH-DEHYDROG-A-RXN",    "1.6.5.3", "NADH:ubiquinone oxidoreductase (H+-transporting)", "EV-COMP",
                     "GDQC-10787", "Zm00001d031431", "Zm00001d031431",                    "NADH-DEHYDROG-A-RXN",    "1.6.5.3", "NADH:ubiquinone oxidoreductase (H+-transporting)", "EV-COMP",
                     "GDQC-10816", "Zm00001d028181", "Zm00001d028181",                    "NADH-DEHYDROG-A-RXN",    "1.6.5.3", "NADH:ubiquinone oxidoreductase (H+-transporting)", "EV-COMP",
                     "GDQC-11057", "Zm00001d008215", "Zm00001d008215",                    "NADH-DEHYDROG-A-RXN",    "1.6.5.3", "NADH:ubiquinone oxidoreductase (H+-transporting)", "EV-COMP",
                     "GDQC-11141", "Zm00001d027389", "Zm00001d027389",                    "NADH-DEHYDROG-A-RXN",    "1.6.5.3", "NADH:ubiquinone oxidoreductase (H+-transporting)", "EV-COMP",
                     "GDQC-11406",  "GRMZM5G839924",  "GRMZM5G839924",                    "NADH-DEHYDROG-A-RXN",    "1.6.5.3", "NADH:ubiquinone oxidoreductase (H+-transporting)", "EV-COMP",
                     "GDQC-11550", "Zm00001d025423", "Zm00001d025423",                    "NADH-DEHYDROG-A-RXN",    "1.6.5.3", "NADH:ubiquinone oxidoreductase (H+-transporting)", "EV-COMP",
                     "GDQC-11822", "Zm00001d027493", "Zm00001d027493",                    "NADH-DEHYDROG-A-RXN",    "1.6.5.3", "NADH:ubiquinone oxidoreductase (H+-transporting)", "EV-COMP",
                     "GDQC-11838", "Zm00001d048204", "Zm00001d048204",                    "NADH-DEHYDROG-A-RXN",    "1.6.5.3", "NADH:ubiquinone oxidoreductase (H+-transporting)", "EV-COMP",
                     "GDQC-11971", "Zm00001d006714", "Zm00001d006714",                    "NADH-DEHYDROG-A-RXN",    "1.6.5.3", "NADH:ubiquinone oxidoreductase (H+-transporting)", "EV-COMP",
                     "GDQC-11997",  "GRMZM5G866223",  "GRMZM5G866223",                    "NADH-DEHYDROG-A-RXN",    "1.6.5.3", "NADH:ubiquinone oxidoreductase (H+-transporting)", "EV-COMP",
                     "GDQC-12325", "Zm00001d031940", "Zm00001d031940",                    "NADH-DEHYDROG-A-RXN",    "1.6.5.3", "NADH:ubiquinone oxidoreductase (H+-transporting)", "EV-COMP",
                     "GDQC-12385",  "GRMZM5G854065",  "GRMZM5G854065",                    "NADH-DEHYDROG-A-RXN",    "1.6.5.3", "NADH:ubiquinone oxidoreductase (H+-transporting)", "EV-COMP",
                     "GDQC-12456", "Zm00001d016566", "Zm00001d016566",                    "NADH-DEHYDROG-A-RXN",    "1.6.5.3", "NADH:ubiquinone oxidoreductase (H+-transporting)", "EV-COMP",
                     "GDQC-12457", "Zm00001d011732", "Zm00001d011732",                    "NADH-DEHYDROG-A-RXN",    "1.6.5.3", "NADH:ubiquinone oxidoreductase (H+-transporting)", "EV-COMP",
                     "GDQC-12465", "Zm00001d014428", "Zm00001d014428",                    "NADH-DEHYDROG-A-RXN",    "1.6.5.3", "NADH:ubiquinone oxidoreductase (H+-transporting)", "EV-COMP",
                     "GDQC-12515",  "GRMZM5G864407",  "GRMZM5G864407",                    "NADH-DEHYDROG-A-RXN",    "1.6.5.3", "NADH:ubiquinone oxidoreductase (H+-transporting)", "EV-COMP",
                     "GDQC-12527", "Zm00001d035824", "Zm00001d035824",                    "NADH-DEHYDROG-A-RXN",    "1.6.5.3", "NADH:ubiquinone oxidoreductase (H+-transporting)", "EV-COMP",
                     "GDQC-12528",  "GRMZM5G800096",  "GRMZM5G800096",                    "NADH-DEHYDROG-A-RXN",    "1.6.5.3", "NADH:ubiquinone oxidoreductase (H+-transporting)", "EV-COMP",
                     "GDQC-12621", "Zm00001d047441", "Zm00001d047441",                    "NADH-DEHYDROG-A-RXN",    "1.6.5.3", "NADH:ubiquinone oxidoreductase (H+-transporting)", "EV-COMP",
                     "GDQC-12714", "Zm00001d038658", "Zm00001d038658",                    "NADH-DEHYDROG-A-RXN",    "1.6.5.3", "NADH:ubiquinone oxidoreductase (H+-transporting)", "EV-COMP",
                     "GDQC-13003", "Zm00001d038441", "Zm00001d038441",                    "NADH-DEHYDROG-A-RXN",    "1.6.5.3", "NADH:ubiquinone oxidoreductase (H+-transporting)", "EV-COMP",
                     "GDQC-13045",  "GRMZM5G800101",  "GRMZM5G800101",                    "NADH-DEHYDROG-A-RXN",    "1.6.5.3", "NADH:ubiquinone oxidoreductase (H+-transporting)", "EV-COMP",
                     "GDQC-13070", "Zm00001d021536", "Zm00001d021536",                    "NADH-DEHYDROG-A-RXN",    "1.6.5.3", "NADH:ubiquinone oxidoreductase (H+-transporting)", "EV-COMP",
                     "GDQC-13127",  "GRMZM5G804358",  "GRMZM5G804358",                    "NADH-DEHYDROG-A-RXN",    "1.6.5.3", "NADH:ubiquinone oxidoreductase (H+-transporting)", "EV-COMP",
                     "GDQC-13187", "Zm00001d012469", "Zm00001d012469",                    "NADH-DEHYDROG-A-RXN",    "1.6.5.3", "NADH:ubiquinone oxidoreductase (H+-transporting)", "EV-COMP",
                     "GDQC-13403", "Zm00001d013156", "Zm00001d013156",                    "NADH-DEHYDROG-A-RXN",    "1.6.5.3", "NADH:ubiquinone oxidoreductase (H+-transporting)", "EV-COMP",
                     "GDQC-13497", "Zm00001d046260", "Zm00001d046260",                    "NADH-DEHYDROG-A-RXN",    "1.6.5.3", "NADH:ubiquinone oxidoreductase (H+-transporting)", "EV-COMP",
                     "GDQC-13999", "Zm00001d013424", "Zm00001d013424",                    "NADH-DEHYDROG-A-RXN",    "1.6.5.3", "NADH:ubiquinone oxidoreductase (H+-transporting)", "EV-COMP",
                     "GDQC-14251",  "GRMZM5G887911",  "GRMZM5G887911",                    "NADH-DEHYDROG-A-RXN",    "1.6.5.3", "NADH:ubiquinone oxidoreductase (H+-transporting)", "EV-COMP",
                     "GDQC-14301", "Zm00001d033530", "Zm00001d033530",                    "NADH-DEHYDROG-A-RXN",    "1.6.5.3", "NADH:ubiquinone oxidoreductase (H+-transporting)", "EV-COMP",
                     "GDQC-14435", "Zm00001d040515", "Zm00001d040515",                    "NADH-DEHYDROG-A-RXN",    "1.6.5.3", "NADH:ubiquinone oxidoreductase (H+-transporting)", "EV-COMP",
                     "GDQC-14466", "Zm00001d020930", "Zm00001d020930",                    "NADH-DEHYDROG-A-RXN",    "1.6.5.3", "NADH:ubiquinone oxidoreductase (H+-transporting)", "EV-COMP",
                     "GDQC-14513", "Zm00001d040819", "Zm00001d040819",                    "NADH-DEHYDROG-A-RXN",    "1.6.5.3", "NADH:ubiquinone oxidoreductase (H+-transporting)", "EV-COMP",
                     "GDQC-14576", "Zm00001d018479", "Zm00001d018479",                    "NADH-DEHYDROG-A-RXN",    "1.6.5.3", "NADH:ubiquinone oxidoreductase (H+-transporting)", "EV-COMP",
                     "GDQC-14831",  "GRMZM5G874448",  "GRMZM5G874448",                    "NADH-DEHYDROG-A-RXN",    "1.6.5.3", "NADH:ubiquinone oxidoreductase (H+-transporting)", "EV-COMP",
                     "GDQC-14957",  "GRMZM5G897755",  "GRMZM5G897755",                    "NADH-DEHYDROG-A-RXN",    "1.6.5.3", "NADH:ubiquinone oxidoreductase (H+-transporting)", "EV-COMP",
                      "GDQC-8836", "Zm00001d007854", "Zm00001d007854",                           "1.10.2.2-RXN",   "1.10.2.2",            "ubiquinol-cytochrome C oxidoreductase", "EV-COMP",
                     "GDQC-11343",  "GRMZM5G884707",  "GRMZM5G884707",                           "1.10.2.2-RXN",   "1.10.2.2",            "ubiquinol-cytochrome C oxidoreductase", "EV-COMP",
                     "GDQC-11624", "Zm00001d011706", "Zm00001d011706",                           "1.10.2.2-RXN",   "1.10.2.2",            "ubiquinol-cytochrome C oxidoreductase", "EV-COMP",
                     "GDQC-11783", "Zm00001d028194", "Zm00001d028194",                           "1.10.2.2-RXN",   "1.10.2.2",            "ubiquinol-cytochrome C oxidoreductase", "EV-COMP",
                     "GDQC-11991", "Zm00001d007245", "Zm00001d007245",                           "1.10.2.2-RXN",   "1.10.2.2",            "ubiquinol-cytochrome C oxidoreductase", "EV-COMP",
                     "GDQC-12007", "Zm00001d019992", "Zm00001d019992",                           "1.10.2.2-RXN",   "1.10.2.2",            "ubiquinol-cytochrome C oxidoreductase", "EV-COMP",
                     "GDQC-12142", "Zm00001d048102", "Zm00001d048102",                           "1.10.2.2-RXN",   "1.10.2.2",            "ubiquinol-cytochrome C oxidoreductase", "EV-COMP",
                     "GDQC-10407", "Zm00001d034218", "Zm00001d034218",               "CYTOCHROME-C-OXIDASE-RXN",    "1.9.3.1",                             "cytochrome c oxidase", "EV-COMP",
                     "GDQC-10798", "Zm00001d009727", "Zm00001d009727",               "CYTOCHROME-C-OXIDASE-RXN",    "1.9.3.1",                             "cytochrome c oxidase", "EV-COMP",
                     "GDQC-10845",  "GRMZM5G862955",  "GRMZM5G862955",               "CYTOCHROME-C-OXIDASE-RXN",    "1.9.3.1",                             "cytochrome c oxidase", "EV-COMP",
                     "GDQC-10913",  "GRMZM5G884538",  "GRMZM5G884538",               "CYTOCHROME-C-OXIDASE-RXN",    "1.9.3.1",                             "cytochrome c oxidase", "EV-COMP",
                     "GDQC-11087", "Zm00001d051055", "Zm00001d051055",               "CYTOCHROME-C-OXIDASE-RXN",    "1.9.3.1",                             "cytochrome c oxidase", "EV-COMP",
                     "GDQC-11121", "Zm00001d029393", "Zm00001d029393",               "CYTOCHROME-C-OXIDASE-RXN",    "1.9.3.1",                             "cytochrome c oxidase", "EV-COMP",
                     "GDQC-11211", "Zm00001d047218", "Zm00001d047218",               "CYTOCHROME-C-OXIDASE-RXN",    "1.9.3.1",                             "cytochrome c oxidase", "EV-COMP",
                     "GDQC-11963", "Zm00001d004057", "Zm00001d004057",               "CYTOCHROME-C-OXIDASE-RXN",    "1.9.3.1",                             "cytochrome c oxidase", "EV-COMP",
                     "GDQC-12521", "Zm00001d011330", "Zm00001d011330",               "CYTOCHROME-C-OXIDASE-RXN",    "1.9.3.1",                             "cytochrome c oxidase", "EV-COMP",
                     "GDQC-14076", "Zm00001d044459", "Zm00001d044459",               "CYTOCHROME-C-OXIDASE-RXN",    "1.9.3.1",                             "cytochrome c oxidase", "EV-COMP"
                     )

```
```{r}
CornCycAerRespI <- as.character(CornCycAerRespI.Tib$Gene.Accession)
```
```{r}
CornCycAerRespIII.Tib <- tibble::tribble(
                           ~Gene.ID,  ~Gene.Accession,       ~Gene.name,                             ~Reaction.id, ~Reaction.EC,                                ~Enzymatic.activity, ~Evidence,
                        "GDQC-7770", "Zm00001d005695", "Zm00001d005695", "SUCCINATE-DEHYDROGENASE-UBIQUINONE-RXN",    "1.3.5.1",                 "succinate:quinone oxidoreductase", "EV-COMP",
                        "GDQC-7955", "Zm00001d045520", "Zm00001d045520", "SUCCINATE-DEHYDROGENASE-UBIQUINONE-RXN",    "1.3.5.1",                 "succinate:quinone oxidoreductase", "EV-COMP",
                        "GDQC-8018", "Zm00001d013966", "Zm00001d013966", "SUCCINATE-DEHYDROGENASE-UBIQUINONE-RXN",    "1.3.5.1",                 "succinate:quinone oxidoreductase", "EV-COMP",
                        "GDQC-8096", "Zm00001d032588", "Zm00001d032588", "SUCCINATE-DEHYDROGENASE-UBIQUINONE-RXN",    "1.3.5.1",                 "succinate:quinone oxidoreductase", "EV-COMP",
                        "GDQC-8158", "Zm00001d010491", "Zm00001d010491", "SUCCINATE-DEHYDROGENASE-UBIQUINONE-RXN",    "1.3.5.1",                 "succinate:quinone oxidoreductase", "EV-COMP",
                        "GDQC-8318", "Zm00001d013526", "Zm00001d013526", "SUCCINATE-DEHYDROGENASE-UBIQUINONE-RXN",    "1.3.5.1",                 "succinate:quinone oxidoreductase", "EV-COMP",
                        "GDQC-9088", "Zm00001d049691", "Zm00001d049691", "SUCCINATE-DEHYDROGENASE-UBIQUINONE-RXN",    "1.3.5.1",                 "succinate:quinone oxidoreductase", "EV-COMP",
                        "GDQC-9442", "Zm00001d020354", "Zm00001d020354", "SUCCINATE-DEHYDROGENASE-UBIQUINONE-RXN",    "1.3.5.1",                 "succinate:quinone oxidoreductase", "EV-COMP",
                       "GDQC-10013", "Zm00001d033554", "Zm00001d033554", "SUCCINATE-DEHYDROGENASE-UBIQUINONE-RXN",    "1.3.5.1",                 "succinate:quinone oxidoreductase", "EV-COMP",
                       "GDQC-11333", "Zm00001d035054", "Zm00001d035054", "SUCCINATE-DEHYDROGENASE-UBIQUINONE-RXN",    "1.3.5.1",                 "succinate:quinone oxidoreductase", "EV-COMP",
                       "GDQC-12265", "Zm00001d037899", "Zm00001d037899", "SUCCINATE-DEHYDROGENASE-UBIQUINONE-RXN",    "1.3.5.1",                 "succinate:quinone oxidoreductase", "EV-COMP",
                       "GDQC-14714", "Zm00001d005663", "Zm00001d005663", "SUCCINATE-DEHYDROGENASE-UBIQUINONE-RXN",    "1.3.5.1",                          "succinate dehydrogenase", "EV-COMP",
                       "GDQC-13911", "Zm00001d029316", "Zm00001d029316", "SUCCINATE-DEHYDROGENASE-UBIQUINONE-RXN",    "1.3.5.1",                          "succinate dehydrogenase", "EV-COMP",
                       "GDQC-13630", "Zm00001d029250", "Zm00001d029250", "SUCCINATE-DEHYDROGENASE-UBIQUINONE-RXN",    "1.3.5.1",                          "succinate dehydrogenase", "EV-COMP",
                       "GDQC-13185", "Zm00001d035136", "Zm00001d035136", "SUCCINATE-DEHYDROGENASE-UBIQUINONE-RXN",    "1.3.5.1",                          "succinate dehydrogenase", "EV-COMP",
                       "GDQC-13152", "Zm00001d038435", "Zm00001d038435", "SUCCINATE-DEHYDROGENASE-UBIQUINONE-RXN",    "1.3.5.1",                 "succinate:quinone oxidoreductase", "EV-COMP",
                       "GDQC-12667", "Zm00001d007966", "Zm00001d007966", "SUCCINATE-DEHYDROGENASE-UBIQUINONE-RXN",    "1.3.5.1",                          "succinate dehydrogenase", "EV-COMP",
                       "GDQC-12436", "Zm00001d014927", "Zm00001d014927", "SUCCINATE-DEHYDROGENASE-UBIQUINONE-RXN",    "1.3.5.1",                          "succinate dehydrogenase", "EV-COMP",
                       "GDQC-12094", "Zm00001d008855", "Zm00001d008855", "SUCCINATE-DEHYDROGENASE-UBIQUINONE-RXN",    "1.3.5.1",                 "succinate:quinone oxidoreductase", "EV-COMP",
                        "GDQC-9931", "Zm00001d049649", "Zm00001d049649", "SUCCINATE-DEHYDROGENASE-UBIQUINONE-RXN",    "1.3.5.1",                          "succinate dehydrogenase", "EV-COMP",
                        "GDQC-9553", "Zm00001d018758", "Zm00001d018758", "SUCCINATE-DEHYDROGENASE-UBIQUINONE-RXN",    "1.3.5.1",                          "succinate dehydrogenase", "EV-COMP",
                        "GDQC-9434", "Zm00001d038502", "Zm00001d038502", "SUCCINATE-DEHYDROGENASE-UBIQUINONE-RXN",    "1.3.5.1",                          "succinate dehydrogenase", "EV-COMP",
                        "GDQC-9381", "Zm00001d045148", "Zm00001d045148", "SUCCINATE-DEHYDROGENASE-UBIQUINONE-RXN",    "1.3.5.1",                          "succinate dehydrogenase", "EV-COMP",
                        "GDQC-7819", "Zm00001d025385", "Zm00001d025385", "SUCCINATE-DEHYDROGENASE-UBIQUINONE-RXN",    "1.3.5.1",                 "succinate:quinone oxidoreductase", "EV-COMP",
                        "GDQC-7642", "Zm00001d046083", "Zm00001d046083",                    "NADH-DEHYDROG-A-RXN",    "1.6.5.3", "NADH:ubiquinone oxidoreductase (H+-transporting)", "EV-COMP",
                        "GDQC-7715", "Zm00001d012599", "Zm00001d012599",                    "NADH-DEHYDROG-A-RXN",    "1.6.5.3", "NADH:ubiquinone oxidoreductase (H+-transporting)", "EV-COMP",
                        "GDQC-7803", "Zm00001d050384", "Zm00001d050384",                    "NADH-DEHYDROG-A-RXN",    "1.6.5.3", "NADH:ubiquinone oxidoreductase (H+-transporting)", "EV-COMP",
                        "GDQC-7811",  "GRMZM5G810298",  "GRMZM5G810298",                    "NADH-DEHYDROG-A-RXN",    "1.6.5.3", "NADH:ubiquinone oxidoreductase (H+-transporting)", "EV-COMP",
                        "GDQC-7818", "Zm00001d030138", "Zm00001d030138",                    "NADH-DEHYDROG-A-RXN",    "1.6.5.3", "NADH:ubiquinone oxidoreductase (H+-transporting)", "EV-COMP",
                        "GDQC-7876", "Zm00001d022324", "Zm00001d022324",                    "NADH-DEHYDROG-A-RXN",    "1.6.5.3", "NADH:ubiquinone oxidoreductase (H+-transporting)", "EV-COMP",
                        "GDQC-7926", "Zm00001d022341", "Zm00001d022341",                    "NADH-DEHYDROG-A-RXN",    "1.6.5.3", "NADH:ubiquinone oxidoreductase (H+-transporting)", "EV-COMP",
                        "GDQC-7930",  "GRMZM5G876106",  "GRMZM5G876106",                    "NADH-DEHYDROG-A-RXN",    "1.6.5.3", "NADH:ubiquinone oxidoreductase (H+-transporting)", "EV-COMP",
                        "GDQC-8001", "Zm00001d019605", "Zm00001d019605",                    "NADH-DEHYDROG-A-RXN",    "1.6.5.3", "NADH:ubiquinone oxidoreductase (H+-transporting)", "EV-COMP",
                        "GDQC-8044",  "GRMZM5G800980",  "GRMZM5G800980",                    "NADH-DEHYDROG-A-RXN",    "1.6.5.3", "NADH:ubiquinone oxidoreductase (H+-transporting)", "EV-COMP",
                        "GDQC-8072",  "GRMZM5G851708",  "GRMZM5G851708",                    "NADH-DEHYDROG-A-RXN",    "1.6.5.3", "NADH:ubiquinone oxidoreductase (H+-transporting)", "EV-COMP",
                        "GDQC-8084", "Zm00001d007048", "Zm00001d007048",                    "NADH-DEHYDROG-A-RXN",    "1.6.5.3", "NADH:ubiquinone oxidoreductase (H+-transporting)", "EV-COMP",
                        "GDQC-8179", "Zm00001d012742", "Zm00001d012742",                    "NADH-DEHYDROG-A-RXN",    "1.6.5.3", "NADH:ubiquinone oxidoreductase (H+-transporting)", "EV-COMP",
                        "GDQC-8190",  "GRMZM5G856027",  "GRMZM5G856027",                    "NADH-DEHYDROG-A-RXN",    "1.6.5.3", "NADH:ubiquinone oxidoreductase (H+-transporting)", "EV-COMP",
                        "GDQC-8324", "Zm00001d042061", "Zm00001d042061",                    "NADH-DEHYDROG-A-RXN",    "1.6.5.3", "NADH:ubiquinone oxidoreductase (H+-transporting)", "EV-COMP",
                        "GDQC-8488", "Zm00001d043620", "Zm00001d043620",                    "NADH-DEHYDROG-A-RXN",    "1.6.5.3", "NADH:ubiquinone oxidoreductase (H+-transporting)", "EV-COMP",
                        "GDQC-8628", "Zm00001d033531", "Zm00001d033531",                    "NADH-DEHYDROG-A-RXN",    "1.6.5.3", "NADH:ubiquinone oxidoreductase (H+-transporting)", "EV-COMP",
                        "GDQC-8828", "Zm00001d011845", "Zm00001d011845",                    "NADH-DEHYDROG-A-RXN",    "1.6.5.3", "NADH:ubiquinone oxidoreductase (H+-transporting)", "EV-COMP",
                        "GDQC-9059", "Zm00001d043224", "Zm00001d043224",                    "NADH-DEHYDROG-A-RXN",    "1.6.5.3", "NADH:ubiquinone oxidoreductase (H+-transporting)", "EV-COMP",
                        "GDQC-9143", "Zm00001d028918", "Zm00001d028918",                    "NADH-DEHYDROG-A-RXN",    "1.6.5.3", "NADH:ubiquinone oxidoreductase (H+-transporting)", "EV-COMP",
                        "GDQC-9148",  "GRMZM5G876913",  "GRMZM5G876913",                    "NADH-DEHYDROG-A-RXN",    "1.6.5.3", "NADH:ubiquinone oxidoreductase (H+-transporting)", "EV-COMP",
                        "GDQC-9316", "Zm00001d021958", "Zm00001d021958",                    "NADH-DEHYDROG-A-RXN",    "1.6.5.3", "NADH:ubiquinone oxidoreductase (H+-transporting)", "EV-COMP",
                        "GDQC-9414", "Zm00001d043619", "Zm00001d043619",                    "NADH-DEHYDROG-A-RXN",    "1.6.5.3", "NADH:ubiquinone oxidoreductase (H+-transporting)", "EV-COMP",
                        "GDQC-9512", "Zm00001d033753", "Zm00001d033753",                    "NADH-DEHYDROG-A-RXN",    "1.6.5.3", "NADH:ubiquinone oxidoreductase (H+-transporting)", "EV-COMP",
                        "GDQC-9798", "Zm00001d038508", "Zm00001d038508",                    "NADH-DEHYDROG-A-RXN",    "1.6.5.3", "NADH:ubiquinone oxidoreductase (H+-transporting)", "EV-COMP",
                        "GDQC-9807", "Zm00001d006421", "Zm00001d006421",                    "NADH-DEHYDROG-A-RXN",    "1.6.5.3", "NADH:ubiquinone oxidoreductase (H+-transporting)", "EV-COMP",
                       "GDQC-10140", "Zm00001d031975", "Zm00001d031975",                    "NADH-DEHYDROG-A-RXN",    "1.6.5.3", "NADH:ubiquinone oxidoreductase (H+-transporting)", "EV-COMP",
                       "GDQC-10308", "Zm00001d034244", "Zm00001d034244",                    "NADH-DEHYDROG-A-RXN",    "1.6.5.3", "NADH:ubiquinone oxidoreductase (H+-transporting)", "EV-COMP",
                       "GDQC-10380", "Zm00001d012741", "Zm00001d012741",                    "NADH-DEHYDROG-A-RXN",    "1.6.5.3", "NADH:ubiquinone oxidoreductase (H+-transporting)", "EV-COMP",
                       "GDQC-10398", "Zm00001d016705", "Zm00001d016705",                    "NADH-DEHYDROG-A-RXN",    "1.6.5.3", "NADH:ubiquinone oxidoreductase (H+-transporting)", "EV-COMP",
                       "GDQC-10504",  "GRMZM5G834128",  "GRMZM5G834128",                    "NADH-DEHYDROG-A-RXN",    "1.6.5.3", "NADH:ubiquinone oxidoreductase (H+-transporting)", "EV-COMP",
                       "GDQC-10787", "Zm00001d031431", "Zm00001d031431",                    "NADH-DEHYDROG-A-RXN",    "1.6.5.3", "NADH:ubiquinone oxidoreductase (H+-transporting)", "EV-COMP",
                       "GDQC-10816", "Zm00001d028181", "Zm00001d028181",                    "NADH-DEHYDROG-A-RXN",    "1.6.5.3", "NADH:ubiquinone oxidoreductase (H+-transporting)", "EV-COMP",
                       "GDQC-11057", "Zm00001d008215", "Zm00001d008215",                    "NADH-DEHYDROG-A-RXN",    "1.6.5.3", "NADH:ubiquinone oxidoreductase (H+-transporting)", "EV-COMP",
                       "GDQC-11141", "Zm00001d027389", "Zm00001d027389",                    "NADH-DEHYDROG-A-RXN",    "1.6.5.3", "NADH:ubiquinone oxidoreductase (H+-transporting)", "EV-COMP",
                       "GDQC-11406",  "GRMZM5G839924",  "GRMZM5G839924",                    "NADH-DEHYDROG-A-RXN",    "1.6.5.3", "NADH:ubiquinone oxidoreductase (H+-transporting)", "EV-COMP",
                       "GDQC-11550", "Zm00001d025423", "Zm00001d025423",                    "NADH-DEHYDROG-A-RXN",    "1.6.5.3", "NADH:ubiquinone oxidoreductase (H+-transporting)", "EV-COMP",
                       "GDQC-11822", "Zm00001d027493", "Zm00001d027493",                    "NADH-DEHYDROG-A-RXN",    "1.6.5.3", "NADH:ubiquinone oxidoreductase (H+-transporting)", "EV-COMP",
                       "GDQC-11838", "Zm00001d048204", "Zm00001d048204",                    "NADH-DEHYDROG-A-RXN",    "1.6.5.3", "NADH:ubiquinone oxidoreductase (H+-transporting)", "EV-COMP",
                       "GDQC-11971", "Zm00001d006714", "Zm00001d006714",                    "NADH-DEHYDROG-A-RXN",    "1.6.5.3", "NADH:ubiquinone oxidoreductase (H+-transporting)", "EV-COMP",
                       "GDQC-11997",  "GRMZM5G866223",  "GRMZM5G866223",                    "NADH-DEHYDROG-A-RXN",    "1.6.5.3", "NADH:ubiquinone oxidoreductase (H+-transporting)", "EV-COMP",
                       "GDQC-12325", "Zm00001d031940", "Zm00001d031940",                    "NADH-DEHYDROG-A-RXN",    "1.6.5.3", "NADH:ubiquinone oxidoreductase (H+-transporting)", "EV-COMP",
                       "GDQC-12385",  "GRMZM5G854065",  "GRMZM5G854065",                    "NADH-DEHYDROG-A-RXN",    "1.6.5.3", "NADH:ubiquinone oxidoreductase (H+-transporting)", "EV-COMP",
                       "GDQC-12456", "Zm00001d016566", "Zm00001d016566",                    "NADH-DEHYDROG-A-RXN",    "1.6.5.3", "NADH:ubiquinone oxidoreductase (H+-transporting)", "EV-COMP",
                       "GDQC-12457", "Zm00001d011732", "Zm00001d011732",                    "NADH-DEHYDROG-A-RXN",    "1.6.5.3", "NADH:ubiquinone oxidoreductase (H+-transporting)", "EV-COMP",
                       "GDQC-12465", "Zm00001d014428", "Zm00001d014428",                    "NADH-DEHYDROG-A-RXN",    "1.6.5.3", "NADH:ubiquinone oxidoreductase (H+-transporting)", "EV-COMP",
                       "GDQC-12515",  "GRMZM5G864407",  "GRMZM5G864407",                    "NADH-DEHYDROG-A-RXN",    "1.6.5.3", "NADH:ubiquinone oxidoreductase (H+-transporting)", "EV-COMP",
                       "GDQC-12527", "Zm00001d035824", "Zm00001d035824",                    "NADH-DEHYDROG-A-RXN",    "1.6.5.3", "NADH:ubiquinone oxidoreductase (H+-transporting)", "EV-COMP",
                       "GDQC-12528",  "GRMZM5G800096",  "GRMZM5G800096",                    "NADH-DEHYDROG-A-RXN",    "1.6.5.3", "NADH:ubiquinone oxidoreductase (H+-transporting)", "EV-COMP",
                       "GDQC-12621", "Zm00001d047441", "Zm00001d047441",                    "NADH-DEHYDROG-A-RXN",    "1.6.5.3", "NADH:ubiquinone oxidoreductase (H+-transporting)", "EV-COMP",
                       "GDQC-12714", "Zm00001d038658", "Zm00001d038658",                    "NADH-DEHYDROG-A-RXN",    "1.6.5.3", "NADH:ubiquinone oxidoreductase (H+-transporting)", "EV-COMP",
                       "GDQC-13003", "Zm00001d038441", "Zm00001d038441",                    "NADH-DEHYDROG-A-RXN",    "1.6.5.3", "NADH:ubiquinone oxidoreductase (H+-transporting)", "EV-COMP",
                       "GDQC-13045",  "GRMZM5G800101",  "GRMZM5G800101",                    "NADH-DEHYDROG-A-RXN",    "1.6.5.3", "NADH:ubiquinone oxidoreductase (H+-transporting)", "EV-COMP",
                       "GDQC-13070", "Zm00001d021536", "Zm00001d021536",                    "NADH-DEHYDROG-A-RXN",    "1.6.5.3", "NADH:ubiquinone oxidoreductase (H+-transporting)", "EV-COMP",
                       "GDQC-13127",  "GRMZM5G804358",  "GRMZM5G804358",                    "NADH-DEHYDROG-A-RXN",    "1.6.5.3", "NADH:ubiquinone oxidoreductase (H+-transporting)", "EV-COMP",
                       "GDQC-13187", "Zm00001d012469", "Zm00001d012469",                    "NADH-DEHYDROG-A-RXN",    "1.6.5.3", "NADH:ubiquinone oxidoreductase (H+-transporting)", "EV-COMP",
                       "GDQC-13403", "Zm00001d013156", "Zm00001d013156",                    "NADH-DEHYDROG-A-RXN",    "1.6.5.3", "NADH:ubiquinone oxidoreductase (H+-transporting)", "EV-COMP",
                       "GDQC-13497", "Zm00001d046260", "Zm00001d046260",                    "NADH-DEHYDROG-A-RXN",    "1.6.5.3", "NADH:ubiquinone oxidoreductase (H+-transporting)", "EV-COMP",
                       "GDQC-13999", "Zm00001d013424", "Zm00001d013424",                    "NADH-DEHYDROG-A-RXN",    "1.6.5.3", "NADH:ubiquinone oxidoreductase (H+-transporting)", "EV-COMP",
                       "GDQC-14251",  "GRMZM5G887911",  "GRMZM5G887911",                    "NADH-DEHYDROG-A-RXN",    "1.6.5.3", "NADH:ubiquinone oxidoreductase (H+-transporting)", "EV-COMP",
                       "GDQC-14301", "Zm00001d033530", "Zm00001d033530",                    "NADH-DEHYDROG-A-RXN",    "1.6.5.3", "NADH:ubiquinone oxidoreductase (H+-transporting)", "EV-COMP",
                       "GDQC-14435", "Zm00001d040515", "Zm00001d040515",                    "NADH-DEHYDROG-A-RXN",    "1.6.5.3", "NADH:ubiquinone oxidoreductase (H+-transporting)", "EV-COMP",
                       "GDQC-14466", "Zm00001d020930", "Zm00001d020930",                    "NADH-DEHYDROG-A-RXN",    "1.6.5.3", "NADH:ubiquinone oxidoreductase (H+-transporting)", "EV-COMP",
                       "GDQC-14513", "Zm00001d040819", "Zm00001d040819",                    "NADH-DEHYDROG-A-RXN",    "1.6.5.3", "NADH:ubiquinone oxidoreductase (H+-transporting)", "EV-COMP",
                       "GDQC-14576", "Zm00001d018479", "Zm00001d018479",                    "NADH-DEHYDROG-A-RXN",    "1.6.5.3", "NADH:ubiquinone oxidoreductase (H+-transporting)", "EV-COMP",
                       "GDQC-14831",  "GRMZM5G874448",  "GRMZM5G874448",                    "NADH-DEHYDROG-A-RXN",    "1.6.5.3", "NADH:ubiquinone oxidoreductase (H+-transporting)", "EV-COMP",
                       "GDQC-14957",  "GRMZM5G897755",  "GRMZM5G897755",                    "NADH-DEHYDROG-A-RXN",    "1.6.5.3", "NADH:ubiquinone oxidoreductase (H+-transporting)", "EV-COMP",
                        "GDQC-8383", "Zm00001d002436", "Zm00001d002436",                               "RXN-6883",  "1.10.3.11",                  "ubiquinol:oxygen oxidoreductase", "EV-COMP",
                        "GDQC-8893", "Zm00001d002435", "Zm00001d002435",                               "RXN-6883",  "1.10.3.11",                  "ubiquinol:oxygen oxidoreductase", "EV-COMP",
                        "GDQC-8921", "Zm00001d002434", "Zm00001d002434",                               "RXN-6883",  "1.10.3.11",                  "ubiquinol:oxygen oxidoreductase", "EV-COMP",
                       "GDQC-12692", "Zm00001d017727", "Zm00001d017727",                               "RXN-6883",  "1.10.3.11",                  "ubiquinol:oxygen oxidoreductase", "EV-COMP",
                       "GDQC-13150", "Zm00001d001908", "Zm00001d001908",                               "RXN-6883",  "1.10.3.11",                  "ubiquinol:oxygen oxidoreductase", "EV-COMP",
                       "GDQC-13700", "Zm00001d001909", "Zm00001d001909",                               "RXN-6883",  "1.10.3.11",                  "ubiquinol:oxygen oxidoreductase", "EV-COMP"
                       )

```
```{r}
CornCycAerRespIII <- as.character(CornCycAerRespIII.Tib$Gene.Accession)
```
```{r}
OxPhos <- unique(c(KEGG.OxPhos.V4,CornCycAerRespI,CornCycAerRespIII))
```
```{r}
Nuc.MtRibo <- tmt.withPolymorphic$Gene[grepl("ribosom", tmt.withPolymorphic$Protein)&grepl("mito", tmt.withPolymorphic$Protein)]
```



```{r}
tmt.mt <- tmt.withPolymorphic[grep("ZeamMp", tmt.withPolymorphic$Accession),]
```
```{r}
Mt.OxPhos <- tmt.mt$Gene[grep("nad|ndh|atp|cob|cox", tmt.mt$Protein)]
```
```{r}
Mt.ribo <- tmt.mt$Gene[grep("rps|rpl", tmt.mt$Protein)]
```
```{r}
Mt.other <- tmt.mt$Gene[!(tmt.mt$Gene%in%c(Mt.OxPhos, Mt.ribo))]
```


##GOI dataframe
```{r}
tmt.GOI <- tmt.withPolymorphic[,c(183,176,182)]

tmt.GOI$Category <- "other"
tmt.GOI$Order <- 26
tmt.GOI[(tmt.GOI$Gene %in% lightrxn.PSI.kc),"Category"] <- "NE.PSI.kc"
tmt.GOI[(tmt.GOI$Gene %in% lightrxn.PSI.kc),"Order"] <- 1
tmt.GOI[(tmt.GOI$Gene %in% lightrxn.PSII.kc),"Category"] <- "NE.PSII.kc"
tmt.GOI[(tmt.GOI$Gene %in% lightrxn.PSII.kc),"Order"] <- 2
tmt.GOI[(tmt.GOI$Gene %in% lightrxn.ATPase.k),"Category"] <- "NE.ATPase.k"
tmt.GOI[(tmt.GOI$Gene %in% lightrxn.ATPase.k),"Order"] <- 3
tmt.GOI[(tmt.GOI$Gene %in% lightrxn.NDH),"Category"] <- "NE.NDH"
tmt.GOI[(tmt.GOI$Gene %in% lightrxn.NDH),"Order"] <- 4
tmt.GOI[(tmt.GOI$Gene %in% lightrxn.ETC.kc),"Category"] <- "NE.ETC.kc"
tmt.GOI[(tmt.GOI$Gene %in% lightrxn.ETC.kc),"Order"] <- 5
tmt.GOI[(tmt.GOI$Gene %in% darkrxn.C4.c),"Category"] <- "NE.C4photo.c"
tmt.GOI[(tmt.GOI$Gene %in% darkrxn.C4.c),"Order"] <- 6
tmt.GOI[(tmt.GOI$Gene %in% darkrxn.Calvin.kc),"Category"] <- "NE.CalvinCycle.kc"
tmt.GOI[(tmt.GOI$Gene %in% darkrxn.Calvin.kc),"Order"] <- 7
tmt.GOI[(tmt.GOI$Gene %in% darkrxn.Calvin.RubiscoSS),"Category"] <- "NE.Rubisco"
tmt.GOI[(tmt.GOI$Gene %in% darkrxn.Calvin.RubiscoSS),"Order"] <- 8
tmt.GOI[(tmt.GOI$Gene %in% Pt.PS.PSI),"Category"] <- "Pt.PSI"
tmt.GOI[(tmt.GOI$Gene %in% Pt.PS.PSI),"Order"] <- 9
tmt.GOI[(tmt.GOI$Gene %in% Pt.PS.PSII),"Category"] <- "Pt.PSII"
tmt.GOI[(tmt.GOI$Gene %in% Pt.PS.PSII),"Order"] <- 10
tmt.GOI[(tmt.GOI$Gene %in% Pt.PS.ATP),"Category"] <- "Pt.ATP"
tmt.GOI[(tmt.GOI$Gene %in% Pt.PS.ATP),"Order"] <- 11
tmt.GOI[(tmt.GOI$Gene %in% Pt.PS.NADH),"Category"] <- "Pt.NADH"
tmt.GOI[(tmt.GOI$Gene %in% Pt.PS.NADH),"Order"] <- 12
tmt.GOI[(tmt.GOI$Gene %in% Pt.PS.b6f),"Category"] <- "Pt.b6f"
tmt.GOI[(tmt.GOI$Gene %in% Pt.PS.b6f),"Order"] <- 13
tmt.GOI[(tmt.GOI$Gene %in% Pt.PS.rubisco),"Category"] <- "Pt.Rubisco"
tmt.GOI[(tmt.GOI$Gene %in% Pt.PS.rubisco),"Order"] <- 14
tmt.GOI[(tmt.GOI$Gene %in% Pt.rpo),"Category"] <- "Pt.RNApol"
tmt.GOI[(tmt.GOI$Gene %in% Pt.rpo),"Order"] <- 15
tmt.GOI[(tmt.GOI$Gene %in% Pt.other),"Category"] <- "Pt.Other"
tmt.GOI[(tmt.GOI$Gene %in% Pt.other),"Order"] <- 16
tmt.GOI[(tmt.GOI$Gene %in% Pt.ribo),"Category"] <- "Pt.Ribo"
tmt.GOI[(tmt.GOI$Gene %in% Pt.ribo),"Order"] <- 17
tmt.GOI[(tmt.GOI$Gene %in% Nuc.PtRibo),"Category"] <- "NE.PlastidRibo"
tmt.GOI[(tmt.GOI$Gene %in% Nuc.PtRibo),"Order"] <- 18
tmt.GOI[(tmt.GOI$Gene %in% eukRibo),"Category"] <- "NE.EukRibo"
tmt.GOI[(tmt.GOI$Gene %in% eukRibo),"Order"] <- 19
tmt.GOI[(tmt.GOI$Gene %in% Nuc.MtRibo),"Category"] <- "NE.MitoRibo"
tmt.GOI[(tmt.GOI$Gene %in% Nuc.MtRibo),"Order"] <- 20
tmt.GOI[(tmt.GOI$Gene %in% Mt.ribo),"Category"] <- "Mt.Ribo"
tmt.GOI[(tmt.GOI$Gene %in% Mt.ribo),"Order"] <- 21
tmt.GOI[(tmt.GOI$Gene %in% Mt.OxPhos),"Category"] <- "Mt.OxPhos"
tmt.GOI[(tmt.GOI$Gene %in% Mt.OxPhos),"Order"] <- 22
tmt.GOI[(tmt.GOI$Gene %in% Mt.other),"Category"] <- "Mt.Other"
tmt.GOI[(tmt.GOI$Gene %in% Mt.other),"Order"] <- 23
tmt.GOI[(tmt.GOI$Gene %in% OxPhos),"Category"] <- "NE.OxPhos.kc"
tmt.GOI[(tmt.GOI$Gene %in% OxPhos),"Order"] <- 24
tmt.GOI[(tmt.GOI$Gene %in% EtBio),"Category"] <- "NE.EtBio.c"
tmt.GOI[(tmt.GOI$Gene %in% EtBio),"Order"] <- 25

tmt.GOI$Organelle <- NA
tmt.GOI[(tmt.GOI$Gene %in% plastidCC),"Organelle"] <- "GO:plastid"
tmt.GOI[(tmt.GOI$Gene %in% mitoCC),"Organelle"] <- "GO:mitochondria"

tmt.GOI <- tmt.GOI[order(tmt.GOI$Order),]
```
```{r}
cornCycplastid <- readr::read_tsv("plastid_instances_of_Genes_in_Zea_mays_mays (1).txt")
tmt.GOI <- merge.data.frame(tmt.GOI, cornCycplastid, by.x = "Gene", by.y = "Genes", plastid.x = T)
colnames(tmt.GOI)[7] <- "CornCycPathways"
```
```{r}
kegg <- gage::kegg.gsets(species = "zma")
kegg.df <- as.data.frame(unlist(kegg$kg.sets))
colnames(kegg.df) <- "Entrez"
kegg.df$Pathway <- rownames(kegg.df)
row.names(kegg.df) <- NULL
kegg.df$Entrez <- as.character(kegg.df$Entrez)
```
```{r}
tmt.GOI <- merge.data.frame(tmt.GOI, mgdb[,c(1,12)], by.x = "Gene", by.y = "v4_gene_model", plastid.x = T, plastid.y = F)
tmt.GOI <- distinct(tmt.GOI)
```
```{r}
tmt.GOI$Entrez[tmt.GOI$Accession=="ZemaCp003.1"] <- "845178"
tmt.GOI$Entrez[tmt.GOI$Accession=="ZemaCp004.1"] <- "845232"
tmt.GOI$Entrez[tmt.GOI$Accession=="ZemaCp008.1"] <- "845201"
tmt.GOI$Entrez[tmt.GOI$Accession=="ZemaCp013.1"] <- "845225"
tmt.GOI$Entrez[tmt.GOI$Accession=="ZemaCp014.1"] <- "845226"
tmt.GOI$Entrez[tmt.GOI$Accession=="ZemaCp016.1"] <- "845236"
tmt.GOI$Entrez[tmt.GOI$Accession=="ZemaCp017.1"] <- "845174"
tmt.GOI$Entrez[tmt.GOI$Accession=="ZemaCp018.1"] <- "845173"
tmt.GOI$Entrez[tmt.GOI$Accession=="ZemaCp019.1"] <- "845172"
tmt.GOI$Entrez[tmt.GOI$Accession=="ZemaCp019.1"] <- "845172"
tmt.GOI$Entrez[tmt.GOI$Accession=="ZemaCp021.1"] <- "845229"
tmt.GOI$Entrez[tmt.GOI$Accession=="ZemaCp022.1"] <- "845196"
tmt.GOI$Entrez[tmt.GOI$Accession=="ZemaCp023.1"] <- "845195"
tmt.GOI$Entrez[tmt.GOI$Accession=="ZemaCp024.1"] <- "1466361"
tmt.GOI$Entrez[tmt.GOI$Accession=="ZemaCp026.1"] <- "845238"
tmt.GOI$Entrez[tmt.GOI$Accession=="ZemaCp027.1"] <- "845189"
tmt.GOI$Entrez[tmt.GOI$Accession=="ZemaCp028.1"] <- "845190"
tmt.GOI$Entrez[tmt.GOI$Accession=="ZemaCp029.1"] <- "845182"
tmt.GOI$Entrez[tmt.GOI$Accession=="ZemaCp030.1"] <- "845171"
tmt.GOI$Entrez[tmt.GOI$Accession=="ZemaCp032.1"] <- "845212"
tmt.GOI$Entrez[tmt.GOI$Accession=="ZemaCp035.1"] <- "845175"
tmt.GOI$Entrez[tmt.GOI$Accession=="ZemaCp036.1"] <- "845191"
tmt.GOI$Entrez[tmt.GOI$Accession=="ZemaCp039.1"] <- "845204"
tmt.GOI$Entrez[tmt.GOI$Accession=="ZemaCp040.1"] <- "845203"
tmt.GOI$Entrez[tmt.GOI$Accession=="ZemaCp043.1"] <- "845198"
tmt.GOI$Entrez[tmt.GOI$Accession=="ZemaCp044.1"] <- "845222"
tmt.GOI$Entrez[tmt.GOI$Accession=="ZemaCp045.1"] <- "845233"
tmt.GOI$Entrez[tmt.GOI$Accession=="ZemaCp046.1"] <- "845217"
tmt.GOI$Entrez[tmt.GOI$Accession=="ZemaCp047.1"] <- "845176"
tmt.GOI$Entrez[tmt.GOI$Accession=="ZemaCp048.1"] <- "845200"
tmt.GOI$Entrez[tmt.GOI$Accession=="ZemaCp050.1"] <- "845211"
tmt.GOI$Entrez[tmt.GOI$Accession=="ZemaCp051.1"] <- "845205"
tmt.GOI$Entrez[tmt.GOI$Accession=="ZemaCp052.1"] <- "845192"
tmt.GOI$Entrez[tmt.GOI$Accession=="ZemaCp053.1"] <- "845193"
tmt.GOI$Entrez[tmt.GOI$Accession=="ZemaCp054.1"] <- "845224"
tmt.GOI$Entrez[tmt.GOI$Accession=="ZemaCp055.1"] <- "845228"
tmt.GOI$Entrez[tmt.GOI$Accession=="ZemaCp056.1"] <- "845223"
tmt.GOI$Entrez[tmt.GOI$Accession=="ZemaCp057.1"] <- "845177"
tmt.GOI$Entrez[tmt.GOI$Accession=="ZemaCp058.1"] <- "845241"
tmt.GOI$Entrez[tmt.GOI$Accession=="ZemaCp059.1"] <- "845213"
tmt.GOI$Entrez[tmt.GOI$Accession=="ZemaCp060.1"] <- "845214"
tmt.GOI$Entrez[tmt.GOI$Accession=="ZemaCp061.1"] <- "845237"
tmt.GOI$Entrez[tmt.GOI$Accession=="ZemaCp062.1"] <- "845218"
tmt.GOI$Entrez[tmt.GOI$Accession=="ZemaCp063.1"] <- "845235"
tmt.GOI$Entrez[tmt.GOI$Accession=="ZemaCp065.1"] <- "845216"
tmt.GOI$Entrez[tmt.GOI$Accession=="ZemaCp067.1"] <- "845220"
tmt.GOI$Entrez[tmt.GOI$Accession=="ZemaCp075.1"] <- "845181"
tmt.GOI$Entrez[tmt.GOI$Accession=="ZemaCp082.1"] <- "845230"
tmt.GOI$Entrez[tmt.GOI$Accession=="ZemaCp083.1"] <- "845185"
tmt.GOI$Entrez[tmt.GOI$Accession=="ZemaCp084.1"] <- "845221"
tmt.GOI$Entrez[tmt.GOI$Accession=="ZemaCp085.1"] <- "1466378"
tmt.GOI$Entrez[tmt.GOI$Accession=="ZemaCp086.1"] <- "845183"
tmt.GOI$Entrez[tmt.GOI$Accession=="ZemaCp087.1"] <- "1466379"
tmt.GOI$Entrez[tmt.GOI$Accession=="ZemaCp088.1"] <- "845184"
tmt.GOI$Entrez[tmt.GOI$Accession=="ZemaCp090.1"] <- "845188"
tmt.GOI$Entrez[tmt.GOI$Accession=="ZemaCp091.1"] <- "845179"
tmt.GOI$Entrez[tmt.GOI$Accession=="ZemaCp092.1"] <- "845187"

tmt.GOI$Entrez[tmt.GOI$Accession=="ZeamMp008.1"] <- "4055864"
tmt.GOI$Entrez[tmt.GOI$Accession=="ZeamMp024.1"] <- "4055860"
tmt.GOI$Entrez[tmt.GOI$Accession=="ZeamMp032.1"] <- "4055888"
tmt.GOI$Entrez[tmt.GOI$Accession=="ZeamMp033.1"] <- "4055938"
tmt.GOI$Entrez[tmt.GOI$Accession=="ZeamMp036.1"] <- "4055894"
tmt.GOI$Entrez[tmt.GOI$Accession=="ZeamMp050.1"] <- "4055956"
tmt.GOI$Entrez[tmt.GOI$Accession=="ZeamMp051.1"] <- "4055881"
tmt.GOI$Entrez[tmt.GOI$Accession=="ZeamMp057.1"] <- "4055973"
tmt.GOI$Entrez[tmt.GOI$Accession=="ZeamMp081.1"] <- "4055890"
tmt.GOI$Entrez[tmt.GOI$Accession=="ZeamMp092.1"] <- "4055837"
tmt.GOI$Entrez[tmt.GOI$Accession=="ZeamMp097.1"] <- "4055857"
tmt.GOI$Entrez[tmt.GOI$Accession=="ZeamMp098.1"] <- "4055896"
tmt.GOI$Entrez[tmt.GOI$Accession=="ZeamMp113.1"] <- "4055802"
tmt.GOI$Entrez[tmt.GOI$Accession=="ZeamMp119.1"] <- "4055928"
tmt.GOI$Entrez[tmt.GOI$Accession=="ZeamMp129.1"] <- "4055878"
tmt.GOI$Entrez[tmt.GOI$Accession=="ZeamMp136.1"] <- "4056004"
tmt.GOI$Entrez[tmt.GOI$Accession=="ZeamMp159.1"] <- "4055794"
tmt.GOI$Entrez[tmt.GOI$Accession=="ZeamMp167.1"] <- "4056007"
tmt.GOI$Entrez[tmt.GOI$Accession=="ZeamMp179.1"] <- "4056006"
tmt.GOI$Entrez[tmt.GOI$Accession=="ZeamMp185.1"] <- "4055892"
```
```{r}
tmt.GOI <- merge.data.frame(tmt.GOI, kegg.df, by.x = "Entrez", by.y = "Entrez", plastid.x = T, plastid.y = F)
colnames(tmt.GOI)[9] <- "KeggPathway"
```

```{r}
tmt.GOI.w <- data.table::dcast(setDT(tmt.GOI), Accession~rowid(Accession, prefix="KeggPathway"), value.var="KeggPathway")
```
```{r}
tmt.GOI <- merge.data.frame(tmt.GOI[,1:8], tmt.GOI.w, by.x = "Accession", by.y = "Accession")
tmt.GOI <- distinct(tmt.GOI)
```
```{r}
library(tidyr)
```
```{r}
tmt.GOI <- tmt.GOI[,c(1:9,16:23,10:15)]
```

```{r}
tmt.GOI <- unite(tmt.GOI, "KeggPathway", KeggPathway1:KeggPathway15)
```

```{r}
write.csv(tmt.GOI[order(tmt.GOI$Order),], "FinplastideafGOIs.csv", row.names = F)
```


#Scatterplots
##Seedling Leaf - B73xMo17 - OG
###Scatterplot - plastid
####Set limits
```{r}
tmt.OG.SL.lims1 <- tmt.OG.SL
tmt.OG.SL.lims1[,c(28,30)][tmt.OG.SL.lims1[,c(28,30)] <= -1.0] <- -1.0
tmt.OG.SL.lims1[,c(28,30)][tmt.OG.SL.lims1[,c(28,30)] >= 1.0] <- 1.0

tmt.OG.SL.lims5 <- tmt.OG.SL
tmt.OG.SL.lims5[,c(28,30)][tmt.OG.SL.lims5[,c(28,30)] <= -0.5] <- -0.5
tmt.OG.SL.lims5[,c(28,30)][tmt.OG.SL.lims5[,c(28,30)] >= 0.5] <- 0.5
```
####Set GOIs
```{r}
tmt.OG.SL.lims1.plastid <- tmt.OG.SL.lims1

tmt.OG.SL.lims1.plastid$Category <- "Other"
tmt.OG.SL.lims1.plastid$Order <- 1

tmt.OG.SL.lims1.plastid[(tmt.OG.SL.lims1.plastid$Gene %in% eukRibo),"Category"] <- "Cytosolic ribosome"
tmt.OG.SL.lims1.plastid[(tmt.OG.SL.lims1.plastid$Gene %in% eukRibo),"Order"] <- 2

tmt.OG.SL.lims1.plastid[(tmt.OG.SL.lims1.plastid$Gene %in% Nuc.PtRibo),"Category"] <- "NE plastid ribosome"
tmt.OG.SL.lims1.plastid[(tmt.OG.SL.lims1.plastid$Gene %in% Nuc.PtRibo),"Order"] <- 3

tmt.OG.SL.lims1.plastid[(tmt.OG.SL.lims1.plastid$Gene %in% Pt.ribo),"Category"] <- "PE plastid ribosome"
tmt.OG.SL.lims1.plastid[(tmt.OG.SL.lims1.plastid$Gene %in% Pt.ribo),"Order"] <- 4

tmt.OG.SL.lims1.plastid[(tmt.OG.SL.lims1.plastid$Gene %in% lightrxn.NucComplexes),"Category"] <- "NE digenomic complexes"
tmt.OG.SL.lims1.plastid[(tmt.OG.SL.lims1.plastid$Gene %in% lightrxn.NucComplexes),"Order"] <- 5

tmt.OG.SL.lims1.plastid[(tmt.OG.SL.lims1.plastid$Gene %in% Pt.Complexes),"Category"] <- "PE digenomic complexes"
tmt.OG.SL.lims1.plastid[(tmt.OG.SL.lims1.plastid$Gene %in% Pt.Complexes),"Order"] <- 6
#Re-order
tmt.OG.SL.lims1.plastid <- tmt.OG.SL.lims1.plastid[(order(tmt.OG.SL.lims1.plastid$Order)),]
#factor
tmt.OG.SL.lims1.plastid$Category <- factor(tmt.OG.SL.lims1.plastid$Category, levels = c("Other", "Cytosolic ribosome", "NE plastid ribosome","PE plastid ribosome", "NE digenomic complexes", "PE digenomic complexes"))
```
####Scatterplot
```{r}
tmt.OG.SL.lims1.plastid.scat <- ggplot(tmt.OG.SL.lims1.plastid, mapping=aes(x=l2.BM.MP, y=l2.B73.Mo17, color=Category))+
    geom_point(aes(color=Category), size=3, alpha=0.8)+
  scale_color_manual(values=c("Other"="#000000", "Cytosolic ribosome"="#CC79A7", "PE plastid ribosome"="#D55E00", "NE plastid ribosome"="#E69F00", "PE digenomic complexes"="#56B4E9", "NE digenomic complexes"="#009E73"), name="Category")+
    theme_classic()+
          xlim(c(-1,1))+
          ylim(c(-1,1))+
          geom_vline(xintercept=0, linetype="dashed", color = "gray50")+
          geom_hline(yintercept=0, linetype="dashed", color = "gray50")+
          theme(axis.text=element_text(size=10), legend.text=element_text(size=10),
                legend.position="bottom", legend.title = element_blank())+
          guides(col=guide_legend(nrow=3))+
    xlab("log2 BxM/MP")+
    ylab("log2 B73/Mo17")+
    coord_fixed(expand = F)
```
####Add density curves
```{r}
tmt.OG.SL.lims1.plastid.scat
```

```{r}
    densityplotX<-axis_canvas(tmt.OG.SL.lims1.plastid.scat, axis="x")+
      geom_density(aes(tmt.OG.SL.lims1.plastid[,"l2.BM.MP"], color=tmt.OG.SL.lims1.plastid[,"Category"]), size=1)+
    scale_color_manual(values=c("Other"="#000000", "Cytosolic ribosome"="#CC79A7", "PE plastid ribosome"="#D55E00", "NE plastid ribosome"="#E69F00", "PE digenomic complexes"="#56B4E9", "NE digenomic complexes"="#009E73"), name="Category")+
      geom_vline(xintercept=0, linetype="dashed", color = "gray50")+
      theme_void()+
      theme(legend.title = element_blank())
    #set up marginal density plot Y axis
    densityplotY<-axis_canvas(tmt.OG.SL.lims1.plastid.scat, axis="y", coord_flip = T)+
      geom_density(aes(tmt.OG.SL.lims1.plastid[,"l2.B73.Mo17"], color=tmt.OG.SL.lims1.plastid[,"Category"]), size=1)+
      scale_color_manual(values=c("Other"="#000000", "Cytosolic ribosome"="#CC79A7", "PE plastid ribosome"="#D55E00", "NE plastid ribosome"="#E69F00", "PE digenomic complexes"="#56B4E9", "NE digenomic complexes"="#009E73"), name="Category")+
      scale_linetype_manual(values=c("dotted", "solid"))+
      geom_vline(xintercept=0, linetype="dashed", color = "gray50")+
      theme_void()+
      theme(legend.title = element_blank())+
      coord_flip()
    #put two plots together
      q <- insert_xaxis_grob(tmt.OG.SL.lims1.plastid.scat, densityplotX, grid::unit(0.275, "null"), position = "top")
      q <- insert_yaxis_grob(q, densityplotY, grid::unit(0.275, "null"), position = "right")
    #save
     # ggsave(plot=q, filename = "tmt.OG.SL.lims1.plastid.scat.png")
  ggsave(plot=q, filename = "tmt.OG.SL.lims1.plastid.scat.pdf",  dpi=300, width=4.75, height = 5, scale = 1, units = "in")
```


###Scatterplot - Mitochondria
####Set GOIs
```{r}
tmt.OG.SL.lims1.mito <- tmt.OG.SL.lims1

tmt.OG.SL.lims1.mito$Category <- "Other"
tmt.OG.SL.lims1.mito$Order <- 1

tmt.OG.SL.lims1.mito[(tmt.OG.SL.lims1.mito$Gene %in% OxPhos),"Category"] <- "NE oxidative phosphorylation"
tmt.OG.SL.lims1.mito[(tmt.OG.SL.lims1.mito$Gene %in% OxPhos),"Order"] <- 2

tmt.OG.SL.lims1.mito[(tmt.OG.SL.lims1.mito$Gene %in% Mt.OxPhos),"Category"] <- "ME oxidative phosphorylation"
tmt.OG.SL.lims1.mito[(tmt.OG.SL.lims1.mito$Gene %in% Mt.OxPhos),"Order"] <- 3

tmt.OG.SL.lims1.mito[(tmt.OG.SL.lims1.mito$Gene %in% Nuc.MtRibo),"Category"] <- "NE mitochondrial ribosome"
tmt.OG.SL.lims1.mito[(tmt.OG.SL.lims1.mito$Gene %in% Nuc.MtRibo),"Order"] <- 4

tmt.OG.SL.lims1.mito[(tmt.OG.SL.lims1.mito$Gene %in% Mt.ribo),"Category"] <- "ME ribosome"
tmt.OG.SL.lims1.mito[(tmt.OG.SL.lims1.mito$Gene %in% Mt.ribo),"Order"] <- 5


#Re-order
tmt.OG.SL.lims1.mito <- tmt.OG.SL.lims1.mito[(order(tmt.OG.SL.lims1.mito$Order)),]
#factor
tmt.OG.SL.lims1.mito$Category <- factor(tmt.OG.SL.lims1.mito$Category, levels = c("Other", "NE oxidative phosphorylation", "ME oxidative phosphorylation", "NE mitochondrial ribosome", "ME ribosome"))
```
####Scatterplot
```{r}
tmt.OG.SL.lims1.mito.scat <- ggplot(tmt.OG.SL.lims1.mito, mapping=aes(x=l2.BM.MP, y=l2.B73.Mo17, color=Category))+
    geom_point(aes(color=Category), size=3, alpha=0.8)+
  scale_color_manual(values=c("Other"="#000000", "ME ribosome"="#D55E00", "NE mitochondrial ribosome"="#E69F00", "ME oxidative phosphorylation"="#56B4E9", "NE oxidative phosphorylation"="#009E73"), name="Category")+
    theme_classic()+
          xlim(c(-1,1))+
          ylim(c(-1,1))+
          geom_vline(xintercept=0, linetype="dashed", color = "gray50")+
          geom_hline(yintercept=0, linetype="dashed", color = "gray50")+
          theme(axis.text=element_text(size=10), legend.text=element_text(size=10),
                legend.position="bottom", legend.title = element_blank())+
          guides(col=guide_legend(nrow = 3))+
    xlab("log2 BxM/MP")+
    ylab("log2 B73/Mo17")+
    coord_fixed(expand = F)
```
####Add density curves
```{r}
    densityplotX<-axis_canvas(tmt.OG.SL.lims1.mito.scat, axis="x")+
      geom_density(aes(tmt.OG.SL.lims1.mito[,"l2.BM.MP"], color=tmt.OG.SL.lims1.mito[,"Category"]), size=1)+
    scale_color_manual(values=c("Other"="#000000", "ME ribosome"="#D55E00", "NE mitochondrial ribosome"="#E69F00", "ME oxidative phosphorylation"="#56B4E9", "NE oxidative phosphorylation"="#009E73"), name="Category")+
      geom_vline(xintercept=0, linetype="dashed", color = "gray50")+
      theme_void()+
      theme(legend.title = element_blank())
    #set up marginal density plot Y axis
    densityplotY<-axis_canvas(tmt.OG.SL.lims1.mito.scat, axis="y", coord_flip = T)+
      geom_density(aes(tmt.OG.SL.lims1.mito[,"l2.B73.Mo17"], color=tmt.OG.SL.lims1.mito[,"Category"]), size=1)+
      scale_color_manual(values=c("Other"="#000000", "ME ribosome"="#D55E00", "NE mitochondrial ribosome"="#E69F00", "ME oxidative phosphorylation"="#56B4E9", "NE oxidative phosphorylation"="#009E73"), name="Category")+
      scale_linetype_manual(values=c("dotted", "solid"))+
      geom_vline(xintercept=0, linetype="dashed", color = "gray50")+
      theme_void()+
      theme(legend.title = element_blank())+
      coord_flip()
    #put two plots together
      q <- insert_xaxis_grob(tmt.OG.SL.lims1.mito.scat, densityplotX, grid::unit(0.275, "null"), position = "top")
      q <- insert_yaxis_grob(q, densityplotY, grid::unit(0.275, "null"), position = "right")
    #save
    #  ggsave(plot=q, filename = "tmt.OG.SL.lims1.mito.scat.png")
  ggsave(plot=q, filename = "tmt.OG.SL.lims1.mito.scat.pdf", dpi=300, width=4.75, height = 5, scale = 1, units = "in")
```


##Seedling Leaf - B73xMo17 - OG + acs
####Set limits
```{r}
tmt.OG.acs.SL.lims1 <- tmt.OG.acs.SL
tmt.OG.acs.SL.lims1[,c(40:41)][tmt.OG.acs.SL.lims1[,c(40:41)] <= -1.0] <- -1.0
tmt.OG.acs.SL.lims1[,c(40:41)][tmt.OG.acs.SL.lims1[,c(40:41)] >= 1.0] <- 1.0

tmt.OG.acs.SL.lims5 <- tmt.OG.acs.SL
tmt.OG.acs.SL.lims5[,c(40:41)][tmt.OG.acs.SL.lims5[,c(40:41)] <= -0.5] <- -0.5
tmt.OG.acs.SL.lims5[,c(40:41)][tmt.OG.acs.SL.lims5[,c(40:41)] >= 0.5] <- 0.5
```
####Set GOIs
```{r}
tmt.OG.acs.SL.lims1.et <- tmt.OG.acs.SL.lims1

tmt.OG.acs.SL.lims1.et$Category <- "Other"
tmt.OG.acs.SL.lims1.et$Order <- 1

tmt.OG.acs.SL.lims1.et[(tmt.OG.acs.SL.lims1.et$Gene %in% eukRibo),"Category"] <- "Cytosolic ribosome"
tmt.OG.acs.SL.lims1.et[(tmt.OG.acs.SL.lims1.et$Gene %in% eukRibo),"Order"] <- 2

tmt.OG.acs.SL.lims1.et[(tmt.OG.acs.SL.lims1.et$Gene %in% Nuc.PtRibo),"Category"] <- "NE plastid ribosome"
tmt.OG.acs.SL.lims1.et[(tmt.OG.acs.SL.lims1.et$Gene %in% Nuc.PtRibo),"Order"] <- 3

tmt.OG.acs.SL.lims1.et[(tmt.OG.acs.SL.lims1.et$Gene %in% Pt.ribo),"Category"] <- "PE plastid ribosome"
tmt.OG.acs.SL.lims1.et[(tmt.OG.acs.SL.lims1.et$Gene %in% Pt.ribo),"Order"] <- 4

tmt.OG.acs.SL.lims1.et[(tmt.OG.acs.SL.lims1.et$Gene %in% lightrxn.NucComplexes),"Category"] <- "NE digenomic complexes"
tmt.OG.acs.SL.lims1.et[(tmt.OG.acs.SL.lims1.et$Gene %in% lightrxn.NucComplexes),"Order"] <- 5

tmt.OG.acs.SL.lims1.et[(tmt.OG.acs.SL.lims1.et$Gene %in% Pt.Complexes),"Category"] <- "PE digenomic complexes"
tmt.OG.acs.SL.lims1.et[(tmt.OG.acs.SL.lims1.et$Gene %in% Pt.Complexes),"Order"] <- 6

tmt.OG.acs.SL.lims1.et[(tmt.OG.acs.SL.lims1.et$Gene %in% EtBio),"Category"] <- "Ethylene biosynthesis"
tmt.OG.acs.SL.lims1.et[(tmt.OG.acs.SL.lims1.et$Gene %in% EtBio),"Order"] <- 7

#Re-order
tmt.OG.acs.SL.lims1.et <- tmt.OG.acs.SL.lims1.et[(order(tmt.OG.acs.SL.lims1.et$Order)),]
#factor
tmt.OG.acs.SL.lims1.et$Category <- factor(tmt.OG.acs.SL.lims1.et$Category, levels = c("Other", "Cytosolic ribosome", "NE plastid ribosome","PE plastid ribosome", "NE digenomic complexes", "PE digenomic complexes", "Ethylene biosynthesis"))
```
####Scatterplot
```{r}
tmt.OG.acs.SL.lims1.et.scat <- ggplot(tmt.OG.acs.SL.lims1.et, mapping=aes(x=l2.HYB.MP, y=l2.ACS.B73, color=Category))+
    geom_point(aes(color=Category), size=3, alpha=0.8)+
  scale_color_manual(values=c("Other"="#000000", "Cytosolic ribosome"="#CC79A7", "PE plastid ribosome"="#D55E00", "NE plastid ribosome"="#E69F00", "PE digenomic complexes"="#56B4E9", "NE digenomic complexes"="#009E73", "Ethylene biosynthesis"="#F0E442"), name="Category")+
    theme_classic()+
          xlim(c(-1,1))+
          ylim(c(-1,1))+
          geom_vline(xintercept=0, linetype="dashed", color = "gray50")+
          geom_hline(yintercept=0, linetype="dashed", color = "gray50")+
          theme(axis.text=element_text(size=10), legend.text=element_text(size=10),
                legend.position="bottom", legend.title = element_blank())+
          guides(col=guide_legend(nrow=4))+
    xlab("log2 BxM/MP")+
    ylab("log2 acs/B73")+
    coord_fixed(expand = F)
```
####Add density curves
```{r}
    densityplotX<-axis_canvas(tmt.OG.acs.SL.lims1.et.scat, axis="x")+
      geom_density(aes(tmt.OG.acs.SL.lims1.et[,"l2.HYB.MP"], color=tmt.OG.acs.SL.lims1.et[,"Category"]), size=1)+
    scale_color_manual(values=c("Other"="#000000", "Cytosolic ribosome"="#CC79A7", "PE plastid ribosome"="#D55E00", "NE plastid ribosome"="#E69F00", "PE digenomic complexes"="#56B4E9", "NE digenomic complexes"="#009E73", "Ethylene biosynthesis"="#F0E442"), name="Category")+
      geom_vline(xintercept=0, linetype="dashed", color = "gray50")+
      theme_void()+
      theme(legend.title = element_blank())
    #set up marginal density plot Y axis
    densityplotY<-axis_canvas(tmt.OG.acs.SL.lims1.et.scat, axis="y", coord_flip = T)+
      geom_density(aes(tmt.OG.acs.SL.lims1.et[,"l2.ACS.B73"], color=tmt.OG.acs.SL.lims1.et[,"Category"]), size=1)+
      scale_color_manual(values=c("Other"="#000000", "Cytosolic ribosome"="#CC79A7", "PE plastid ribosome"="#D55E00", "NE plastid ribosome"="#E69F00", "PE digenomic complexes"="#56B4E9", "NE digenomic complexes"="#009E73", "Ethylene biosynthesis"="#F0E442"), name="Category")+
      scale_linetype_manual(values=c("dotted", "solid"))+
      geom_vline(xintercept=0, linetype="dashed", color = "gray50")+
      theme_void()+
      theme(legend.title = element_blank())+
      coord_flip()
    #put two plots together
      q <- insert_xaxis_grob(tmt.OG.acs.SL.lims1.et.scat, densityplotX, grid::unit(0.275, "null"), position = "top")
      q <- insert_yaxis_grob(q, densityplotY, grid::unit(0.275, "null"), position = "right")
    #save
     # ggsave(plot=q, filename = "tmt.OG.acs.SL.lims1.et.scat.png")
  ggsave(plot=q, filename = "tmt.OG.SL.lims1.et.scat.pdf", dpi=300, width=4.75, height = 5, scale = 1, units = "in")
```


##Seedling Leaf - B73xMo17 - DN
###Scatterplot - plastid
####Set limits
```{r}
tmt.dn.SL.lims1 <- tmt.dn.SL
tmt.dn.SL.lims1[,c(28,30)][tmt.dn.SL.lims1[,c(28,30)] <= -1.0] <- -1.0
tmt.dn.SL.lims1[,c(28,30)][tmt.dn.SL.lims1[,c(28,30)] >= 1.0] <- 1.0

tmt.dn.SL.lims5 <- tmt.dn.SL
tmt.dn.SL.lims5[,c(28,30)][tmt.dn.SL.lims5[,c(28,30)] <= -0.5] <- -0.5
tmt.dn.SL.lims5[,c(28,30)][tmt.dn.SL.lims5[,c(28,30)] >= 0.5] <- 0.5
```
####Set GOIs
```{r}
tmt.dn.SL.lims1.plastid <- tmt.dn.SL.lims1

tmt.dn.SL.lims1.plastid$Category <- "Other"
tmt.dn.SL.lims1.plastid$Order <- 1

tmt.dn.SL.lims1.plastid[(tmt.dn.SL.lims1.plastid$Gene %in% eukRibo),"Category"] <- "Cytosolic ribosome"
tmt.dn.SL.lims1.plastid[(tmt.dn.SL.lims1.plastid$Gene %in% eukRibo),"Order"] <- 2

tmt.dn.SL.lims1.plastid[(tmt.dn.SL.lims1.plastid$Gene %in% Nuc.PtRibo),"Category"] <- "NE plastid ribosome"
tmt.dn.SL.lims1.plastid[(tmt.dn.SL.lims1.plastid$Gene %in% Nuc.PtRibo),"Order"] <- 3

tmt.dn.SL.lims1.plastid[(tmt.dn.SL.lims1.plastid$Gene %in% Pt.ribo),"Category"] <- "PE plastid ribosome"
tmt.dn.SL.lims1.plastid[(tmt.dn.SL.lims1.plastid$Gene %in% Pt.ribo),"Order"] <- 4

tmt.dn.SL.lims1.plastid[(tmt.dn.SL.lims1.plastid$Gene %in% lightrxn.NucComplexes),"Category"] <- "NE digenomic complexes"
tmt.dn.SL.lims1.plastid[(tmt.dn.SL.lims1.plastid$Gene %in% lightrxn.NucComplexes),"Order"] <- 5

tmt.dn.SL.lims1.plastid[(tmt.dn.SL.lims1.plastid$Gene %in% Pt.Complexes),"Category"] <- "PE digenomic complexes"
tmt.dn.SL.lims1.plastid[(tmt.dn.SL.lims1.plastid$Gene %in% Pt.Complexes),"Order"] <- 6
#Re-order
tmt.dn.SL.lims1.plastid <- tmt.dn.SL.lims1.plastid[(order(tmt.dn.SL.lims1.plastid$Order)),]
#factor
tmt.dn.SL.lims1.plastid$Category <- factor(tmt.dn.SL.lims1.plastid$Category, levels = c("Other", "Cytosolic ribosome", "NE plastid ribosome","PE plastid ribosome", "NE digenomic complexes", "PE digenomic complexes"))
```
####Scatterplot
```{r}
tmt.dn.SL.lims1.plastid.scat <- ggplot(tmt.dn.SL.lims1.plastid, mapping=aes(x=l2.BM.MP, y=l2.B73.Mo17, color=Category))+
    geom_point(aes(color=Category), size=3, alpha=0.8)+
  scale_color_manual(values=c("Other"="#000000", "Cytosolic ribosome"="#CC79A7", "PE plastid ribosome"="#D55E00", "NE plastid ribosome"="#E69F00", "PE digenomic complexes"="#56B4E9", "NE digenomic complexes"="#009E73"), name="Category")+
    theme_classic()+
          xlim(c(-1,1))+
          ylim(c(-1,1))+
          geom_vline(xintercept=0, linetype="dashed", color = "gray50")+
          geom_hline(yintercept=0, linetype="dashed", color = "gray50")+
          theme(axis.text=element_text(size=10), legend.text=element_text(size=10),
                legend.position="bottom", legend.title = element_blank())+
          guides(col=guide_legend(nrow=3))+
    xlab("log2 BxM/MP")+
    ylab("log2 B73/Mo17")+
    coord_fixed(expand = F)
```
####Add density curves
```{r}
    densityplotX<-axis_canvas(tmt.dn.SL.lims1.plastid.scat, axis="x")+
      geom_density(aes(tmt.dn.SL.lims1.plastid[,"l2.BM.MP"], color=tmt.dn.SL.lims1.plastid[,"Category"]), size=1)+
    scale_color_manual(values=c("Other"="#000000", "Cytosolic ribosome"="#CC79A7", "PE plastid ribosome"="#D55E00", "NE plastid ribosome"="#E69F00", "PE digenomic complexes"="#56B4E9", "NE digenomic complexes"="#009E73"), name="Category")+
      geom_vline(xintercept=0, linetype="dashed", color = "gray50")+
      theme_void()+
      theme(legend.title = element_blank())
    #set up marginal density plot Y axis
    densityplotY<-axis_canvas(tmt.dn.SL.lims1.plastid.scat, axis="y", coord_flip = T)+
      geom_density(aes(tmt.dn.SL.lims1.plastid[,"l2.B73.Mo17"], color=tmt.dn.SL.lims1.plastid[,"Category"]), size=1)+
      scale_color_manual(values=c("Other"="#000000", "Cytosolic ribosome"="#CC79A7", "PE plastid ribosome"="#D55E00", "NE plastid ribosome"="#E69F00", "PE digenomic complexes"="#56B4E9", "NE digenomic complexes"="#009E73"), name="Category")+
      scale_linetype_manual(values=c("dotted", "solid"))+
      geom_vline(xintercept=0, linetype="dashed", color = "gray50")+
      theme_void()+
      theme(legend.title = element_blank())+
      coord_flip()
    #put two plots together
      q <- insert_xaxis_grob(tmt.dn.SL.lims1.plastid.scat, densityplotX, grid::unit(0.275, "null"), position = "top")
      q <- insert_yaxis_grob(q, densityplotY, grid::unit(0.275, "null"), position = "right")
    #save
     # ggsave(plot=q, filename = "tmt.dn.SL.lims1.plastid.scat.png")
  ggsave(plot=q, filename = "tmt.dn.SL.lims1.plastid.scat.pdf", dpi=300, width=4.75, height = 5, scale = 1, units = "in")
```


###Scatterplot - Mitochondria
####Set GOIs
```{r}
tmt.dn.SL.lims1.mito <- tmt.dn.SL.lims1

tmt.dn.SL.lims1.mito$Category <- "Other"
tmt.dn.SL.lims1.mito$Order <- 1

tmt.dn.SL.lims1.mito[(tmt.dn.SL.lims1.mito$Gene %in% OxPhos),"Category"] <- "NE oxidative phosphorylation"
tmt.dn.SL.lims1.mito[(tmt.dn.SL.lims1.mito$Gene %in% OxPhos),"Order"] <- 2

tmt.dn.SL.lims1.mito[(tmt.dn.SL.lims1.mito$Gene %in% Mt.OxPhos),"Category"] <- "ME oxidative phosphorylation"
tmt.dn.SL.lims1.mito[(tmt.dn.SL.lims1.mito$Gene %in% Mt.OxPhos),"Order"] <- 3

tmt.dn.SL.lims1.mito[(tmt.dn.SL.lims1.mito$Gene %in% Nuc.MtRibo),"Category"] <- "NE mitochondrial ribosome"
tmt.dn.SL.lims1.mito[(tmt.dn.SL.lims1.mito$Gene %in% Nuc.MtRibo),"Order"] <- 4

tmt.dn.SL.lims1.mito[(tmt.dn.SL.lims1.mito$Gene %in% Mt.ribo),"Category"] <- "ME ribosome"
tmt.dn.SL.lims1.mito[(tmt.dn.SL.lims1.mito$Gene %in% Mt.ribo),"Order"] <- 5


#Re-order
tmt.dn.SL.lims1.mito <- tmt.dn.SL.lims1.mito[(order(tmt.dn.SL.lims1.mito$Order)),]
#factor
tmt.dn.SL.lims1.mito$Category <- factor(tmt.dn.SL.lims1.mito$Category, levels = c("Other", "NE oxidative phosphorylation", "ME oxidative phosphorylation", "NE mitochondrial ribosome", "ME ribosome"))
```
####Scatterplot
```{r}
tmt.dn.SL.lims1.mito.scat <- ggplot(tmt.dn.SL.lims1.mito, mapping=aes(x=l2.BM.MP, y=l2.B73.Mo17, color=Category))+
    geom_point(aes(color=Category), size=3, alpha=0.8)+
  scale_color_manual(values=c("Other"="#000000", "ME ribosome"="#D55E00", "NE mitochondrial ribosome"="#E69F00", "ME oxidative phosphorylation"="#56B4E9", "NE oxidative phosphorylation"="#009E73"), name="Category")+
    theme_classic()+
          xlim(c(-1,1))+
          ylim(c(-1,1))+
          geom_vline(xintercept=0, linetype="dashed", color = "gray50")+
          geom_hline(yintercept=0, linetype="dashed", color = "gray50")+
          theme(axis.text=element_text(size=10), legend.text=element_text(size=10),
                legend.position="bottom", legend.title = element_blank())+
          guides(col=guide_legend(nrow = 3))+
    xlab("log2 BxM/MP")+
    ylab("log2 B73/Mo17")+
    coord_fixed(expand = F)
```
####Add density curves
```{r}
    densityplotX<-axis_canvas(tmt.dn.SL.lims1.mito.scat, axis="x")+
      geom_density(aes(tmt.dn.SL.lims1.mito[,"l2.BM.MP"], color=tmt.dn.SL.lims1.mito[,"Category"]), size=1)+
    scale_color_manual(values=c("Other"="#000000", "ME ribosome"="#D55E00", "NE mitochondrial ribosome"="#E69F00", "ME oxidative phosphorylation"="#56B4E9", "NE oxidative phosphorylation"="#009E73"), name="Category")+
      geom_vline(xintercept=0, linetype="dashed", color = "gray50")+
      theme_void()+
      theme(legend.title = element_blank())
    #set up marginal density plot Y axis
    densityplotY<-axis_canvas(tmt.dn.SL.lims1.mito.scat, axis="y", coord_flip = T)+
      geom_density(aes(tmt.dn.SL.lims1.mito[,"l2.B73.Mo17"], color=tmt.dn.SL.lims1.mito[,"Category"]), size=1)+
      scale_color_manual(values=c("Other"="#000000", "ME ribosome"="#D55E00", "NE mitochondrial ribosome"="#E69F00", "ME oxidative phosphorylation"="#56B4E9", "NE oxidative phosphorylation"="#009E73"), name="Category")+
      scale_linetype_manual(values=c("dotted", "solid"))+
      geom_vline(xintercept=0, linetype="dashed", color = "gray50")+
      theme_void()+
      theme(legend.title = element_blank())+
      coord_flip()
    #put two plots together
      q <- insert_xaxis_grob(tmt.dn.SL.lims1.mito.scat, densityplotX, grid::unit(0.275, "null"), position = "top")
      q <- insert_yaxis_grob(q, densityplotY, grid::unit(0.275, "null"), position = "right")
    #save
    #  ggsave(plot=q, filename = "tmt.dn.SL.lims1.mito.scat.png")
  ggsave(plot=q, filename = "tmt.dn.SL.lims1.mito.scat.pdf", dpi=300, width=4.75, height = 5, scale = 1, units = "in")
```


##Seedling Leaf - B73xMo17 - DN + acs
####Set limits
```{r}
tmt.dn.acs.SL.lims1 <- tmt.dn.acs.SL
tmt.dn.acs.SL.lims1[,c(40:41)][tmt.dn.acs.SL.lims1[,c(40:41)] <= -1.0] <- -1.0
tmt.dn.acs.SL.lims1[,c(40:41)][tmt.dn.acs.SL.lims1[,c(40:41)] >= 1.0] <- 1.0

tmt.dn.acs.SL.lims5 <- tmt.dn.acs.SL
tmt.dn.acs.SL.lims5[,c(40:41)][tmt.dn.acs.SL.lims5[,c(40:41)] <= -0.5] <- -0.5
tmt.dn.acs.SL.lims5[,c(40:41)][tmt.dn.acs.SL.lims5[,c(40:41)] >= 0.5] <- 0.5
```
####Set GOIs
```{r}
tmt.dn.acs.SL.lims1.et <- tmt.dn.acs.SL.lims1

tmt.dn.acs.SL.lims1.et$Category <- "Other"
tmt.dn.acs.SL.lims1.et$Order <- 1

tmt.dn.acs.SL.lims1.et[(tmt.dn.acs.SL.lims1.et$Gene %in% eukRibo),"Category"] <- "Cytosolic ribosome"
tmt.dn.acs.SL.lims1.et[(tmt.dn.acs.SL.lims1.et$Gene %in% eukRibo),"Order"] <- 2

tmt.dn.acs.SL.lims1.et[(tmt.dn.acs.SL.lims1.et$Gene %in% Nuc.PtRibo),"Category"] <- "NE plastid ribosome"
tmt.dn.acs.SL.lims1.et[(tmt.dn.acs.SL.lims1.et$Gene %in% Nuc.PtRibo),"Order"] <- 3

tmt.dn.acs.SL.lims1.et[(tmt.dn.acs.SL.lims1.et$Gene %in% Pt.ribo),"Category"] <- "PE plastid ribosome"
tmt.dn.acs.SL.lims1.et[(tmt.dn.acs.SL.lims1.et$Gene %in% Pt.ribo),"Order"] <- 4

tmt.dn.acs.SL.lims1.et[(tmt.dn.acs.SL.lims1.et$Gene %in% lightrxn.NucComplexes),"Category"] <- "NE digenomic complexes"
tmt.dn.acs.SL.lims1.et[(tmt.dn.acs.SL.lims1.et$Gene %in% lightrxn.NucComplexes),"Order"] <- 5

tmt.dn.acs.SL.lims1.et[(tmt.dn.acs.SL.lims1.et$Gene %in% Pt.Complexes),"Category"] <- "PE digenomic complexes"
tmt.dn.acs.SL.lims1.et[(tmt.dn.acs.SL.lims1.et$Gene %in% Pt.Complexes),"Order"] <- 6

tmt.dn.acs.SL.lims1.et[(tmt.dn.acs.SL.lims1.et$Gene %in% EtBio),"Category"] <- "Ethylene biosynthesis"
tmt.dn.acs.SL.lims1.et[(tmt.dn.acs.SL.lims1.et$Gene %in% EtBio),"Order"] <- 7

#Re-order
tmt.dn.acs.SL.lims1.et <- tmt.dn.acs.SL.lims1.et[(order(tmt.dn.acs.SL.lims1.et$Order)),]
#factor
tmt.dn.acs.SL.lims1.et$Category <- factor(tmt.dn.acs.SL.lims1.et$Category, levels = c("Other", "Cytosolic ribosome", "NE plastid ribosome","PE plastid ribosome", "NE digenomic complexes", "PE digenomic complexes", "Ethylene biosynthesis"))
```
####Scatterplot
```{r}
tmt.dn.acs.SL.lims1.et.scat <- ggplot(tmt.dn.acs.SL.lims1.et, mapping=aes(x=l2.HYB.MP, y=l2.ACS.B73, color=Category))+
    geom_point(aes(color=Category), size=3, alpha=0.8)+
  scale_color_manual(values=c("Other"="#000000", "Cytosolic ribosome"="#CC79A7", "PE plastid ribosome"="#D55E00", "NE plastid ribosome"="#E69F00", "PE digenomic complexes"="#56B4E9", "NE digenomic complexes"="#009E73", "Ethylene biosynthesis"="#F0E442"), name="Category")+
    theme_classic()+
          xlim(c(-1,1))+
          ylim(c(-1,1))+
          geom_vline(xintercept=0, linetype="dashed", color = "gray50")+
          geom_hline(yintercept=0, linetype="dashed", color = "gray50")+
          theme(axis.text=element_text(size=10), legend.text=element_text(size=10),
                legend.position="bottom", legend.title = element_blank())+
          guides(col=guide_legend(nrow=4))+
    xlab("log2 BxM/MP")+
    ylab("log2 acs/B73")+
    coord_fixed(expand = F)
```
####Add density curves
```{r}
    densityplotX<-axis_canvas(tmt.dn.acs.SL.lims1.et.scat, axis="x")+
      geom_density(aes(tmt.dn.acs.SL.lims1.et[,"l2.HYB.MP"], color=tmt.dn.acs.SL.lims1.et[,"Category"]), size=1)+
    scale_color_manual(values=c("Other"="#000000", "Cytosolic ribosome"="#CC79A7", "PE plastid ribosome"="#D55E00", "NE plastid ribosome"="#E69F00", "PE digenomic complexes"="#56B4E9", "NE digenomic complexes"="#009E73", "Ethylene biosynthesis"="#F0E442"), name="Category")+
      geom_vline(xintercept=0, linetype="dashed", color = "gray50")+
      theme_void()+
      theme(legend.title = element_blank())
    #set up marginal density plot Y axis
    densityplotY<-axis_canvas(tmt.dn.acs.SL.lims1.et.scat, axis="y", coord_flip = T)+
      geom_density(aes(tmt.dn.acs.SL.lims1.et[,"l2.ACS.B73"], color=tmt.dn.acs.SL.lims1.et[,"Category"]), size=1)+
      scale_color_manual(values=c("Other"="#000000", "Cytosolic ribosome"="#CC79A7", "PE plastid ribosome"="#D55E00", "NE plastid ribosome"="#E69F00", "PE digenomic complexes"="#56B4E9", "NE digenomic complexes"="#009E73", "Ethylene biosynthesis"="#F0E442"), name="Category")+
      scale_linetype_manual(values=c("dotted", "solid"))+
      geom_vline(xintercept=0, linetype="dashed", color = "gray50")+
      theme_void()+
      theme(legend.title = element_blank())+
      coord_flip()
    #put two plots together
      q <- insert_xaxis_grob(tmt.dn.acs.SL.lims1.et.scat, densityplotX, grid::unit(0.275, "null"), position = "top")
      q <- insert_yaxis_grob(q, densityplotY, grid::unit(0.275, "null"), position = "right")
    #save
     # ggsave(plot=q, filename = "tmt.dn.acs.SL.lims1.et.scat.png")
  ggsave(plot=q, filename = "tmt.dn.SL.lims1.et.scat.pdf", dpi=300, width=4.75, height = 5, scale = 1, units = "in")
```


##Seedling Leaf - B73xMo17 - 6H
###Scatterplot - plastid
####Set limits
```{r}
tmt.6h.SL.lims1 <- tmt.6h.SL
tmt.6h.SL.lims1[,c(28,30)][tmt.6h.SL.lims1[,c(28,30)] <= -1.0] <- -1.0
tmt.6h.SL.lims1[,c(28,30)][tmt.6h.SL.lims1[,c(28,30)] >= 1.0] <- 1.0

tmt.6h.SL.lims5 <- tmt.6h.SL
tmt.6h.SL.lims5[,c(28,30)][tmt.6h.SL.lims5[,c(28,30)] <= -0.5] <- -0.5
tmt.6h.SL.lims5[,c(28,30)][tmt.6h.SL.lims5[,c(28,30)] >= 0.5] <- 0.5
```
####Set GOIs
```{r}
tmt.6h.SL.lims1.plastid <- tmt.6h.SL.lims1

tmt.6h.SL.lims1.plastid$Category <- "Other"
tmt.6h.SL.lims1.plastid$Order <- 1

tmt.6h.SL.lims1.plastid[(tmt.6h.SL.lims1.plastid$Gene %in% eukRibo),"Category"] <- "Cytosolic ribosome"
tmt.6h.SL.lims1.plastid[(tmt.6h.SL.lims1.plastid$Gene %in% eukRibo),"Order"] <- 2

tmt.6h.SL.lims1.plastid[(tmt.6h.SL.lims1.plastid$Gene %in% Nuc.PtRibo),"Category"] <- "NE plastid ribosome"
tmt.6h.SL.lims1.plastid[(tmt.6h.SL.lims1.plastid$Gene %in% Nuc.PtRibo),"Order"] <- 3

tmt.6h.SL.lims1.plastid[(tmt.6h.SL.lims1.plastid$Gene %in% Pt.ribo),"Category"] <- "PE plastid ribosome"
tmt.6h.SL.lims1.plastid[(tmt.6h.SL.lims1.plastid$Gene %in% Pt.ribo),"Order"] <- 4

tmt.6h.SL.lims1.plastid[(tmt.6h.SL.lims1.plastid$Gene %in% lightrxn.NucComplexes),"Category"] <- "NE digenomic complexes"
tmt.6h.SL.lims1.plastid[(tmt.6h.SL.lims1.plastid$Gene %in% lightrxn.NucComplexes),"Order"] <- 5

tmt.6h.SL.lims1.plastid[(tmt.6h.SL.lims1.plastid$Gene %in% Pt.Complexes),"Category"] <- "PE digenomic complexes"
tmt.6h.SL.lims1.plastid[(tmt.6h.SL.lims1.plastid$Gene %in% Pt.Complexes),"Order"] <- 6
#Re-order
tmt.6h.SL.lims1.plastid <- tmt.6h.SL.lims1.plastid[(order(tmt.6h.SL.lims1.plastid$Order)),]
#factor
tmt.6h.SL.lims1.plastid$Category <- factor(tmt.6h.SL.lims1.plastid$Category, levels = c("Other", "Cytosolic ribosome", "NE plastid ribosome","PE plastid ribosome", "NE digenomic complexes", "PE digenomic complexes"))
```
####Scatterplot
```{r}
tmt.6h.SL.lims1.plastid.scat <- ggplot(tmt.6h.SL.lims1.plastid, mapping=aes(x=l2.BM.MP, y=l2.B73.Mo17, color=Category))+
    geom_point(aes(color=Category), size=3, alpha=0.8)+
  scale_color_manual(values=c("Other"="#000000", "Cytosolic ribosome"="#CC79A7", "PE plastid ribosome"="#D55E00", "NE plastid ribosome"="#E69F00", "PE digenomic complexes"="#56B4E9", "NE digenomic complexes"="#009E73"), name="Category")+
    theme_classic()+
          xlim(c(-1,1))+
          ylim(c(-1,1))+
          geom_vline(xintercept=0, linetype="dashed", color = "gray50")+
          geom_hline(yintercept=0, linetype="dashed", color = "gray50")+
          theme(axis.text=element_text(size=10), legend.text=element_text(size=10),
                legend.position="bottom", legend.title = element_blank())+
          guides(col=guide_legend(nrow=3))+
    xlab("log2 BxM/MP")+
    ylab("log2 B73/Mo17")+
    coord_fixed(expand = F)
```
####Add density curves
```{r}
    densityplotX<-axis_canvas(tmt.6h.SL.lims1.plastid.scat, axis="x")+
      geom_density(aes(tmt.6h.SL.lims1.plastid[,"l2.BM.MP"], color=tmt.6h.SL.lims1.plastid[,"Category"]), size=1)+
    scale_color_manual(values=c("Other"="#000000", "Cytosolic ribosome"="#CC79A7", "PE plastid ribosome"="#D55E00", "NE plastid ribosome"="#E69F00", "PE digenomic complexes"="#56B4E9", "NE digenomic complexes"="#009E73"), name="Category")+
      geom_vline(xintercept=0, linetype="dashed", color = "gray50")+
      theme_void()+
      theme(legend.title = element_blank())
    #set up marginal density plot Y axis
    densityplotY<-axis_canvas(tmt.6h.SL.lims1.plastid.scat, axis="y", coord_flip = T)+
      geom_density(aes(tmt.6h.SL.lims1.plastid[,"l2.B73.Mo17"], color=tmt.6h.SL.lims1.plastid[,"Category"]), size=1)+
      scale_color_manual(values=c("Other"="#000000", "Cytosolic ribosome"="#CC79A7", "PE plastid ribosome"="#D55E00", "NE plastid ribosome"="#E69F00", "PE digenomic complexes"="#56B4E9", "NE digenomic complexes"="#009E73"), name="Category")+
      scale_linetype_manual(values=c("dotted", "solid"))+
      geom_vline(xintercept=0, linetype="dashed", color = "gray50")+
      theme_void()+
      theme(legend.title = element_blank())+
      coord_flip()
    #put two plots together
      q <- insert_xaxis_grob(tmt.6h.SL.lims1.plastid.scat, densityplotX, grid::unit(0.275, "null"), position = "top")
      q <- insert_yaxis_grob(q, densityplotY, grid::unit(0.275, "null"), position = "right")
    #save
     # ggsave(plot=q, filename = "tmt.6h.SL.lims1.plastid.scat.png")
  ggsave(plot=q, filename = "tmt.6h.SL.lims1.plastid.scat.pdf", dpi=300, width=4.75, height = 5, scale = 1, units = "in")
```


###Scatterplot - Mitochondria
####Set GOIs
```{r}
tmt.6h.SL.lims1.mito <- tmt.6h.SL.lims1

tmt.6h.SL.lims1.mito$Category <- "Other"
tmt.6h.SL.lims1.mito$Order <- 1

tmt.6h.SL.lims1.mito[(tmt.6h.SL.lims1.mito$Gene %in% OxPhos),"Category"] <- "NE oxidative phosphorylation"
tmt.6h.SL.lims1.mito[(tmt.6h.SL.lims1.mito$Gene %in% OxPhos),"Order"] <- 2

tmt.6h.SL.lims1.mito[(tmt.6h.SL.lims1.mito$Gene %in% Mt.OxPhos),"Category"] <- "ME oxidative phosphorylation"
tmt.6h.SL.lims1.mito[(tmt.6h.SL.lims1.mito$Gene %in% Mt.OxPhos),"Order"] <- 3

tmt.6h.SL.lims1.mito[(tmt.6h.SL.lims1.mito$Gene %in% Nuc.MtRibo),"Category"] <- "NE mitochondrial ribosome"
tmt.6h.SL.lims1.mito[(tmt.6h.SL.lims1.mito$Gene %in% Nuc.MtRibo),"Order"] <- 4

tmt.6h.SL.lims1.mito[(tmt.6h.SL.lims1.mito$Gene %in% Mt.ribo),"Category"] <- "ME ribosome"
tmt.6h.SL.lims1.mito[(tmt.6h.SL.lims1.mito$Gene %in% Mt.ribo),"Order"] <- 5


#Re-order
tmt.6h.SL.lims1.mito <- tmt.6h.SL.lims1.mito[(order(tmt.6h.SL.lims1.mito$Order)),]
#factor
tmt.6h.SL.lims1.mito$Category <- factor(tmt.6h.SL.lims1.mito$Category, levels = c("Other", "NE oxidative phosphorylation", "ME oxidative phosphorylation", "NE mitochondrial ribosome", "ME ribosome"))
```
####Scatterplot
```{r}
tmt.6h.SL.lims1.mito.scat <- ggplot(tmt.6h.SL.lims1.mito, mapping=aes(x=l2.BM.MP, y=l2.B73.Mo17, color=Category))+
    geom_point(aes(color=Category), size=3, alpha=0.8)+
  scale_color_manual(values=c("Other"="#000000", "ME ribosome"="#D55E00", "NE mitochondrial ribosome"="#E69F00", "ME oxidative phosphorylation"="#56B4E9", "NE oxidative phosphorylation"="#009E73"), name="Category")+
    theme_classic()+
          xlim(c(-1,1))+
          ylim(c(-1,1))+
          geom_vline(xintercept=0, linetype="dashed", color = "gray50")+
          geom_hline(yintercept=0, linetype="dashed", color = "gray50")+
          theme(axis.text=element_text(size=10), legend.text=element_text(size=10),
                legend.position="bottom", legend.title = element_blank())+
          guides(col=guide_legend(nrow = 3))+
    xlab("log2 BxM/MP")+
    ylab("log2 B73/Mo17")+
    coord_fixed(expand = F)
```
####Add density curves
```{r}
    densityplotX<-axis_canvas(tmt.6h.SL.lims1.mito.scat, axis="x")+
      geom_density(aes(tmt.6h.SL.lims1.mito[,"l2.BM.MP"], color=tmt.6h.SL.lims1.mito[,"Category"]), size=1)+
    scale_color_manual(values=c("Other"="#000000", "ME ribosome"="#D55E00", "NE mitochondrial ribosome"="#E69F00", "ME oxidative phosphorylation"="#56B4E9", "NE oxidative phosphorylation"="#009E73"), name="Category")+
      geom_vline(xintercept=0, linetype="dashed", color = "gray50")+
      theme_void()+
      theme(legend.title = element_blank())
    #set up marginal density plot Y axis
    densityplotY<-axis_canvas(tmt.6h.SL.lims1.mito.scat, axis="y", coord_flip = T)+
      geom_density(aes(tmt.6h.SL.lims1.mito[,"l2.B73.Mo17"], color=tmt.6h.SL.lims1.mito[,"Category"]), size=1)+
      scale_color_manual(values=c("Other"="#000000", "ME ribosome"="#D55E00", "NE mitochondrial ribosome"="#E69F00", "ME oxidative phosphorylation"="#56B4E9", "NE oxidative phosphorylation"="#009E73"), name="Category")+
      scale_linetype_manual(values=c("dotted", "solid"))+
      geom_vline(xintercept=0, linetype="dashed", color = "gray50")+
      theme_void()+
      theme(legend.title = element_blank())+
      coord_flip()
    #put two plots together
      q <- insert_xaxis_grob(tmt.6h.SL.lims1.mito.scat, densityplotX, grid::unit(0.275, "null"), position = "top")
      q <- insert_yaxis_grob(q, densityplotY, grid::unit(0.275, "null"), position = "right")
    #save
    #  ggsave(plot=q, filename = "tmt.6h.SL.lims1.mito.scat.png")
  ggsave(plot=q, filename = "tmt.6h.SL.lims1.mito.scat.pdf", dpi=300, width=4.75, height = 5, scale = 1, units = "in")
```


##Seedling Leaf - B73xMo17 - 6H + acs
####Set limits
```{r}
tmt.6h.acs.SL.lims1 <- tmt.6h.acs.SL
tmt.6h.acs.SL.lims1[,c(40:41)][tmt.6h.acs.SL.lims1[,c(40:41)] <= -1.0] <- -1.0
tmt.6h.acs.SL.lims1[,c(40:41)][tmt.6h.acs.SL.lims1[,c(40:41)] >= 1.0] <- 1.0

tmt.6h.acs.SL.lims5 <- tmt.6h.acs.SL
tmt.6h.acs.SL.lims5[,c(40:41)][tmt.6h.acs.SL.lims5[,c(40:41)] <= -0.5] <- -0.5
tmt.6h.acs.SL.lims5[,c(40:41)][tmt.6h.acs.SL.lims5[,c(40:41)] >= 0.5] <- 0.5
```
####Set GOIs
```{r}
tmt.6h.acs.SL.lims1.et <- tmt.6h.acs.SL.lims1

tmt.6h.acs.SL.lims1.et$Category <- "Other"
tmt.6h.acs.SL.lims1.et$Order <- 1

tmt.6h.acs.SL.lims1.et[(tmt.6h.acs.SL.lims1.et$Gene %in% eukRibo),"Category"] <- "Cytosolic ribosome"
tmt.6h.acs.SL.lims1.et[(tmt.6h.acs.SL.lims1.et$Gene %in% eukRibo),"Order"] <- 2

tmt.6h.acs.SL.lims1.et[(tmt.6h.acs.SL.lims1.et$Gene %in% Nuc.PtRibo),"Category"] <- "NE plastid ribosome"
tmt.6h.acs.SL.lims1.et[(tmt.6h.acs.SL.lims1.et$Gene %in% Nuc.PtRibo),"Order"] <- 3

tmt.6h.acs.SL.lims1.et[(tmt.6h.acs.SL.lims1.et$Gene %in% Pt.ribo),"Category"] <- "PE plastid ribosome"
tmt.6h.acs.SL.lims1.et[(tmt.6h.acs.SL.lims1.et$Gene %in% Pt.ribo),"Order"] <- 4

tmt.6h.acs.SL.lims1.et[(tmt.6h.acs.SL.lims1.et$Gene %in% lightrxn.NucComplexes),"Category"] <- "NE digenomic complexes"
tmt.6h.acs.SL.lims1.et[(tmt.6h.acs.SL.lims1.et$Gene %in% lightrxn.NucComplexes),"Order"] <- 5

tmt.6h.acs.SL.lims1.et[(tmt.6h.acs.SL.lims1.et$Gene %in% Pt.Complexes),"Category"] <- "PE digenomic complexes"
tmt.6h.acs.SL.lims1.et[(tmt.6h.acs.SL.lims1.et$Gene %in% Pt.Complexes),"Order"] <- 6

tmt.6h.acs.SL.lims1.et[(tmt.6h.acs.SL.lims1.et$Gene %in% EtBio),"Category"] <- "Ethylene biosynthesis"
tmt.6h.acs.SL.lims1.et[(tmt.6h.acs.SL.lims1.et$Gene %in% EtBio),"Order"] <- 7

#Re-order
tmt.6h.acs.SL.lims1.et <- tmt.6h.acs.SL.lims1.et[(order(tmt.6h.acs.SL.lims1.et$Order)),]
#factor
tmt.6h.acs.SL.lims1.et$Category <- factor(tmt.6h.acs.SL.lims1.et$Category, levels = c("Other", "Cytosolic ribosome", "NE plastid ribosome","PE plastid ribosome", "NE digenomic complexes", "PE digenomic complexes", "Ethylene biosynthesis"))
```
####Scatterplot
```{r}
tmt.6h.acs.SL.lims1.et.scat <- ggplot(tmt.6h.acs.SL.lims1.et, mapping=aes(x=l2.HYB.MP, y=l2.ACS.B73, color=Category))+
    geom_point(aes(color=Category), size=3, alpha=0.8)+
  scale_color_manual(values=c("Other"="#000000", "Cytosolic ribosome"="#CC79A7", "PE plastid ribosome"="#D55E00", "NE plastid ribosome"="#E69F00", "PE digenomic complexes"="#56B4E9", "NE digenomic complexes"="#009E73", "Ethylene biosynthesis"="#F0E442"), name="Category")+
    theme_classic()+
          xlim(c(-1,1))+
          ylim(c(-1,1))+
          geom_vline(xintercept=0, linetype="dashed", color = "gray50")+
          geom_hline(yintercept=0, linetype="dashed", color = "gray50")+
          theme(axis.text=element_text(size=10), legend.text=element_text(size=10),
                legend.position="bottom", legend.title = element_blank())+
          guides(col=guide_legend(nrow=4))+
    xlab("log2 BxM/MP")+
    ylab("log2 acs/B73")+
    coord_fixed(expand = F)
```
####Add density curves
```{r}
    densityplotX<-axis_canvas(tmt.6h.acs.SL.lims1.et.scat, axis="x")+
      geom_density(aes(tmt.6h.acs.SL.lims1.et[,"l2.HYB.MP"], color=tmt.6h.acs.SL.lims1.et[,"Category"]), size=1)+
    scale_color_manual(values=c("Other"="#000000", "Cytosolic ribosome"="#CC79A7", "PE plastid ribosome"="#D55E00", "NE plastid ribosome"="#E69F00", "PE digenomic complexes"="#56B4E9", "NE digenomic complexes"="#009E73", "Ethylene biosynthesis"="#F0E442"), name="Category")+
      geom_vline(xintercept=0, linetype="dashed", color = "gray50")+
      theme_void()+
      theme(legend.title = element_blank())
    #set up marginal density plot Y axis
    densityplotY<-axis_canvas(tmt.6h.acs.SL.lims1.et.scat, axis="y", coord_flip = T)+
      geom_density(aes(tmt.6h.acs.SL.lims1.et[,"l2.ACS.B73"], color=tmt.6h.acs.SL.lims1.et[,"Category"]), size=1)+
      scale_color_manual(values=c("Other"="#000000", "Cytosolic ribosome"="#CC79A7", "PE plastid ribosome"="#D55E00", "NE plastid ribosome"="#E69F00", "PE digenomic complexes"="#56B4E9", "NE digenomic complexes"="#009E73", "Ethylene biosynthesis"="#F0E442"), name="Category")+
      scale_linetype_manual(values=c("dotted", "solid"))+
      geom_vline(xintercept=0, linetype="dashed", color = "gray50")+
      theme_void()+
      theme(legend.title = element_blank())+
      coord_flip()
    #put two plots together
      q <- insert_xaxis_grob(tmt.6h.acs.SL.lims1.et.scat, densityplotX, grid::unit(0.275, "null"), position = "top")
      q <- insert_yaxis_grob(q, densityplotY, grid::unit(0.275, "null"), position = "right")
    #save
     # ggsave(plot=q, filename = "tmt.6h.acs.SL.lims1.et.scat.png")
  ggsave(plot=q, filename = "tmt.6h.SL.lims1.et.scat.pdf", dpi=300, width=4.75, height = 5, scale = 1, units = "in")
```


##Seedling Leaf - B73xMo17 - RIL
###Scatterplot - plastid
####Set limits
```{r}
tmt.ril.SL.lims1 <- tmt.ril.SL
tmt.ril.SL.lims1[,c(28,30)][tmt.ril.SL.lims1[,c(28,30)] <= -1.0] <- -1.0
tmt.ril.SL.lims1[,c(28,30)][tmt.ril.SL.lims1[,c(28,30)] >= 1.0] <- 1.0

tmt.ril.SL.lims5 <- tmt.ril.SL
tmt.ril.SL.lims5[,c(28,30)][tmt.ril.SL.lims5[,c(28,30)] <= -0.5] <- -0.5
tmt.ril.SL.lims5[,c(28,30)][tmt.ril.SL.lims5[,c(28,30)] >= 0.5] <- 0.5
```
####Set GOIs
```{r}
tmt.ril.SL.lims1.plastid <- tmt.ril.SL.lims1

tmt.ril.SL.lims1.plastid$Category <- "Other"
tmt.ril.SL.lims1.plastid$Order <- 1

tmt.ril.SL.lims1.plastid[(tmt.ril.SL.lims1.plastid$Gene %in% eukRibo),"Category"] <- "Cytosolic ribosome"
tmt.ril.SL.lims1.plastid[(tmt.ril.SL.lims1.plastid$Gene %in% eukRibo),"Order"] <- 2

tmt.ril.SL.lims1.plastid[(tmt.ril.SL.lims1.plastid$Gene %in% Nuc.PtRibo),"Category"] <- "NE plastid ribosome"
tmt.ril.SL.lims1.plastid[(tmt.ril.SL.lims1.plastid$Gene %in% Nuc.PtRibo),"Order"] <- 3

tmt.ril.SL.lims1.plastid[(tmt.ril.SL.lims1.plastid$Gene %in% Pt.ribo),"Category"] <- "PE plastid ribosome"
tmt.ril.SL.lims1.plastid[(tmt.ril.SL.lims1.plastid$Gene %in% Pt.ribo),"Order"] <- 4

tmt.ril.SL.lims1.plastid[(tmt.ril.SL.lims1.plastid$Gene %in% lightrxn.NucComplexes),"Category"] <- "NE digenomic complexes"
tmt.ril.SL.lims1.plastid[(tmt.ril.SL.lims1.plastid$Gene %in% lightrxn.NucComplexes),"Order"] <- 5

tmt.ril.SL.lims1.plastid[(tmt.ril.SL.lims1.plastid$Gene %in% Pt.Complexes),"Category"] <- "PE digenomic complexes"
tmt.ril.SL.lims1.plastid[(tmt.ril.SL.lims1.plastid$Gene %in% Pt.Complexes),"Order"] <- 6
#Re-order
tmt.ril.SL.lims1.plastid <- tmt.ril.SL.lims1.plastid[(order(tmt.ril.SL.lims1.plastid$Order)),]
#factor
tmt.ril.SL.lims1.plastid$Category <- factor(tmt.ril.SL.lims1.plastid$Category, levels = c("Other", "Cytosolic ribosome", "NE plastid ribosome","PE plastid ribosome", "NE digenomic complexes", "PE digenomic complexes"))
```
####Scatterplot
```{r}
tmt.ril.SL.lims1.plastid.scat <- ggplot(tmt.ril.SL.lims1.plastid, mapping=aes(x=l2.BM.MP, y=l2.B73.Mo17, color=Category))+
    geom_point(aes(color=Category), size=3, alpha=0.8)+
  scale_color_manual(values=c("Other"="#000000", "Cytosolic ribosome"="#CC79A7", "PE plastid ribosome"="#D55E00", "NE plastid ribosome"="#E69F00", "PE digenomic complexes"="#56B4E9", "NE digenomic complexes"="#009E73"), name="Category")+
    theme_classic()+
          xlim(c(-1,1))+
          ylim(c(-1,1))+
          geom_vline(xintercept=0, linetype="dashed", color = "gray50")+
          geom_hline(yintercept=0, linetype="dashed", color = "gray50")+
          theme(axis.text=element_text(size=10), legend.text=element_text(size=10),
                legend.position="bottom", legend.title = element_blank())+
          guides(col=guide_legend(nrow=3))+
    xlab("log2 BxM/MP")+
    ylab("log2 B73/Mo17")+
    coord_fixed(expand = F)
```
####Add density curves
```{r}
    densityplotX<-axis_canvas(tmt.ril.SL.lims1.plastid.scat, axis="x")+
      geom_density(aes(tmt.ril.SL.lims1.plastid[,"l2.BM.MP"], color=tmt.ril.SL.lims1.plastid[,"Category"]), size=1)+
    scale_color_manual(values=c("Other"="#000000", "Cytosolic ribosome"="#CC79A7", "PE plastid ribosome"="#D55E00", "NE plastid ribosome"="#E69F00", "PE digenomic complexes"="#56B4E9", "NE digenomic complexes"="#009E73"), name="Category")+
      geom_vline(xintercept=0, linetype="dashed", color = "gray50")+
      theme_void()+
      theme(legend.title = element_blank())
    #set up marginal density plot Y axis
    densityplotY<-axis_canvas(tmt.ril.SL.lims1.plastid.scat, axis="y", coord_flip = T)+
      geom_density(aes(tmt.ril.SL.lims1.plastid[,"l2.B73.Mo17"], color=tmt.ril.SL.lims1.plastid[,"Category"]), size=1)+
      scale_color_manual(values=c("Other"="#000000", "Cytosolic ribosome"="#CC79A7", "PE plastid ribosome"="#D55E00", "NE plastid ribosome"="#E69F00", "PE digenomic complexes"="#56B4E9", "NE digenomic complexes"="#009E73"), name="Category")+
      scale_linetype_manual(values=c("dotted", "solid"))+
      geom_vline(xintercept=0, linetype="dashed", color = "gray50")+
      theme_void()+
      theme(legend.title = element_blank())+
      coord_flip()
    #put two plots together
      q <- insert_xaxis_grob(tmt.ril.SL.lims1.plastid.scat, densityplotX, grid::unit(0.275, "null"), position = "top")
      q <- insert_yaxis_grob(q, densityplotY, grid::unit(0.275, "null"), position = "right")
    #save
     # ggsave(plot=q, filename = "tmt.ril.SL.lims1.plastid.scat.png")
  ggsave(plot=q, filename = "tmt.ril.SL.lims1.plastid.scat.pdf",dpi=300, width=4.75, height = 5, scale = 1, units = "in")
```


###Scatterplot - Mitochondria
####Set GOIs
```{r}
tmt.ril.SL.lims1.mito <- tmt.ril.SL.lims1

tmt.ril.SL.lims1.mito$Category <- "Other"
tmt.ril.SL.lims1.mito$Order <- 1

tmt.ril.SL.lims1.mito[(tmt.ril.SL.lims1.mito$Gene %in% OxPhos),"Category"] <- "NE oxidative phosphorylation"
tmt.ril.SL.lims1.mito[(tmt.ril.SL.lims1.mito$Gene %in% OxPhos),"Order"] <- 2

tmt.ril.SL.lims1.mito[(tmt.ril.SL.lims1.mito$Gene %in% Mt.OxPhos),"Category"] <- "ME oxidative phosphorylation"
tmt.ril.SL.lims1.mito[(tmt.ril.SL.lims1.mito$Gene %in% Mt.OxPhos),"Order"] <- 3

tmt.ril.SL.lims1.mito[(tmt.ril.SL.lims1.mito$Gene %in% Nuc.MtRibo),"Category"] <- "NE mitochondrial ribosome"
tmt.ril.SL.lims1.mito[(tmt.ril.SL.lims1.mito$Gene %in% Nuc.MtRibo),"Order"] <- 4

tmt.ril.SL.lims1.mito[(tmt.ril.SL.lims1.mito$Gene %in% Mt.ribo),"Category"] <- "ME ribosome"
tmt.ril.SL.lims1.mito[(tmt.ril.SL.lims1.mito$Gene %in% Mt.ribo),"Order"] <- 5


#Re-order
tmt.ril.SL.lims1.mito <- tmt.ril.SL.lims1.mito[(order(tmt.ril.SL.lims1.mito$Order)),]
#factor
tmt.ril.SL.lims1.mito$Category <- factor(tmt.ril.SL.lims1.mito$Category, levels = c("Other", "NE oxidative phosphorylation", "ME oxidative phosphorylation", "NE mitochondrial ribosome", "ME ribosome"))
```
####Scatterplot
```{r}
tmt.ril.SL.lims1.mito.scat <- ggplot(tmt.ril.SL.lims1.mito, mapping=aes(x=l2.BM.MP, y=l2.B73.Mo17, color=Category))+
    geom_point(aes(color=Category), size=3, alpha=0.8)+
  scale_color_manual(values=c("Other"="#000000", "ME ribosome"="#D55E00", "NE mitochondrial ribosome"="#E69F00", "ME oxidative phosphorylation"="#56B4E9", "NE oxidative phosphorylation"="#009E73"), name="Category")+
    theme_classic()+
          xlim(c(-1,1))+
          ylim(c(-1,1))+
          geom_vline(xintercept=0, linetype="dashed", color = "gray50")+
          geom_hline(yintercept=0, linetype="dashed", color = "gray50")+
          theme(axis.text=element_text(size=10), legend.text=element_text(size=10),
                legend.position="bottom", legend.title = element_blank())+
          guides(col=guide_legend(nrow = 3))+
    xlab("log2 BxM/MP")+
    ylab("log2 B73/Mo17")+
    coord_fixed(expand = F)
```
####Add density curves
```{r}
    densityplotX<-axis_canvas(tmt.ril.SL.lims1.mito.scat, axis="x")+
      geom_density(aes(tmt.ril.SL.lims1.mito[,"l2.BM.MP"], color=tmt.ril.SL.lims1.mito[,"Category"]), size=1)+
    scale_color_manual(values=c("Other"="#000000", "ME ribosome"="#D55E00", "NE mitochondrial ribosome"="#E69F00", "ME oxidative phosphorylation"="#56B4E9", "NE oxidative phosphorylation"="#009E73"), name="Category")+
      geom_vline(xintercept=0, linetype="dashed", color = "gray50")+
      theme_void()+
      theme(legend.title = element_blank())
    #set up marginal density plot Y axis
    densityplotY<-axis_canvas(tmt.ril.SL.lims1.mito.scat, axis="y", coord_flip = T)+
      geom_density(aes(tmt.ril.SL.lims1.mito[,"l2.B73.Mo17"], color=tmt.ril.SL.lims1.mito[,"Category"]), size=1)+
      scale_color_manual(values=c("Other"="#000000", "ME ribosome"="#D55E00", "NE mitochondrial ribosome"="#E69F00", "ME oxidative phosphorylation"="#56B4E9", "NE oxidative phosphorylation"="#009E73"), name="Category")+
      scale_linetype_manual(values=c("dotted", "solid"))+
      geom_vline(xintercept=0, linetype="dashed", color = "gray50")+
      theme_void()+
      theme(legend.title = element_blank())+
      coord_flip()
    #put two plots together
      q <- insert_xaxis_grob(tmt.ril.SL.lims1.mito.scat, densityplotX, grid::unit(0.275, "null"), position = "top")
      q <- insert_yaxis_grob(q, densityplotY, grid::unit(0.275, "null"), position = "right")
    #save
    #  ggsave(plot=q, filename = "tmt.ril.SL.lims1.mito.scat.png")
  ggsave(plot=q, filename = "tmt.ril.SL.lims1.mito.scat.pdf", dpi=300, width=4.75, height = 5, scale = 1, units = "in")
```


##Seedling Leaf - B73xMo17 - RIL + acs
####Set limits
```{r}
tmt.ril.acs.SL.lims1 <- tmt.ril.acs.SL
tmt.ril.acs.SL.lims1[,c(40:41)][tmt.ril.acs.SL.lims1[,c(40:41)] <= -1.0] <- -1.0
tmt.ril.acs.SL.lims1[,c(40:41)][tmt.ril.acs.SL.lims1[,c(40:41)] >= 1.0] <- 1.0

tmt.ril.acs.SL.lims5 <- tmt.ril.acs.SL
tmt.ril.acs.SL.lims5[,c(40:41)][tmt.ril.acs.SL.lims5[,c(40:41)] <= -0.5] <- -0.5
tmt.ril.acs.SL.lims5[,c(40:41)][tmt.ril.acs.SL.lims5[,c(40:41)] >= 0.5] <- 0.5
```
####Set GOIs
```{r}
tmt.ril.acs.SL.lims1.et <- tmt.ril.acs.SL.lims1

tmt.ril.acs.SL.lims1.et$Category <- "Other"
tmt.ril.acs.SL.lims1.et$Order <- 1

tmt.ril.acs.SL.lims1.et[(tmt.ril.acs.SL.lims1.et$Gene %in% eukRibo),"Category"] <- "Cytosolic ribosome"
tmt.ril.acs.SL.lims1.et[(tmt.ril.acs.SL.lims1.et$Gene %in% eukRibo),"Order"] <- 2

tmt.ril.acs.SL.lims1.et[(tmt.ril.acs.SL.lims1.et$Gene %in% Nuc.PtRibo),"Category"] <- "NE plastid ribosome"
tmt.ril.acs.SL.lims1.et[(tmt.ril.acs.SL.lims1.et$Gene %in% Nuc.PtRibo),"Order"] <- 3

tmt.ril.acs.SL.lims1.et[(tmt.ril.acs.SL.lims1.et$Gene %in% Pt.ribo),"Category"] <- "PE plastid ribosome"
tmt.ril.acs.SL.lims1.et[(tmt.ril.acs.SL.lims1.et$Gene %in% Pt.ribo),"Order"] <- 4

tmt.ril.acs.SL.lims1.et[(tmt.ril.acs.SL.lims1.et$Gene %in% lightrxn.NucComplexes),"Category"] <- "NE digenomic complexes"
tmt.ril.acs.SL.lims1.et[(tmt.ril.acs.SL.lims1.et$Gene %in% lightrxn.NucComplexes),"Order"] <- 5

tmt.ril.acs.SL.lims1.et[(tmt.ril.acs.SL.lims1.et$Gene %in% Pt.Complexes),"Category"] <- "PE digenomic complexes"
tmt.ril.acs.SL.lims1.et[(tmt.ril.acs.SL.lims1.et$Gene %in% Pt.Complexes),"Order"] <- 6

tmt.ril.acs.SL.lims1.et[(tmt.ril.acs.SL.lims1.et$Gene %in% EtBio),"Category"] <- "Ethylene biosynthesis"
tmt.ril.acs.SL.lims1.et[(tmt.ril.acs.SL.lims1.et$Gene %in% EtBio),"Order"] <- 7

#Re-order
tmt.ril.acs.SL.lims1.et <- tmt.ril.acs.SL.lims1.et[(order(tmt.ril.acs.SL.lims1.et$Order)),]
#factor
tmt.ril.acs.SL.lims1.et$Category <- factor(tmt.ril.acs.SL.lims1.et$Category, levels = c("Other", "Cytosolic ribosome", "NE plastid ribosome","PE plastid ribosome", "NE digenomic complexes", "PE digenomic complexes", "Ethylene biosynthesis"))
```
####Scatterplot
```{r}
tmt.ril.acs.SL.lims1.et.scat <- ggplot(tmt.ril.acs.SL.lims1.et, mapping=aes(x=l2.HYB.MP, y=l2.ACS.B73, color=Category))+
    geom_point(aes(color=Category), size=3, alpha=0.8)+
  scale_color_manual(values=c("Other"="#000000", "Cytosolic ribosome"="#CC79A7", "PE plastid ribosome"="#D55E00", "NE plastid ribosome"="#E69F00", "PE digenomic complexes"="#56B4E9", "NE digenomic complexes"="#009E73", "Ethylene biosynthesis"="#F0E442"), name="Category")+
    theme_classic()+
          xlim(c(-1,1))+
          ylim(c(-1,1))+
          geom_vline(xintercept=0, linetype="dashed", color = "gray50")+
          geom_hline(yintercept=0, linetype="dashed", color = "gray50")+
          theme(axis.text=element_text(size=10), legend.text=element_text(size=10),
                legend.position="bottom", legend.title = element_blank())+
          guides(col=guide_legend(nrow=4))+
    xlab("log2 BxM/MP")+
    ylab("log2 acs/B73")+
    coord_fixed(expand = F)
```
####Add density curves
```{r}
    densityplotX<-axis_canvas(tmt.ril.acs.SL.lims1.et.scat, axis="x")+
      geom_density(aes(tmt.ril.acs.SL.lims1.et[,"l2.HYB.MP"], color=tmt.ril.acs.SL.lims1.et[,"Category"]), size=1)+
    scale_color_manual(values=c("Other"="#000000", "Cytosolic ribosome"="#CC79A7", "PE plastid ribosome"="#D55E00", "NE plastid ribosome"="#E69F00", "PE digenomic complexes"="#56B4E9", "NE digenomic complexes"="#009E73", "Ethylene biosynthesis"="#F0E442"), name="Category")+
      geom_vline(xintercept=0, linetype="dashed", color = "gray50")+
      theme_void()+
      theme(legend.title = element_blank())
    #set up marginal density plot Y axis
    densityplotY<-axis_canvas(tmt.ril.acs.SL.lims1.et.scat, axis="y", coord_flip = T)+
      geom_density(aes(tmt.ril.acs.SL.lims1.et[,"l2.ACS.B73"], color=tmt.ril.acs.SL.lims1.et[,"Category"]), size=1)+
      scale_color_manual(values=c("Other"="#000000", "Cytosolic ribosome"="#CC79A7", "PE plastid ribosome"="#D55E00", "NE plastid ribosome"="#E69F00", "PE digenomic complexes"="#56B4E9", "NE digenomic complexes"="#009E73", "Ethylene biosynthesis"="#F0E442"), name="Category")+
      scale_linetype_manual(values=c("dotted", "solid"))+
      geom_vline(xintercept=0, linetype="dashed", color = "gray50")+
      theme_void()+
      theme(legend.title = element_blank())+
      coord_flip()
    #put two plots together
      q <- insert_xaxis_grob(tmt.ril.acs.SL.lims1.et.scat, densityplotX, grid::unit(0.275, "null"), position = "top")
      q <- insert_yaxis_grob(q, densityplotY, grid::unit(0.275, "null"), position = "right")
    #save
     # ggsave(plot=q, filename = "tmt.ril.acs.SL.lims1.et.scat.png")
  ggsave(plot=q, filename = "tmt.ril.SL.lims1.et.scat.pdf", dpi=300, width=4.75, height = 5, scale = 1, units = "in")
```


#PCA
```{r}
tmt.OG.SL.reps <- tmt.OG.SL
tmt.OG.SL.reps$OG.BM.MP.1 <- tmt.OG.SL.reps$SeedlingLeaf.BM.r1/tmt.OG.SL.reps$MP.1
tmt.OG.SL.reps$OG.BM.MP.2 <- tmt.OG.SL.reps$SeedlingLeaf.BM.r2/tmt.OG.SL.reps$MP.2
tmt.OG.SL.reps$OG.BM.MP.3 <- tmt.OG.SL.reps$SeedlingLeaf.BM.r3/tmt.OG.SL.reps$MP.3

tmt.6h.SL.reps <- tmt.6h.SL
tmt.6h.SL.reps$H6.BM.MP.1 <- tmt.6h.SL.reps$X6Hybrid.B73xMo17.r1/tmt.6h.SL.reps$MP.1
tmt.6h.SL.reps$H6.BM.MP.2 <- tmt.6h.SL.reps$X6Hybrid.B73xMo17.r2/tmt.6h.SL.reps$MP.2
tmt.6h.SL.reps$H6.BM.MP.3 <- tmt.6h.SL.reps$X6Hybrid.B73xMo17.r3/tmt.6h.SL.reps$MP.3

tmt.dn.SL.reps <- tmt.dn.SL
tmt.dn.SL.reps$DN.BM.MP.1 <- tmt.dn.SL.reps$Day.BM.r1/tmt.dn.SL.reps$MP.1
tmt.dn.SL.reps$DN.BM.MP.2 <- tmt.dn.SL.reps$Day.BM.r2/tmt.dn.SL.reps$MP.2
tmt.dn.SL.reps$DN.BM.MP.3 <- tmt.dn.SL.reps$Day.BM.r3/tmt.dn.SL.reps$MP.3

tmt.ril.SL.reps <- tmt.ril.SL
tmt.ril.SL.reps$ril.BM.MP.1 <- tmt.ril.SL.reps$BM.1/tmt.ril.SL.reps$MP.1
tmt.ril.SL.reps$ril.BM.MP.2 <- tmt.ril.SL.reps$BM.3/tmt.ril.SL.reps$MP.2
tmt.ril.SL.reps$ril.BM.MP.3 <- tmt.ril.SL.reps$BM.4/tmt.ril.SL.reps$MP.3
```
```{r}
tmt.plastid4reps <- Reduce(function(x,y) merge(x = x, y = y, by = "Accession"), 
       list(tmt.ril.SL.reps[,c(6,31:33)], tmt.OG.SL.reps[,c(6,31:33)], tmt.6h.SL.reps[,c(6,31:33)], tmt.dn.SL.reps[,c(6,31:33)]))

rownames(tmt.plastid4reps) <- tmt.plastid4reps$Accession
tmt.plastid4reps <- tmt.plastid4reps[,-1]

tmt.plastid4reps <- na.omit(tmt.plastid4reps)

tmt.plastid4reps.t <- as.data.frame(t(tmt.plastid4reps))

tmt.plastid4reps.pca <- prcomp(tmt.plastid4reps.t, center = TRUE,scale. = TRUE)

tmt.plastid4reps.groups <- c(rep("RIL",3), rep("OG",3), rep("6H", 3), rep("DN", 3))

ggbiplot::ggbiplot(tmt.plastid4reps.pca,ellipse=TRUE, circle=TRUE, obs.scale = 1, var.scale = 1,var.axes=FALSE, labels=rownames(tmt.plastid4reps.t), groups=tmt.plastid4reps.groups)
ggsave("PCA_plastid4_plastidProteins.png")
```


#Combine plastid 4 for BioVis
```{r}
tmt.plastid4 <- Reduce(function(x,y) merge(x = x, y = y, by = "Accession", plastid=T), 
       list(tmt.OG.SL[,c(6,8,26)], tmt.ril.SL[,c(6,8,26)], tmt.6h.SL[,c(6,8,26)], tmt.dn.SL[,c(6,8,26)]))

colnames(tmt.plastid4) <- c("Accession", "OG.Protein", "OG.BM.MP", "RIL.Protein", "RIL.BM.MP", "H6.Protein", "H6.BM.MP", "DN.Protein", "DN.BM.MP")
```
```{r}
tmt.plastid4$Protein <- dplyr::coalesce(!!!list(tmt.plastid4$RIL.Protein, tmt.plastid4$OG.Protein, tmt.plastid4$H6.Protein, tmt.plastid4$DN.Protein))
```
```{r}
tmt.plastid4 <- tmt.plastid4[,c(1, 10, 3, 5, 7, 9)]
```
```{r}
tmt.plastid4$Gene <- substr(tmt.plastid4$Accession, start = 1, stop = 14)
tmt.plastid4.entrez <- merge.data.frame(tmt.plastid4, mgdb[,c(1,12)], by.x = "Gene", by.y = "v4_gene_model", plastid.x = T, plastid.y = F)
tmt.plastid4.entrez <- distinct(tmt.plastid4.entrez)
```
```{r}
tmt.plastid4.entrez$Entrez[tmt.plastid4.entrez$Accession=="ZemaCp003.1"] <- "845178"
tmt.plastid4.entrez$Entrez[tmt.plastid4.entrez$Accession=="ZemaCp004.1"] <- "845232"
tmt.plastid4.entrez$Entrez[tmt.plastid4.entrez$Accession=="ZemaCp008.1"] <- "845201"
tmt.plastid4.entrez$Entrez[tmt.plastid4.entrez$Accession=="ZemaCp013.1"] <- "845225"
tmt.plastid4.entrez$Entrez[tmt.plastid4.entrez$Accession=="ZemaCp014.1"] <- "845226"
tmt.plastid4.entrez$Entrez[tmt.plastid4.entrez$Accession=="ZemaCp016.1"] <- "845236"
tmt.plastid4.entrez$Entrez[tmt.plastid4.entrez$Accession=="ZemaCp017.1"] <- "845174"
tmt.plastid4.entrez$Entrez[tmt.plastid4.entrez$Accession=="ZemaCp018.1"] <- "845173"
tmt.plastid4.entrez$Entrez[tmt.plastid4.entrez$Accession=="ZemaCp019.1"] <- "845172"
tmt.plastid4.entrez$Entrez[tmt.plastid4.entrez$Accession=="ZemaCp019.1"] <- "845172"
tmt.plastid4.entrez$Entrez[tmt.plastid4.entrez$Accession=="ZemaCp021.1"] <- "845229"
tmt.plastid4.entrez$Entrez[tmt.plastid4.entrez$Accession=="ZemaCp022.1"] <- "845196"
tmt.plastid4.entrez$Entrez[tmt.plastid4.entrez$Accession=="ZemaCp023.1"] <- "845195"
tmt.plastid4.entrez$Entrez[tmt.plastid4.entrez$Accession=="ZemaCp024.1"] <- "1466361"
tmt.plastid4.entrez$Entrez[tmt.plastid4.entrez$Accession=="ZemaCp026.1"] <- "845238"
tmt.plastid4.entrez$Entrez[tmt.plastid4.entrez$Accession=="ZemaCp027.1"] <- "845189"
tmt.plastid4.entrez$Entrez[tmt.plastid4.entrez$Accession=="ZemaCp028.1"] <- "845190"
tmt.plastid4.entrez$Entrez[tmt.plastid4.entrez$Accession=="ZemaCp029.1"] <- "845182"
tmt.plastid4.entrez$Entrez[tmt.plastid4.entrez$Accession=="ZemaCp030.1"] <- "845171"
tmt.plastid4.entrez$Entrez[tmt.plastid4.entrez$Accession=="ZemaCp032.1"] <- "845212"
tmt.plastid4.entrez$Entrez[tmt.plastid4.entrez$Accession=="ZemaCp035.1"] <- "845175"
tmt.plastid4.entrez$Entrez[tmt.plastid4.entrez$Accession=="ZemaCp036.1"] <- "845191"
tmt.plastid4.entrez$Entrez[tmt.plastid4.entrez$Accession=="ZemaCp039.1"] <- "845204"
tmt.plastid4.entrez$Entrez[tmt.plastid4.entrez$Accession=="ZemaCp040.1"] <- "845203"
tmt.plastid4.entrez$Entrez[tmt.plastid4.entrez$Accession=="ZemaCp043.1"] <- "845198"
tmt.plastid4.entrez$Entrez[tmt.plastid4.entrez$Accession=="ZemaCp044.1"] <- "845222"
tmt.plastid4.entrez$Entrez[tmt.plastid4.entrez$Accession=="ZemaCp045.1"] <- "845233"
tmt.plastid4.entrez$Entrez[tmt.plastid4.entrez$Accession=="ZemaCp046.1"] <- "845217"
tmt.plastid4.entrez$Entrez[tmt.plastid4.entrez$Accession=="ZemaCp047.1"] <- "845176"
tmt.plastid4.entrez$Entrez[tmt.plastid4.entrez$Accession=="ZemaCp048.1"] <- "845200"
tmt.plastid4.entrez$Entrez[tmt.plastid4.entrez$Accession=="ZemaCp050.1"] <- "845211"
tmt.plastid4.entrez$Entrez[tmt.plastid4.entrez$Accession=="ZemaCp051.1"] <- "845205"
tmt.plastid4.entrez$Entrez[tmt.plastid4.entrez$Accession=="ZemaCp052.1"] <- "845192"
tmt.plastid4.entrez$Entrez[tmt.plastid4.entrez$Accession=="ZemaCp053.1"] <- "845193"
tmt.plastid4.entrez$Entrez[tmt.plastid4.entrez$Accession=="ZemaCp054.1"] <- "845224"
tmt.plastid4.entrez$Entrez[tmt.plastid4.entrez$Accession=="ZemaCp055.1"] <- "845228"
tmt.plastid4.entrez$Entrez[tmt.plastid4.entrez$Accession=="ZemaCp056.1"] <- "845223"
tmt.plastid4.entrez$Entrez[tmt.plastid4.entrez$Accession=="ZemaCp057.1"] <- "845177"
tmt.plastid4.entrez$Entrez[tmt.plastid4.entrez$Accession=="ZemaCp058.1"] <- "845241"
tmt.plastid4.entrez$Entrez[tmt.plastid4.entrez$Accession=="ZemaCp059.1"] <- "845213"
tmt.plastid4.entrez$Entrez[tmt.plastid4.entrez$Accession=="ZemaCp060.1"] <- "845214"
tmt.plastid4.entrez$Entrez[tmt.plastid4.entrez$Accession=="ZemaCp061.1"] <- "845237"
tmt.plastid4.entrez$Entrez[tmt.plastid4.entrez$Accession=="ZemaCp062.1"] <- "845218"
tmt.plastid4.entrez$Entrez[tmt.plastid4.entrez$Accession=="ZemaCp063.1"] <- "845235"
tmt.plastid4.entrez$Entrez[tmt.plastid4.entrez$Accession=="ZemaCp065.1"] <- "845216"
tmt.plastid4.entrez$Entrez[tmt.plastid4.entrez$Accession=="ZemaCp067.1"] <- "845220"
tmt.plastid4.entrez$Entrez[tmt.plastid4.entrez$Accession=="ZemaCp075.1"] <- "845181"
tmt.plastid4.entrez$Entrez[tmt.plastid4.entrez$Accession=="ZemaCp082.1"] <- "845230"
tmt.plastid4.entrez$Entrez[tmt.plastid4.entrez$Accession=="ZemaCp083.1"] <- "845185"
tmt.plastid4.entrez$Entrez[tmt.plastid4.entrez$Accession=="ZemaCp084.1"] <- "845221"
tmt.plastid4.entrez$Entrez[tmt.plastid4.entrez$Accession=="ZemaCp085.1"] <- "1466378"
tmt.plastid4.entrez$Entrez[tmt.plastid4.entrez$Accession=="ZemaCp086.1"] <- "845183"
tmt.plastid4.entrez$Entrez[tmt.plastid4.entrez$Accession=="ZemaCp087.1"] <- "1466379"
tmt.plastid4.entrez$Entrez[tmt.plastid4.entrez$Accession=="ZemaCp088.1"] <- "845184"
tmt.plastid4.entrez$Entrez[tmt.plastid4.entrez$Accession=="ZemaCp090.1"] <- "845188"
tmt.plastid4.entrez$Entrez[tmt.plastid4.entrez$Accession=="ZemaCp091.1"] <- "845179"
tmt.plastid4.entrez$Entrez[tmt.plastid4.entrez$Accession=="ZemaCp092.1"] <- "845187"

tmt.plastid4.entrez$Entrez[tmt.plastid4.entrez$Accession=="ZeamMp008.1"] <- "4055864"
tmt.plastid4.entrez$Entrez[tmt.plastid4.entrez$Accession=="ZeamMp024.1"] <- "4055860"
tmt.plastid4.entrez$Entrez[tmt.plastid4.entrez$Accession=="ZeamMp032.1"] <- "4055888"
tmt.plastid4.entrez$Entrez[tmt.plastid4.entrez$Accession=="ZeamMp033.1"] <- "4055938"
tmt.plastid4.entrez$Entrez[tmt.plastid4.entrez$Accession=="ZeamMp036.1"] <- "4055894"
tmt.plastid4.entrez$Entrez[tmt.plastid4.entrez$Accession=="ZeamMp050.1"] <- "4055956"
tmt.plastid4.entrez$Entrez[tmt.plastid4.entrez$Accession=="ZeamMp051.1"] <- "4055881"
tmt.plastid4.entrez$Entrez[tmt.plastid4.entrez$Accession=="ZeamMp057.1"] <- "4055973"
tmt.plastid4.entrez$Entrez[tmt.plastid4.entrez$Accession=="ZeamMp081.1"] <- "4055890"
tmt.plastid4.entrez$Entrez[tmt.plastid4.entrez$Accession=="ZeamMp092.1"] <- "4055837"
tmt.plastid4.entrez$Entrez[tmt.plastid4.entrez$Accession=="ZeamMp097.1"] <- "4055857"
tmt.plastid4.entrez$Entrez[tmt.plastid4.entrez$Accession=="ZeamMp098.1"] <- "4055896"
tmt.plastid4.entrez$Entrez[tmt.plastid4.entrez$Accession=="ZeamMp113.1"] <- "4055802"
tmt.plastid4.entrez$Entrez[tmt.plastid4.entrez$Accession=="ZeamMp119.1"] <- "4055928"
tmt.plastid4.entrez$Entrez[tmt.plastid4.entrez$Accession=="ZeamMp129.1"] <- "4055878"
tmt.plastid4.entrez$Entrez[tmt.plastid4.entrez$Accession=="ZeamMp136.1"] <- "4056004"
tmt.plastid4.entrez$Entrez[tmt.plastid4.entrez$Accession=="ZeamMp159.1"] <- "4055794"
tmt.plastid4.entrez$Entrez[tmt.plastid4.entrez$Accession=="ZeamMp167.1"] <- "4056007"
tmt.plastid4.entrez$Entrez[tmt.plastid4.entrez$Accession=="ZeamMp179.1"] <- "4056006"
tmt.plastid4.entrez$Entrez[tmt.plastid4.entrez$Accession=="ZeamMp185.1"] <- "4055892"
```
```{r}
saveRDS(tmt.plastid4.entrez, "tmt.plastid4.entrez")
```
```{r}
tmt.plastid4.entrez <- readRDS("tmt.plastid4.entrez")
```
##amp/bmp
```{r}
tmt.plastid4.entrez.trim <- tmt.plastid4.entrez[!duplicated(tmt.plastid4.entrez$Entrez),]
tmt.plastid4.entrez.trim <- na.omit(tmt.plastid4.entrez.trim)
tmt.plastid4.entrez.trim <- tmt.plastid4.entrez.trim[!tmt.plastid4.entrez.trim$Entrez=="",]

tmt.plastid4.entrez.trim$Avg.BM.MP <- rowMeans(tmt.plastid4.entrez.trim[,4:7])
tmt.plastid4.entrez.trim$l2.Avg.BM.MP <- log2(tmt.plastid4.entrez.trim$Avg.BM.MP)

tmt.plastid4.entrez.trim.pos <- tmt.plastid4.entrez.trim[(tmt.plastid4.entrez.trim$l2.Avg.BM.MP>0),]

tmt.plastid4.entrez.trim.neg <- tmt.plastid4.entrez.trim[(tmt.plastid4.entrez.trim$l2.Avg.BM.MP<0),]
tmt.plastid4.entrez.trim.neg$l2.Avg.BM.MP <- 0-tmt.plastid4.entrez.trim.neg$l2.Avg.BM.MP

tmt.plastid4.entrez.trim.amp <- tmt.plastid4.entrez.trim[(tmt.plastid4.entrez.trim$RIL.BM.MP>1.05 & tmt.plastid4.entrez.trim$OG.BM.MP>1.05 & tmt.plastid4.entrez.trim$H6.BM.MP>1.05 & tmt.plastid4.entrez.trim$DN.BM.MP>1.05),]

tmt.plastid4.entrez.trim.bmp <- tmt.plastid4.entrez.trim[(tmt.plastid4.entrez.trim$RIL.BM.MP<0.95 & tmt.plastid4.entrez.trim$OG.BM.MP<0.95 & tmt.plastid4.entrez.trim$H6.BM.MP<0.95 & tmt.plastid4.entrez.trim$DN.BM.MP<0.95),]
tmt.plastid4.entrez.trim.bmp$l2.Avg.BM.MP <- 0-tmt.plastid4.entrez.trim.bmp$l2.Avg.BM.MP
```


```{r}
readr::write_tsv(tmt.plastid4.entrez.trim.pos[,c(8,10)], "tmt.plastid4.entrez.trim.pos.tsv", col_names = F)
readr::write_tsv(tmt.plastid4.entrez.trim.neg[,c(8,10)], "tmt.plastid4.entrez.trim.neg.tsv", col_names = F)
readr::write_tsv(tmt.plastid4.entrez.trim.amp[,c(8,10)], "tmt.plastid4.entrez.trim.amp.tsv", col_names = F)
readr::write_tsv(tmt.plastid4.entrez.trim.bmp[,c(8,10)], "tmt.plastid4.entrez.trim.bmp.tsv", col_names = F)
```


#RNA
##OG
```{r}
cpm <- read_tsv("data/cpm.tsv")
```
```{r}
cpm.OG.SL <- as.data.frame(cpm[,c(1,178:186)])
```
```{r}
colnames(cpm.OG.SL) <- c("Gene", "B73.1", "B73.2", "B73.3", "Mo17.1", "Mo17.2", "Mo17.3", "BM.1", "BM.2", "BM.3")
```

```{r}
#Remove missing genes
cpm.OG.SL <- cpm.OG.SL[!(cpm.OG.SL$B73.1==0),]
cpm.OG.SL <- cpm.OG.SL[!(cpm.OG.SL$Mo17.1==0),]
cpm.OG.SL <- cpm.OG.SL[!(cpm.OG.SL$BM.1==0),]
#Calculate averages of bio reps
cpm.OG.SL$B73avg <- rowMeans(cpm.OG.SL[,2:4])
cpm.OG.SL$Mo17avg <- rowMeans(cpm.OG.SL[,5:7])
cpm.OG.SL$HYBavg <- rowMeans(cpm.OG.SL[,8:10])
cpm.OG.SL$MPavg <- rowMeans(cpm.OG.SL[,c(2:7)])
cpm.OG.SL$MP.1 <- rowMeans(cpm.OG.SL[,c(2,5)])
cpm.OG.SL$MP.2 <- rowMeans(cpm.OG.SL[,c(3,6)])
cpm.OG.SL$MP.3 <- rowMeans(cpm.OG.SL[,c(4,7)])
#calculate ratio
cpm.OG.SL$HYB.MP <- cpm.OG.SL$HYBavg/cpm.OG.SL$MPavg
#calculate log2(hyb/MP)
cpm.OG.SL$l2.BM.MP <- log2(cpm.OG.SL$HYB.MP)
#Calculate log2(B73/Mo17)
cpm.OG.SL$B73.Mo17 <- cpm.OG.SL$B73avg/cpm.OG.SL$Mo17avg
cpm.OG.SL$l2.B73.Mo17 <-log2(cpm.OG.SL$B73.Mo17)
saveRDS(cpm.OG.SL, "cpm.OG.SL")
```

###Scatterplots
####Scatterplot - plastid
#####Set limits
```{r}
cpm.OG.SL.lims1 <- cpm.OG.SL
cpm.OG.SL.lims1[,c(19,21)][cpm.OG.SL.lims1[,c(19,21)] <= -1.0] <- -1.0
cpm.OG.SL.lims1[,c(19,21)][cpm.OG.SL.lims1[,c(19,21)] >= 1.0] <- 1.0

cpm.OG.SL.lims5 <- cpm.OG.SL
cpm.OG.SL.lims5[,c(19,21)][cpm.OG.SL.lims5[,c(19,21)] <= -0.5] <- -0.5
cpm.OG.SL.lims5[,c(19,21)][cpm.OG.SL.lims5[,c(19,21)] >= 0.5] <- 0.5
```
#####Set GOIs
```{r}
cpm.OG.SL.lims1.plastid <- cpm.OG.SL.lims1

cpm.OG.SL.lims1.plastid$Category <- "Other"
cpm.OG.SL.lims1.plastid$Order <- 1

cpm.OG.SL.lims1.plastid[(cpm.OG.SL.lims1.plastid$Gene %in% eukRibo),"Category"] <- "Cytosolic ribosome"
cpm.OG.SL.lims1.plastid[(cpm.OG.SL.lims1.plastid$Gene %in% eukRibo),"Order"] <- 2

cpm.OG.SL.lims1.plastid[(cpm.OG.SL.lims1.plastid$Gene %in% Nuc.PtRibo),"Category"] <- "NE plastid ribosome"
cpm.OG.SL.lims1.plastid[(cpm.OG.SL.lims1.plastid$Gene %in% Nuc.PtRibo),"Order"] <- 3

cpm.OG.SL.lims1.plastid[(cpm.OG.SL.lims1.plastid$Gene %in% lightrxn.NucComplexes),"Category"] <- "NE digenomic complexes"
cpm.OG.SL.lims1.plastid[(cpm.OG.SL.lims1.plastid$Gene %in% lightrxn.NucComplexes),"Order"] <- 4

#Re-order
cpm.OG.SL.lims1.plastid <- cpm.OG.SL.lims1.plastid[(order(cpm.OG.SL.lims1.plastid$Order)),]
#factor
cpm.OG.SL.lims1.plastid$Category <- factor(cpm.OG.SL.lims1.plastid$Category, levels = c("Other", "Cytosolic ribosome", "NE plastid ribosome","NE digenomic complexes"))
```

#####Scatterplot
```{r}
cpm.OG.SL.lims1.plastid.scat <- ggplot(cpm.OG.SL.lims1.plastid, mapping=aes(x=l2.BM.MP, y=l2.B73.Mo17, color=Category))+
    geom_point(aes(color=Category), size=3, alpha=0.8)+
  scale_color_manual(values=c("Other"="#000000", "Cytosolic ribosome"="#CC79A7", "NE plastid ribosome"="#E69F00", "NE digenomic complexes"="#009E73"), name="Category")+
    theme_classic()+
          xlim(c(-1,1))+
          ylim(c(-1,1))+
          geom_vline(xintercept=0, linetype="dashed", color = "gray50")+
          geom_hline(yintercept=0, linetype="dashed", color = "gray50")+
          theme(axis.text=element_text(size=10), legend.text=element_text(size=10),
                legend.position="bottom", legend.title = element_blank())+
          guides(col=guide_legend(nrow=2))+
    xlab("log2 BxM/MP")+
    ylab("log2 B73/Mo17")+
    coord_fixed(expand = F)
```
#####Add density curves
```{r}
    densityplotX<-axis_canvas(cpm.OG.SL.lims1.plastid.scat, axis="x")+
      geom_density(aes(cpm.OG.SL.lims1.plastid[,"l2.BM.MP"], color=cpm.OG.SL.lims1.plastid[,"Category"]), size=1)+
    scale_color_manual(values=c("Other"="#000000", "Cytosolic ribosome"="#CC79A7", "NE plastid ribosome"="#E69F00","NE digenomic complexes"="#009E73"), name="Category")+
      geom_vline(xintercept=0, linetype="dashed", color = "gray50")+
      theme_void()+
      theme(legend.title = element_blank())
    #set up marginal density plot Y axis
    densityplotY<-axis_canvas(cpm.OG.SL.lims1.plastid.scat, axis="y", coord_flip = T)+
      geom_density(aes(cpm.OG.SL.lims1.plastid[,"l2.B73.Mo17"], color=cpm.OG.SL.lims1.plastid[,"Category"]), size=1)+
      scale_color_manual(values=c("Other"="#000000", "Cytosolic ribosome"="#CC79A7", "NE plastid ribosome"="#E69F00", "NE digenomic complexes"="#009E73"), name="Category")+
      scale_linetype_manual(values=c("dotted", "solid"))+
      geom_vline(xintercept=0, linetype="dashed", color = "gray50")+
      theme_void()+
      theme(legend.title = element_blank())+
      coord_flip()
    #put two plots together
      q <- insert_xaxis_grob(cpm.OG.SL.lims1.plastid.scat, densityplotX, grid::unit(0.275, "null"), position = "top")
      q <- insert_yaxis_grob(q, densityplotY, grid::unit(0.275, "null"), position = "right")
    #save
     # ggsave(plot=q, filename = "cpm.OG.SL.lims1.plastid.scat.png")
  ggsave(plot=q, filename = "cpm.OG.SL.lims1.plastid.scat.pdf",  dpi=300, width=4.75, height = 5, scale = 1, units = "in")
```


####Scatterplot - Mitochondria
#####Set GOIs
```{r}
cpm.OG.SL.lims1.mito <- cpm.OG.SL.lims1

cpm.OG.SL.lims1.mito$Category <- "Other"
cpm.OG.SL.lims1.mito$Order <- 1

cpm.OG.SL.lims1.mito[(cpm.OG.SL.lims1.mito$Gene %in% OxPhos),"Category"] <- "NE oxidative phosphorylation"
cpm.OG.SL.lims1.mito[(cpm.OG.SL.lims1.mito$Gene %in% OxPhos),"Order"] <- 2

cpm.OG.SL.lims1.mito[(cpm.OG.SL.lims1.mito$Gene %in% Nuc.MtRibo),"Category"] <- "NE mitochondrial ribosome"
cpm.OG.SL.lims1.mito[(cpm.OG.SL.lims1.mito$Gene %in% Nuc.MtRibo),"Order"] <- 3

#Re-order
cpm.OG.SL.lims1.mito <- cpm.OG.SL.lims1.mito[(order(cpm.OG.SL.lims1.mito$Order)),]
#factor
cpm.OG.SL.lims1.mito$Category <- factor(cpm.OG.SL.lims1.mito$Category, levels = c("Other", "NE oxidative phosphorylation", "NE mitochondrial ribosome"))
```
#####Scatterplot
```{r}
cpm.OG.SL.lims1.mito.scat <- ggplot(cpm.OG.SL.lims1.mito, mapping=aes(x=l2.BM.MP, y=l2.B73.Mo17, color=Category))+
    geom_point(aes(color=Category), size=3, alpha=0.8)+
  scale_color_manual(values=c("Other"="#000000", "NE mitochondrial ribosome"="#E69F00", "NE oxidative phosphorylation"="#009E73"), name="Category")+
    theme_classic()+
          xlim(c(-1,1))+
          ylim(c(-1,1))+
          geom_vline(xintercept=0, linetype="dashed", color = "gray50")+
          geom_hline(yintercept=0, linetype="dashed", color = "gray50")+
          theme(axis.text=element_text(size=10), legend.text=element_text(size=10),
                legend.position="bottom", legend.title = element_blank())+
          guides(col=guide_legend(nrow = 2))+
    xlab("log2 BxM/MP")+
    ylab("log2 B73/Mo17")+
    coord_fixed(expand = F)
```
#####Add density curves
```{r}
    densityplotX<-axis_canvas(cpm.OG.SL.lims1.mito.scat, axis="x")+
      geom_density(aes(cpm.OG.SL.lims1.mito[,"l2.BM.MP"], color=cpm.OG.SL.lims1.mito[,"Category"]), size=1)+
    scale_color_manual(values=c("Other"="#000000", "NE mitochondrial ribosome"="#E69F00", "NE oxidative phosphorylation"="#009E73"), name="Category")+
      geom_vline(xintercept=0, linetype="dashed", color = "gray50")+
      theme_void()+
      theme(legend.title = element_blank())
    #set up marginal density plot Y axis
    densityplotY<-axis_canvas(cpm.OG.SL.lims1.mito.scat, axis="y", coord_flip = T)+
      geom_density(aes(cpm.OG.SL.lims1.mito[,"l2.B73.Mo17"], color=cpm.OG.SL.lims1.mito[,"Category"]), size=1)+
      scale_color_manual(values=c("Other"="#000000", "NE mitochondrial ribosome"="#E69F00", "NE oxidative phosphorylation"="#009E73"), name="Category")+
      scale_linetype_manual(values=c("dotted", "solid"))+
      geom_vline(xintercept=0, linetype="dashed", color = "gray50")+
      theme_void()+
      theme(legend.title = element_blank())+
      coord_flip()
    #put two plots together
      q <- insert_xaxis_grob(cpm.OG.SL.lims1.mito.scat, densityplotX, grid::unit(0.275, "null"), position = "top")
      q <- insert_yaxis_grob(q, densityplotY, grid::unit(0.275, "null"), position = "right")
    #save
    #  ggsave(plot=q, filename = "cpm.OG.SL.lims1.mito.scat.png")
  ggsave(plot=q, filename = "cpm.OG.SL.lims1.mito.scat.pdf", dpi=300, width=4.75, height = 5, scale = 1, units = "in")
```




##6H
###Setup data
```{r}
rna.6h <- readRDS("data/rn20b.rds")
```
```{r}
rna.6h.th <- rna.6h$th
rna.6h.fcnts <- rna.6h$fcnts
rna.6h.salmon <- rna.6h$salmon
rna.6h.salmontranscript <- rna.6h$salmon_transcript
rna.6h.tl <- rna.6h$tl
rna.6h.tm <- rna.6h$tm
rna.6h.th_m <- rna.6h$th_m
rna.6h.tm_m <- rna.6h$tm_m
```

```{r}
rna.6h.tm_m.B73 <- rna.6h.tm_m[rna.6h.tm_m$SampleID=="s007",]
rna.6h.tm_m.Mo17 <- rna.6h.tm_m[rna.6h.tm_m$SampleID=="s017",]
rna.6h.tm_m.BM <- rna.6h.tm_m[rna.6h.tm_m$SampleID=="s009",]
```
```{r}
cpm.6h.SL <- data.frame("Gene"=rna.6h.tm_m.B73$gid, "B73"=rna.6h.tm_m.B73$CPM, "Mo17"=rna.6h.tm_m.Mo17$CPM, "BM"=rna.6h.tm_m.BM$CPM)
```
```{r}
#Remove genes not detected
cpm.6h.SL <- cpm.6h.SL[!(rowSums(cpm.6h.SL[,2:4])==0),]
#Calculate log2 hybrid/MP values
cpm.6h.SL$l2.BM.MP <- log2(cpm.6h.SL$BM/(rowMeans(cpm.6h.SL[,2:3])))
#Calculate log2 parent/parent values
cpm.6h.SL$l2.B73.Mo17 <- log2(cpm.6h.SL$B73/cpm.6h.SL$Mo17)
```
###Scatterplots
####Scatterplot - plastid
#####Set limits
```{r}
colnames(cpm.6h.SL)
```

```{r}
cpm.6h.SL.lims1 <- cpm.6h.SL
cpm.6h.SL.lims1[,c(5:6)][cpm.6h.SL.lims1[,c(5:6)] <= -1.0] <- -1.0
cpm.6h.SL.lims1[,c(5:6)][cpm.6h.SL.lims1[,c(5:6)] >= 1.0] <- 1.0

cpm.6h.SL.lims5 <- cpm.6h.SL
cpm.6h.SL.lims5[,c(5:6)][cpm.6h.SL.lims5[,c(5:6)] <= -0.5] <- -0.5
cpm.6h.SL.lims5[,c(5:6)][cpm.6h.SL.lims5[,c(5:6)] >= 0.5] <- 0.5
```
#####Set GOIs
```{r}
cpm.6h.SL.lims1.plastid <- cpm.6h.SL.lims1

cpm.6h.SL.lims1.plastid$Category <- "Other"
cpm.6h.SL.lims1.plastid$Order <- 1

cpm.6h.SL.lims1.plastid[(cpm.6h.SL.lims1.plastid$Gene %in% eukRibo),"Category"] <- "Cytosolic ribosome"
cpm.6h.SL.lims1.plastid[(cpm.6h.SL.lims1.plastid$Gene %in% eukRibo),"Order"] <- 2

cpm.6h.SL.lims1.plastid[(cpm.6h.SL.lims1.plastid$Gene %in% Nuc.PtRibo),"Category"] <- "NE plastid ribosome"
cpm.6h.SL.lims1.plastid[(cpm.6h.SL.lims1.plastid$Gene %in% Nuc.PtRibo),"Order"] <- 3

cpm.6h.SL.lims1.plastid[(cpm.6h.SL.lims1.plastid$Gene %in% lightrxn.NucComplexes),"Category"] <- "NE digenomic complexes"
cpm.6h.SL.lims1.plastid[(cpm.6h.SL.lims1.plastid$Gene %in% lightrxn.NucComplexes),"Order"] <- 4

#Re-order
cpm.6h.SL.lims1.plastid <- cpm.6h.SL.lims1.plastid[(order(cpm.6h.SL.lims1.plastid$Order)),]
#factor
cpm.6h.SL.lims1.plastid$Category <- factor(cpm.6h.SL.lims1.plastid$Category, levels = c("Other", "Cytosolic ribosome", "NE plastid ribosome","NE digenomic complexes"))
```

#####Scatterplot
```{r}
cpm.6h.SL.lims1.plastid.scat <- ggplot(cpm.6h.SL.lims1.plastid, mapping=aes(x=l2.BM.MP, y=l2.B73.Mo17, color=Category))+
    geom_point(aes(color=Category), size=3, alpha=0.8)+
  scale_color_manual(values=c("Other"="#000000", "Cytosolic ribosome"="#CC79A7", "NE plastid ribosome"="#E69F00", "NE digenomic complexes"="#009E73"), name="Category")+
    theme_classic()+
          xlim(c(-1,1))+
          ylim(c(-1,1))+
          geom_vline(xintercept=0, linetype="dashed", color = "gray50")+
          geom_hline(yintercept=0, linetype="dashed", color = "gray50")+
          theme(axis.text=element_text(size=10), legend.text=element_text(size=10),
                legend.position="bottom", legend.title = element_blank())+
          guides(col=guide_legend(nrow=2))+
    xlab("log2 BxM/MP")+
    ylab("log2 B73/Mo17")+
    coord_fixed(expand = F)
```
#####Add density curves
```{r}
    densityplotX<-axis_canvas(cpm.6h.SL.lims1.plastid.scat, axis="x")+
      geom_density(aes(cpm.6h.SL.lims1.plastid[,"l2.BM.MP"], color=cpm.6h.SL.lims1.plastid[,"Category"]), size=1)+
    scale_color_manual(values=c("Other"="#000000", "Cytosolic ribosome"="#CC79A7", "NE plastid ribosome"="#E69F00","NE digenomic complexes"="#009E73"), name="Category")+
      geom_vline(xintercept=0, linetype="dashed", color = "gray50")+
      theme_void()+
      theme(legend.title = element_blank())
    #set up marginal density plot Y axis
    densityplotY<-axis_canvas(cpm.6h.SL.lims1.plastid.scat, axis="y", coord_flip = T)+
      geom_density(aes(cpm.6h.SL.lims1.plastid[,"l2.B73.Mo17"], color=cpm.6h.SL.lims1.plastid[,"Category"]), size=1)+
      scale_color_manual(values=c("Other"="#000000", "Cytosolic ribosome"="#CC79A7", "NE plastid ribosome"="#E69F00", "NE digenomic complexes"="#009E73"), name="Category")+
      scale_linetype_manual(values=c("dotted", "solid"))+
      geom_vline(xintercept=0, linetype="dashed", color = "gray50")+
      theme_void()+
      theme(legend.title = element_blank())+
      coord_flip()
    #put two plots together
      q <- insert_xaxis_grob(cpm.6h.SL.lims1.plastid.scat, densityplotX, grid::unit(0.275, "null"), position = "top")
      q <- insert_yaxis_grob(q, densityplotY, grid::unit(0.275, "null"), position = "right")
    #save
     # ggsave(plot=q, filename = "cpm.6h.SL.lims1.plastid.scat.png")
  ggsave(plot=q, filename = "cpm.6h.SL.lims1.plastid.scat.pdf",  dpi=300, width=4.75, height = 5, scale = 1, units = "in")
```
####Scatterplot - Mitochondria
#####Set GOIs
```{r}
cpm.6h.SL.lims1.mito <- cpm.6h.SL.lims1

cpm.6h.SL.lims1.mito$Category <- "Other"
cpm.6h.SL.lims1.mito$Order <- 1

cpm.6h.SL.lims1.mito[(cpm.6h.SL.lims1.mito$Gene %in% OxPhos),"Category"] <- "NE oxidative phosphorylation"
cpm.6h.SL.lims1.mito[(cpm.6h.SL.lims1.mito$Gene %in% OxPhos),"Order"] <- 2

cpm.6h.SL.lims1.mito[(cpm.6h.SL.lims1.mito$Gene %in% Nuc.MtRibo),"Category"] <- "NE mitochondrial ribosome"
cpm.6h.SL.lims1.mito[(cpm.6h.SL.lims1.mito$Gene %in% Nuc.MtRibo),"Order"] <- 3

#Re-order
cpm.6h.SL.lims1.mito <- cpm.6h.SL.lims1.mito[(order(cpm.6h.SL.lims1.mito$Order)),]
#factor
cpm.6h.SL.lims1.mito$Category <- factor(cpm.6h.SL.lims1.mito$Category, levels = c("Other", "NE oxidative phosphorylation", "NE mitochondrial ribosome"))
```
#####Scatterplot
```{r}
cpm.6h.SL.lims1.mito.scat <- ggplot(cpm.6h.SL.lims1.mito, mapping=aes(x=l2.BM.MP, y=l2.B73.Mo17, color=Category))+
    geom_point(aes(color=Category), size=3, alpha=0.8)+
  scale_color_manual(values=c("Other"="#000000", "NE mitochondrial ribosome"="#E69F00", "NE oxidative phosphorylation"="#009E73"), name="Category")+
    theme_classic()+
          xlim(c(-1,1))+
          ylim(c(-1,1))+
          geom_vline(xintercept=0, linetype="dashed", color = "gray50")+
          geom_hline(yintercept=0, linetype="dashed", color = "gray50")+
          theme(axis.text=element_text(size=10), legend.text=element_text(size=10),
                legend.position="bottom", legend.title = element_blank())+
          guides(col=guide_legend(nrow = 2))+
    xlab("log2 BxM/MP")+
    ylab("log2 B73/Mo17")+
    coord_fixed(expand = F)
```
#####Add density curves
```{r}
    densityplotX<-axis_canvas(cpm.6h.SL.lims1.mito.scat, axis="x")+
      geom_density(aes(cpm.6h.SL.lims1.mito[,"l2.BM.MP"], color=cpm.6h.SL.lims1.mito[,"Category"]), size=1)+
    scale_color_manual(values=c("Other"="#000000", "NE mitochondrial ribosome"="#E69F00", "NE oxidative phosphorylation"="#009E73"), name="Category")+
      geom_vline(xintercept=0, linetype="dashed", color = "gray50")+
      theme_void()+
      theme(legend.title = element_blank())
    #set up marginal density plot Y axis
    densityplotY<-axis_canvas(cpm.6h.SL.lims1.mito.scat, axis="y", coord_flip = T)+
      geom_density(aes(cpm.6h.SL.lims1.mito[,"l2.B73.Mo17"], color=cpm.6h.SL.lims1.mito[,"Category"]), size=1)+
      scale_color_manual(values=c("Other"="#000000", "NE mitochondrial ribosome"="#E69F00", "NE oxidative phosphorylation"="#009E73"), name="Category")+
      scale_linetype_manual(values=c("dotted", "solid"))+
      geom_vline(xintercept=0, linetype="dashed", color = "gray50")+
      theme_void()+
      theme(legend.title = element_blank())+
      coord_flip()
    #put two plots together
      q <- insert_xaxis_grob(cpm.6h.SL.lims1.mito.scat, densityplotX, grid::unit(0.275, "null"), position = "top")
      q <- insert_yaxis_grob(q, densityplotY, grid::unit(0.275, "null"), position = "right")
    #save
    #  ggsave(plot=q, filename = "cpm.6h.SL.lims1.mito.scat.png")
  ggsave(plot=q, filename = "cpm.6h.SL.lims1.mito.scat.pdf", dpi=300, width=4.75, height = 5, scale = 1, units = "in")
```









##RIL
###Setup data
```{r}
rna.ril <- readRDS("data/rn20b2.rds")
```
```{r}
rna.ril.th <- rna.ril$th
rna.ril.fcnts <- rna.ril$fcnts
rna.ril.salmon <- rna.ril$salmon
rna.ril.salmontranscript <- rna.ril$salmon_transcript
rna.ril.tl <- rna.ril$tl
rna.ril.tm <- rna.ril$tm
rna.ril.th_m <- rna.ril$th_m
rna.ril.tm_m <- rna.ril$tm_m
```
```{r}
rna.ril.tm_m.B73 <- rna.ril.tm_m[rna.ril.tm_m$SampleID=="s001",]
rna.ril.tm_m.BM <- rna.ril.tm_m[rna.ril.tm_m$SampleID=="s002",]
rna.ril.tm_m.14 <- rna.ril.tm_m[rna.ril.tm_m$SampleID=="s003",]
rna.ril.tm_m.14B <- rna.ril.tm_m[rna.ril.tm_m$SampleID=="s004",]
rna.ril.tm_m.14M <- rna.ril.tm_m[rna.ril.tm_m$SampleID=="s005",]
rna.ril.tm_m.16 <- rna.ril.tm_m[rna.ril.tm_m$SampleID=="s006",]
rna.ril.tm_m.16B <- rna.ril.tm_m[rna.ril.tm_m$SampleID=="s007",]
rna.ril.tm_m.16M <- rna.ril.tm_m[rna.ril.tm_m$SampleID=="s008",]
rna.ril.tm_m.21 <- rna.ril.tm_m[rna.ril.tm_m$SampleID=="s009",]
rna.ril.tm_m.21B <- rna.ril.tm_m[rna.ril.tm_m$SampleID=="s010",]
rna.ril.tm_m.21M <- rna.ril.tm_m[rna.ril.tm_m$SampleID=="s011",]
rna.ril.tm_m.317 <- rna.ril.tm_m[rna.ril.tm_m$SampleID=="s012",]
rna.ril.tm_m.317B <- rna.ril.tm_m[rna.ril.tm_m$SampleID=="s013",]
rna.ril.tm_m.317M <- rna.ril.tm_m[rna.ril.tm_m$SampleID=="s014",]
rna.ril.tm_m.Mo17 <- rna.ril.tm_m[rna.ril.tm_m$SampleID=="s015",]
```
```{r}
cpm.ril.SL <- data.frame("Gene"=rna.ril.tm_m.B73$gid, "B73"=rna.ril.tm_m.B73$CPM, "Mo17"=rna.ril.tm_m.Mo17$CPM, "BM"=rna.ril.tm_m.BM$CPM, "M0014"=rna.ril.tm_m.14$CPM, "M0014xB"=rna.ril.tm_m.14B$CPM, "M0014xM"=rna.ril.tm_m.14M$CPM, "M0016"=rna.ril.tm_m.14$CPM, "M0016xB"=rna.ril.tm_m.16B$CPM, "M0016xM"=rna.ril.tm_m.16M$CPM, "M0021"=rna.ril.tm_m.14$CPM, "M0021xB"=rna.ril.tm_m.21B$CPM, "M0021xM"=rna.ril.tm_m.21M$CPM, "M0317"=rna.ril.tm_m.14$CPM, "M0317xB"=rna.ril.tm_m.317B$CPM, "M0317xM"=rna.ril.tm_m.317M$CPM)
```
```{r}
#Remove genes not detected
cpm.ril.SL <- cpm.ril.SL[!(rowSums(cpm.ril.SL[,2:8])==0),]
```
```{r}
#Calculate log2 hybrid/MP values
cpm.ril.SL$l2.BM.MP <- log2(cpm.ril.SL$BM/(rowMeans(cpm.ril.SL[,2:3])))
cpm.ril.SL$l2.M0014xB.MP <- log2(cpm.ril.SL$M0014xB/(rowMeans(cpm.ril.SL[,c(2,5)])))
cpm.ril.SL$l2.M0014xM.MP <- log2(cpm.ril.SL$M0014xM/(rowMeans(cpm.ril.SL[,c(3,5)])))
cpm.ril.SL$l2.M0016xB.MP <- log2(cpm.ril.SL$M0016xB/(rowMeans(cpm.ril.SL[,c(2,8)])))
cpm.ril.SL$l2.M0016xM.MP <- log2(cpm.ril.SL$M0016xM/(rowMeans(cpm.ril.SL[,c(3,8)])))
cpm.ril.SL$l2.M0021xB.MP <- log2(cpm.ril.SL$M0021xB/(rowMeans(cpm.ril.SL[,c(2,11)])))
cpm.ril.SL$l2.M0021xM.MP <- log2(cpm.ril.SL$M0021xM/(rowMeans(cpm.ril.SL[,c(3,11)])))
cpm.ril.SL$l2.M0317xB.MP <- log2(cpm.ril.SL$M0317xB/(rowMeans(cpm.ril.SL[,c(2,14)])))
cpm.ril.SL$l2.M0317xM.MP <- log2(cpm.ril.SL$M0317xM/(rowMeans(cpm.ril.SL[,c(3,14)])))
```
```{r}
#Calculate log2 parent/parent values
cpm.ril.SL$l2.B73.Mo17 <- log2(cpm.ril.SL$B73/cpm.ril.SL$Mo17)
cpm.ril.SL$l2.M0014.B73 <- log2(cpm.ril.SL$M0014/cpm.ril.SL$B73)
cpm.ril.SL$l2.M0014.Mo17 <- log2(cpm.ril.SL$M0014/cpm.ril.SL$Mo17)
cpm.ril.SL$l2.M0016.B73 <- log2(cpm.ril.SL$M0016/cpm.ril.SL$B73)
cpm.ril.SL$l2.M0016.Mo17 <- log2(cpm.ril.SL$M0016/cpm.ril.SL$Mo17)
cpm.ril.SL$l2.M0021.B73 <- log2(cpm.ril.SL$M0021/cpm.ril.SL$B73)
cpm.ril.SL$l2.M0021.Mo17 <- log2(cpm.ril.SL$M0021/cpm.ril.SL$Mo17)
cpm.ril.SL$l2.M0317.B73 <- log2(cpm.ril.SL$M0317/cpm.ril.SL$B73)
cpm.ril.SL$l2.M0317.Mo17 <- log2(cpm.ril.SL$M0317/cpm.ril.SL$Mo17)
```
###Scatterplots
####Scatterplot - plastid
#####Set limits
```{r}
colnames(cpm.ril.SL)
```

```{r}
cpm.ril.SL.lims1 <- cpm.ril.SL
cpm.ril.SL.lims1[,c(17:34)][cpm.ril.SL.lims1[,c(17:34)] <= -1.0] <- -1.0
cpm.ril.SL.lims1[,c(17:34)][cpm.ril.SL.lims1[,c(17:34)] >= 1.0] <- 1.0

cpm.ril.SL.lims5 <- cpm.ril.SL
cpm.ril.SL.lims5[,c(17:34)][cpm.ril.SL.lims5[,c(17:34)] <= -0.5] <- -0.5
cpm.ril.SL.lims5[,c(17:34)][cpm.ril.SL.lims5[,c(17:34)] >= 0.5] <- 0.5
```
#####Set GOIs
```{r}
cpm.ril.SL.lims1.plastid <- cpm.ril.SL.lims1

cpm.ril.SL.lims1.plastid$Category <- "Other"
cpm.ril.SL.lims1.plastid$Order <- 1

cpm.ril.SL.lims1.plastid[(cpm.ril.SL.lims1.plastid$Gene %in% eukRibo),"Category"] <- "Cytosolic ribosome"
cpm.ril.SL.lims1.plastid[(cpm.ril.SL.lims1.plastid$Gene %in% eukRibo),"Order"] <- 2

cpm.ril.SL.lims1.plastid[(cpm.ril.SL.lims1.plastid$Gene %in% Nuc.PtRibo),"Category"] <- "NE plastid ribosome"
cpm.ril.SL.lims1.plastid[(cpm.ril.SL.lims1.plastid$Gene %in% Nuc.PtRibo),"Order"] <- 3

cpm.ril.SL.lims1.plastid[(cpm.ril.SL.lims1.plastid$Gene %in% lightrxn.NucComplexes),"Category"] <- "NE digenomic complexes"
cpm.ril.SL.lims1.plastid[(cpm.ril.SL.lims1.plastid$Gene %in% lightrxn.NucComplexes),"Order"] <- 4

#Re-order
cpm.ril.SL.lims1.plastid <- cpm.ril.SL.lims1.plastid[(order(cpm.ril.SL.lims1.plastid$Order)),]
#factor
cpm.ril.SL.lims1.plastid$Category <- factor(cpm.ril.SL.lims1.plastid$Category, levels = c("Other", "Cytosolic ribosome", "NE plastid ribosome","NE digenomic complexes"))
```

#####Scatterplot
```{r}
cpm.ril.SL.lims1.plastid.scat <- ggplot(cpm.ril.SL.lims1.plastid, mapping=aes(x=l2.BM.MP, y=l2.B73.Mo17, color=Category))+
    geom_point(aes(color=Category), size=3, alpha=0.8)+
  scale_color_manual(values=c("Other"="#000000", "Cytosolic ribosome"="#CC79A7", "NE plastid ribosome"="#E69F00", "NE digenomic complexes"="#009E73"), name="Category")+
    theme_classic()+
          xlim(c(-1,1))+
          ylim(c(-1,1))+
          geom_vline(xintercept=0, linetype="dashed", color = "gray50")+
          geom_hline(yintercept=0, linetype="dashed", color = "gray50")+
          theme(axis.text=element_text(size=10), legend.text=element_text(size=10),
                legend.position="bottom", legend.title = element_blank())+
          guides(col=guide_legend(nrow=2))+
    xlab("log2 BxM/MP")+
    ylab("log2 B73/Mo17")+
    coord_fixed(expand = F)
```
#####Add density curves
```{r}
    densityplotX<-axis_canvas(cpm.ril.SL.lims1.plastid.scat, axis="x")+
      geom_density(aes(cpm.ril.SL.lims1.plastid[,"l2.BM.MP"], color=cpm.ril.SL.lims1.plastid[,"Category"]), size=1)+
    scale_color_manual(values=c("Other"="#000000", "Cytosolic ribosome"="#CC79A7", "NE plastid ribosome"="#E69F00","NE digenomic complexes"="#009E73"), name="Category")+
      geom_vline(xintercept=0, linetype="dashed", color = "gray50")+
      theme_void()+
      theme(legend.title = element_blank())
    #set up marginal density plot Y axis
    densityplotY<-axis_canvas(cpm.ril.SL.lims1.plastid.scat, axis="y", coord_flip = T)+
      geom_density(aes(cpm.ril.SL.lims1.plastid[,"l2.B73.Mo17"], color=cpm.ril.SL.lims1.plastid[,"Category"]), size=1)+
      scale_color_manual(values=c("Other"="#000000", "Cytosolic ribosome"="#CC79A7", "NE plastid ribosome"="#E69F00", "NE digenomic complexes"="#009E73"), name="Category")+
      scale_linetype_manual(values=c("dotted", "solid"))+
      geom_vline(xintercept=0, linetype="dashed", color = "gray50")+
      theme_void()+
      theme(legend.title = element_blank())+
      coord_flip()
    #put two plots together
      q <- insert_xaxis_grob(cpm.ril.SL.lims1.plastid.scat, densityplotX, grid::unit(0.275, "null"), position = "top")
      q <- insert_yaxis_grob(q, densityplotY, grid::unit(0.275, "null"), position = "right")
    #save
     # ggsave(plot=q, filename = "cpm.ril.SL.lims1.plastid.scat.png")
  ggsave(plot=q, filename = "cpm.ril.SL.lims1.plastid.scat.pdf",  dpi=300, width=4.75, height = 5, scale = 1, units = "in")
```
####Scatterplot - Mitochondria
#####Set GOIs
```{r}
cpm.ril.SL.lims1.mito <- cpm.ril.SL.lims1

cpm.ril.SL.lims1.mito$Category <- "Other"
cpm.ril.SL.lims1.mito$Order <- 1

cpm.ril.SL.lims1.mito[(cpm.ril.SL.lims1.mito$Gene %in% OxPhos),"Category"] <- "NE oxidative phosphorylation"
cpm.ril.SL.lims1.mito[(cpm.ril.SL.lims1.mito$Gene %in% OxPhos),"Order"] <- 2

cpm.ril.SL.lims1.mito[(cpm.ril.SL.lims1.mito$Gene %in% Nuc.MtRibo),"Category"] <- "NE mitochondrial ribosome"
cpm.ril.SL.lims1.mito[(cpm.ril.SL.lims1.mito$Gene %in% Nuc.MtRibo),"Order"] <- 3

#Re-order
cpm.ril.SL.lims1.mito <- cpm.ril.SL.lims1.mito[(order(cpm.ril.SL.lims1.mito$Order)),]
#factor
cpm.ril.SL.lims1.mito$Category <- factor(cpm.ril.SL.lims1.mito$Category, levels = c("Other", "NE oxidative phosphorylation", "NE mitochondrial ribosome"))
```
#####Scatterplot
```{r}
cpm.ril.SL.lims1.mito.scat <- ggplot(cpm.ril.SL.lims1.mito, mapping=aes(x=l2.BM.MP, y=l2.B73.Mo17, color=Category))+
    geom_point(aes(color=Category), size=3, alpha=0.8)+
  scale_color_manual(values=c("Other"="#000000", "NE mitochondrial ribosome"="#E69F00", "NE oxidative phosphorylation"="#009E73"), name="Category")+
    theme_classic()+
          xlim(c(-1,1))+
          ylim(c(-1,1))+
          geom_vline(xintercept=0, linetype="dashed", color = "gray50")+
          geom_hline(yintercept=0, linetype="dashed", color = "gray50")+
          theme(axis.text=element_text(size=10), legend.text=element_text(size=10),
                legend.position="bottom", legend.title = element_blank())+
          guides(col=guide_legend(nrow = 2))+
    xlab("log2 BxM/MP")+
    ylab("log2 B73/Mo17")+
    coord_fixed(expand = F)
```
#####Add density curves
```{r}
    densityplotX<-axis_canvas(cpm.ril.SL.lims1.mito.scat, axis="x")+
      geom_density(aes(cpm.ril.SL.lims1.mito[,"l2.BM.MP"], color=cpm.ril.SL.lims1.mito[,"Category"]), size=1)+
    scale_color_manual(values=c("Other"="#000000", "NE mitochondrial ribosome"="#E69F00", "NE oxidative phosphorylation"="#009E73"), name="Category")+
      geom_vline(xintercept=0, linetype="dashed", color = "gray50")+
      theme_void()+
      theme(legend.title = element_blank())
    #set up marginal density plot Y axis
    densityplotY<-axis_canvas(cpm.ril.SL.lims1.mito.scat, axis="y", coord_flip = T)+
      geom_density(aes(cpm.ril.SL.lims1.mito[,"l2.B73.Mo17"], color=cpm.ril.SL.lims1.mito[,"Category"]), size=1)+
      scale_color_manual(values=c("Other"="#000000", "NE mitochondrial ribosome"="#E69F00", "NE oxidative phosphorylation"="#009E73"), name="Category")+
      scale_linetype_manual(values=c("dotted", "solid"))+
      geom_vline(xintercept=0, linetype="dashed", color = "gray50")+
      theme_void()+
      theme(legend.title = element_blank())+
      coord_flip()
    #put two plots together
      q <- insert_xaxis_grob(cpm.ril.SL.lims1.mito.scat, densityplotX, grid::unit(0.275, "null"), position = "top")
      q <- insert_yaxis_grob(q, densityplotY, grid::unit(0.275, "null"), position = "right")
    #save
    #  ggsave(plot=q, filename = "cpm.ril.SL.lims1.mito.scat.png")
  ggsave(plot=q, filename = "cpm.ril.SL.lims1.mito.scat.pdf", dpi=300, width=4.75, height = 5, scale = 1, units = "in")
```





