---
title: "Final version of Leaf Heterosis proteomics for Paper"
output: github_document
---
#Setup
```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
set.seed(586)
```
```{r libraries}
library(matrixStats)
library(topGO)
library(ggplot2)
library(AnnotationDbi)
library(cowplot)
```

#Import data
```{r read csv}
df <- read.csv("data/Leaf paper-total proteome final 121319.csv", stringsAsFactors = F)
#make new column for gene accession without transcript id
df$Gene <- substr(df$Accession, start = 1, stop = 14)
df.withPolymorphic <- df
#Remove polymorphic proteins
df <- df[-(grep("only", df$Accession)),]
```

```{r T-test function}
ttestFun <- function(df, col1, col2){
  result <- NULL
  for(i in 1:(nrow(df))){
  v1 = (as.numeric(df[i, col1]))
  v2 = (as.numeric(df[i, col2]))
  t_test <- t.test(v1, v2, var.equal = TRUE)
  pval <- t_test$p.value
  result[i] <- pval
  }
  result
}
```

##Gamer GO

```{r read in gamer}
# Read in Maize GAMER GO dataset, which lists maize V4 accessions with all associated GO terms (Wimalanathan et al. 2018) downloaded from https://datacommons.cyverse.org/browse/iplant/home/shared/commons_repo/curated/Carolyn_Lawrence-Dill_maize-GAMER_maize.B73_RefGen_v4_Zm00001d.2_Oct_2017.r1/e.agg_data
GAMER <- read.delim("maize.B73.AGPv4.aggregate.gaf", sep = "\t", header = FALSE)
colnames(GAMER) <- c("db", "db_object_id", "db_object_symbol", "qualifier", "term_accession", "db_reference", "evidence_code", "with", "aspect", "db_object_name", "db_object_synonym", "db_object_type", "taxon", "date", "assigned_by", "annotation_extension", "gene_product_form_id")
gamer <- GAMER[-(1:2),]
gamer$Term <- Term(as.character(gamer$term_accession))
```


#Define topGO functions
```{r GOanalysisGOdata_Function}
## Build GOdata object
# EnrichedDF = AMP/BMP output or other subset dataframe of proteins of interest
# GamerCut = the dataframe created above
# ont = "BP", "MF", or "CC"
GOanalysisGodata <- function(EnrichedDF, GamerCut, ont, nodesize = 1){
  gene2GO_TissueCut <- split((as.character(GamerCut$term_accession)),GamerCut$db_object_id)
  gene2GOb_TissueCut <- gene2GO_TissueCut[-2]
  str(head(gene2GOb_TissueCut))
  geneNames_tissuecut <- names(gene2GOb_TissueCut)
  myInterestingGenes <- EnrichedDF$Gene
  geneList <- factor(as.integer(geneNames_tissuecut %in% myInterestingGenes))
  names(geneList) <- geneNames_tissuecut
  GOdata <- new("topGOdata", ontology = ont, allGenes = geneList, annot = annFUN.gene2GO, nodeSize=nodesize, gene2GO = gene2GOb_TissueCut)
  GOdata
}
```

```{r GOanalysisMap_Function}
## GO analysis function to generate map of GO terms
# Godata = Godata object (output of GoanalysisGodata function)
# pdfname = and desired output pdf name in quotes
GOanalysisMap <- function(GOdata, pdfname, signodes=5){
  resultFis <- runTest(GOdata, algorithm = "classic", statistic = "fisher")
  printGraph(GOdata, resultFis, firstSigNodes = signodes, fn.prefix = pdfname, useInfo = "all", pdfSW = TRUE)
}
```

```{r GOanalysisDF_Function}
## GO analysis function, same as above, except to generate dataframe summary of top 20 of GO terms instead of map
# Godata = Godata object (output of GoanalysisGodata function)
# pdfname = and desired output pdf name in quotes
GOanalysisDF <- function(GOdata){
resultFis <- runTest(GOdata, algorithm = "classic", statistic = "fisher")
test.statF <- new("classicCount", testStatistic = GOFisherTest, name = "Fisher test")
resultFisher <- getSigGroups(GOdata, test.statF)
test.statKS <- new("classicScore", testStatistic = GOKSTest, name = "KS tests")
resultKS <- getSigGroups(GOdata, test.statKS)
test.statRW <- new("weightCount", testStatistic = GOFisherTest, name = "Fisher test", sigRatio = "ratio")
resultWeight <- getSigGroups(GOdata, test.statRW)
DF <- GenTable(GOdata, classic = resultFis, KS = resultKS, weight = resultWeight, orderBy = "classic", ranksOf = "classic", topNodes = 20)
DF
}
```


#acs
##Format df
```{r format df}
#Subset
df.acs <- df[,c(1:18,89)]
#Remove missing genes
df.acs <- df.acs[!(df.acs$B73.acs.B73.r1==0),]
#Calculate averages of bio reps
df.acs$B73avg <- rowMeans(df.acs[,9:13])
df.acs$ACSavg <- rowMeans(df.acs[,14:18])
#calculate ratio
df.acs$ACS.B73 <- df.acs$ACSavg/df.acs$B73avg
#calculate t-test
df.acs$Sig <- ttestFun(df.acs, 9:13, 14:18)
#calculate log2(acs/b73)
df.acs$l2fc <- log2(df.acs$ACS.B73)
#Calculate standard deviation
df.acs$SD <- rowSds(as.matrix(df.acs[,14:18]))
#subset to 10% significantly above or below mid-parent
df.acs.amp <- df.acs[(df.acs$Sig<0.05)&(df.acs$ACS.B73>1.1),]
df.acs.bmp <- df.acs[(df.acs$Sig<0.05)&(df.acs$ACS.B73<0.9),]
```
##GO term enrichment
```{r trim gamer}
#Trim gamer to only those genes detected in our data, to use as background for enrichment
gamerCutACS <- gamer[(gamer$db_object_id %in% df.acs$Gene),]
```
```{r run GoanalysisGodata}
df.acs.amp_BP_GOdata <- GOanalysisGodata(df.acs.amp, gamerCutACS, "BP", nodesize = 5)
df.acs.amp_MF_GOdata <- GOanalysisGodata(df.acs.amp, gamerCutACS, "MF", nodesize = 5)
df.acs.amp_CC_GOdata <- GOanalysisGodata(df.acs.amp, gamerCutACS, "CC", nodesize = 5)

df.acs.bmp_BP_GOdata <- GOanalysisGodata(df.acs.bmp, gamerCutACS, "BP", nodesize = 5)
df.acs.bmp_MF_GOdata <- GOanalysisGodata(df.acs.bmp, gamerCutACS, "MF", nodesize = 5)
df.acs.bmp_CC_GOdata <- GOanalysisGodata(df.acs.bmp, gamerCutACS, "CC", nodesize = 5)
```
```{r run GoanalysisMap}
dir.create("GOmaps")
GOanalysisMap(df.acs.amp_BP_GOdata, "GOmaps/df.acs.amp_BP_GOdata")
GOanalysisMap(df.acs.amp_MF_GOdata, "GOmaps/df.acs.amp_MF_GOdata")
GOanalysisMap(df.acs.amp_CC_GOdata, "GOmaps/df.acs.amp_CC_GOdata")

GOanalysisMap(df.acs.bmp_BP_GOdata, "GOmaps/df.acs.bmp_BP_GOdata")
GOanalysisMap(df.acs.bmp_MF_GOdata, "GOmaps/df.acs.bmp_MF_GOdata")
GOanalysisMap(df.acs.bmp_CC_GOdata, "GOmaps/df.acs.bmp_CC_GOdata")
```
```{r run GoanalysisDF}
df.acs.amp_BP_GOdata.DF <- GOanalysisDF(df.acs.amp_BP_GOdata)
df.acs.amp_MF_GOdata.DF <- GOanalysisDF(df.acs.amp_MF_GOdata)
df.acs.amp_CC_GOdata.DF <- GOanalysisDF(df.acs.amp_CC_GOdata)

df.acs.bmp_BP_GOdata.DF <- GOanalysisDF(df.acs.bmp_BP_GOdata)
df.acs.bmp_MF_GOdata.DF <- GOanalysisDF(df.acs.bmp_MF_GOdata)
df.acs.bmp_CC_GOdata.DF <- GOanalysisDF(df.acs.bmp_CC_GOdata)
```

#Seedling Leaf - B73xMo17
##Format df
###AMP/BMP
```{r format df}
#Subset
df.SL <- df[,c(1:8,19:28,89)]
#Remove missing genes
df.SL <- df.SL[!(df.SL$SeedlingLeaf.B73.r1==0),]
#Calculate averages of bio reps
df.SL$B73avg <- rowMeans(df.SL[,10:12])
df.SL$Mo17avg <- rowMeans(df.SL[,13:15])
df.SL$HYBavg <- rowMeans(df.SL[,16:18])
df.SL$MPavg <- rowMeans(df.SL[,c(10:15)])
df.SL$MP.1 <- rowMeans(df.SL[,c(10,13)])
df.SL$MP.2 <- rowMeans(df.SL[,c(11,14)])
df.SL$MP.3 <- rowMeans(df.SL[,c(12,15)])
#calculate ratio
df.SL$HYB.MP <- df.SL$HYBavg/df.SL$MPavg
#calculate t-test
df.SL$Sig <- ttestFun(df.SL, 10:12, 24:26)
#calculate log2(hyb/MP)
df.SL$l2fc <- log2(df.SL$HYB.MP)
#subset to 10% significantly above or below mid-parent
df.SL.amp <- df.SL[(df.SL$Sig<0.05)&(df.SL$HYB.MP>1.1),]
df.SL.bmp <- df.SL[(df.SL$Sig<0.05)&(df.SL$HYB.MP<0.9),]
```
###MP+HP
```{r Subset to mid-parent proteins where parents are different}
df.SL.2 <- df.SL

df.SL.2.b73hp <- df.SL.2[(df.SL.2$B73avg>1.1*(df.SL.2$Mo17avg)),]
df.SL.2.mo17hp <- df.SL.2[(df.SL.2$Mo17avg>1.1*(df.SL.2$B73avg)),]

df.SL.2.b73hp.mp <- df.SL.2.b73hp[!(df.SL.2.b73hp$Accession%in%c(df.SL.amp$Accession, df.SL.bmp$Accession)),]

df.SL.2.mo17hp.mp <- df.SL.2.mo17hp[!(df.SL.2.mo17hp$Accession%in%c(df.SL.amp$Accession, df.SL.bmp$Accession)),]

df.SL.2.b73hp.mp.translation <- df.SL.2.b73hp.mp[(grep("translation", df.SL.2.b73hp.mp$Protein)),]
df.SL.2.mo17hp.mp.translation <- df.SL.2.mo17hp.mp[(grep("translation", df.SL.2.mo17hp.mp$Protein)),]

df.SL.2.b73hp.mp.translation$b73_mo17 <- df.SL.2.b73hp.mp.translation$B73avg/df.SL.2.b73hp.mp.translation$Mo17avg
df.SL.2.mo17hp.mp.translation$mo17_b73 <- df.SL.2.mo17hp.mp.translation$Mo17avg/df.SL.2.mo17hp.mp.translation$B73avg

write.csv(df.SL.2.b73hp.mp.translation, "df.SL.2.b73hp.mp.translation.csv")
write.csv(df.SL.2.mo17hp.mp.translation, "df.SL.2.mo17hp.mp.translation.csv")
```

##GO term enrichment
###AMP/BMP
```{r trim gamer}
#Trim gamer to only those genes detected in our data, to use as background for enrichment
gamerCutSL <- gamer[(gamer$db_object_id %in% df.SL$Gene),]
```
```{r run GoanalysisGodata for amp/bmp}
df.SL.amp_BP_GOdata <- GOanalysisGodata(df.SL.amp, gamerCutSL, "BP", nodesize = 5)
df.SL.amp_MF_GOdata <- GOanalysisGodata(df.SL.amp, gamerCutSL, "MF", nodesize = 5)
df.SL.amp_CC_GOdata <- GOanalysisGodata(df.SL.amp, gamerCutSL, "CC", nodesize = 5)

df.SL.bmp_BP_GOdata <- GOanalysisGodata(df.SL.bmp, gamerCutSL, "BP", nodesize = 5)
df.SL.bmp_MF_GOdata <- GOanalysisGodata(df.SL.bmp, gamerCutSL, "MF", nodesize = 5)
df.SL.bmp_CC_GOdata <- GOanalysisGodata(df.SL.bmp, gamerCutSL, "CC", nodesize = 5)
```
```{r run GoanalysisMap for amp/bmp}
dir.create("GOmaps")
GOanalysisMap(df.SL.amp_BP_GOdata, "GOmaps/df.SL.amp_BP_GOdata")
GOanalysisMap(df.SL.amp_MF_GOdata, "GOmaps/df.SL.amp_MF_GOdata")
GOanalysisMap(df.SL.amp_CC_GOdata, "GOmaps/df.SL.amp_CC_GOdata")

GOanalysisMap(df.SL.bmp_BP_GOdata, "GOmaps/df.SL.bmp_BP_GOdata")
GOanalysisMap(df.SL.bmp_MF_GOdata, "GOmaps/df.SL.bmp_MF_GOdata")
GOanalysisMap(df.SL.bmp_CC_GOdata, "GOmaps/df.SL.bmp_CC_GOdata")
```
```{r run GoanalysisDF for amp/bmp}
df.SL.amp_BP_GOdata.DF <- GOanalysisDF(df.SL.amp_BP_GOdata)
df.SL.amp_MF_GOdata.DF <- GOanalysisDF(df.SL.amp_MF_GOdata)
df.SL.amp_CC_GOdata.DF <- GOanalysisDF(df.SL.amp_CC_GOdata)

df.SL.bmp_BP_GOdata.DF <- GOanalysisDF(df.SL.bmp_BP_GOdata)
df.SL.bmp_MF_GOdata.DF <- GOanalysisDF(df.SL.bmp_MF_GOdata)
df.SL.bmp_CC_GOdata.DF <- GOanalysisDF(df.SL.bmp_CC_GOdata)
```
```{r view enrichment}
df.SL.bmp_CC_GOdata.DF
```

###MP+HP
```{r run GoanalysisGodata for hp+mp}
df.SL.2.b73hp.mp_BP_GOdata <- GOanalysisGodata(df.SL.2.b73hp.mp, gamerCutSL, "BP")
df.SL.2.b73hp.mp_CC_GOdata <- GOanalysisGodata(df.SL.2.b73hp.mp, gamerCutSL, "CC")
df.SL.2.b73hp.mp_MF_GOdata <- GOanalysisGodata(df.SL.2.b73hp.mp, gamerCutSL, "MF")

df.SL.2.mo17hp.mp_BP_GOdata <- GOanalysisGodata(df.SL.2.mo17hp.mp, gamerCutSL, "BP")
df.SL.2.mo17hp.mp_CC_GOdata <- GOanalysisGodata(df.SL.2.mo17hp.mp, gamerCutSL, "CC")
df.SL.2.mo17hp.mp_MF_GOdata <- GOanalysisGodata(df.SL.2.mo17hp.mp, gamerCutSL, "MF")
```
```{r run GoanalysisDF for hp+mp}
df.SL.2.b73hp.mp_BP_GOdata.DF <- GOanalysisDF(df.SL.2.b73hp.mp_BP_GOdata)
df.SL.2.b73hp.mp_CC_GOdata.DF <- GOanalysisDF(df.SL.2.b73hp.mp_CC_GOdata)
df.SL.2.b73hp.mp_MF_GOdata.DF <- GOanalysisDF(df.SL.2.b73hp.mp_MF_GOdata)

df.SL.2.mo17hp.mp_BP_GOdata.DF <- GOanalysisDF(df.SL.2.mo17hp.mp_BP_GOdata)
df.SL.2.mo17hp.mp_CC_GOdata.DF <- GOanalysisDF(df.SL.2.mo17hp.mp_CC_GOdata)
df.SL.2.mo17hp.mp_MF_GOdata.DF <- GOanalysisDF(df.SL.2.mo17hp.mp_MF_GOdata)
```
```{r view enrichment}
df.SL.2.b73hp.mp_BP_GOdata.DF
```
#ACS and BxM enrichment overlap
```{r subset to proteins amp/bmp in both the mutant and hybrid}
df.SL.acs.amp <- df.SL.amp[(df.SL.amp$Accession%in%df.acs.amp$Accession),]
df.SL.acs.bmp <- df.SL.bmp[(df.SL.bmp$Accession%in%df.acs.bmp$Accession),]
df.SL.acs <- df.SL[(df.SL$Gene%in%df.acs$Gene),]
```
```{r trim gamer}
#Trim gamer to only those genes detected in our data, to use as background for enrichment
gamerCutSLacs <- gamer[(gamer$db_object_id %in% df.SL.acs$Gene),]
```
```{r run GOanalysisGodata on hyb+acs overlapping enrichment}
df.SL.acs.amp_BP_GOdata <- GOanalysisGodata(df.SL.acs.amp, gamerCutSLacs, "BP", nodesize = 5)
df.SL.acs.amp_MF_GOdata <- GOanalysisGodata(df.SL.acs.amp, gamerCutSLacs, "MF", nodesize = 5)
df.SL.acs.amp_CC_GOdata <- GOanalysisGodata(df.SL.acs.amp, gamerCutSLacs, "CC", nodesize = 5)

df.SL.acs.bmp_BP_GOdata <- GOanalysisGodata(df.SL.acs.bmp, gamerCutSLacs, "BP", nodesize = 5)
df.SL.acs.bmp_MF_GOdata <- GOanalysisGodata(df.SL.acs.bmp, gamerCutSLacs, "MF", nodesize = 5)
df.SL.acs.bmp_CC_GOdata <- GOanalysisGodata(df.SL.acs.bmp, gamerCutSLacs, "CC", nodesize = 5)
```
```{r run GOanalysisMap on hyb+acs overlapping enrichment}
dir.create("GOmaps")
GOanalysisMap(df.SL.acs.amp_BP_GOdata, "GOmaps/df.SL.acs.amp_BP_GOdata")
GOanalysisMap(df.SL.acs.amp_MF_GOdata, "GOmaps/df.SL.acs.amp_MF_GOdata")
GOanalysisMap(df.SL.acs.amp_CC_GOdata, "GOmaps/df.SL.acs.amp_CC_GOdata")

GOanalysisMap(df.SL.acs.bmp_BP_GOdata, "GOmaps/df.SL.acs.bmp_BP_GOdata")
GOanalysisMap(df.SL.acs.bmp_MF_GOdata, "GOmaps/df.SL.acs.bmp_MF_GOdata")
GOanalysisMap(df.SL.acs.bmp_CC_GOdata, "GOmaps/df.SL.acs.bmp_CC_GOdata")
```
```{r run GOanalysisDF on hyb+acs overlapping enrichment}
df.SL.acs.amp_BP_GOdata.DF <- GOanalysisDF(df.SL.acs.amp_BP_GOdata)
df.SL.acs.amp_MF_GOdata.DF <- GOanalysisDF(df.SL.acs.amp_MF_GOdata)
df.SL.acs.amp_CC_GOdata.DF <- GOanalysisDF(df.SL.acs.amp_CC_GOdata)

df.SL.acs.bmp_BP_GOdata.DF <- GOanalysisDF(df.SL.acs.bmp_BP_GOdata)
df.SL.acs.bmp_MF_GOdata.DF <- GOanalysisDF(df.SL.acs.bmp_MF_GOdata)
df.SL.acs.bmp_CC_GOdata.DF <- GOanalysisDF(df.SL.acs.bmp_CC_GOdata)
```
```{r view enrichment}
df.SL.acs.bmp_BP_GOdata.DF
```
#Merge acs and Seedling Leaf BxM data
```{r merge data for scatterplots}
df.acs.SL <- merge.data.frame(df.acs[,c(1:8,19,24:25)], df.SL[,c(6,c(27,29))], by="Accession", all = F, suffixes = c(".acs", ".BM"))
```

#Define GOIs
Gene lists and sources:
1. Plastid encoded photosynthesis associated proteins (Annotation)
2. Photosynthesis associated nuclear encoded proteins (GO GAMER+CornCyc)
3. Plastid encoded ribosomal proteins + 3"other"(= matK, infA, and clpP, added based on expression grouping) (Annotation)
4. Plastid encoded RNA polymerases (Annotation)
5. Nuclear encoded plastid ribosomal proteins (GO GAMER and Annotation)
6. Eukaryotic ribosomal proteins (Annotation)
7. Ethylene biosynthesis (CornCyc)
8. Response to ethylene (GAMER GO)
9. Pentose-phosphate shunt (GAMER GO)
```{r 1}
#Plastid encoded photosynthesis associated proteins  (EnsemblPlants Annotation)
#generate vector of Pt encoded genes
PtGenes<-df.withPolymorphic[grep("chromosome:B73_RefGen_v4:Pt:", df.withPolymorphic$Protein),]

#define list of plastid-encoded photosynthesis-associated genes
PtPS <- c("ZemaCp008.1", "ZemaCp022.1", "ZemaCp023.1", "ZemaCp032.1", "ZemaCp008.1", "ZemaCp039.1", "ZemaCp040.1", "ZemaCp048.1", "ZemaCp051.1", "ZemaCp087.1", "ZemaCp017.1", "ZemaCp018.1", "ZemaCp019.1", "ZemaCp024.1", "ZemaCp027.1", "ZemaCp028.1", "ZemaCp029.1", "ZemaCp030.1", "ZemaCp035.1", "ZemaCp036.1", "ZemaCp052.1", "ZemaCp053.1", "ZemaCp075.1", "ZemaCp083.1", "ZemaCp085.1", "ZemaCp086.1", "ZemaCp090.1", "ZemaCp091.1", "ZemaCp092.1", "ZemaCp050.1", "ZemaCp043.1", "ZemaCp088.1")
```

```{r 2}
#Photosynthesis associated nuclear encoded proteins (PhANGs)  (GO GAMER+CornCyc)
PHANG<-as.vector(unique(gamer$db_object_id[grep("GO:0015979", gamer$term_accession)]))
#Add CornCyc annotated PhANGs
#accessed from https://corncyc-b73-v4.maizegdb.org/ on 1/4/2020
cornCyc<-as.data.frame(readr::read_csv(file = "cornCycGeneLists01042020.csv",
                         col_names = TRUE
                         ))
cornCyc.PHANG <- unique(c(cornCyc$Gene.Accession[(cornCyc$Pathway=="Oxygenic_Photosynthesis_SuperPath")], cornCyc$Gene.Accession[(cornCyc$Pathway=="Light_Rxn")]))
PHANG <- unique(c(PHANG, cornCyc.PHANG))
```

```{r 3}
#Plastid encoded ribosomal proteins (EnsemblPlants Annotation)
#get subset of Pt encoded genes annotated as ribosomal proteins
PtEncodeRibo<- c(PtGenes$Gene[(grep("description:rps", PtGenes$Protein))], PtGenes$Gene[(grep("description:rpl", PtGenes$Protein))], PtGenes$Gene[(grep("description:ribosom", PtGenes$Protein))])
```

```{r 4}
#Check that no plastid-encoded photosynthesis/ribosomal proteins were missed
Pt <- df.withPolymorphic[(grep("B73_RefGen_v4:Pt", df.withPolymorphic$Protein)),]
Pt.rpo <- Pt$Gene[(grep("rpo", Pt$Protein))]
Pt.other <- Pt$Gene[!(Pt$Accession%in%c(PtEncodeRibo, PtPS, Pt.rpo))]
```

```{r 5}
#Nuclear encoded plastid ribosomal proteins (GO GAMER and EnsemblPlants Annotation)
PtRibo<-as.vector(unique(gamer$db_object_id[grep("GO:0009547", gamer$term_accession)]))
#Add proteins annotated as "ribosom" and "plastid/chloroplast" to the nuclear-encoded plastid-localized list
ribosom <- df.withPolymorphic[(grep("ribosom", df.withPolymorphic$Protein)),]
ribosom.pt <- ribosom$Gene[(grep("plastid|Plastid|Chloropl|chloropl", ribosom$Protein))]
PtRibo <- unique(c(PtRibo,ribosom.pt))

#Check that there are no missing prokaryotic ribosomal proteins with plastid localization
ProkRibo <- df.withPolymorphic[(grep("30S|60S", df.withPolymorphic$Protein)),]
ProkRiboNotPt <- ProkRibo[!(ProkRibo$Gene%in%PtRibo),]
GOplastid<-as.vector(unique(gamer$db_object_id[grep("GO:0009536", gamer$term_accession)]))
ProkRiboNotPt.plastid <- ProkRiboNotPt[(ProkRiboNotPt$Gene%in%GOplastid),]
PtRibo <- c(PtRibo, ProkRiboNotPt.plastid$Gene)
```

```{r 6}
#Eukaryotic ribosomal proteins (EnsemblPlants Annotation)
eukRibo<-df.withPolymorphic[(grep("40S|60S", df.withPolymorphic$Protein)),]
#remove the mitochondrial ones
eukRibo<- eukRibo$Gene[-(grep("mito|Mito", eukRibo$Protein))]
```

```{r 7}
MAT <- cornCyc$`Gene Accession`[(cornCyc$`Enzymatic activity`=="methionine adenosyltransferase")]
ACS <- cornCyc$`Gene Accession`[(cornCyc$`Enzymatic activity`=="1-aminocyclopropane-1-carboxylate synthase")]
ACO <- cornCyc$`Gene Accession`[(cornCyc$`Enzymatic activity`=="1-aminocyclopropane-1-carboxylate oxidase")]
EtBio <- c(MAT,ACS,ACO)
```

```{r 8}
ResponseToEt <- as.vector(unique(gamer$db_object_id[grep("GO:0009723", gamer$term_accession)]))
```

```{r 9}
PPS <- as.vector(unique(gamer$db_object_id[grep("GO:0006098", gamer$term_accession)]))
```
##Make csv with GOIs
```{r}
df2 <- df.withPolymorphic
df2$Category <- "Other"
df2$Order <- 9
df2[(df2$Gene %in% PPS),"Category"] <- "Pentose Phosphate Shunt"
df2[(df2$Gene %in% PPS),"Order"] <- 8
df2[(df2$Gene %in% PtPS),"Category"] <- "Plastid-encoded photosynthesis-associated"
df2[(df2$Gene %in% PtPS),"Order"] <- 3
df2[(df2$Gene %in% PHANG),"Category"] <- "Nuclear-encoded photosynthesis-associated"
df2[(df2$Gene %in% PHANG),"Order"] <- 4
df2[(df2$Gene %in% PtEncodeRibo),"Category"] <- "Plastid-encoded ribosome"
df2[(df2$Gene %in% PtEncodeRibo),"Order"] <- 1
df2[(df2$Gene %in% PtRibo),"Category"] <- "Nuclear-encoded plastid ribosome"
df2[(df2$Gene %in% PtRibo),"Order"] <- 2
df2[(df2$Gene %in% eukRibo),"Category"] <- "Eukaryotic ribosome"
df2[(df2$Gene %in% eukRibo),"Order"] <- 7
df2[(df2$Gene %in% Pt.rpo),"Category"] <- "Plastid-encoded RNA polymerase"
df2[(df2$Gene %in% Pt.rpo),"Order"] <- 5
df2[(df2$Gene %in% Pt.other),"Category"] <- "Plastid-encoded other"
df2[(df2$Gene %in% Pt.other),"Order"] <- 6
#Re-order
df2 <- df2[(order(df2$Order)),]
```
```{r}
write.csv(df2, "HeterosisGOIs_DSB.csv")
```


#Scatterplot1
##Make GOI description column
```{r define categories}
df.acs.SL.scat1 <- df.acs.SL
df.acs.SL.scat1$Category <- "Other"
df.acs.SL.scat1$Order <- 1
df.acs.SL.scat1[(df.acs.SL.scat1$Gene %in% PPS),"Category"] <- "PPS"
df.acs.SL.scat1[(df.acs.SL.scat1$Gene %in% PPS),"Order"] <- 2
df.acs.SL.scat1[(df.acs.SL.scat1$Gene %in% PtPS),"Category"] <- "PlastidEncodedPS"
df.acs.SL.scat1[(df.acs.SL.scat1$Gene %in% PtPS),"Order"] <- 3
df.acs.SL.scat1[(df.acs.SL.scat1$Gene %in% PHANG),"Category"] <- "PHANGs"
df.acs.SL.scat1[(df.acs.SL.scat1$Gene %in% PHANG),"Order"] <- 4
df.acs.SL.scat1[(df.acs.SL.scat1$Gene %in% c(PtEncodeRibo,Pt.other)),"Category"] <- "PlastidEncodedRibo+Other"
df.acs.SL.scat1[(df.acs.SL.scat1$Gene %in% c(PtEncodeRibo,Pt.other)),"Order"] <- 5
df.acs.SL.scat1[(df.acs.SL.scat1$Gene %in% PtRibo),"Category"] <- "NuclearEncodedPlastidRibo"
df.acs.SL.scat1[(df.acs.SL.scat1$Gene %in% PtRibo),"Order"] <- 6
df.acs.SL.scat1[(df.acs.SL.scat1$Gene %in% eukRibo),"Category"] <- "CytosolicRibo"
df.acs.SL.scat1[(df.acs.SL.scat1$Gene %in% eukRibo),"Order"] <- 2
df.acs.SL.scat1[(df.acs.SL.scat1$Gene %in% Pt.rpo),"Category"] <- "Pt.rpo"
df.acs.SL.scat1[(df.acs.SL.scat1$Gene %in% Pt.rpo),"Order"] <- 8
#Re-order
df.acs.SL.scat1 <- df.acs.SL.scat1[(order(df.acs.SL.scat1$Order)),]
```
##Set limits for plot
```{r}
df.acs.SL.scat1.lims <- df.acs.SL.scat1

df.acs.SL.scat1.lims$l2fc.acs[df.acs.SL.scat1.lims$l2fc.acs > 1.0] = 1.0
df.acs.SL.scat1.lims$l2fc.acs[df.acs.SL.scat1.lims$l2fc.acs < -1.0] = -1.0

df.acs.SL.scat1.lims$l2fc.BM[df.acs.SL.scat1.lims$l2fc.BM > 1.0] = 1.0
df.acs.SL.scat1.lims$l2fc.BM[df.acs.SL.scat1.lims$l2fc.BM < -1.0] = -1.0
```
##Make scatterplot
```{r}
fig4 <- ggplot(df.acs.SL.scat1.lims, mapping=aes(x=l2fc.BM, y=l2fc.acs, color=Category))+
  geom_point(size=.8)+
  scale_color_manual(values=c("Other"="grey70","PPS"="#efb806", "PlastidEncodedPS"="red3", "PHANGs"="#ef06db", "PlastidEncodedRibo+Other"="blue4", "NuclearEncodedPlastidRibo"="dodgerblue1", "CytosolicRibo"="#019d01", "Pt.rpo"="#653f01"), name="Protein Category")+
    theme_classic()+
    xlim(c(-1,1))+
    ylim(c(-1,1))+
    geom_vline(xintercept=0, linetype="dashed", color = "gray50")+
    geom_hline(yintercept=0, linetype="dashed", color = "gray50")+
    #geom_abline(intercept =0 , slope = 1, linetype="dashed", color="gray50")+
    theme(axis.text=element_text(size=10),
              legend.position="none")+
    guides(fill=guide_legend(nrow=2,byrow=TRUE))+
    xlab("log2 hyb/MP protein abundance")+
    ylab("log2 acs/B73 protein abundance")+
    coord_fixed(expand = F)

fig4
```
##Add density curves
```{r}
#add density curves
#add marginal density plots and resave
    densityplotX<-axis_canvas(fig4, axis="x")+
      geom_density(aes(df.acs.SL.scat1.lims[,"l2fc.BM"], color=df.acs.SL.scat1.lims[,"Category"]), df.acs.SL.scat1.lims, size=0.7)+
      scale_color_manual(values=c("Other"="grey70","PPS"="#efb806", "PlastidEncodedPS"="red3", "PHANGs"="#ef06db", "PlastidEncodedRibo+Other"="blue4", "NuclearEncodedPlastidRibo"="dodgerblue1", "CytosolicRibo"="#019d01", "Pt.rpo"="#653f01"))+
      geom_vline(xintercept=0, linetype="dashed", color = "gray50")+
      theme_void()
    #set up marginal density plot Y axis
    densityplotY<-axis_canvas(fig4, axis="y", coord_flip = T)+
      geom_density(aes(df.acs.SL.scat1.lims[,"l2fc.acs"], color=df.acs.SL.scat1.lims[,"Category"]), df.acs.SL.scat1.lims, size=0.7)+
      scale_color_manual(values=c("Other"="grey70","PPS"="#efb806", "PlastidEncodedPS"="red3", "PHANGs"="#ef06db", "PlastidEncodedRibo+Other"="blue4", "NuclearEncodedPlastidRibo"="dodgerblue1", "CytosolicRibo"="#019d01", "Pt.rpo"="#653f01"))+
      geom_vline(xintercept=0, linetype="dashed", color = "gray50")+
      theme_void()+
      coord_flip()
    #put two plots together
    q <- insert_xaxis_grob(fig4, densityplotX, grid::unit(0.275, "null"), position = "top")
    q <- insert_yaxis_grob(q, densityplotY, grid::unit(0.275, "null"), position = "right")
    #save
  ggsave(plot=q, filename = "Fig4new.tiff",
        device="tiff", dpi=300, width=4.5, height=4.5, compression = "lzw")
```
```{r}
ggplot(df.acs.SL.scat1.lims)+
  geom_density(aes(df.acs.SL.scat1.lims[,"l2fc.BM"], color=df.acs.SL.scat1.lims[,"Category"]), df.acs.SL.scat1.lims, size=1.2)+
      scale_color_manual(values=c("Other"="black","PPS"="#efb806", "PlastidEncodedPS"="red3", "PHANGs"="#ef06db", "PlastidEncodedRibo+Other"="blue4", "NuclearEncodedPlastidRibo"="dodgerblue1", "CytosolicRibo"="#019d01", "Pt.rpo"="#855b05"))+
      geom_vline(xintercept=0, linetype="dashed", color = "gray70")+
      theme_void()+
  theme(legend.position = "none")+
  xlim(-1,1)
ggsave("l2BM_density.tiff", dpi = 300, width = 8, height = 2)
```
```{r}
df.acs.SL.scat1.lims.noOther <- df.acs.SL.scat1.lims[!(df.acs.SL.scat1.lims$Category=="Other"),]
```
```{r}
coln
```

```{r}
ggplot(df.acs.SL.scat1.lims.noOther)+
  geom_density(aes(df.acs.SL.scat1.lims.noOther[,"l2fc.acs"], color=df.acs.SL.scat1.lims.noOther[,"Category"]), df.acs.SL.scat1.lims.noOther, size=1.2)+
      scale_color_manual(values=c("PPS"="#efb806", "PlastidEncodedPS"="red3", "PHANGs"="#ef06db", "PlastidEncodedRibo+Other"="blue4", "NuclearEncodedPlastidRibo"="dodgerblue1", "CytosolicRibo"="#019d01", "Pt.rpo"="#855b05"))+
      geom_vline(xintercept=0, linetype="dashed", color = "gray70")+
      theme_void()+
  theme(legend.position = "none")+
  xlim(-1,1)
#ggsave("l2acs_density.tiff", dpi = 300, width = 8, height = 2)
```

#Scatterplot2
##Make GOI description column
```{r}
df.acs.SL.scat2<- df.acs.SL
df.acs.SL.scat2$Category <- "Other"
df.acs.SL.scat2$Order <- 1
df.acs.SL.scat2[(df.acs.SL.scat2$Gene %in% EtBio),"Category"] <- "EtBio"
df.acs.SL.scat2[(df.acs.SL.scat2$Gene %in% EtBio),"Order"] <- 2
df.acs.SL.scat2[(df.acs.SL.scat2$Gene %in% ResponseToEt),"Category"] <- "ResponseToEt"
df.acs.SL.scat2[(df.acs.SL.scat2$Gene %in% ResponseToEt),"Order"] <- 3

#Re-order
df.acs.SL.scat2 <- df.acs.SL.scat2[(order(df.acs.SL.scat2$Order)),]
```
##Set limits for plot
```{r}
df.acs.SL.scat2.lims <- df.acs.SL.scat2

df.acs.SL.scat2.lims$l2fc.acs[df.acs.SL.scat2.lims$l2fc.acs > 1.0] = 1.0
df.acs.SL.scat2.lims$l2fc.acs[df.acs.SL.scat2.lims$l2fc.acs < -1.0] = -1.0

df.acs.SL.scat2.lims$l2fc.BM[df.acs.SL.scat2.lims$l2fc.BM > 1.0] = 1.0
df.acs.SL.scat2.lims$l2fc.BM[df.acs.SL.scat2.lims$l2fc.BM < -1.0] = -1.0
```
##Make scatterplot
```{r}
fig4.2 <- ggplot(df.acs.SL.scat2.lims, mapping=aes(x=l2fc.acs, y=l2fc.BM, color=Category))+
  geom_point(size=.8)+
  scale_color_manual(values=c("Other"="grey80", "EtBio"="blue", "ResponseToEt"="red"), name="Protein Category")+
  theme_classic()+
  xlim(c(-1,1))+
  ylim(c(-1,1))+
  geom_vline(xintercept=0, linetype="dashed", color = "gray50")+
  geom_hline(yintercept=0, linetype="dashed", color = "gray50")+
  #geom_abline(intercept =0 , slope = 1, linetype="dashed", color="gray50")+
  theme(axis.text=element_text(size=10),
          legend.position="none")+
  guides(fill=guide_legend(nrow=2,byrow=TRUE))+
  xlab("log2 hyb/MP protein abundance")+
  ylab("log2 acs/B73 protein abundance")+
  coord_fixed(expand = F)

fig4.2
```
##Add density curves
```{r}
#add density curves
#add marginal density plots and resave
    densityplotX<-axis_canvas(fig4.2, axis="x")+
      geom_density(aes(df.acs.SL.scat2.lims[,"l2fc.BM"], color=df.acs.SL.scat2.lims[,"Category"]), df.acs.SL.scat2.lims, size=0.7)+
      scale_color_manual(values=c("Other"="grey80", "EtBio"="blue", "ResponseToEt"="red"), name="Protein Category")+
      geom_vline(xintercept=0, linetype="dashed", color = "gray50")+
      theme_void()
    #set up marginal density plot Y axis
    densityplotY<-axis_canvas(fig4.2, axis="y", coord_flip = T)+
      geom_density(aes(df.acs.SL.scat2.lims[,"l2fc.acs"], color=df.acs.SL.scat2.lims[,"Category"]), df.acs.SL.scat2.lims, size=0.7)+
      scale_color_manual(values=c("Other"="grey80", "EtBio"="blue", "ResponseToEt"="red"), name="Protein Category")+
      geom_vline(xintercept=0, linetype="dashed", color = "gray50")+
      theme_void()+
      coord_flip()
    #put two plots together
    q <- insert_xaxis_grob(fig4.2, densityplotX, grid::unit(0.275, "null"), position = "top")
    q <- insert_yaxis_grob(q, densityplotY, grid::unit(0.275, "null"), position = "right")
    #save
    ggsave(plot=q, filename = "Fig4_2new.tiff",
           device="tiff", dpi=300, width=4.5, height=4.5, compression = "lzw")
```


