---
title: "Final version of Leaf Heterosis proteomics for Paper"
output: github_document
---
#Setup
```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
set.seed(586)
```
#Import data
```{r}
df <- read.csv("data/Leaf paper-total proteome final 121319.csv", stringsAsFactors = F)

df$Gene <- substr(df$Accession, start = 1, stop = 14)
```
```{r}
#Remove polymorphic proteins
df <- df[-(grep("only", df$Accession)),]
```

```{r}
ttestFun <- function(df, col1, col2){
  result <- NULL
  for(i in 1:(nrow(df))){
  v1 = (as.numeric(df[i, col1]))
  v2 = (as.numeric(df[i, col2]))
  t_test <- t.test(v1, v2, var.equal = TRUE)
  pval <- t_test$p.value
  result[i] <- pval
  }
  result
}
```
##Gamer GO
```{r}
library(AnnotationDbi)
```

```{r}
# Read in Maize GAMER GO dataset, which lists maize V4 accessions with all associated GO terms (Wimalanathan et al. 2018) downloaded from https://datacommons.cyverse.org/browse/iplant/home/shared/commons_repo/curated/Carolyn_Lawrence-Dill_maize-GAMER_maize.B73_RefGen_v4_Zm00001d.2_Oct_2017.r1/e.agg_data
GAMER <- read.delim("maize.B73.AGPv4.aggregate.gaf", sep = "\t", header = FALSE)
colnames(GAMER) <- c("db", "db_object_id", "db_object_symbol", "qualifier", "term_accession", "db_reference", "evidence_code", "with", "aspect", "db_object_name", "db_object_synonym", "db_object_type", "taxon", "date", "assigned_by", "annotation_extension", "gene_product_form_id")
gamer <- GAMER[-(1:2),]
gamer$Term <- Term(as.character(gamer$term_accession))
```

#Define topGO functions
```{r GOanalysisGOdata_Function}
library(topGO)
## Build GOdata object
# EnrichedDF = AHP/BLP/AMP/BMP output
# GamerCut = the dataframe created above
# ont = "BP", "MF", or "CC"
GOanalysisGodata <- function(EnrichedDF, GamerCut, ont, nodesize = 1){
  gene2GO_TissueCut <- split((as.character(GamerCut$term_accession)),GamerCut$db_object_id)
  gene2GOb_TissueCut <- gene2GO_TissueCut[-2]
  str(head(gene2GOb_TissueCut))
  geneNames_tissuecut <- names(gene2GOb_TissueCut)
  myInterestingGenes <- EnrichedDF$Gene
  geneList <- factor(as.integer(geneNames_tissuecut %in% myInterestingGenes))
  names(geneList) <- geneNames_tissuecut
  GOdata <- new("topGOdata", ontology = ont, allGenes = geneList, annot = annFUN.gene2GO, nodeSize=nodesize, gene2GO = gene2GOb_TissueCut)
  GOdata
}
```
```{r GOanalysisMap_Function}
## GO analysis function to generate map of GO terms
# Godata = Godata object (output of GoanalysisGodata function)
# pdfname = and desired output pdf name in quotes
GOanalysisMap <- function(GOdata, pdfname, signodes=5){
  resultFis <- runTest(GOdata, algorithm = "classic", statistic = "fisher")
  printGraph(GOdata, resultFis, firstSigNodes = signodes, fn.prefix = pdfname, useInfo = "all", pdfSW = TRUE)
}
```
```{r GOanalysisDF_Function}
## GO analysis function, same as above, except to generate dataframe summary of top 20 of GO terms instead of map
# Godata = Godata object (output of GoanalysisGodata function)
# pdfname = and desired output pdf name in quotes
GOanalysisDF <- function(GOdata){
resultFis <- runTest(GOdata, algorithm = "classic", statistic = "fisher")
test.statF <- new("classicCount", testStatistic = GOFisherTest, name = "Fisher test")
resultFisher <- getSigGroups(GOdata, test.statF)
test.statKS <- new("classicScore", testStatistic = GOKSTest, name = "KS tests")
resultKS <- getSigGroups(GOdata, test.statKS)
test.statRW <- new("weightCount", testStatistic = GOFisherTest, name = "Fisher test", sigRatio = "ratio")
resultWeight <- getSigGroups(GOdata, test.statRW)
DF <- GenTable(GOdata, classic = resultFis, KS = resultKS, weight = resultWeight, orderBy = "classic", ranksOf = "classic", topNodes = 20)
DF
}
```
#acs
##Format df
```{r}
library(matrixStats)
```
```{r}
#Subset
df.acs <- df[,c(1:18,89)]
#Remove missing genes
df.acs <- df.acs[!(df.acs$B73.acs.B73.r1==0),]
#Calculate averages of bio reps
df.acs$B73avg <- rowMeans(df.acs[,9:13])
df.acs$ACSavg <- rowMeans(df.acs[,14:18])
#calculate ratio
df.acs$ACS.B73 <- df.acs$ACSavg/df.acs$B73avg
#calculate t-test
df.acs$Sig <- ttestFun(df.acs, 9:13, 14:18)
#calculate log2(acs/b73)
df.acs$l2fc <- log2(df.acs$ACS.B73)
#Calculate standard deviation
df.acs$SD <- rowSds(as.matrix(df.acs[,14:18]))
#subset to 10% significantly above or below mid-parent
df.acs.amp <- df.acs[(df.acs$Sig<0.05)&(df.acs$ACS.B73>1.1),]
df.acs.bmp <- df.acs[(df.acs$Sig<0.05)&(df.acs$ACS.B73<0.9),]
```
##GO term enrichment
```{r}
#Trim gamer to only those genes detected in our data, to use as background for enrichment
gamerCutACS <- gamer[(gamer$db_object_id %in% df.acs$Gene),]
```
```{r}
df.acs.amp_BP_GOdata <- GOanalysisGodata(df.acs.amp, gamerCutACS, "BP", nodesize = 5)
df.acs.amp_MF_GOdata <- GOanalysisGodata(df.acs.amp, gamerCutACS, "MF", nodesize = 5)
df.acs.amp_CC_GOdata <- GOanalysisGodata(df.acs.amp, gamerCutACS, "CC", nodesize = 5)

df.acs.bmp_BP_GOdata <- GOanalysisGodata(df.acs.bmp, gamerCutACS, "BP", nodesize = 5)
df.acs.bmp_MF_GOdata <- GOanalysisGodata(df.acs.bmp, gamerCutACS, "MF", nodesize = 5)
df.acs.bmp_CC_GOdata <- GOanalysisGodata(df.acs.bmp, gamerCutACS, "CC", nodesize = 5)
```
```{r}
dir.create("GOmaps")
GOanalysisMap(df.acs.amp_BP_GOdata, "GOmaps/df.acs.amp_BP_GOdata")
GOanalysisMap(df.acs.amp_MF_GOdata, "GOmaps/df.acs.amp_MF_GOdata")
GOanalysisMap(df.acs.amp_CC_GOdata, "GOmaps/df.acs.amp_CC_GOdata")

GOanalysisMap(df.acs.bmp_BP_GOdata, "GOmaps/df.acs.bmp_BP_GOdata")
GOanalysisMap(df.acs.bmp_MF_GOdata, "GOmaps/df.acs.bmp_MF_GOdata")
GOanalysisMap(df.acs.bmp_CC_GOdata, "GOmaps/df.acs.bmp_CC_GOdata")
```
```{r}
df.acs.amp_BP_GOdata.DF <- GOanalysisDF(df.acs.amp_BP_GOdata)
df.acs.amp_MF_GOdata.DF <- GOanalysisDF(df.acs.amp_MF_GOdata)
df.acs.amp_CC_GOdata.DF <- GOanalysisDF(df.acs.amp_CC_GOdata)

df.acs.bmp_BP_GOdata.DF <- GOanalysisDF(df.acs.bmp_BP_GOdata)
df.acs.bmp_MF_GOdata.DF <- GOanalysisDF(df.acs.bmp_MF_GOdata)
df.acs.bmp_CC_GOdata.DF <- GOanalysisDF(df.acs.bmp_CC_GOdata)
```
#Seedling Leaf - B73xMo17
##Format df
```{r}
#Subset
df.SL <- df[,c(1:8,19:28,89)]
#Remove missing genes
df.SL <- df.SL[!(df.SL$SeedlingLeaf.B73.r1==0),]
#Calculate averages of bio reps
df.SL$B73avg <- rowMeans(df.SL[,10:12])
df.SL$Mo17avg <- rowMeans(df.SL[,13:15])
df.SL$HYBavg <- rowMeans(df.SL[,16:18])
df.SL$MPavg <- rowMeans(df.SL[,c(10:15)])
df.SL$MP.1 <- rowMeans(df.SL[,c(10,13)])
df.SL$MP.2 <- rowMeans(df.SL[,c(11,14)])
df.SL$MP.3 <- rowMeans(df.SL[,c(12,15)])
#calculate ratio
df.SL$HYB.MP <- df.SL$HYBavg/df.SL$MPavg
#calculate t-test
df.SL$Sig <- ttestFun(df.SL, 10:12, 24:26)
#calculate log2(hyb/MP)
df.SL$l2fc <- log2(df.SL$HYB.MP)
#subset to 10% significantly above or below mid-parent
df.SL.amp <- df.SL[(df.SL$Sig<0.05)&(df.SL$HYB.MP>1.1),]
df.SL.bmp <- df.SL[(df.SL$Sig<0.05)&(df.SL$HYB.MP<0.9),]
```

##GO term enrichment
```{r}
#Trim gamer to only those genes detected in our data, to use as background for enrichment
gamerCutSL <- gamer[(gamer$db_object_id %in% df.SL$Gene),]
```

```{r}
df.SL.amp_BP_GOdata <- GOanalysisGodata(df.SL.amp, gamerCutSL, "BP", nodesize = 5)
df.SL.amp_MF_GOdata <- GOanalysisGodata(df.SL.amp, gamerCutSL, "MF", nodesize = 5)
df.SL.amp_CC_GOdata <- GOanalysisGodata(df.SL.amp, gamerCutSL, "CC", nodesize = 5)

df.SL.bmp_BP_GOdata <- GOanalysisGodata(df.SL.bmp, gamerCutSL, "BP", nodesize = 5)
df.SL.bmp_MF_GOdata <- GOanalysisGodata(df.SL.bmp, gamerCutSL, "MF", nodesize = 5)
df.SL.bmp_CC_GOdata <- GOanalysisGodata(df.SL.bmp, gamerCutSL, "CC", nodesize = 5)
```
```{r}
dir.create("GOmaps")
GOanalysisMap(df.SL.amp_BP_GOdata, "GOmaps/df.SL.amp_BP_GOdata")
GOanalysisMap(df.SL.amp_MF_GOdata, "GOmaps/df.SL.amp_MF_GOdata")
GOanalysisMap(df.SL.amp_CC_GOdata, "GOmaps/df.SL.amp_CC_GOdata")

GOanalysisMap(df.SL.bmp_BP_GOdata, "GOmaps/df.SL.bmp_BP_GOdata")
GOanalysisMap(df.SL.bmp_MF_GOdata, "GOmaps/df.SL.bmp_MF_GOdata")
GOanalysisMap(df.SL.bmp_CC_GOdata, "GOmaps/df.SL.bmp_CC_GOdata")
```
```{r}
df.SL.amp_BP_GOdata.DF <- GOanalysisDF(df.SL.amp_BP_GOdata)
df.SL.amp_MF_GOdata.DF <- GOanalysisDF(df.SL.amp_MF_GOdata)
df.SL.amp_CC_GOdata.DF <- GOanalysisDF(df.SL.amp_CC_GOdata)

df.SL.bmp_BP_GOdata.DF <- GOanalysisDF(df.SL.bmp_BP_GOdata)
df.SL.bmp_MF_GOdata.DF <- GOanalysisDF(df.SL.bmp_MF_GOdata)
df.SL.bmp_CC_GOdata.DF <- GOanalysisDF(df.SL.bmp_CC_GOdata)
```
#ACS and BxM enrichment overlap
```{r}
df.SL.acs.amp <- df.SL.amp[(df.SL.amp$Accession%in%df.acs.amp$Accession),]
df.SL.acs.bmp <- df.SL.bmp[(df.SL.bmp$Accession%in%df.acs.bmp$Accession),]
df.SL.acs <- df.SL[(df.SL$Gene%in%df.acs$Gene),]
```
```{r}
#Trim gamer to only those genes detected in our data, to use as background for enrichment
gamerCutSLacs <- gamer[(gamer$db_object_id %in% df.SL.acs$Gene),]
```
```{r}
df.SL.acs.amp_BP_GOdata <- GOanalysisGodata(df.SL.acs.amp, gamerCutSLacs, "BP", nodesize = 5)
df.SL.acs.amp_MF_GOdata <- GOanalysisGodata(df.SL.acs.amp, gamerCutSLacs, "MF", nodesize = 5)
df.SL.acs.amp_CC_GOdata <- GOanalysisGodata(df.SL.acs.amp, gamerCutSLacs, "CC", nodesize = 5)

df.SL.acs.bmp_BP_GOdata <- GOanalysisGodata(df.SL.acs.bmp, gamerCutSLacs, "BP", nodesize = 5)
df.SL.acs.bmp_MF_GOdata <- GOanalysisGodata(df.SL.acs.bmp, gamerCutSLacs, "MF", nodesize = 5)
df.SL.acs.bmp_CC_GOdata <- GOanalysisGodata(df.SL.acs.bmp, gamerCutSLacs, "CC", nodesize = 5)
```
```{r}
dir.create("GOmaps")
GOanalysisMap(df.SL.acs.amp_BP_GOdata, "GOmaps/df.SL.acs.amp_BP_GOdata")
GOanalysisMap(df.SL.acs.amp_MF_GOdata, "GOmaps/df.SL.acs.amp_MF_GOdata")
GOanalysisMap(df.SL.acs.amp_CC_GOdata, "GOmaps/df.SL.acs.amp_CC_GOdata")

GOanalysisMap(df.SL.acs.bmp_BP_GOdata, "GOmaps/df.SL.acs.bmp_BP_GOdata")
GOanalysisMap(df.SL.acs.bmp_MF_GOdata, "GOmaps/df.SL.acs.bmp_MF_GOdata")
GOanalysisMap(df.SL.acs.bmp_CC_GOdata, "GOmaps/df.SL.acs.bmp_CC_GOdata")
```
```{r}
df.SL.acs.amp_BP_GOdata.DF <- GOanalysisDF(df.SL.acs.amp_BP_GOdata)
df.SL.acs.amp_MF_GOdata.DF <- GOanalysisDF(df.SL.acs.amp_MF_GOdata)
df.SL.acs.amp_CC_GOdata.DF <- GOanalysisDF(df.SL.acs.amp_CC_GOdata)

df.SL.acs.bmp_BP_GOdata.DF <- GOanalysisDF(df.SL.acs.bmp_BP_GOdata)
df.SL.acs.bmp_MF_GOdata.DF <- GOanalysisDF(df.SL.acs.bmp_MF_GOdata)
df.SL.acs.bmp_CC_GOdata.DF <- GOanalysisDF(df.SL.acs.bmp_CC_GOdata)
```
```{r}
df.SL.acs.bmp_BP_GOdata.DF
```

#Merge acs and Seedling Leaf BxM data
```{r}
df.acs.SL <- merge.data.frame(df.acs[,c(1:8,19,24:25)], df.SL[,c(6,c(27,29))], by="Accession", all = F, suffixes = c(".acs", ".BM"))
```
#Define GOIs
Gene lists and sources:
1. Plastid encoded photosynthesis associated proteins  (EnsemblPlants Annotation)
2. Photosynthesis associated nuclear encoded proteins (PhANGs)  (GO GAMER+CornCyc)
3. Plastid encoded ribosomal proteins (EnsemblPlants Annotation)
4. Nuclear encoded plastid ribosomal proteins (GO GAMER and EnsemblPlants Annotation)
5. Eukaryotic ribosomal proteins (EnsemblPlants Annotation)
```{r}
#Plastid encoded photosynthesis associated proteins  (EnsemblPlants Annotation)
#generate vector of Pt encoded genes
PtGenes<-df[grep("chromosome:B73_RefGen_v4:Pt:", df$Protein),]

#define list of plastid-encoded photosynthesis-associated genes
PtPS <- c("ZemaCp008.1", "ZemaCp022.1", "ZemaCp023.1", "ZemaCp032.1", "ZemaCp008.1", "ZemaCp039.1", "ZemaCp040.1", "ZemaCp048.1", "ZemaCp051.1", "ZemaCp087.1", "ZemaCp017.1", "ZemaCp018.1", "ZemaCp019.1", "ZemaCp024.1", "ZemaCp027.1", "ZemaCp028.1", "ZemaCp029.1", "ZemaCp030.1", "ZemaCp035.1", "ZemaCp036.1", "ZemaCp052.1", "ZemaCp053.1", "ZemaCp075.1", "ZemaCp083.1", "ZemaCp085.1", "ZemaCp086.1", "ZemaCp090.1", "ZemaCp091.1", "ZemaCp092.1", "ZemaCp050.1", "ZemaCp043.1", "ZemaCp088.1")
```

```{r}
#Photosynthesis associated nuclear encoded proteins (PhANGs)  (GO GAMER+CornCyc)
PHANG<-as.vector(unique(gamer$db_object_id[grep("GO:0015979", gamer$term_accession)]))
#Add CornCyc annotated PhANGs
#accessed from https://corncyc-b73-v4.maizegdb.org/ on 1/4/2020
cornCyc<-as.data.frame(readr::read_csv(file = "cornCycGeneLists01042020.csv",
                         col_names = TRUE
                         ))
cornCyc.PHANG <- unique(c(cornCyc$Gene.Accession[(cornCyc$Pathway=="Oxygenic_Photosynthesis_SuperPath")], cornCyc$Gene.Accession[(cornCyc$Pathway=="Light_Rxn")]))
PHANG <- unique(c(PHANG, cornCyc.PHANG))
```

```{r}
#Plastid encoded ribosomal proteins (EnsemblPlants Annotation)
#get subset of Pt encoded genes annotated as ribosomal proteins
PtEncodeRibo<- c(PtGenes$Gene[(grep("description:rps", PtGenes$Protein))], PtGenes$Gene[(grep("description:rpl", PtGenes$Protein))], PtGenes$Gene[(grep("description:ribosom", PtGenes$Protein))])
```
```{r}
#Check that no plastid-encoded photosynthesis/ribosomal proteins were missed
Pt <- df[(grep("B73_RefGen_v4:Pt", df$Protein)),]
Pt.rpo <- Pt$Gene[(grep("rpo", Pt$Protein))]
Pt.other <- Pt$Gene[!(Pt$Accession%in%c(PtEncodeRibo, PtPS, Pt.rpo))]
```


```{r}
#Nuclear encoded plastid ribosomal proteins (GO GAMER and EnsemblPlants Annotation)
PtRibo<-as.vector(unique(gamer$db_object_id[grep("GO:0009547|GO:0043253", gamer$term_accession)]))
#Add proteins annotated as "ribosom" and "plastid/chloroplast" to the nuclear-encoded plastid-localized list
ribosom <- df[(grep("ribosom", df$Protein)),]
ribosom.pt <- ribosom$Gene[(grep("plastid|Plastid|Chloropl|chloropl", ribosom$Protein))]
PtRibo <- unique(c(PtRibo,ribosom.pt))

#Check that there are no missing prokaryotic ribosomal proteins with plastid localization
ProkRibo <- df[(grep("30S|60S", df$Protein)),]
ProkRiboNotPt <- ProkRibo[!(ProkRibo$Gene%in%PtRibo),]
GOplastid<-as.vector(unique(gamer$db_object_id[grep("GO:0009536", gamer$term_accession)]))
ProkRiboNotPt.plastid <- ProkRiboNotPt[(ProkRiboNotPt$Gene%in%GOplastid),]
PtRibo <- c(PtRibo, ProkRiboNotPt.plastid$Gene)
```
```{r}
#Eukaryotic ribosomal proteins (EnsemblPlants Annotation)
eukRibo<-df$Gene[(grep("40S|60S", df$Protein))]
```

```{r}
MAT <- cornCyc$`Gene Accession`[(cornCyc$`Enzymatic activity`=="methionine adenosyltransferase")]
ACS <- cornCyc$`Gene Accession`[(cornCyc$`Enzymatic activity`=="1-aminocyclopropane-1-carboxylate synthase")]
ACO <- cornCyc$`Gene Accession`[(cornCyc$`Enzymatic activity`=="1-aminocyclopropane-1-carboxylate oxidase")]
EtBio <- c(MAT,ACS,ACO)
```

```{r}
#Glyceraldehyde-3-Phosphate Metabolic Process
G3P <- as.vector(unique(gamer$db_object_id[grep("GO:0006098", gamer$term_accession)]))

#Response to Ethylene
ResponseToEt <- as.vector(unique(gamer$db_object_id[grep("GO:0009723", gamer$term_accession)]))

#Response to water
ResponseToWater <- as.vector(unique(gamer$db_object_id[grep("GO:0009415", gamer$term_accession)]))

#Pentose Phosphate Shunt
PPP <- as.vector(unique(gamer$db_object_id[grep("GO:0006098", gamer$term_accession)]))

#Glutathione Met. Process
Glutathione <- as.vector(unique(gamer$db_object_id[grep("GO:0006749", gamer$term_accession)]))

#Response to dessication
ResponseToDessication <- as.vector(unique(gamer$db_object_id[grep("GO:0009269", gamer$term_accession)]))
```
```{r}
ResponseToStress <- as.vector(unique(gamer$db_object_id[grep("GO:0006950", gamer$term_accession)]))
```



#Scatterplot1
##Make GOI description column
```{r}
df.acs.SL.scat1 <- df.acs.SL
df.acs.SL.scat1$Category <- "Other"
df.acs.SL.scat1$Order <- 1
df.acs.SL.scat1[(df.acs.SL.scat1$Gene %in% PPP),"Category"] <- "PPP"
df.acs.SL.scat1[(df.acs.SL.scat1$Gene %in% PPP),"Order"] <- 2
df.acs.SL.scat1[(df.acs.SL.scat1$Gene %in% PtPS),"Category"] <- "PlastidEncodedPS"
df.acs.SL.scat1[(df.acs.SL.scat1$Gene %in% PtPS),"Order"] <- 3
df.acs.SL.scat1[(df.acs.SL.scat1$Gene %in% PHANG),"Category"] <- "PHANGs"
df.acs.SL.scat1[(df.acs.SL.scat1$Gene %in% PHANG),"Order"] <- 4
df.acs.SL.scat1[(df.acs.SL.scat1$Gene %in% c(PtEncodeRibo,Pt.other)),"Category"] <- "PlastidEncodedRibo+Other"
df.acs.SL.scat1[(df.acs.SL.scat1$Gene %in% c(PtEncodeRibo,Pt.other)),"Order"] <- 5
df.acs.SL.scat1[(df.acs.SL.scat1$Gene %in% PtRibo),"Category"] <- "NuclearEncodedPlastidRibo"
df.acs.SL.scat1[(df.acs.SL.scat1$Gene %in% PtRibo),"Order"] <- 6
df.acs.SL.scat1[(df.acs.SL.scat1$Gene %in% eukRibo),"Category"] <- "CytosolicRibo"
df.acs.SL.scat1[(df.acs.SL.scat1$Gene %in% eukRibo),"Order"] <- 7
df.acs.SL.scat1[(df.acs.SL.scat1$Gene %in% Pt.rpo),"Category"] <- "Pt.rpo"
df.acs.SL.scat1[(df.acs.SL.scat1$Gene %in% Pt.rpo),"Order"] <- 8

#Re-order
df.acs.SL.scat1 <- df.acs.SL.scat1[(order(df.acs.SL.scat1$Order)),]
```
##Set limits for plot
```{r}
df.acs.SL.scat1.lims <- df.acs.SL.scat1

df.acs.SL.scat1.lims$l2fc.acs[df.acs.SL.scat1.lims$l2fc.acs > 1.0] = 1.0
df.acs.SL.scat1.lims$l2fc.acs[df.acs.SL.scat1.lims$l2fc.acs < -1.0] = -1.0

df.acs.SL.scat1.lims$l2fc.BM[df.acs.SL.scat1.lims$l2fc.BM > 1.0] = 1.0
df.acs.SL.scat1.lims$l2fc.BM[df.acs.SL.scat1.lims$l2fc.BM < -1.0] = -1.0
```
##Make scatterplot
```{r}
library(ggplot2)
```
```{r}
fig4 <- ggplot(df.acs.SL.scat1.lims, mapping=aes(x=l2fc.acs, y=l2fc.BM, color=Category))+
  geom_point(size=.8)+
  scale_color_manual(values=c("Other"="grey70","PPP"="#efb806", "PlastidEncodedPS"="red3", "PHANGs"="#ef06db", "PlastidEncodedRibo+Other"="blue4", "NuclearEncodedPlastidRibo"="dodgerblue1", "CytosolicRibo"="#019d01", "Pt.rpo"="#653f01"), labels=c("Other"="Other", "PPP"="Pentose Phosphate Shunt", "PlastidEncodedPS"="Plastid-encoded photosynthesis-associated", "PHANGs"="Nuclear-encoded photosynthesis-associated", "PlastidEncodedRibo+Other"="Plastid-encoded ribosome +other", "NuclearEncodedPlastidRibo"="Nuclear-encoded plastid ribosome", "CytosolicRibo"="Cytosolic ribosome","Pt.rpo"="Plastid-encoded RNA polymerase"), name="Protein Category")+
  theme_bw()+
  xlab("log2(acs/B73)")+
  ylab("log2(BxM/MP)")+
  geom_abline(slope = 1, color="grey50", size=.5)+
  geom_vline(color="grey50", size=.5, xintercept = 0)+
  geom_hline(color="grey50", size=.5, yintercept = 0)+
  coord_fixed(expand = F)+
  xlim(-1,1)+
  ylim(-1,1)+
  theme(legend.position = "bottom", legend.text=element_text(size=7), legend.title = element_text(size=9))+
  theme(legend.position = "none")+
  theme(axis.title = element_text(size=20))

fig4
```
##Add density curves
```{r}
library(cowplot)
```
```{r}
#add density curves
#add marginal density plots and resave
    densityplotX<-axis_canvas(fig4, axis="x")+
      geom_density(aes(df.acs.SL.scat1.lims[,"l2fc.acs"], color=df.acs.SL.scat1.lims[,"Category"]), df.acs.SL.scat1.lims, size=0.9)+
      scale_color_manual(values=c("Other"="grey70","PPP"="#efb806", "PlastidEncodedPS"="red3", "PHANGs"="#ef06db", "PlastidEncodedRibo+Other"="blue4", "NuclearEncodedPlastidRibo"="dodgerblue1", "CytosolicRibo"="#019d01", "Pt.rpo"="#653f01"))+
      geom_vline(xintercept=0, linetype="dashed", color = "gray50")+
      theme_void()+
      theme(legend.title = element_blank())+
      ylim(0,15)
    #set up marginal density plot Y axis
    densityplotY<-axis_canvas(fig4, axis="y", coord_flip = TRUE)+
      geom_density(aes(df.acs.SL.scat1.lims[,"l2fc.BM"], color=df.acs.SL.scat1.lims[,"Category"]), df.acs.SL.scat1.lims, size=0.9)+
      scale_color_manual(values=c("Other"="grey70","PPP"="#efb806", "PlastidEncodedPS"="red3", "PHANGs"="#ef06db", "PlastidEncodedRibo+Other"="blue4", "NuclearEncodedPlastidRibo"="dodgerblue1", "CytosolicRibo"="#019d01", "Pt.rpo"="#653f01"))+
      geom_vline(xintercept=0, linetype="dashed", color = "gray50")+
      theme_void()+
      theme(legend.title = element_blank())+
      coord_flip()+
      ylim(0,15)
    #put two plots together
    q <- insert_xaxis_grob(fig4, densityplotX, position = "top")
    q <- insert_yaxis_grob(q, densityplotY, position = "right")
    #save
    ggsave(plot=q, filename = "Fig4.tiff",
           device="tiff", dpi=600, width=9, height=8, compression = "lzw")
```
#Scatterplot1b
##Make GOI description column
```{r}
df.acs.SL.scat1b <- df.acs.SL
df.acs.SL.scat1b$Category <- "Other"
df.acs.SL.scat1b$Order <- 1
#df.acs.SL.scat1b[(df.acs.SL.scat1b$Gene %in% PPP),"Category"] <- "PPP"
#df.acs.SL.scat1b[(df.acs.SL.scat1b$Gene %in% PPP),"Order"] <- 2
df.acs.SL.scat1b[(df.acs.SL.scat1b$Gene %in% PtPS),"Category"] <- "PlastidEncodedPS"
df.acs.SL.scat1b[(df.acs.SL.scat1b$Gene %in% PtPS),"Order"] <- 3
#df.acs.SL.scat1b[(df.acs.SL.scat1b$Gene %in% PHANG),"Category"] <- "PHANGs"
#df.acs.SL.scat1b[(df.acs.SL.scat1b$Gene %in% PHANG),"Order"] <- 4
#df.acs.SL.scat1b[(df.acs.SL.scat1b$Gene %in% c(PtEncodeRibo,Pt.other)),"Category"] <- "PlastidEncodedRibo+Other"
#df.acs.SL.scat1b[(df.acs.SL.scat1b$Gene %in% c(PtEncodeRibo,Pt.other)),"Order"] <- 5
#df.acs.SL.scat1b[(df.acs.SL.scat1b$Gene %in% PtRibo),"Category"] <- "NuclearEncodedPlastidRibo"
#df.acs.SL.scat1b[(df.acs.SL.scat1b$Gene %in% PtRibo),"Order"] <- 6
#df.acs.SL.scat1b[(df.acs.SL.scat1b$Gene %in% eukRibo),"Category"] <- "CytosolicRibo"
#df.acs.SL.scat1b[(df.acs.SL.scat1b$Gene %in% eukRibo),"Order"] <- 7
#df.acs.SL.scat1b[(df.acs.SL.scat1b$Gene %in% Pt.rpo),"Category"] <- "Pt.rpo"
#df.acs.SL.scat1b[(df.acs.SL.scat1b$Gene %in% Pt.rpo),"Order"] <- 8

#Re-order
df.acs.SL.scat1b <- df.acs.SL.scat1b[(order(df.acs.SL.scat1b$Order)),]
```
##Set limits for plot
```{r}
df.acs.SL.scat1b.lims <- df.acs.SL.scat1b

df.acs.SL.scat1b.lims$l2fc.acs[df.acs.SL.scat1b.lims$l2fc.acs > 1.0] = 1.0
df.acs.SL.scat1b.lims$l2fc.acs[df.acs.SL.scat1b.lims$l2fc.acs < -1.0] = -1.0

df.acs.SL.scat1b.lims$l2fc.BM[df.acs.SL.scat1b.lims$l2fc.BM > 1.0] = 1.0
df.acs.SL.scat1b.lims$l2fc.BM[df.acs.SL.scat1b.lims$l2fc.BM < -1.0] = -1.0
```
##Make scatterplot
```{r}
library(ggplot2)
```
```{r}
fig4b <- ggplot(df.acs.SL.scat1b.lims, mapping=aes(x=l2fc.acs, y=l2fc.BM, color=Category))+
  geom_point(size=.8)+
  scale_color_manual(values=c("Other"="grey70","PlastidEncodedPS"="red3"), labels=c("Other"="Other","PlastidEncodedPS"="Plastid-encoded photosynthesis-associated"), name="Protein Category")+
  theme_bw()+
  xlab("log2(acs/B73)")+
  ylab("log2(BxM/MP)")+
  geom_abline(slope = 1, color="grey50", size=.5)+
  geom_vline(color="grey50", size=.5, xintercept = 0)+
  geom_hline(color="grey50", size=.5, yintercept = 0)+
  coord_fixed(expand = F)+
  xlim(-1,1)+
  ylim(-1,1)+
  theme(legend.position = "bottom", legend.text=element_text(size=7), legend.title = element_text(size=9))+
  theme(legend.position = "none")+
  theme(axis.title = element_text(size=20))

fig4b
```
##Add density curves
```{r}
library(cowplot)
```
```{r}
#add density curves
#add marginal density plots and resave
    densityplotX<-axis_canvas(fig4b, axis="x")+
      geom_density(aes(df.acs.SL.scat1b.lims[,"l2fc.acs"], color=df.acs.SL.scat1b.lims[,"Category"]), df.acs.SL.scat1b.lims, size=0.9)+
      scale_color_manual(values=c("Other"="grey70", "PlastidEncodedPS"="red3"))+
      geom_vline(xintercept=0, linetype="dashed", color = "gray50")+
      theme_void()+
      theme(legend.title = element_blank())+
      ylim(0,15)
    #set up marginal density plot Y axis
    densityplotY<-axis_canvas(fig4b, axis="y", coord_flip = TRUE)+
      geom_density(aes(df.acs.SL.scat1b.lims[,"l2fc.BM"], color=df.acs.SL.scat1b.lims[,"Category"]), df.acs.SL.scat1b.lims, size=0.9)+
      scale_color_manual(values=c("Other"="grey70", "PlastidEncodedPS"="red3"))+
      geom_vline(xintercept=0, linetype="dashed", color = "gray50")+
      theme_void()+
      theme(legend.title = element_blank())+
      coord_flip()+
      ylim(0,15)
    #put two plots together
    q <- insert_xaxis_grob(fig4b, densityplotX, position = "top")
    q <- insert_yaxis_grob(q, densityplotY, position = "right")
    #save
    ggsave(plot=q, filename = "Fig4b.tiff",
           device="tiff", dpi=600, width=9, height=8, compression = "lzw")
```
#Scatterplot1c
##Make GOI description column
```{r}
df.acs.SL.scat1c <- df.acs.SL
df.acs.SL.scat1c$Category <- "Other"
df.acs.SL.scat1c$Order <- 1
#df.acs.SL.scat1c[(df.acs.SL.scat1c$Gene %in% PPP),"Category"] <- "PPP"
#df.acs.SL.scat1c[(df.acs.SL.scat1c$Gene %in% PPP),"Order"] <- 2
df.acs.SL.scat1c[(df.acs.SL.scat1c$Gene %in% PtPS),"Category"] <- "PlastidEncodedPS"
df.acs.SL.scat1c[(df.acs.SL.scat1c$Gene %in% PtPS),"Order"] <- 3
df.acs.SL.scat1c[(df.acs.SL.scat1c$Gene %in% PHANG),"Category"] <- "PHANGs"
df.acs.SL.scat1c[(df.acs.SL.scat1c$Gene %in% PHANG),"Order"] <- 4
#df.acs.SL.scat1c[(df.acs.SL.scat1c$Gene %in% c(PtEncodeRibo,Pt.other)),"Category"] <- "PlastidEncodedRibo+Other"
#df.acs.SL.scat1c[(df.acs.SL.scat1c$Gene %in% c(PtEncodeRibo,Pt.other)),"Order"] <- 5
#df.acs.SL.scat1c[(df.acs.SL.scat1c$Gene %in% PtRibo),"Category"] <- "NuclearEncodedPlastidRibo"
#df.acs.SL.scat1c[(df.acs.SL.scat1c$Gene %in% PtRibo),"Order"] <- 6
#df.acs.SL.scat1c[(df.acs.SL.scat1c$Gene %in% eukRibo),"Category"] <- "CytosolicRibo"
#df.acs.SL.scat1c[(df.acs.SL.scat1c$Gene %in% eukRibo),"Order"] <- 7
#df.acs.SL.scat1c[(df.acs.SL.scat1c$Gene %in% Pt.rpo),"Category"] <- "Pt.rpo"
#df.acs.SL.scat1c[(df.acs.SL.scat1c$Gene %in% Pt.rpo),"Order"] <- 8

#Re-order
df.acs.SL.scat1c <- df.acs.SL.scat1c[(order(df.acs.SL.scat1c$Order)),]
```
##Set limits for plot
```{r}
df.acs.SL.scat1c.lims <- df.acs.SL.scat1c

df.acs.SL.scat1c.lims$l2fc.acs[df.acs.SL.scat1c.lims$l2fc.acs > 1.0] = 1.0
df.acs.SL.scat1c.lims$l2fc.acs[df.acs.SL.scat1c.lims$l2fc.acs < -1.0] = -1.0

df.acs.SL.scat1c.lims$l2fc.BM[df.acs.SL.scat1c.lims$l2fc.BM > 1.0] = 1.0
df.acs.SL.scat1c.lims$l2fc.BM[df.acs.SL.scat1c.lims$l2fc.BM < -1.0] = -1.0
```
##Make scatterplot
```{r}
library(ggplot2)
```
```{r}
fig4c <- ggplot(df.acs.SL.scat1c.lims, mapping=aes(x=l2fc.acs, y=l2fc.BM, color=Category))+
  geom_point(size=.8)+
  scale_color_manual(values=c("Other"="grey70", "PlastidEncodedPS"="red3", "PHANGs"="#ef06db"), labels=c("Other"="Other", "PlastidEncodedPS"="Plastid-encoded photosynthesis-associated", "PHANGs"="Nuclear-encoded photosynthesis-associated"), name="Protein Category")+
  theme_bw()+
  xlab("log2(acs/B73)")+
  ylab("log2(BxM/MP)")+
  geom_abline(slope = 1, color="grey50", size=.5)+
  geom_vline(color="grey50", size=.5, xintercept = 0)+
  geom_hline(color="grey50", size=.5, yintercept = 0)+
  coord_fixed(expand = F)+
  xlim(-1,1)+
  ylim(-1,1)+
  theme(legend.position = "bottom", legend.text=element_text(size=7), legend.title = element_text(size=9))+
  theme(legend.position = "none")+
  theme(axis.title = element_text(size=20))

fig4c
```
##Add density curves
```{r}
library(cowplot)
```
```{r}
#add density curves
#add marginal density plots and resave
    densityplotX<-axis_canvas(fig4c, axis="x")+
      geom_density(aes(df.acs.SL.scat1c.lims[,"l2fc.acs"], color=df.acs.SL.scat1c.lims[,"Category"]), df.acs.SL.scat1c.lims, size=0.9)+
      scale_color_manual(values=c("Other"="grey70", "PlastidEncodedPS"="red3", "PHANGs"="#ef06db"))+
      geom_vline(xintercept=0, linetype="dashed", color = "gray50")+
      theme_void()+
      theme(legend.title = element_blank())+
      ylim(0,15)
    #set up marginal density plot Y axis
    densityplotY<-axis_canvas(fig4c, axis="y", coord_flip = TRUE)+
      geom_density(aes(df.acs.SL.scat1c.lims[,"l2fc.BM"], color=df.acs.SL.scat1c.lims[,"Category"]), df.acs.SL.scat1c.lims, size=0.9)+
      scale_color_manual(values=c("Other"="grey70", "PlastidEncodedPS"="red3", "PHANGs"="#ef06db"))+
      geom_vline(xintercept=0, linetype="dashed", color = "gray50")+
      theme_void()+
      theme(legend.title = element_blank())+
      coord_flip()+
      ylim(0,15)
    #put two plots together
    q <- insert_xaxis_grob(fig4c, densityplotX, position = "top")
    q <- insert_yaxis_grob(q, densityplotY, position = "right")
    #save
    ggsave(plot=q, filename = "Fig4c.tiff",
           device="tiff", dpi=600, width=9, height=8, compression = "lzw")
```
#Scatterplot1d
##Make GOI description column
```{r}
df.acs.SL.scat1d <- df.acs.SL
df.acs.SL.scat1d$Category <- "Other"
df.acs.SL.scat1d$Order <- 1
#df.acs.SL.scat1d[(df.acs.SL.scat1d$Gene %in% PPP),"Category"] <- "PPP"
#df.acs.SL.scat1d[(df.acs.SL.scat1d$Gene %in% PPP),"Order"] <- 2
df.acs.SL.scat1d[(df.acs.SL.scat1d$Gene %in% PtPS),"Category"] <- "PlastidEncodedPS"
df.acs.SL.scat1d[(df.acs.SL.scat1d$Gene %in% PtPS),"Order"] <- 3
df.acs.SL.scat1d[(df.acs.SL.scat1d$Gene %in% PHANG),"Category"] <- "PHANGs"
df.acs.SL.scat1d[(df.acs.SL.scat1d$Gene %in% PHANG),"Order"] <- 4
df.acs.SL.scat1d[(df.acs.SL.scat1d$Gene %in% c(PtEncodeRibo,Pt.other)),"Category"] <- "PlastidEncodedRibo+Other"
df.acs.SL.scat1d[(df.acs.SL.scat1d$Gene %in% c(PtEncodeRibo,Pt.other)),"Order"] <- 5
#df.acs.SL.scat1d[(df.acs.SL.scat1d$Gene %in% PtRibo),"Category"] <- "NuclearEncodedPlastidRibo"
#df.acs.SL.scat1d[(df.acs.SL.scat1d$Gene %in% PtRibo),"Order"] <- 6
#df.acs.SL.scat1d[(df.acs.SL.scat1d$Gene %in% eukRibo),"Category"] <- "CytosolicRibo"
#df.acs.SL.scat1d[(df.acs.SL.scat1d$Gene %in% eukRibo),"Order"] <- 7
#df.acs.SL.scat1d[(df.acs.SL.scat1d$Gene %in% Pt.rpo),"Category"] <- "Pt.rpo"
#df.acs.SL.scat1d[(df.acs.SL.scat1d$Gene %in% Pt.rpo),"Order"] <- 8

#Re-order
df.acs.SL.scat1d <- df.acs.SL.scat1d[(order(df.acs.SL.scat1d$Order)),]
```
##Set limits for plot
```{r}
df.acs.SL.scat1d.lims <- df.acs.SL.scat1d

df.acs.SL.scat1d.lims$l2fc.acs[df.acs.SL.scat1d.lims$l2fc.acs > 1.0] = 1.0
df.acs.SL.scat1d.lims$l2fc.acs[df.acs.SL.scat1d.lims$l2fc.acs < -1.0] = -1.0

df.acs.SL.scat1d.lims$l2fc.BM[df.acs.SL.scat1d.lims$l2fc.BM > 1.0] = 1.0
df.acs.SL.scat1d.lims$l2fc.BM[df.acs.SL.scat1d.lims$l2fc.BM < -1.0] = -1.0
```
##Make scatterplot
```{r}
library(ggplot2)
```
```{r}
fig4d <- ggplot(df.acs.SL.scat1d.lims, mapping=aes(x=l2fc.acs, y=l2fc.BM, color=Category))+
  geom_point(size=.8)+
  scale_color_manual(values=c("Other"="grey70", "PlastidEncodedPS"="red3", "PHANGs"="#ef06db", "PlastidEncodedRibo+Other"="blue4"), labels=c("Other"="Other",  "PlastidEncodedPS"="Plastid-encoded photosynthesis-associated", "PHANGs"="Nuclear-encoded photosynthesis-associated", "PlastidEncodedRibo+Other"="Plastid-encoded ribosome +other"), name="Protein Category")+
  theme_bw()+
  xlab("log2(acs/B73)")+
  ylab("log2(BxM/MP)")+
  geom_abline(slope = 1, color="grey50", size=.5)+
  geom_vline(color="grey50", size=.5, xintercept = 0)+
  geom_hline(color="grey50", size=.5, yintercept = 0)+
  coord_fixed(expand = F)+
  xlim(-1,1)+
  ylim(-1,1)+
  theme(legend.position = "bottom", legend.text=element_text(size=7), legend.title = element_text(size=9))+
  theme(legend.position = "none")+
  theme(axis.title = element_text(size=20))

fig4d
```
##Add density curves
```{r}
library(cowplot)
```
```{r}
#add density curves
#add marginal density plots and resave
    densityplotX<-axis_canvas(fig4d, axis="x")+
      geom_density(aes(df.acs.SL.scat1d.lims[,"l2fc.acs"], color=df.acs.SL.scat1d.lims[,"Category"]), df.acs.SL.scat1d.lims, size=0.9)+
      scale_color_manual(values=c("Other"="grey70", "PlastidEncodedPS"="red3", "PHANGs"="#ef06db", "PlastidEncodedRibo+Other"="blue4"))+
      geom_vline(xintercept=0, linetype="dashed", color = "gray50")+
      theme_void()+
      theme(legend.title = element_blank())+
      ylim(0,15)
    #set up marginal density plot Y axis
    densityplotY<-axis_canvas(fig4d, axis="y", coord_flip = TRUE)+
      geom_density(aes(df.acs.SL.scat1d.lims[,"l2fc.BM"], color=df.acs.SL.scat1d.lims[,"Category"]), df.acs.SL.scat1d.lims, size=0.9)+
      scale_color_manual(values=c("Other"="grey70", "PlastidEncodedPS"="red3", "PHANGs"="#ef06db", "PlastidEncodedRibo+Other"="blue4"))+
      geom_vline(xintercept=0, linetype="dashed", color = "gray50")+
      theme_void()+
      theme(legend.title = element_blank())+
      coord_flip()+
      ylim(0,15)
    #put two plots together
    q <- insert_xaxis_grob(fig4d, densityplotX, position = "top")
    q <- insert_yaxis_grob(q, densityplotY, position = "right")
    #save
    ggsave(plot=q, filename = "Fig4d.tiff",
           device="tiff", dpi=600, width=9, height=8, compression = "lzw")
```
#Scatterplot1e
##Make GOI description column
```{r}
df.acs.SL.scat1e <- df.acs.SL
df.acs.SL.scat1e$Category <- "Other"
df.acs.SL.scat1e$Order <- 1
#df.acs.SL.scat1e[(df.acs.SL.scat1e$Gene %in% PPP),"Category"] <- "PPP"
#df.acs.SL.scat1e[(df.acs.SL.scat1e$Gene %in% PPP),"Order"] <- 2
df.acs.SL.scat1e[(df.acs.SL.scat1e$Gene %in% PtPS),"Category"] <- "PlastidEncodedPS"
df.acs.SL.scat1e[(df.acs.SL.scat1e$Gene %in% PtPS),"Order"] <- 3
df.acs.SL.scat1e[(df.acs.SL.scat1e$Gene %in% PHANG),"Category"] <- "PHANGs"
df.acs.SL.scat1e[(df.acs.SL.scat1e$Gene %in% PHANG),"Order"] <- 4
df.acs.SL.scat1e[(df.acs.SL.scat1e$Gene %in% c(PtEncodeRibo,Pt.other)),"Category"] <- "PlastidEncodedRibo+Other"
df.acs.SL.scat1e[(df.acs.SL.scat1e$Gene %in% c(PtEncodeRibo,Pt.other)),"Order"] <- 5
df.acs.SL.scat1e[(df.acs.SL.scat1e$Gene %in% PtRibo),"Category"] <- "NuclearEncodedPlastidRibo"
df.acs.SL.scat1e[(df.acs.SL.scat1e$Gene %in% PtRibo),"Order"] <- 6
#df.acs.SL.scat1e[(df.acs.SL.scat1e$Gene %in% eukRibo),"Category"] <- "CytosolicRibo"
#df.acs.SL.scat1e[(df.acs.SL.scat1e$Gene %in% eukRibo),"Order"] <- 7
#df.acs.SL.scat1e[(df.acs.SL.scat1e$Gene %in% Pt.rpo),"Category"] <- "Pt.rpo"
#df.acs.SL.scat1e[(df.acs.SL.scat1e$Gene %in% Pt.rpo),"Order"] <- 8

#Re-order
df.acs.SL.scat1e <- df.acs.SL.scat1e[(order(df.acs.SL.scat1e$Order)),]
```
##Set limits for plot
```{r}
df.acs.SL.scat1e.lims <- df.acs.SL.scat1e

df.acs.SL.scat1e.lims$l2fc.acs[df.acs.SL.scat1e.lims$l2fc.acs > 1.0] = 1.0
df.acs.SL.scat1e.lims$l2fc.acs[df.acs.SL.scat1e.lims$l2fc.acs < -1.0] = -1.0

df.acs.SL.scat1e.lims$l2fc.BM[df.acs.SL.scat1e.lims$l2fc.BM > 1.0] = 1.0
df.acs.SL.scat1e.lims$l2fc.BM[df.acs.SL.scat1e.lims$l2fc.BM < -1.0] = -1.0
```
##Make scatterplot
```{r}
library(ggplot2)
```
```{r}
fig4e <- ggplot(df.acs.SL.scat1e.lims, mapping=aes(x=l2fc.acs, y=l2fc.BM, color=Category))+
  geom_point(size=.8)+
  scale_color_manual(values=c("Other"="grey70", "PlastidEncodedPS"="red3", "PHANGs"="#ef06db", "PlastidEncodedRibo+Other"="blue4", "NuclearEncodedPlastidRibo"="dodgerblue1"), labels=c("Other"="Other", "PlastidEncodedPS"="Plastid-encoded photosynthesis-associated", "PHANGs"="Nuclear-encoded photosynthesis-associated", "PlastidEncodedRibo+Other"="Plastid-encoded ribosome +other", "NuclearEncodedPlastidRibo"="Nuclear-encoded plastid ribosome"), name="Protein Category")+
  theme_bw()+
  xlab("log2(acs/B73)")+
  ylab("log2(BxM/MP)")+
  geom_abline(slope = 1, color="grey50", size=.5)+
  geom_vline(color="grey50", size=.5, xintercept = 0)+
  geom_hline(color="grey50", size=.5, yintercept = 0)+
  coord_fixed(expand = F)+
  xlim(-1,1)+
  ylim(-1,1)+
  theme(legend.position = "bottom", legend.text=element_text(size=7), legend.title = element_text(size=9))+
  theme(legend.position = "none")+
  theme(axis.title = element_text(size=20))

fig4e
```
##Add density curves
```{r}
library(cowplot)
```
```{r}
#add density curves
#add marginal density plots and resave
    densityplotX<-axis_canvas(fig4e, axis="x")+
      geom_density(aes(df.acs.SL.scat1e.lims[,"l2fc.acs"], color=df.acs.SL.scat1e.lims[,"Category"]), df.acs.SL.scat1e.lims, size=0.9)+
      scale_color_manual(values=c("Other"="grey70", "PlastidEncodedPS"="red3", "PHANGs"="#ef06db", "PlastidEncodedRibo+Other"="blue4", "NuclearEncodedPlastidRibo"="dodgerblue1"))+
      geom_vline(xintercept=0, linetype="dashed", color = "gray50")+
      theme_void()+
      theme(legend.title = element_blank())+
      ylim(0,15)
    #set up marginal density plot Y axis
    densityplotY<-axis_canvas(fig4e, axis="y", coord_flip = TRUE)+
      geom_density(aes(df.acs.SL.scat1e.lims[,"l2fc.BM"], color=df.acs.SL.scat1e.lims[,"Category"]), df.acs.SL.scat1e.lims, size=0.9)+
      scale_color_manual(values=c("Other"="grey70", "PlastidEncodedPS"="red3", "PHANGs"="#ef06db", "PlastidEncodedRibo+Other"="blue4", "NuclearEncodedPlastidRibo"="dodgerblue1"))+
      geom_vline(xintercept=0, linetype="dashed", color = "gray50")+
      theme_void()+
      theme(legend.title = element_blank())+
      coord_flip()+
      ylim(0,15)
    #put two plots together
    q <- insert_xaxis_grob(fig4e, densityplotX, position = "top")
    q <- insert_yaxis_grob(q, densityplotY, position = "right")
    #save
    ggsave(plot=q, filename = "Fig4e.tiff",
           device="tiff", dpi=600, width=9, height=8, compression = "lzw")
```
#Scatterplot1f
##Make GOI description column
```{r}
df.acs.SL.scat1f <- df.acs.SL
df.acs.SL.scat1f$Category <- "Other"
df.acs.SL.scat1f$Order <- 1
#df.acs.SL.scat1f[(df.acs.SL.scat1f$Gene %in% PPP),"Category"] <- "PPP"
#df.acs.SL.scat1f[(df.acs.SL.scat1f$Gene %in% PPP),"Order"] <- 2
df.acs.SL.scat1f[(df.acs.SL.scat1f$Gene %in% PtPS),"Category"] <- "PlastidEncodedPS"
df.acs.SL.scat1f[(df.acs.SL.scat1f$Gene %in% PtPS),"Order"] <- 3
df.acs.SL.scat1f[(df.acs.SL.scat1f$Gene %in% PHANG),"Category"] <- "PHANGs"
df.acs.SL.scat1f[(df.acs.SL.scat1f$Gene %in% PHANG),"Order"] <- 4
df.acs.SL.scat1f[(df.acs.SL.scat1f$Gene %in% c(PtEncodeRibo,Pt.other)),"Category"] <- "PlastidEncodedRibo+Other"
df.acs.SL.scat1f[(df.acs.SL.scat1f$Gene %in% c(PtEncodeRibo,Pt.other)),"Order"] <- 5
df.acs.SL.scat1f[(df.acs.SL.scat1f$Gene %in% PtRibo),"Category"] <- "NuclearEncodedPlastidRibo"
df.acs.SL.scat1f[(df.acs.SL.scat1f$Gene %in% PtRibo),"Order"] <- 6
#df.acs.SL.scat1f[(df.acs.SL.scat1f$Gene %in% eukRibo),"Category"] <- "CytosolicRibo"
#df.acs.SL.scat1f[(df.acs.SL.scat1f$Gene %in% eukRibo),"Order"] <- 7
df.acs.SL.scat1f[(df.acs.SL.scat1f$Gene %in% Pt.rpo),"Category"] <- "Pt.rpo"
df.acs.SL.scat1f[(df.acs.SL.scat1f$Gene %in% Pt.rpo),"Order"] <- 8

#Re-order
df.acs.SL.scat1f <- df.acs.SL.scat1f[(order(df.acs.SL.scat1f$Order)),]
```
##Set limits for plot
```{r}
df.acs.SL.scat1f.lims <- df.acs.SL.scat1f

df.acs.SL.scat1f.lims$l2fc.acs[df.acs.SL.scat1f.lims$l2fc.acs > 1.0] = 1.0
df.acs.SL.scat1f.lims$l2fc.acs[df.acs.SL.scat1f.lims$l2fc.acs < -1.0] = -1.0

df.acs.SL.scat1f.lims$l2fc.BM[df.acs.SL.scat1f.lims$l2fc.BM > 1.0] = 1.0
df.acs.SL.scat1f.lims$l2fc.BM[df.acs.SL.scat1f.lims$l2fc.BM < -1.0] = -1.0
```
##Make scatterplot
```{r}
library(ggplot2)
```
```{r}
fig4f <- ggplot(df.acs.SL.scat1f.lims, mapping=aes(x=l2fc.acs, y=l2fc.BM, color=Category))+
  geom_point(size=.8)+
  scale_color_manual(values=c("Other"="grey70", "PlastidEncodedPS"="red3", "PHANGs"="#ef06db", "PlastidEncodedRibo+Other"="blue4", "NuclearEncodedPlastidRibo"="dodgerblue1", "Pt.rpo"="#653f01"), labels=c("Other"="Other", "PlastidEncodedPS"="Plastid-encoded photosynthesis-associated", "PHANGs"="Nuclear-encoded photosynthesis-associated", "PlastidEncodedRibo+Other"="Plastid-encoded ribosome +other", "NuclearEncodedPlastidRibo"="Nuclear-encoded plastid ribosome","Pt.rpo"="Plastid-encoded RNA polymerase"), name="Protein Category")+
  theme_bw()+
  xlab("log2(acs/B73)")+
  ylab("log2(BxM/MP)")+
  geom_abline(slope = 1, color="grey50", size=.5)+
  geom_vline(color="grey50", size=.5, xintercept = 0)+
  geom_hline(color="grey50", size=.5, yintercept = 0)+
  coord_fixed(expand = F)+
  xlim(-1,1)+
  ylim(-1,1)+
  theme(legend.position = "bottom", legend.text=element_text(size=7), legend.title = element_text(size=9))+
  theme(legend.position = "none")+
  theme(axis.title = element_text(size=20))

fig4f
```
##Add density curves
```{r}
library(cowplot)
```
```{r}
#add density curves
#add marginal density plots and resave
    densityplotX<-axis_canvas(fig4f, axis="x")+
      geom_density(aes(df.acs.SL.scat1f.lims[,"l2fc.acs"], color=df.acs.SL.scat1f.lims[,"Category"]), df.acs.SL.scat1f.lims, size=0.9)+
      scale_color_manual(values=c("Other"="grey70", "PlastidEncodedPS"="red3", "PHANGs"="#ef06db", "PlastidEncodedRibo+Other"="blue4", "NuclearEncodedPlastidRibo"="dodgerblue1", "Pt.rpo"="#653f01"))+
      geom_vline(xintercept=0, linetype="dashed", color = "gray50")+
      theme_void()+
      theme(legend.title = element_blank())+
      ylim(0,15)
    #set up marginal density plot Y axis
    densityplotY<-axis_canvas(fig4f, axis="y", coord_flip = TRUE)+
      geom_density(aes(df.acs.SL.scat1f.lims[,"l2fc.BM"], color=df.acs.SL.scat1f.lims[,"Category"]), df.acs.SL.scat1f.lims, size=0.9)+
      scale_color_manual(values=c("Other"="grey70", "PlastidEncodedPS"="red3", "PHANGs"="#ef06db", "PlastidEncodedRibo+Other"="blue4", "NuclearEncodedPlastidRibo"="dodgerblue1", "Pt.rpo"="#653f01"))+
      geom_vline(xintercept=0, linetype="dashed", color = "gray50")+
      theme_void()+
      theme(legend.title = element_blank())+
      coord_flip()+
      ylim(0,15)
    #put two plots together
    q <- insert_xaxis_grob(fig4f, densityplotX, position = "top")
    q <- insert_yaxis_grob(q, densityplotY, position = "right")
    #save
    ggsave(plot=q, filename = "Fig4f.tiff",
           device="tiff", dpi=600, width=9, height=8, compression = "lzw")
```
#Scatterplot1g
##Make GOI description column
```{r}
df.acs.SL.scat1g <- df.acs.SL
df.acs.SL.scat1g$Category <- "Other"
df.acs.SL.scat1g$Order <- 1
df.acs.SL.scat1g[(df.acs.SL.scat1g$Gene %in% PPP),"Category"] <- "PPP"
df.acs.SL.scat1g[(df.acs.SL.scat1g$Gene %in% PPP),"Order"] <- 2
df.acs.SL.scat1g[(df.acs.SL.scat1g$Gene %in% PtPS),"Category"] <- "PlastidEncodedPS"
df.acs.SL.scat1g[(df.acs.SL.scat1g$Gene %in% PtPS),"Order"] <- 3
df.acs.SL.scat1g[(df.acs.SL.scat1g$Gene %in% PHANG),"Category"] <- "PHANGs"
df.acs.SL.scat1g[(df.acs.SL.scat1g$Gene %in% PHANG),"Order"] <- 4
df.acs.SL.scat1g[(df.acs.SL.scat1g$Gene %in% c(PtEncodeRibo,Pt.other)),"Category"] <- "PlastidEncodedRibo+Other"
df.acs.SL.scat1g[(df.acs.SL.scat1g$Gene %in% c(PtEncodeRibo,Pt.other)),"Order"] <- 5
df.acs.SL.scat1g[(df.acs.SL.scat1g$Gene %in% PtRibo),"Category"] <- "NuclearEncodedPlastidRibo"
df.acs.SL.scat1g[(df.acs.SL.scat1g$Gene %in% PtRibo),"Order"] <- 6
#df.acs.SL.scat1g[(df.acs.SL.scat1g$Gene %in% eukRibo),"Category"] <- "CytosolicRibo"
#df.acs.SL.scat1g[(df.acs.SL.scat1g$Gene %in% eukRibo),"Order"] <- 7
df.acs.SL.scat1g[(df.acs.SL.scat1g$Gene %in% Pt.rpo),"Category"] <- "Pt.rpo"
df.acs.SL.scat1g[(df.acs.SL.scat1g$Gene %in% Pt.rpo),"Order"] <- 8

#Re-order
df.acs.SL.scat1g <- df.acs.SL.scat1g[(order(df.acs.SL.scat1g$Order)),]
```
##Set limits for plot
```{r}
df.acs.SL.scat1g.lims <- df.acs.SL.scat1g

df.acs.SL.scat1g.lims$l2fc.acs[df.acs.SL.scat1g.lims$l2fc.acs > 1.0] = 1.0
df.acs.SL.scat1g.lims$l2fc.acs[df.acs.SL.scat1g.lims$l2fc.acs < -1.0] = -1.0

df.acs.SL.scat1g.lims$l2fc.BM[df.acs.SL.scat1g.lims$l2fc.BM > 1.0] = 1.0
df.acs.SL.scat1g.lims$l2fc.BM[df.acs.SL.scat1g.lims$l2fc.BM < -1.0] = -1.0
```
##Make scatterplot
```{r}
library(ggplot2)
```
```{r}
fig4g <- ggplot(df.acs.SL.scat1g.lims, mapping=aes(x=l2fc.acs, y=l2fc.BM, color=Category))+
  geom_point(size=.8)+
  scale_color_manual(values=c("Other"="grey70","PPP"="#efb806", "PlastidEncodedPS"="red3", "PHANGs"="#ef06db", "PlastidEncodedRibo+Other"="blue4", "NuclearEncodedPlastidRibo"="dodgerblue1", "Pt.rpo"="#653f01"), labels=c("Other"="Other", "PPP"="Pentose Phosphate Shunt", "PlastidEncodedPS"="Plastid-encoded photosynthesis-associated", "PHANGs"="Nuclear-encoded photosynthesis-associated", "PlastidEncodedRibo+Other"="Plastid-encoded ribosome +other", "NuclearEncodedPlastidRibo"="Nuclear-encoded plastid ribosome", "Pt.rpo"="Plastid-encoded RNA polymerase"), name="Protein Category")+
  theme_bw()+
  xlab("log2(acs/B73)")+
  ylab("log2(BxM/MP)")+
  geom_abline(slope = 1, color="grey50", size=.5)+
  geom_vline(color="grey50", size=.5, xintercept = 0)+
  geom_hline(color="grey50", size=.5, yintercept = 0)+
  coord_fixed(expand = F)+
  xlim(-1,1)+
  ylim(-1,1)+
  theme(legend.position = "bottom", legend.text=element_text(size=7), legend.title = element_text(size=9))+
  theme(legend.position = "none")+
  theme(axis.title = element_text(size=20))

fig4g
```
##Add density curves
```{r}
library(cowplot)
```
```{r}
#add density curves
#add marginal density plots and resave
    densityplotX<-axis_canvas(fig4g, axis="x")+
      geom_density(aes(df.acs.SL.scat1g.lims[,"l2fc.acs"], color=df.acs.SL.scat1g.lims[,"Category"]), df.acs.SL.scat1g.lims, size=0.9)+
      scale_color_manual(values=c("Other"="grey70","PPP"="#efb806", "PlastidEncodedPS"="red3", "PHANGs"="#ef06db", "PlastidEncodedRibo+Other"="blue4", "NuclearEncodedPlastidRibo"="dodgerblue1", "Pt.rpo"="#653f01"))+
      geom_vline(xintercept=0, linetype="dashed", color = "gray50")+
      theme_void()+
      theme(legend.title = element_blank())+
      ylim(0,15)
    #set up marginal density plot Y axis
    densityplotY<-axis_canvas(fig4g, axis="y", coord_flip = TRUE)+
      geom_density(aes(df.acs.SL.scat1g.lims[,"l2fc.BM"], color=df.acs.SL.scat1g.lims[,"Category"]), df.acs.SL.scat1g.lims, size=0.9)+
      scale_color_manual(values=c("Other"="grey70","PPP"="#efb806", "PlastidEncodedPS"="red3", "PHANGs"="#ef06db", "PlastidEncodedRibo+Other"="blue4", "NuclearEncodedPlastidRibo"="dodgerblue1", "Pt.rpo"="#653f01"))+
      geom_vline(xintercept=0, linetype="dashed", color = "gray50")+
      theme_void()+
      theme(legend.title = element_blank())+
      coord_flip()+
      ylim(0,15)
    #put two plots together
    q <- insert_xaxis_grob(fig4g, densityplotX, position = "top")
    q <- insert_yaxis_grob(q, densityplotY, position = "right")
    #save
    ggsave(plot=q, filename = "Fig4g.tiff",
           device="tiff", dpi=600, width=9, height=8, compression = "lzw")
```
#Scatterplot2
##Make GOI description column
```{r}
df.acs.SL.scat2<- df.acs.SL
df.acs.SL.scat2$Category <- "Other"
df.acs.SL.scat2$Order <- 1
df.acs.SL.scat2[(df.acs.SL.scat2$Gene %in% EtBio),"Category"] <- "EtBio"
df.acs.SL.scat2[(df.acs.SL.scat2$Gene %in% EtBio),"Order"] <- 2
df.acs.SL.scat2[(df.acs.SL.scat2$Gene %in% ResponseToEt),"Category"] <- "ResponseToEt"
df.acs.SL.scat2[(df.acs.SL.scat2$Gene %in% ResponseToEt),"Order"] <- 3

#Re-order
df.acs.SL.scat2 <- df.acs.SL.scat2[(order(df.acs.SL.scat2$Order)),]
```
##Set limits for plot
```{r}
df.acs.SL.scat2.lims <- df.acs.SL.scat2

df.acs.SL.scat2.lims$l2fc.acs[df.acs.SL.scat2.lims$l2fc.acs > 1.0] = 1.0
df.acs.SL.scat2.lims$l2fc.acs[df.acs.SL.scat2.lims$l2fc.acs < -1.0] = -1.0

df.acs.SL.scat2.lims$l2fc.BM[df.acs.SL.scat2.lims$l2fc.BM > 1.0] = 1.0
df.acs.SL.scat2.lims$l2fc.BM[df.acs.SL.scat2.lims$l2fc.BM < -1.0] = -1.0
```
##Make scatterplot
```{r}
library(ggplot2)
```
```{r}
fig4.2 <- ggplot(df.acs.SL.scat2.lims, mapping=aes(x=l2fc.acs, y=l2fc.BM, color=Category))+
  geom_point(size=.8)+
  scale_color_manual(values=c("Other"="grey80", "EtBio"="blue", "ResponseToEt"="red"), name="Protein Category")+
  theme_bw()+
  xlab("log2(acs/B73)")+
  ylab("log2(BxM/MP)")+
  geom_abline(slope = 1, color="grey50", size=.5)+
  geom_vline(color="grey50", size=.5, xintercept = 0)+
  geom_hline(color="grey50", size=.5, yintercept = 0)+
  coord_fixed(expand = F)+
  xlim(-1,1)+
  ylim(-1,1)+
  theme(legend.position = "bottom", legend.text=element_text(size=7), legend.title = element_text(size=9))+
  theme(legend.position = "none")+
  theme(axis.title = element_text(size=20))

fig4.2
```
##Add density curves
```{r}
library(cowplot)
```
```{r}
#add density curves
#add marginal density plots and resave
    densityplotX<-axis_canvas(fig4.2, axis="x")+
      geom_density(aes(df.acs.SL.scat2.lims[,"l2fc.acs"], color=df.acs.SL.scat2.lims[,"Category"]), df.acs.SL.scat2.lims, size=0.9)+
      scale_color_manual(values=c("Other"="grey80", "EtBio"="blue", "ResponseToEt"="red"), name="Protein Category")+
      geom_vline(xintercept=0, linetype="dashed", color = "gray50")+
      theme_void()+
      theme(legend.title = element_blank())+
  ylim(0,15)
    #set up marginal density plot Y axis
    densityplotY<-axis_canvas(fig4.2, axis="y", coord_flip = TRUE)+
      geom_density(aes(df.acs.SL.scat2.lims[,"l2fc.BM"], color=df.acs.SL.scat2.lims[,"Category"]), df.acs.SL.scat2.lims, size=0.9)+
      scale_color_manual(values=c("Other"="grey80", "EtBio"="blue", "ResponseToEt"="red"), name="Protein Category")+
      geom_vline(xintercept=0, linetype="dashed", color = "gray50")+
      theme_void()+
      theme(legend.title = element_blank())+
      coord_flip()+
      ylim(0,15)
    #put two plots together
    q <- insert_xaxis_grob(fig4.2, densityplotX, position = "top")
    q <- insert_yaxis_grob(q, densityplotY, position = "right")
    #save
    ggsave(plot=q, filename = "Fig4_2.tiff",
           device="tiff", dpi=600, width=9, height=8, compression = "lzw")
```




