---
title: "Final version of Leaf Heterosis proteomics for Paper"
output: github_document
---
#Setup
```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
set.seed(586)
```
```{r libraries}
library(matrixStats)
library(topGO)
library(ggplot2)
library(AnnotationDbi)
library(cowplot)
library(dplyr)
```

#Import data
```{r read csv}
df <- read.csv("data/Leaf paper-total proteome final 121319.csv", stringsAsFactors = F)
#make new column for gene accession without transcript id
df$Gene <- substr(df$Accession, start = 1, stop = 14)
df.withPolymorphic <- df
#Remove polymorphic proteins
df <- df[-(grep("only", df$Accession)),]
```

```{r T-test function}
ttestFun <- function(df, col1, col2){
  result <- NULL
  for(i in 1:(nrow(df))){
  v1 = (as.numeric(df[i, col1]))
  v2 = (as.numeric(df[i, col2]))
  t_test <- t.test(v1, v2, var.equal = TRUE)
  pval <- t_test$p.value
  result[i] <- pval
  }
  result
}
```

##Gamer GO

```{r read in gamer}
# Read in Maize GAMER GO dataset, which lists maize V4 accessions with all associated GO terms (Wimalanathan et al. 2018) downloaded from https://datacommons.cyverse.org/browse/iplant/home/shared/commons_repo/curated/Carolyn_Lawrence-Dill_maize-GAMER_maize.B73_RefGen_v4_Zm00001d.2_Oct_2017.r1/e.agg_data
GAMER <- read.delim("maize.B73.AGPv4.aggregate.gaf", sep = "\t", header = FALSE)
colnames(GAMER) <- c("db", "db_object_id", "db_object_symbol", "qualifier", "term_accession", "db_reference", "evidence_code", "with", "aspect", "db_object_name", "db_object_synonym", "db_object_type", "taxon", "date", "assigned_by", "annotation_extension", "gene_product_form_id")
gamer <- GAMER[-(1:2),]
gamer$Term <- Term(as.character(gamer$term_accession))
```


#Define topGO functions
```{r GOanalysisGOdata_Function}
## Build GOdata object
# EnrichedDF = AMP/BMP output or other subset dataframe of proteins of interest
# GamerCut = the dataframe created above
# ont = "BP", "MF", or "CC"
GOanalysisGodata <- function(EnrichedDF, GamerCut, ont, nodesize = 1){
  gene2GO_TissueCut <- split((as.character(GamerCut$term_accession)),GamerCut$db_object_id)
  gene2GOb_TissueCut <- gene2GO_TissueCut[-2]
  str(head(gene2GOb_TissueCut))
  geneNames_tissuecut <- names(gene2GOb_TissueCut)
  myInterestingGenes <- EnrichedDF$Gene
  geneList <- factor(as.integer(geneNames_tissuecut %in% myInterestingGenes))
  names(geneList) <- geneNames_tissuecut
  GOdata <- new("topGOdata", ontology = ont, allGenes = geneList, annot = annFUN.gene2GO, nodeSize=nodesize, gene2GO = gene2GOb_TissueCut)
  GOdata
}
```

```{r GOanalysisMap_Function}
## GO analysis function to generate map of GO terms
# Godata = Godata object (output of GoanalysisGodata function)
# pdfname = and desired output pdf name in quotes
GOanalysisMap <- function(GOdata, pdfname, signodes=5){
  resultFis <- runTest(GOdata, algorithm = "classic", statistic = "fisher")
  printGraph(GOdata, resultFis, firstSigNodes = signodes, fn.prefix = pdfname, useInfo = "all", pdfSW = TRUE)
}
```

```{r GOanalysisDF_Function}
## GO analysis function, same as above, except to generate dataframe summary of top 20 of GO terms instead of map
# Godata = Godata object (output of GoanalysisGodata function)
# pdfname = and desired output pdf name in quotes
GOanalysisDF <- function(GOdata){
resultFis <- runTest(GOdata, algorithm = "classic", statistic = "fisher")
test.statF <- new("classicCount", testStatistic = GOFisherTest, name = "Fisher test")
resultFisher <- getSigGroups(GOdata, test.statF)
test.statKS <- new("classicScore", testStatistic = GOKSTest, name = "KS tests")
resultKS <- getSigGroups(GOdata, test.statKS)
test.statRW <- new("weightCount", testStatistic = GOFisherTest, name = "Fisher test", sigRatio = "ratio")
resultWeight <- getSigGroups(GOdata, test.statRW)
DF <- GenTable(GOdata, classic = resultFis, KS = resultKS, weight = resultWeight, orderBy = "classic", ranksOf = "classic", topNodes = 20)
DF
}
```

#Define GOIs
##godata
```{r}
godataBP <- GOanalysisGodata(df.SL, gamer, "BP")
godataCC <- GOanalysisGodata(df.SL, gamer, "CC")
```
```{r}
SelectGOgenes <- function(GOdata, GOterm){
 df <- data.frame("Gene"=(topGO::genesInTerm(GOdata, GOterm)))
colnames(df)="Gene"
as.character(df$Gene)
}
```

##Photosynthesis

```{r}
GOphoto<-read.csv("PhotosynthesisGOterms.csv", stringsAsFactors = F)
```
```{r}
GOphoto.all <- GOphoto$go
GOphoto.lightrxn <- GOphoto$go[GOphoto$subpath=="photosynthesis, light reaction"]
GOphoto.lightrxn.lightharvesting <- GOphoto$go[grep("harvesting",GOphoto$term)]
GOphoto.lightrxn.electrontransport <- GOphoto$go[grep("transport",GOphoto$term)]
GOphoto.lightrxn.NADPreduction <- GOphoto$go[grep("NADP",GOphoto$term)]
GOphoto.lightrxn.phosphorylation <- GOphoto$go[grep("phosphorylation",GOphoto$term)]
GOphoto.lightrxn.PSIassembly <- GOphoto$go[grep("photosystem I assembly",GOphoto$term)]
GOphoto.lightrxn.PSIIassembly <- GOphoto$go[grep("photosystem II assembly",GOphoto$term)]
GOphoto.lightrxn.OxEvolassembly <- GOphoto$go[grep("oxygen evolving",GOphoto$term)]

GOphoto.darkrxn <- GOphoto$go[GOphoto$subpath=="photosynthesis, dark reaction"]
GOphoto.darkrxn.rPP <- GOphoto$go[grep("pentose",GOphoto$term)]
```
```{r}
photosynthesis <- as.vector(gamer$db_object_id[gamer$term_accession%in%GOphoto.all])
lightrxn <- as.vector(gamer$db_object_id[gamer$term_accession%in%GOphoto.lightrxn])
lightrxn.lightharvesting <- as.vector(gamer$db_object_id[gamer$term_accession%in%GOphoto.lightrxn.lightharvesting])
lightrxn.electrontransport <- as.vector(gamer$db_object_id[gamer$term_accession%in%GOphoto.lightrxn.electrontransport])
lightrxn.NADPreduction <- as.vector(gamer$db_object_id[gamer$term_accession%in%GOphoto.lightrxn.NADPreduction])
lightrxn.phosphorylation <- as.vector(gamer$db_object_id[gamer$term_accession%in%GOphoto.lightrxn.phosphorylation])
lightrxn.PSIassembly <- as.vector(gamer$db_object_id[gamer$term_accession%in%GOphoto.lightrxn.PSIassembly])
lightrxn.PSIIassembly <- as.vector(gamer$db_object_id[gamer$term_accession%in%GOphoto.lightrxn.PSIIassembly])
lightrxn.OxEvolassembly <- as.vector(gamer$db_object_id[gamer$term_accession%in%GOphoto.lightrxn.OxEvolassembly])
darkrxn.rPP <- as.vector(gamer$db_object_id[gamer$term_accession%in%GOphoto.darkrxn.rPP])


#Add CornCyc annotated PhANGs
#accessed from https://corncyc-b73-v4.maizegdb.org/ on 1/4/2020
cornCyc<-as.data.frame(readr::read_csv(file = "cornCycGeneLists01042020.csv",
                         col_names = TRUE
                         ))
lightrxn.electrontransport<- unique(c(lightrxn.electrontransport, (cornCyc$`Gene Accession`[(cornCyc$`Enzymatic activity`=="plastoquinol-plastocyanin reductase")])))

lightrxn.NADPreduction<- unique(c(lightrxn.NADPreduction, cornCyc$`Gene Accession`[(cornCyc$`Enzymatic activity`=="ferredoxin NADP reductase")]))

lightrxn.lightharvesting<- unique(c(lightrxn.lightharvesting, cornCyc$`Gene Accession`[(cornCyc$`Enzymatic activity`=="photosystem I")]))
lightrxn.lightharvesting<- unique(c(lightrxn.lightharvesting, cornCyc$`Gene Accession`[(cornCyc$`Enzymatic activity`=="photosystem II")]))

darkrxn.rPP <- unique(c(darkrxn.rPP, cornCyc$`Gene Accession`[cornCyc$Pathway=="Calvin_Cycle"]))

#define thylakoid proteins as everything in that cellular compartment, minus our proteins of interest
thylakoid <- SelectGOgenes(godataCC, "GO:0009579")
thylakoid <- thylakoid[!thylakoid%in%photosynthesis]
thylakoid <- thylakoid[!thylakoid%in%cornCyc$`Gene Accession`]

OxEvolComp <- SelectGOgenes(godataCC, "GO:0009654")
```


##Plastid-encoded
```{r}
#Plastid-encoded proteins
Pt<-df.SL[grep("v4:Pt", df.SL$Protein),]
#define list of plastid-encoded photosynthesis-associated genes
PtPS <- Pt[-grep("rps|rpo|rpl|orf", Pt$Protein),]
#Plastid encoded ribosomal proteins (EnsemblPlants Annotation)
#get subset of Pt encoded genes annotated as ribosomal proteins
Pt.ribo <- Pt[grep("rps|rpl", Pt$Protein),]
#Plastid encoded RNA polymerases (EnsemblPlants Annotation)
Pt.rpo <- Pt[grep("rpo", Pt$Protein),]
#re-define to gene vector
Pt <- Pt$Gene
PtPS <- PtPS$Gene
Pt.ribo <- Pt.ribo$Gene
Pt.rpo <- Pt.rpo$Gene
#Define "other" as everything left
Pt.other <- Pt[!(Pt%in%c(Pt.ribo, PtPS, Pt.rpo))]
```

##Nuclear-encoded plastid ribosome
```{r}
#Nuclear encoded plastid ribosomal proteins (GO GAMER and EnsemblPlants Annotation)
Nuc.PtRibo <- SelectGOgenes(godataCC, "GO:0009547")
#Add proteins annotated as "ribosom" and "plastid/chloroplast" to the nuclear-encoded plastid-localized list
ribosom <- df.SL[(grep("ribosom", df.SL$Protein)),]
ribosom.pt <- ribosom[(grep("plastid|Plastid|Chloropl|chloropl", ribosom$Protein)),]
#remove plastid-encoded 
ribosom.pt <- ribosom.pt[-grep("NP_04",ribosom.pt$Protein),]
Nuc.PtRibo <- unique(c(Nuc.PtRibo,ribosom.pt$Gene))
#Check that there are no missing prokaryotic ribosomal proteins with plastid localization
ProkRibo <- df.SL[(grep("30S|60S", df.SL$Protein)),]
ProkRiboNotPt <- ProkRibo[!(ProkRibo$Gene%in%Nuc.PtRibo),]
GOplastid <- SelectGOgenes(godataCC, "GO:0009536")
ProkRiboNotPt.plastid <- ProkRiboNotPt[(ProkRiboNotPt$Gene%in%GOplastid),]
Nuc.PtRibo <- unique(c(Nuc.PtRibo, ProkRiboNotPt.plastid$Gene))
```

##Eukaryotic ribosome
```{r}
#Eukaryotic ribosomal proteins (EnsemblPlants Annotation)
eukRibo<-df.SL[(grep("40S|60S", df.SL$Protein)),]
#remove the mitochondrial ones
eukRibo<- eukRibo$Gene[-(grep("mito|Mito", eukRibo$Protein))]
```

##Etyhlene
```{r}
MAT <- cornCyc$`Gene Accession`[(cornCyc$`Enzymatic activity`=="methionine adenosyltransferase")]
ACS <- cornCyc$`Gene Accession`[(cornCyc$`Enzymatic activity`=="1-aminocyclopropane-1-carboxylate synthase")]
ACO <- cornCyc$`Gene Accession`[(cornCyc$`Enzymatic activity`=="1-aminocyclopropane-1-carboxylate oxidase")]
EtBio <- c(MAT,ACS,ACO)
```
```{r}
ResponseToEt <- SelectGOgenes(godataBP, "GO:0009723")
```
```{r}
gamer.responseToEt <- gamer$db_object_id[gamer$term_accession=="GO:0009723"]
gamer.DetectionOfEt <- gamer$db_object_id[gamer$term_accession=="GO:0009727"]
gamer.CellResponseToEt <- gamer$db_object_id[gamer$term_accession=="GO:0071369"]
gamer.EtSignaling <- gamer$db_object_id[gamer$term_accession=="GO:0009873"]
```
#acs
##Format df
```{r format df}
#Subset
df.acs <- df[,c(1:18,89)]
#Remove missing genes
df.acs <- df.acs[!(df.acs$B73.acs.B73.r1==0),]
#Calculate averages of bio reps
df.acs$B73avg <- rowMeans(df.acs[,9:13])
df.acs$ACSavg <- rowMeans(df.acs[,14:18])
#calculate ratio
df.acs$ACS.B73 <- df.acs$ACSavg/df.acs$B73avg
#calculate t-test
df.acs$Sig <- ttestFun(df.acs, 9:13, 14:18)
#calculate log2(acs/b73)
df.acs$l2fc <- log2(df.acs$ACS.B73)
#Calculate standard deviation
df.acs$SD <- rowSds(as.matrix(df.acs[,14:18]))
#subset to 10% significantly above or below mid-parent
df.acs.amp <- df.acs[(df.acs$Sig<0.05)&(df.acs$ACS.B73>1.1),]
df.acs.bmp <- df.acs[(df.acs$Sig<0.05)&(df.acs$ACS.B73<0.9),]
```
##GO term enrichment
```{r trim gamer}
#Trim gamer to only those genes detected in our data, to use as background for enrichment
gamerCutACS <- gamer[(gamer$db_object_id %in% df.acs$Gene),]
```
```{r run GoanalysisGodata}
df.acs.amp_BP_GOdata <- GOanalysisGodata(df.acs.amp, gamerCutACS, "BP", nodesize = 5)
df.acs.amp_MF_GOdata <- GOanalysisGodata(df.acs.amp, gamerCutACS, "MF", nodesize = 5)
df.acs.amp_CC_GOdata <- GOanalysisGodata(df.acs.amp, gamerCutACS, "CC", nodesize = 5)

df.acs.bmp_BP_GOdata <- GOanalysisGodata(df.acs.bmp, gamerCutACS, "BP", nodesize = 5)
df.acs.bmp_MF_GOdata <- GOanalysisGodata(df.acs.bmp, gamerCutACS, "MF", nodesize = 5)
df.acs.bmp_CC_GOdata <- GOanalysisGodata(df.acs.bmp, gamerCutACS, "CC", nodesize = 5)
```
```{r run GoanalysisMap}
dir.create("GOmaps")
GOanalysisMap(df.acs.amp_BP_GOdata, "GOmaps/df.acs.amp_BP_GOdata")
GOanalysisMap(df.acs.amp_MF_GOdata, "GOmaps/df.acs.amp_MF_GOdata")
GOanalysisMap(df.acs.amp_CC_GOdata, "GOmaps/df.acs.amp_CC_GOdata")

GOanalysisMap(df.acs.bmp_BP_GOdata, "GOmaps/df.acs.bmp_BP_GOdata")
GOanalysisMap(df.acs.bmp_MF_GOdata, "GOmaps/df.acs.bmp_MF_GOdata")
GOanalysisMap(df.acs.bmp_CC_GOdata, "GOmaps/df.acs.bmp_CC_GOdata")
```
```{r run GoanalysisDF}
df.acs.amp_BP_GOdata.DF <- GOanalysisDF(df.acs.amp_BP_GOdata)
df.acs.amp_MF_GOdata.DF <- GOanalysisDF(df.acs.amp_MF_GOdata)
df.acs.amp_CC_GOdata.DF <- GOanalysisDF(df.acs.amp_CC_GOdata)

df.acs.bmp_BP_GOdata.DF <- GOanalysisDF(df.acs.bmp_BP_GOdata)
df.acs.bmp_MF_GOdata.DF <- GOanalysisDF(df.acs.bmp_MF_GOdata)
df.acs.bmp_CC_GOdata.DF <- GOanalysisDF(df.acs.bmp_CC_GOdata)
```

#Seedling Leaf - B73xMo17
##Format df

###AMP/BMP
```{r format df}
#Subset
df.SL <- df[,c(1:8,19:28,89)]
#Remove missing genes
df.SL <- df.SL[!(df.SL$SeedlingLeaf.B73.r1==0),]
#Calculate averages of bio reps
df.SL$B73avg <- rowMeans(df.SL[,10:12])
df.SL$Mo17avg <- rowMeans(df.SL[,13:15])
df.SL$HYBavg <- rowMeans(df.SL[,16:18])
df.SL$MPavg <- rowMeans(df.SL[,c(10:15)])
df.SL$MP.1 <- rowMeans(df.SL[,c(10,13)])
df.SL$MP.2 <- rowMeans(df.SL[,c(11,14)])
df.SL$MP.3 <- rowMeans(df.SL[,c(12,15)])
#calculate ratio
df.SL$HYB.MP <- df.SL$HYBavg/df.SL$MPavg
#calculate t-test
df.SL$Sig <- ttestFun(df.SL, 10:12, 24:26)
#calculate log2(hyb/MP)
df.SL$l2fc <- log2(df.SL$HYB.MP)
#subset to 10% significantly above or below mid-parent
df.SL.amp <- df.SL[(df.SL$Sig<0.05)&(df.SL$HYB.MP>1.1),]
df.SL.bmp <- df.SL[(df.SL$Sig<0.05)&(df.SL$HYB.MP<0.9),]
#Calculate log2(B73/Mo17)
df.SL$B73.Mo17 <- df.SL$B73avg/df.SL$Mo17avg
df.SL$l2fc.B73.Mo17 <-log2(df.SL$B73.Mo17)
```
###MP + HP
```{r Subset to mid-parent proteins where parents are different}
#Copy data
df.SL.2 <- df.SL
#subset to proteins where parents are 10% different from each other
df.SL.2.b73hp <- df.SL.2[(df.SL.2$B73avg>1.1*(df.SL.2$Mo17avg)),]
df.SL.2.mo17hp <- df.SL.2[(df.SL.2$Mo17avg>1.1*(df.SL.2$B73avg)),]
#subset to proteins where hybrid is mid-parent
df.SL.2.b73hp.mp <- df.SL.2.b73hp[!(df.SL.2.b73hp$Accession%in%c(df.SL.amp$Accession, df.SL.bmp$Accession)),]
df.SL.2.mo17hp.mp <- df.SL.2.mo17hp[!(df.SL.2.mo17hp$Accession%in%c(df.SL.amp$Accession, df.SL.bmp$Accession)),]
#add column taking the ratio of the parents
df.SL.2.b73hp.mp$b73_mo17 <- df.SL.2.b73hp.mp$B73avg/df.SL.2.b73hp.mp$Mo17avg
df.SL.2.mo17hp.mp$mo17_b73 <- df.SL.2.mo17hp.mp$Mo17avg/df.SL.2.mo17hp.mp$B73avg
#subset to proteins with "translation" in the annotation
df.SL.2.b73hp.mp.translation <- df.SL.2.b73hp.mp[(grep("translation", df.SL.2.b73hp.mp$Protein)),]
df.SL.2.mo17hp.mp.translation <- df.SL.2.mo17hp.mp[(grep("translation", df.SL.2.mo17hp.mp$Protein)),]
#write csv
write.csv(df.SL.2.b73hp.mp, "df.SL.2.b73hp.mp.csv")
write.csv(df.SL.2.mo17hp.mp, "df.SL.2.mo17hp.mp.csv")
write.csv(df.SL.2.b73hp.mp.translation, "df.SL.2.b73hp.mp.translation.csv")
write.csv(df.SL.2.mo17hp.mp.translation, "df.SL.2.mo17hp.mp.translation.csv")
```
```{r Pie Chart}
df.SL.2<-data.frame("HP"=c("B73", "Mo17"), "MP"=c(nrow(df.SL.2.b73hp), nrow(df.SL.2.mo17hp)))

df.SL.2$HP<- c((paste0(df.SL.2[1,1], " - ", as.character(df.SL.2[1,2]))), (paste0(df.SL.2[2,1], " - ", as.character(df.SL.2[2,2]))))

df.SL.2 <- df.SL.2 %>%
  arrange(desc(HP)) %>%
  mutate(lab.ypos = cumsum(MP) - 0.5*MP)

df.SL.2.bar<- ggplot(df.SL.2, aes(x="", y=MP, fill=HP))+
geom_bar(width = 1, stat = "identity")
df.SL.2.pie <- df.SL.2.bar + coord_polar("y", start=0)+
  theme_void()+
  theme(axis.text=element_blank(), legend.position = "none")+
  geom_text(aes(y = lab.ypos, label = HP), color = "white", size=5)+
  ggtitle("High-parent for MP proteins")+
  theme(plot.title = element_text(hjust = .5))
df.SL.2.pie
ggsave("df.SL.2.pie.png", width = 4, height = 4)
```
###AMP/BMP + HP
```{r Subset to mid-parent proteins where parents are different}
#Copy data
df.SL.3 <- df.SL
#subset to proteins where parents are 10% different from each other
df.SL.3.b73hp <- df.SL.3[(df.SL.3$B73avg>1.1*(df.SL.3$Mo17avg)),]
df.SL.3.mo17hp <- df.SL.3[(df.SL.3$Mo17avg>1.1*(df.SL.3$B73avg)),]
#subset to proteins where hybrid is amp or bmp
df.SL.3.b73hp.amp <- df.SL.3.b73hp[(df.SL.3.b73hp$Accession%in%df.SL.amp$Accession),]
df.SL.3.b73hp.bmp <- df.SL.3.b73hp[(df.SL.3.b73hp$Accession%in%df.SL.bmp$Accession),]
df.SL.3.mo17hp.amp <- df.SL.3.mo17hp[(df.SL.3.mo17hp$Accession%in%df.SL.amp$Accession),]
df.SL.3.mo17hp.bmp <- df.SL.3.mo17hp[(df.SL.3.mo17hp$Accession%in%df.SL.bmp$Accession),]
```
```{r}
write.csv(df.SL.3.b73hp.amp, "df.SL.3.b73hp.amp.csv")
write.csv(df.SL.3.mo17hp.amp, "df.SL.3.mo17hp.amp.csv")
```

```{r pie chart - amp}
df.SL.3.amp<-data.frame("HP"=c("B73", "Mo17"), "AMP"=c(nrow(df.SL.3.b73hp.amp), nrow(df.SL.3.mo17hp.amp)))

df.SL.3.amp$HP<- c((paste0(df.SL.3.amp[1,1], " - ", as.character(df.SL.3.amp[1,2]))), (paste0(df.SL.3.amp[2,1], " - ", as.character(df.SL.3.amp[2,2]))))

df.SL.3.amp <- df.SL.3.amp %>%
  arrange(desc(HP)) %>%
  mutate(lab.ypos = cumsum(AMP) - 0.5*AMP)

df.SL.3.amp.bar<- ggplot(df.SL.3.amp, aes(x="", y=AMP, fill=HP))+
geom_bar(width = 1, stat = "identity")
df.SL.3.amp.pie <- df.SL.3.amp.bar + coord_polar("y", start=0)+
  theme_void()+
  theme(axis.text=element_blank(), legend.position = "none")+
  geom_text(aes(y = lab.ypos, label = HP), color = "white", size=5)+
  ggtitle("High-parent for 10% AMP proteins")+
  theme(plot.title = element_text(hjust = .5))
ggsave("df.SL.3.amp.pie.png", width = 4, height = 4)
```
```{r pie chart - bmp}
df.SL.3.bmp<-data.frame("LP"=c("Mo17", "B73"), "bmp"=c(nrow(df.SL.3.b73hp.bmp), nrow(df.SL.3.mo17hp.bmp)))

df.SL.3.bmp$HP<- c((paste0(df.SL.3.bmp[1,1], " - ", as.character(df.SL.3.bmp[1,2]))), (paste0(df.SL.3.bmp[2,1], " - ", as.character(df.SL.3.bmp[2,2]))))

df.SL.3.bmp <- df.SL.3.bmp %>%
  arrange(desc(HP)) %>%
  mutate(lab.ypos = cumsum(bmp) - 0.5*bmp)

df.SL.3.bmp.bar<- ggplot(df.SL.3.bmp, aes(x="", y=bmp, fill=HP))+
geom_bar(width = 1, stat = "identity")
df.SL.3.bmp.pie <- df.SL.3.bmp.bar + coord_polar("y", start=0)+
  theme_void()+
  theme(axis.text=element_blank(), legend.position = "none")+
  geom_text(aes(y = lab.ypos, label = HP), color = "white", size=5)+
  ggtitle("Low-parent for 10% BMP proteins")+
  theme(plot.title = element_text(hjust = .5))
ggsave("df.SL.3.bmp.pie.png", width = 4, height = 4)
```

##GO term enrichment
```{r trim gamer}
#Trim gamer to only those genes detected in our data, to use as background for enrichment
gamerCutSL <- gamer[(gamer$db_object_id %in% df.SL$Gene),]
```
###AMP/BMP
```{r run GoanalysisGodata for amp/bmp}
df.SL.amp_BP_GOdata <- GOanalysisGodata(df.SL.amp, gamerCutSL, "BP", nodesize = 5)
df.SL.amp_MF_GOdata <- GOanalysisGodata(df.SL.amp, gamerCutSL, "MF", nodesize = 5)
df.SL.amp_CC_GOdata <- GOanalysisGodata(df.SL.amp, gamerCutSL, "CC", nodesize = 5)

df.SL.bmp_BP_GOdata <- GOanalysisGodata(df.SL.bmp, gamerCutSL, "BP", nodesize = 5)
df.SL.bmp_MF_GOdata <- GOanalysisGodata(df.SL.bmp, gamerCutSL, "MF", nodesize = 5)
df.SL.bmp_CC_GOdata <- GOanalysisGodata(df.SL.bmp, gamerCutSL, "CC", nodesize = 5)
```
```{r run GoanalysisMap for amp/bmp}
dir.create("GOmaps")
GOanalysisMap(df.SL.amp_BP_GOdata, "GOmaps/df.SL.amp_BP_GOdata")
GOanalysisMap(df.SL.amp_MF_GOdata, "GOmaps/df.SL.amp_MF_GOdata")
GOanalysisMap(df.SL.amp_CC_GOdata, "GOmaps/df.SL.amp_CC_GOdata")

GOanalysisMap(df.SL.bmp_BP_GOdata, "GOmaps/df.SL.bmp_BP_GOdata")
GOanalysisMap(df.SL.bmp_MF_GOdata, "GOmaps/df.SL.bmp_MF_GOdata")
GOanalysisMap(df.SL.bmp_CC_GOdata, "GOmaps/df.SL.bmp_CC_GOdata")
```
```{r run GoanalysisDF for amp/bmp}
df.SL.amp_BP_GOdata.DF <- GOanalysisDF(df.SL.amp_BP_GOdata)
df.SL.amp_MF_GOdata.DF <- GOanalysisDF(df.SL.amp_MF_GOdata)
df.SL.amp_CC_GOdata.DF <- GOanalysisDF(df.SL.amp_CC_GOdata)

df.SL.bmp_BP_GOdata.DF <- GOanalysisDF(df.SL.bmp_BP_GOdata)
df.SL.bmp_MF_GOdata.DF <- GOanalysisDF(df.SL.bmp_MF_GOdata)
df.SL.bmp_CC_GOdata.DF <- GOanalysisDF(df.SL.bmp_CC_GOdata)
```
```{r view enrichment}
df.SL.bmp_CC_GOdata.DF
```

###MP + HP
```{r run GoanalysisGodata for hp+mp}
df.SL.2.b73hp.mp_BP_GOdata <- GOanalysisGodata(df.SL.2.b73hp.mp, gamerCutSL, "BP")
df.SL.2.b73hp.mp_CC_GOdata <- GOanalysisGodata(df.SL.2.b73hp.mp, gamerCutSL, "CC")
df.SL.2.b73hp.mp_MF_GOdata <- GOanalysisGodata(df.SL.2.b73hp.mp, gamerCutSL, "MF")

df.SL.2.mo17hp.mp_BP_GOdata <- GOanalysisGodata(df.SL.2.mo17hp.mp, gamerCutSL, "BP")
df.SL.2.mo17hp.mp_CC_GOdata <- GOanalysisGodata(df.SL.2.mo17hp.mp, gamerCutSL, "CC")
df.SL.2.mo17hp.mp_MF_GOdata <- GOanalysisGodata(df.SL.2.mo17hp.mp, gamerCutSL, "MF")
```
```{r run GoanalysisDF for hp+mp}
df.SL.2.b73hp.mp_BP_GOdata.DF <- GOanalysisDF(df.SL.2.b73hp.mp_BP_GOdata)
df.SL.2.b73hp.mp_CC_GOdata.DF <- GOanalysisDF(df.SL.2.b73hp.mp_CC_GOdata)
df.SL.2.b73hp.mp_MF_GOdata.DF <- GOanalysisDF(df.SL.2.b73hp.mp_MF_GOdata)

df.SL.2.mo17hp.mp_BP_GOdata.DF <- GOanalysisDF(df.SL.2.mo17hp.mp_BP_GOdata)
df.SL.2.mo17hp.mp_CC_GOdata.DF <- GOanalysisDF(df.SL.2.mo17hp.mp_CC_GOdata)
df.SL.2.mo17hp.mp_MF_GOdata.DF <- GOanalysisDF(df.SL.2.mo17hp.mp_MF_GOdata)
```
```{r view enrichment}
df.SL.2.mo17hp.mp_MF_GOdata.DF
```
###AMP/BMP + HP
```{r run GoanalysisGodata for amp/bmp+mp}
df.SL.3.b73hp.amp_BP_GOdata <- GOanalysisGodata(df.SL.3.b73hp.amp, gamerCutSL, "BP")
df.SL.3.b73hp.amp_CC_GOdata <- GOanalysisGodata(df.SL.3.b73hp.amp, gamerCutSL, "CC")
df.SL.3.b73hp.amp_MF_GOdata <- GOanalysisGodata(df.SL.3.b73hp.amp, gamerCutSL, "MF")

df.SL.3.mo17hp.amp_BP_GOdata <- GOanalysisGodata(df.SL.3.mo17hp.amp, gamerCutSL, "BP")
df.SL.3.mo17hp.amp_CC_GOdata <- GOanalysisGodata(df.SL.3.mo17hp.amp, gamerCutSL, "CC")
df.SL.3.mo17hp.amp_MF_GOdata <- GOanalysisGodata(df.SL.3.mo17hp.amp, gamerCutSL, "MF")

df.SL.3.b73hp.bmp_BP_GOdata <- GOanalysisGodata(df.SL.3.b73hp.bmp, gamerCutSL, "BP")
df.SL.3.b73hp.bmp_CC_GOdata <- GOanalysisGodata(df.SL.3.b73hp.bmp, gamerCutSL, "CC")
df.SL.3.b73hp.bmp_MF_GOdata <- GOanalysisGodata(df.SL.3.b73hp.bmp, gamerCutSL, "MF")

df.SL.3.mo17hp.bmp_BP_GOdata <- GOanalysisGodata(df.SL.3.mo17hp.bmp, gamerCutSL, "BP")
df.SL.3.mo17hp.bmp_CC_GOdata <- GOanalysisGodata(df.SL.3.mo17hp.bmp, gamerCutSL, "CC")
df.SL.3.mo17hp.bmp_MF_GOdata <- GOanalysisGodata(df.SL.3.mo17hp.bmp, gamerCutSL, "MF")
```

```{r run GoanalysisDF for amp/bmp+mp}
df.SL.3.b73hp.amp_BP_GOdata.DF <- GOanalysisDF(df.SL.3.b73hp.amp_BP_GOdata)
df.SL.3.b73hp.amp_CC_GOdata.DF <- GOanalysisDF(df.SL.3.b73hp.amp_CC_GOdata)
df.SL.3.b73hp.amp_MF_GOdata.DF <- GOanalysisDF(df.SL.3.b73hp.amp_MF_GOdata)

df.SL.3.mo17hp.amp_BP_GOdata.DF <- GOanalysisDF(df.SL.3.mo17hp.amp_BP_GOdata)
df.SL.3.mo17hp.amp_CC_GOdata.DF <- GOanalysisDF(df.SL.3.mo17hp.amp_CC_GOdata)
df.SL.3.mo17hp.amp_MF_GOdata.DF <- GOanalysisDF(df.SL.3.mo17hp.amp_MF_GOdata)

df.SL.3.b73hp.bmp_BP_GOdata.DF <- GOanalysisDF(df.SL.3.b73hp.bmp_BP_GOdata)
df.SL.3.b73hp.bmp_CC_GOdata.DF <- GOanalysisDF(df.SL.3.b73hp.bmp_CC_GOdata)
df.SL.3.b73hp.bmp_MF_GOdata.DF <- GOanalysisDF(df.SL.3.b73hp.bmp_MF_GOdata)

df.SL.3.mo17hp.bmp_BP_GOdata.DF <- GOanalysisDF(df.SL.3.mo17hp.bmp_BP_GOdata)
df.SL.3.mo17hp.bmp_CC_GOdata.DF <- GOanalysisDF(df.SL.3.mo17hp.bmp_CC_GOdata)
df.SL.3.mo17hp.bmp_MF_GOdata.DF <- GOanalysisDF(df.SL.3.mo17hp.bmp_MF_GOdata)
```


##Scatterplot - Photosynthesis subterms
###Make GOI description column
```{r define categories}
df.SL.scat1 <- df.SL
df.SL.scat1$Category <- "Other"
df.SL.scat1$Order <- 1
df.SL.scat1[(df.SL.scat1$Gene %in% thylakoid),"Category"] <- "Thylakoid"
df.SL.scat1[(df.SL.scat1$Gene %in% thylakoid),"Order"] <- 2
df.SL.scat1[(df.SL.scat1$Gene %in% lightrxn.PSIIassembly),"Category"] <- "Photosystem II assembly"
df.SL.scat1[(df.SL.scat1$Gene %in% lightrxn.PSIIassembly),"Order"] <- 3
df.SL.scat1[(df.SL.scat1$Gene %in% darkrxn.rPP),"Category"] <- "Calvin Cycle"
df.SL.scat1[(df.SL.scat1$Gene %in% darkrxn.rPP),"Order"] <- 4
df.SL.scat1[(df.SL.scat1$Gene %in% lightrxn.lightharvesting),"Category"] <- "Light harvesting"
df.SL.scat1[(df.SL.scat1$Gene %in% lightrxn.lightharvesting),"Order"] <- 5
df.SL.scat1[(df.SL.scat1$Gene %in% lightrxn.electrontransport),"Category"] <- "Electron transport"
df.SL.scat1[(df.SL.scat1$Gene %in% lightrxn.electrontransport),"Order"] <- 6
df.SL.scat1[(df.SL.scat1$Gene %in% lightrxn.NADPreduction),"Category"] <- "NADP reduction"
df.SL.scat1[(df.SL.scat1$Gene %in% lightrxn.NADPreduction),"Order"] <- 7
df.SL.scat1[(df.SL.scat1$Gene %in% lightrxn.phosphorylation),"Category"] <- "Phosphorylation"
df.SL.scat1[(df.SL.scat1$Gene %in% lightrxn.phosphorylation),"Order"] <- 8
df.SL.scat1[(df.SL.scat1$Gene %in% lightrxn.PSIassembly),"Category"] <- "Photosystem I assembly"
df.SL.scat1[(df.SL.scat1$Gene %in% lightrxn.PSIassembly),"Order"] <- 9
df.SL.scat1[(df.SL.scat1$Gene %in% lightrxn.OxEvolassembly),"Category"] <- "Oxygen Evolving Complex assembly"
df.SL.scat1[(df.SL.scat1$Gene %in% lightrxn.OxEvolassembly),"Order"] <- 10
df.SL.scat1[(df.SL.scat1$Gene %in% OxEvolComp),"Category"] <- "Oxygen Evolving Complex"
df.SL.scat1[(df.SL.scat1$Gene %in% OxEvolComp),"Order"] <- 11
df.SL.scat1[(df.SL.scat1$Gene %in% PtPS),"Category"] <- "Plastid-encoded photosynthesis"
df.SL.scat1[(df.SL.scat1$Gene %in% PtPS),"Order"] <- 12

#Re-order
df.SL.scat1 <- df.SL.scat1[(order(df.SL.scat1$Order)),]
```

###Set limits for plot
```{r}
df.SL.scat1.lims <- df.SL.scat1

df.SL.scat1.lims$l2fc[df.SL.scat1.lims$l2fc > 1.0] = 1.0
df.SL.scat1.lims$l2fc[df.SL.scat1.lims$l2fc < -1.0] = -1.0

df.SL.scat1.lims$l2fc.B73.Mo17[df.SL.scat1.lims$l2fc.B73.Mo17 > 1.0] = 1.0
df.SL.scat1.lims$l2fc.B73.Mo17[df.SL.scat1.lims$l2fc.B73.Mo17 < -1.0] = -1.0
```
###Make scatterplot
```{r}
fig.photosynthesis.bm <- ggplot(df.SL.scat1.lims, mapping=aes(x=l2fc, y=l2fc.B73.Mo17, color=Category))+
  geom_point(size=.8)+
  scale_color_manual(values=c("Other"="grey85","Thylakoid"="grey50", "Light harvesting"="#0bb003", "Electron transport"="#0235de", "NADP reduction"="#f5a800", "Phosphorylation"="#eb0003", "Photosystem I assembly"="#e600f5", "Photosystem II assembly"="brown", "Oxygen Evolving Complex assembly"="black", "Calvin Cycle"="purple", "Oxygen Evolving Complex"="red", "Plastid-encoded photosynthesis"="#1af97c"), name="Protein Category")+
    theme_classic()+
    xlim(c(-1,1))+
    ylim(c(-1,1))+
    geom_vline(xintercept=0, linetype="dashed", color = "gray50")+
    geom_hline(yintercept=0, linetype="dashed", color = "gray50")+
    #geom_abline(intercept =0 , slope = 1, linetype="dashed", color="gray50")+
    theme(axis.text=element_text(size=10),
              legend.position="bottom", legend.title = element_text(size=5), legend.text = element_text(size=5))+
    guides(color=guide_legend(nrow=5,byrow=TRUE))+
    xlab("log2 hyb/MP protein abundance")+
    ylab("log2 B73/Mo17 protein abundance")+
    coord_fixed(expand = F)

fig.photosynthesis.bm
```
###Add density curves
```{r}
#add density curves
#add marginal density plots and resave
    densityplotX<-axis_canvas(fig.photosynthesis.bm, axis="x")+
      geom_density(aes(df.SL.scat1.lims[,"l2fc"], color=df.SL.scat1.lims[,"Category"]), df.SL.scat1.lims, size=0.7)+
      scale_color_manual(values=c("Other"="grey85","Thylakoid"="grey50", "Light harvesting"="#0bb003", "Electron transport"="#0235de", "NADP reduction"="#f5a800", "Phosphorylation"="#eb0003", "Photosystem I assembly"="#e600f5", "Photosystem II assembly"="brown", "Oxygen Evolving Complex assembly"="black", "Calvin Cycle"="purple", "Oxygen Evolving Complex"="red", "Plastid-encoded photosynthesis"="#1af97c"))+
      geom_vline(xintercept=0, linetype="dashed", color = "gray50")+
      theme_void()
    #set up marginal density plot Y axis
    densityplotY<-axis_canvas(fig.photosynthesis.bm, axis="y", coord_flip = T)+
      geom_density(aes(df.SL.scat1.lims[,"l2fc.B73.Mo17"], color=df.SL.scat1.lims[,"Category"]), df.SL.scat1.lims, size=0.7)+
      scale_color_manual(values=c("Other"="grey85","Thylakoid"="grey50", "Light harvesting"="#0bb003", "Electron transport"="#0235de", "NADP reduction"="#f5a800", "Phosphorylation"="#eb0003", "Photosystem I assembly"="#e600f5", "Photosystem II assembly"="brown", "Oxygen Evolving Complex assembly"="black", "Calvin Cycle"="purple", "Oxygen Evolving Complex"="red", "Plastid-encoded photosynthesis"="#1af97c"))+
      geom_vline(xintercept=0, linetype="dashed", color = "gray50")+
      theme_void()+
      coord_flip()
    #put two plots together
    q <- insert_xaxis_grob(fig.photosynthesis.bm, densityplotX, grid::unit(0.275, "null"), position = "top")
    q <- insert_yaxis_grob(q, densityplotY, grid::unit(0.275, "null"), position = "right")
    #save
  ggsave(plot=q, filename = "FigPhotosynthesisBM.tiff",
        device="tiff", dpi=300, width=4.5, height=4.5, compression = "lzw")
```

##Scatterplot - Photosynthesis
###Make GOI description column
```{r define categories}
df.SL.scat1b <- df.SL
df.SL.scat1b$Category <- "Other"
df.SL.scat1b$Order <- 1
df.SL.scat1b[(df.SL.scat1b$Gene %in% thylakoid),"Category"] <- "Thylakoid"
df.SL.scat1b[(df.SL.scat1b$Gene %in% thylakoid),"Order"] <- 2
df.SL.scat1b[(df.SL.scat1b$Gene %in% lightrxn),"Category"] <- "Light Reaction"
df.SL.scat1b[(df.SL.scat1b$Gene %in% lightrxn),"Order"] <- 3
df.SL.scat1b[(df.SL.scat1b$Gene %in% darkrxn),"Category"] <- "Calvin Cycle"
df.SL.scat1b[(df.SL.scat1b$Gene %in% darkrxn),"Order"] <- 4
df.SL.scat1b[(df.SL.scat1b$Gene %in% PtPS),"Category"] <- "Plastid-encoded photosynthesis"
df.SL.scat1b[(df.SL.scat1b$Gene %in% PtPS),"Order"] <- 5

#Re-order
df.SL.scat1b <- df.SL.scat1b[(order(df.SL.scat1b$Order)),]
```

###Set limits for plot
```{r}
df.SL.scat1b.lims <- df.SL.scat1b

df.SL.scat1b.lims$l2fc[df.SL.scat1b.lims$l2fc > 1.0] = 1.0
df.SL.scat1b.lims$l2fc[df.SL.scat1b.lims$l2fc < -1.0] = -1.0

df.SL.scat1b.lims$l2fc.B73.Mo17[df.SL.scat1b.lims$l2fc.B73.Mo17 > 1.0] = 1.0
df.SL.scat1b.lims$l2fc.B73.Mo17[df.SL.scat1b.lims$l2fc.B73.Mo17 < -1.0] = -1.0
```
###Make scatterplot
```{r}
fig.photosynthesis.bm.2 <- ggplot(df.SL.scat1b.lims, mapping=aes(x=l2fc, y=l2fc.B73.Mo17, color=Category))+
  geom_point(size=.8)+
  scale_color_manual(values=c("Other"="grey85","Thylakoid"="grey30", "Light Reaction"="#0bb003", "Calvin Cycle"="#0235de", "Plastid-encoded photosynthesis"="#eb0003"), name="Protein Category")+
    theme_classic()+
    xlim(c(-1,1))+
    ylim(c(-1,1))+
    geom_vline(xintercept=0, linetype="dashed", color = "gray50")+
    geom_hline(yintercept=0, linetype="dashed", color = "gray50")+
    #geom_abline(intercept =0 , slope = 1, linetype="dashed", color="gray50")+
    theme(axis.text=element_text(size=10),
              legend.position="bottom", legend.title = element_text(size=5), legend.text = element_text(size=5))+
    guides(color=guide_legend(nrow=2,byrow=TRUE))+
    xlab("log2 hyb/MP protein abundance")+
    ylab("log2 B73/Mo17 protein abundance")+
    coord_fixed(expand = F)

fig.photosynthesis.bm.2
```
###Add density curves
```{r}
#add density curves
#add marginal density plots and resave
    densityplotX<-axis_canvas(fig.photosynthesis.bm.2, axis="x")+
      geom_density(aes(df.SL.scat1b.lims[,"l2fc"], color=df.SL.scat1b.lims[,"Category"]), df.SL.scat1b.lims, size=0.7)+
      scale_color_manual(values=c("Other"="grey85","Thylakoid"="grey30", "Light Reaction"="#0bb003", "Calvin Cycle"="#0235de", "Plastid-encoded photosynthesis"="#eb0003"))+
      geom_vline(xintercept=0, linetype="dashed", color = "gray50")+
      theme_void()
    #set up marginal density plot Y axis
    densityplotY<-axis_canvas(fig.photosynthesis.bm.2, axis="y", coord_flip = T)+
      geom_density(aes(df.SL.scat1b.lims[,"l2fc.B73.Mo17"], color=df.SL.scat1b.lims[,"Category"]), df.SL.scat1b.lims, size=0.7)+
      scale_color_manual(values=c("Other"="grey85","Thylakoid"="grey30", "Light Reaction"="#0bb003", "Calvin Cycle"="#0235de", "Plastid-encoded photosynthesis"="#eb0003"))+
      geom_vline(xintercept=0, linetype="dashed", color = "gray50")+
      theme_void()+
      coord_flip()
    #put two plots together
    q <- insert_xaxis_grob(fig.photosynthesis.bm.2, densityplotX, grid::unit(0.275, "null"), position = "top")
    q <- insert_yaxis_grob(q, densityplotY, grid::unit(0.275, "null"), position = "right")
    #save
  ggsave(plot=q, filename = "FigPhotosynthesisBM2.tiff",
        device="tiff", dpi=300, width=4.5, height=4.5, compression = "lzw")
```

##Scatterplot - Ribosome
###Make GOI description column
```{r define categories}
df.SL.scat2 <- df.SL
df.SL.scat2$Category <- "Other"
df.SL.scat2$Order <- 1
df.SL.scat2[(df.SL.scat2$Gene %in% eukRibo),"Category"] <- "Eukaryotic ribsome"
df.SL.scat2[(df.SL.scat2$Gene %in% eukRibo),"Order"] <- 2
df.SL.scat2[(df.SL.scat2$Gene %in% Nuc.PtRibo),"Category"] <- "Nuclear-encoded plastid ribosome"
df.SL.scat2[(df.SL.scat2$Gene %in% Nuc.PtRibo),"Order"] <- 3
df.SL.scat2[(df.SL.scat2$Gene %in% Pt.ribo),"Category"] <- "Plastid-encoded ribosome"
df.SL.scat2[(df.SL.scat2$Gene %in% Pt.ribo),"Order"] <- 4
#Re-order
df.SL.scat2 <- df.SL.scat2[(order(df.SL.scat2$Order)),]
```
###Set limits for plot
```{r}
df.SL.scat2.lims <- df.SL.scat2

df.SL.scat2.lims$l2fc[df.SL.scat2.lims$l2fc > 1.0] = 1.0
df.SL.scat2.lims$l2fc[df.SL.scat2.lims$l2fc < -1.0] = -1.0

df.SL.scat2.lims$l2fc.B73.Mo17[df.SL.scat2.lims$l2fc.B73.Mo17 > 1.0] = 1.0
df.SL.scat2.lims$l2fc.B73.Mo17[df.SL.scat2.lims$l2fc.B73.Mo17 < -1.0] = -1.0
```
###Make scatterplot
```{r}
fig.ribosome.bm <- ggplot(df.SL.scat2.lims, mapping=aes(x=l2fc, y=l2fc.B73.Mo17, color=Category))+
  geom_point(size=.8)+
  scale_color_manual(values=c("Other"="grey85","Eukaryotic ribsome"="red", "Nuclear-encoded plastid ribosome"="#0235de", "Plastid-encoded ribosome"="#0bb003"), name="Protein Category")+
    theme_classic()+
    xlim(c(-1,1))+
    ylim(c(-1,1))+
    geom_vline(xintercept=0, linetype="dashed", color = "gray50")+
    geom_hline(yintercept=0, linetype="dashed", color = "gray50")+
    #geom_abline(intercept =0 , slope = 1, linetype="dashed", color="gray50")+
    theme(axis.text=element_text(size=10),
              legend.position="bottom", legend.title = element_text(size=6), legend.text = element_text(size=8))+
    guides(color=guide_legend(nrow=2,byrow=TRUE))+
    xlab("log2 hyb/MP protein abundance")+
    ylab("log2 B73/Mo17 protein abundance")+
    coord_fixed(expand = F)

fig.ribosome.bm
```
###Add density curves
```{r}
#add density curves
#add marginal density plots and resave
    densityplotX<-axis_canvas(fig.ribosome.bm, axis="x")+
      geom_density(aes(df.SL.scat2.lims[,"l2fc"], color=df.SL.scat2.lims[,"Category"]), df.SL.scat2.lims, size=0.7)+
      scale_color_manual(values=c("Other"="grey85","Eukaryotic ribsome"="red", "Nuclear-encoded plastid ribosome"="#0235de", "Plastid-encoded ribosome"="#0bb003"))+
      geom_vline(xintercept=0, linetype="dashed", color = "gray50")+
      theme_void()
    #set up marginal density plot Y axis
    densityplotY<-axis_canvas(fig.ribosome.bm, axis="y", coord_flip = T)+
      geom_density(aes(df.SL.scat2.lims[,"l2fc.B73.Mo17"], color=df.SL.scat2.lims[,"Category"]), df.SL.scat2.lims, size=0.7)+
      scale_color_manual(values=c("Other"="grey85","Eukaryotic ribsome"="red", "Nuclear-encoded plastid ribosome"="#0235de", "Plastid-encoded ribosome"="#0bb003"))+
      geom_vline(xintercept=0, linetype="dashed", color = "gray50")+
      theme_void()+
      coord_flip()
    #put two plots together
    q <- insert_xaxis_grob(fig.ribosome.bm, densityplotX, grid::unit(0.275, "null"), position = "top")
    q <- insert_yaxis_grob(q, densityplotY, grid::unit(0.275, "null"), position = "right")
    #save
  ggsave(plot=q, filename = "FigRibosomeBM.tiff",
        device="tiff", dpi=300, width=4.5, height=4.5, compression = "lzw")
```
#ACS and BxM enrichment overlap
```{r subset to proteins amp/bmp in both the mutant and hybrid}
df.SL.acs.amp <- df.SL.amp[(df.SL.amp$Accession%in%df.acs.amp$Accession),]
df.SL.acs.bmp <- df.SL.bmp[(df.SL.bmp$Accession%in%df.acs.bmp$Accession),]
df.SL.acs <- df.SL[(df.SL$Gene%in%df.acs$Gene),]
```
```{r trim gamer}
#Trim gamer to only those genes detected in our data, to use as background for enrichment
gamerCutSLacs <- gamer[(gamer$db_object_id %in% df.SL.acs$Gene),]
```
```{r run GOanalysisGodata on hyb+acs overlapping enrichment}
df.SL.acs.amp_BP_GOdata <- GOanalysisGodata(df.SL.acs.amp, gamerCutSLacs, "BP", nodesize = 5)
df.SL.acs.amp_MF_GOdata <- GOanalysisGodata(df.SL.acs.amp, gamerCutSLacs, "MF", nodesize = 5)
df.SL.acs.amp_CC_GOdata <- GOanalysisGodata(df.SL.acs.amp, gamerCutSLacs, "CC", nodesize = 5)

df.SL.acs.bmp_BP_GOdata <- GOanalysisGodata(df.SL.acs.bmp, gamerCutSLacs, "BP", nodesize = 5)
df.SL.acs.bmp_MF_GOdata <- GOanalysisGodata(df.SL.acs.bmp, gamerCutSLacs, "MF", nodesize = 5)
df.SL.acs.bmp_CC_GOdata <- GOanalysisGodata(df.SL.acs.bmp, gamerCutSLacs, "CC", nodesize = 5)
```
```{r run GOanalysisMap on hyb+acs overlapping enrichment}
dir.create("GOmaps")
GOanalysisMap(df.SL.acs.amp_BP_GOdata, "GOmaps/df.SL.acs.amp_BP_GOdata")
GOanalysisMap(df.SL.acs.amp_MF_GOdata, "GOmaps/df.SL.acs.amp_MF_GOdata")
GOanalysisMap(df.SL.acs.amp_CC_GOdata, "GOmaps/df.SL.acs.amp_CC_GOdata")

GOanalysisMap(df.SL.acs.bmp_BP_GOdata, "GOmaps/df.SL.acs.bmp_BP_GOdata")
GOanalysisMap(df.SL.acs.bmp_MF_GOdata, "GOmaps/df.SL.acs.bmp_MF_GOdata")
GOanalysisMap(df.SL.acs.bmp_CC_GOdata, "GOmaps/df.SL.acs.bmp_CC_GOdata")
```
```{r run GOanalysisDF on hyb+acs overlapping enrichment}
df.SL.acs.amp_BP_GOdata.DF <- GOanalysisDF(df.SL.acs.amp_BP_GOdata)
df.SL.acs.amp_MF_GOdata.DF <- GOanalysisDF(df.SL.acs.amp_MF_GOdata)
df.SL.acs.amp_CC_GOdata.DF <- GOanalysisDF(df.SL.acs.amp_CC_GOdata)

df.SL.acs.bmp_BP_GOdata.DF <- GOanalysisDF(df.SL.acs.bmp_BP_GOdata)
df.SL.acs.bmp_MF_GOdata.DF <- GOanalysisDF(df.SL.acs.bmp_MF_GOdata)
df.SL.acs.bmp_CC_GOdata.DF <- GOanalysisDF(df.SL.acs.bmp_CC_GOdata)
```
```{r view enrichment}
df.SL.acs.bmp_BP_GOdata.DF
```
##Merge acs and Seedling Leaf BxM data
```{r merge data for scatterplots}
df.acs.SL <- merge.data.frame(df.acs[,c(1:8,19,24:25)], df.SL[,c(6,c(27,29))], by="Accession", all = F, suffixes = c(".acs", ".BM"))
```

##Scatterplot - Photosynthesis
###Make GOI description column
```{r define categories}
df.acs.SL.scat1 <- df.acs.SL
df.acs.SL.scat1$Category <- "Other"
df.acs.SL.scat1$Order <- 1
df.acs.SL.scat1[(df.acs.SL.scat1$Gene %in% thylakoid),"Category"] <- "Thylakoid"
df.acs.SL.scat1[(df.acs.SL.scat1$Gene %in% thylakoid),"Order"] <- 2
df.acs.SL.scat1[(df.acs.SL.scat1$Gene %in% lightrxn.PSIIassembly),"Category"] <- "Photosystem II assembly"
df.acs.SL.scat1[(df.acs.SL.scat1$Gene %in% lightrxn.PSIIassembly),"Order"] <- 3
df.acs.SL.scat1[(df.acs.SL.scat1$Gene %in% darkrxn.rPP),"Category"] <- "Calvin Cycle"
df.acs.SL.scat1[(df.acs.SL.scat1$Gene %in% darkrxn.rPP),"Order"] <- 4
df.acs.SL.scat1[(df.acs.SL.scat1$Gene %in% lightrxn.lightharvesting),"Category"] <- "Light harvesting"
df.acs.SL.scat1[(df.acs.SL.scat1$Gene %in% lightrxn.lightharvesting),"Order"] <- 5
df.acs.SL.scat1[(df.acs.SL.scat1$Gene %in% lightrxn.electrontransport),"Category"] <- "Electron transport"
df.acs.SL.scat1[(df.acs.SL.scat1$Gene %in% lightrxn.electrontransport),"Order"] <- 6
df.acs.SL.scat1[(df.acs.SL.scat1$Gene %in% lightrxn.NADPreduction),"Category"] <- "NADP reduction"
df.acs.SL.scat1[(df.acs.SL.scat1$Gene %in% lightrxn.NADPreduction),"Order"] <- 7
df.acs.SL.scat1[(df.acs.SL.scat1$Gene %in% lightrxn.phosphorylation),"Category"] <- "Phosphorylation"
df.acs.SL.scat1[(df.acs.SL.scat1$Gene %in% lightrxn.phosphorylation),"Order"] <- 8
df.acs.SL.scat1[(df.acs.SL.scat1$Gene %in% lightrxn.PSIassembly),"Category"] <- "Photosystem I assembly"
df.acs.SL.scat1[(df.acs.SL.scat1$Gene %in% lightrxn.PSIassembly),"Order"] <- 9
df.acs.SL.scat1[(df.acs.SL.scat1$Gene %in% lightrxn.OxEvolassembly),"Category"] <- "Oxygen Evolving Complex assembly"
df.acs.SL.scat1[(df.acs.SL.scat1$Gene %in% lightrxn.OxEvolassembly),"Order"] <- 10
df.acs.SL.scat1[(df.acs.SL.scat1$Gene %in% OxEvolComp),"Category"] <- "Oxygen Evolving Complex"
df.acs.SL.scat1[(df.acs.SL.scat1$Gene %in% OxEvolComp),"Order"] <- 11
df.acs.SL.scat1[(df.acs.SL.scat1$Gene %in% PtPS),"Category"] <- "Plastid-encoded photosynthesis"
df.acs.SL.scat1[(df.acs.SL.scat1$Gene %in% PtPS),"Order"] <- 12


#Re-order
df.acs.SL.scat1 <- df.acs.SL.scat1[(order(df.acs.SL.scat1$Order)),]
```


###Set limits for plot
```{r}
df.acs.SL.scat1.lims <- df.acs.SL.scat1

df.acs.SL.scat1.lims$l2fc.acs[df.acs.SL.scat1.lims$l2fc.acs > 1.0] = 1.0
df.acs.SL.scat1.lims$l2fc.acs[df.acs.SL.scat1.lims$l2fc.acs < -1.0] = -1.0

df.acs.SL.scat1.lims$l2fc.BM[df.acs.SL.scat1.lims$l2fc.BM > 1.0] = 1.0
df.acs.SL.scat1.lims$l2fc.BM[df.acs.SL.scat1.lims$l2fc.BM < -1.0] = -1.0
```
###Make scatterplot
```{r}
fig.photosynthesis.acs <- ggplot(df.acs.SL.scat1.lims, mapping=aes(x=l2fc.BM, y=l2fc.acs, color=Category))+
  geom_point(size=.8)+
  scale_color_manual(values=c("Other"="grey85","Thylakoid"="grey50", "Light harvesting"="#0bb003", "Electron transport"="#0235de", "NADP reduction"="#f5a800", "Phosphorylation"="#eb0003", "Photosystem I assembly"="#e600f5", "Photosystem II assembly"="brown", "Oxygen Evolving Complex assembly"="black", "Calvin Cycle"="purple", "Oxygen Evolving Complex"="red", "Plastid-encoded photosynthesis"="#1af97c"), name="Protein Category")+
    theme_classic()+
    xlim(c(-1,1))+
    ylim(c(-1,1))+
    geom_vline(xintercept=0, linetype="dashed", color = "gray50")+
    geom_hline(yintercept=0, linetype="dashed", color = "gray50")+
    #geom_abline(intercept =0 , slope = 1, linetype="dashed", color="gray50")+
    theme(axis.text=element_text(size=10),
              legend.position="bottom", legend.title = element_text(size=6), legend.text = element_text(size=6))+
    guides(color=guide_legend(nrow=4,byrow=TRUE))+
    xlab("log2 hyb/MP protein abundance")+
    ylab("log2 acs/B73 protein abundance")+
    coord_fixed(expand = F)

fig.photosynthesis.acs
```
###Add density curves
```{r}
#add density curves
#add marginal density plots and resave
    densityplotX<-axis_canvas(fig.photosynthesis.acs, axis="x")+
      geom_density(aes(df.acs.SL.scat1.lims[,"l2fc.BM"], color=df.acs.SL.scat1.lims[,"Category"]), df.acs.SL.scat1.lims, size=0.7)+
      scale_color_manual(values=c("Other"="grey85","Thylakoid"="grey50", "Light harvesting"="#0bb003", "Electron transport"="#0235de", "NADP reduction"="#f5a800", "Phosphorylation"="#eb0003", "Photosystem I assembly"="#e600f5", "Photosystem II assembly"="brown", "Oxygen Evolving Complex assembly"="black", "Calvin Cycle"="purple", "Oxygen Evolving Complex"="red", "Plastid-encoded photosynthesis"="#1af97c"))+
      geom_vline(xintercept=0, linetype="dashed", color = "gray50")+
      theme_void()
    #set up marginal density plot Y axis
    densityplotY<-axis_canvas(fig.photosynthesis.acs, axis="y", coord_flip = T)+
      geom_density(aes(df.acs.SL.scat1.lims[,"l2fc.acs"], color=df.acs.SL.scat1.lims[,"Category"]), df.acs.SL.scat1.lims, size=0.7)+
      scale_color_manual(values=c("Other"="grey85","Thylakoid"="grey50", "Light harvesting"="#0bb003", "Electron transport"="#0235de", "NADP reduction"="#f5a800", "Phosphorylation"="#eb0003", "Photosystem I assembly"="#e600f5", "Photosystem II assembly"="brown", "Oxygen Evolving Complex assembly"="black", "Calvin Cycle"="purple", "Oxygen Evolving Complex"="red", "Plastid-encoded photosynthesis"="#1af97c"))+
      geom_vline(xintercept=0, linetype="dashed", color = "gray50")+
      theme_void()+
      coord_flip()
    #put two plots together
    q <- insert_xaxis_grob(fig.photosynthesis.acs, densityplotX, grid::unit(0.275, "null"), position = "top")
    q <- insert_yaxis_grob(q, densityplotY, grid::unit(0.275, "null"), position = "right")
    #save
  ggsave(plot=q, filename = "FigPhotosynthesisACS.tiff",
        device="tiff", dpi=300, width=4.5, height=4.5, compression = "lzw")
```

##Scatterplot - Ribosome
###Make GOI description column
```{r define categories}
df.acs.SL.scat2 <- df.acs.SL
df.acs.SL.scat2$Category <- "Other"
df.acs.SL.scat2$Order <- 1
df.acs.SL.scat2[(df.acs.SL.scat2$Gene %in% eukRibo),"Category"] <- "Eukaryotic ribsome"
df.acs.SL.scat2[(df.acs.SL.scat2$Gene %in% eukRibo),"Order"] <- 2
df.acs.SL.scat2[(df.acs.SL.scat2$Gene %in% Nuc.PtRibo),"Category"] <- "Nuclear-encoded plastid ribosome"
df.acs.SL.scat2[(df.acs.SL.scat2$Gene %in% Nuc.PtRibo),"Order"] <- 3
df.acs.SL.scat2[(df.acs.SL.scat2$Gene %in% Pt.ribo),"Category"] <- "Plastid-encoded ribosome"
df.acs.SL.scat2[(df.acs.SL.scat2$Gene %in% Pt.ribo),"Order"] <- 4
#Re-order
df.acs.SL.scat2 <- df.acs.SL.scat2[(order(df.acs.SL.scat2$Order)),]
```
###Set limits for plot
```{r}
df.acs.SL.scat2.lims <- df.acs.SL.scat2

df.acs.SL.scat2.lims$l2fc.acs[df.acs.SL.scat2.lims$l2fc.acs > 1.0] = 1.0
df.acs.SL.scat2.lims$l2fc.acs[df.acs.SL.scat2.lims$l2fc.acs < -1.0] = -1.0

df.acs.SL.scat2.lims$l2fc.BM[df.acs.SL.scat2.lims$l2fc.BM > 1.0] = 1.0
df.acs.SL.scat2.lims$l2fc.BM[df.acs.SL.scat2.lims$l2fc.BM < -1.0] = -1.0
```
###Make scatterplot
```{r}
fig.ribosome <- ggplot(df.acs.SL.scat2.lims, mapping=aes(x=l2fc.BM, y=l2fc.acs, color=Category))+
  geom_point(size=.8)+
  scale_color_manual(values=c("Other"="grey85","Eukaryotic ribsome"="red", "Nuclear-encoded plastid ribosome"="#0235de", "Plastid-encoded ribosome"="#0bb003"), name="Protein Category")+
    theme_classic()+
    xlim(c(-1,1))+
    ylim(c(-1,1))+
    geom_vline(xintercept=0, linetype="dashed", color = "gray50")+
    geom_hline(yintercept=0, linetype="dashed", color = "gray50")+
    #geom_abline(intercept =0 , slope = 1, linetype="dashed", color="gray50")+
    theme(axis.text=element_text(size=10),
              legend.position="bottom", legend.title = element_text(size=6), legend.text = element_text(size=8))+
    guides(color=guide_legend(nrow=2,byrow=TRUE))+
    xlab("log2 hyb/MP protein abundance")+
    ylab("log2 acs/B73 protein abundance")+
    coord_fixed(expand = F)

fig.ribosome
```
###Add density curves
```{r}
#add density curves
#add marginal density plots and resave
    densityplotX<-axis_canvas(fig.ribosome, axis="x")+
      geom_density(aes(df.acs.SL.scat2.lims[,"l2fc.BM"], color=df.acs.SL.scat2.lims[,"Category"]), df.acs.SL.scat2.lims, size=0.7)+
      scale_color_manual(values=c("Other"="grey85","Eukaryotic ribsome"="red", "Nuclear-encoded plastid ribosome"="#0235de", "Plastid-encoded ribosome"="#0bb003"))+
      geom_vline(xintercept=0, linetype="dashed", color = "gray50")+
      theme_void()
    #set up marginal density plot Y axis
    densityplotY<-axis_canvas(fig.ribosome, axis="y", coord_flip = T)+
      geom_density(aes(df.acs.SL.scat2.lims[,"l2fc.acs"], color=df.acs.SL.scat2.lims[,"Category"]), df.acs.SL.scat2.lims, size=0.7)+
      scale_color_manual(values=c("Other"="grey85","Eukaryotic ribsome"="red", "Nuclear-encoded plastid ribosome"="#0235de", "Plastid-encoded ribosome"="#0bb003"))+
      geom_vline(xintercept=0, linetype="dashed", color = "gray50")+
      theme_void()+
      coord_flip()
    #put two plots together
    q <- insert_xaxis_grob(fig.ribosome, densityplotX, grid::unit(0.275, "null"), position = "top")
    q <- insert_yaxis_grob(q, densityplotY, grid::unit(0.275, "null"), position = "right")
    #save
  ggsave(plot=q, filename = "FigRibosome.tiff",
        device="tiff", dpi=300, width=4.5, height=4.5, compression = "lzw")
```
##Scatterplot - Ethylene
###Make GOI description column
```{r define categories}
df.acs.SL.scat3 <- df.acs.SL
df.acs.SL.scat3$Category <- "Other"
df.acs.SL.scat3$Order <- 1
df.acs.SL.scat3[(df.acs.SL.scat3$Gene %in% gamer.responseToEt),"Category"] <- "Response to ethylene"
df.acs.SL.scat3[(df.acs.SL.scat3$Gene %in% gamer.responseToEt),"Order"] <- 2
df.acs.SL.scat3[(df.acs.SL.scat3$Gene %in% gamer.EtSignaling),"Category"] <- "Ethylene signaling"
df.acs.SL.scat3[(df.acs.SL.scat3$Gene %in% gamer.EtSignaling),"Order"] <- 3
df.acs.SL.scat3[(df.acs.SL.scat3$Gene %in% gamer.DetectionOfEt),"Category"] <- "Detection of ethylene"
df.acs.SL.scat3[(df.acs.SL.scat3$Gene %in% gamer.DetectionOfEt),"Order"] <- 4
df.acs.SL.scat3[(df.acs.SL.scat3$Gene %in% gamer.CellResponseToEt),"Category"] <- "Cellular response to ethylene"
df.acs.SL.scat3[(df.acs.SL.scat3$Gene %in% gamer.CellResponseToEt),"Order"] <- 5

#Re-order
df.acs.SL.scat3 <- df.acs.SL.scat3[(order(df.acs.SL.scat3$Order)),]
```

###Set limits for plot
```{r}
df.acs.SL.scat3.lims <- df.acs.SL.scat3

df.acs.SL.scat3.lims$l2fc.acs[df.acs.SL.scat3.lims$l2fc.acs > 1.0] = 1.0
df.acs.SL.scat3.lims$l2fc.acs[df.acs.SL.scat3.lims$l2fc.acs < -1.0] = -1.0

df.acs.SL.scat3.lims$l2fc.BM[df.acs.SL.scat3.lims$l2fc.BM > 1.0] = 1.0
df.acs.SL.scat3.lims$l2fc.BM[df.acs.SL.scat3.lims$l2fc.BM < -1.0] = -1.0
```
###Make scatterplot
```{r}
fig.ethylene <- ggplot(df.acs.SL.scat3.lims, mapping=aes(x=l2fc.BM, y=l2fc.acs, color=Category))+
  geom_point(size=.8)+
  scale_color_manual(values=c("Other"="grey85","Response to ethylene"="#0bb003", "Ethylene signaling"="#0235de", "Detection of ethylene"="#f5a800", "Cellular response to ethylene"="#eb0003"), name="Protein Category")+
    theme_classic()+
    xlim(c(-1,1))+
    ylim(c(-1,1))+
    geom_vline(xintercept=0, linetype="dashed", color = "gray50")+
    geom_hline(yintercept=0, linetype="dashed", color = "gray50")+
    #geom_abline(intercept =0 , slope = 1, linetype="dashed", color="gray50")+
    theme(axis.text=element_text(size=10),
              legend.position="bottom", legend.title = element_text(size=6), legend.text = element_text(size=8))+
    guides(color=guide_legend(nrow=2,byrow=TRUE))+
    xlab("log2 hyb/MP protein abundance")+
    ylab("log2 acs/B73 protein abundance")+
    coord_fixed(expand = F)

fig.ethylene
```
###Add density curves
```{r}
#add density curves
#add marginal density plots and resave
    densityplotX<-axis_canvas(fig.ethylene, axis="x")+
      geom_density(aes(df.acs.SL.scat3.lims[,"l2fc.BM"], color=df.acs.SL.scat3.lims[,"Category"]), df.acs.SL.scat3.lims, size=0.7)+
      scale_color_manual(values=c("Other"="grey85","Response to ethylene"="red", "Ethylene biosynthesis"="blue"))+
      geom_vline(xintercept=0, linetype="dashed", color = "gray50")+
      theme_void()
    #set up marginal density plot Y axis
    densityplotY<-axis_canvas(fig.ethylene, axis="y", coord_flip = T)+
      geom_density(aes(df.acs.SL.scat3.lims[,"l2fc.acs"], color=df.acs.SL.scat3.lims[,"Category"]), df.acs.SL.scat3.lims, size=0.7)+
      scale_color_manual(values=c("Other"="grey85","Response to ethylene"="red", "Ethylene biosynthesis"="blue"))+
      geom_vline(xintercept=0, linetype="dashed", color = "gray50")+
      theme_void()+
      coord_flip()
    #put two plots together
    q <- insert_xaxis_grob(fig.ethylene, densityplotX, grid::unit(0.275, "null"), position = "top")
    q <- insert_yaxis_grob(q, densityplotY, grid::unit(0.275, "null"), position = "right")
    #save
  ggsave(plot=q, filename = "FigEthylene.tiff",
        device="tiff", dpi=300, width=4.5, height=4.5, compression = "lzw")
```










